--- a/test/CMakeLists.txt	2022-07-28 11:25:13.000000000 -0400
+++ b/test/CMakeLists.txt	2022-07-28 11:26:40.000000000 -0400
@@ -205,7 +205,7 @@
     set(test_exe ${test}.x)
     add_executable(${test_exe} ${test_src})
     add_dependencies(${test_exe} bufr_${kind})
-    target_link_libraries(${test_exe} PRIVATE bufr::bufr_${kind})
+    target_link_libraries(${test_exe} PRIVATE bufr::bufr_${kind} m)
     add_test(NAME ${test} COMMAND ${CMAKE_BINARY_DIR}/test/${test_exe})
   endforeach()
 endforeach()
