diff --git a/Makefile b/Makefile
index 0802f65..3816b6e 100644
--- a/Makefile
+++ b/Makefile
@@ -2,9 +2,9 @@ VERSION=1.0.0
 MINOR_VER=20140314
 CC=gcc
 ifdef DEBUG
-CFLAGS=-g3 -W -Wall -O0 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -mpopcnt -mssse3
+CFLAGS=-g3 -W -Wall -O0 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE
 else
-CFLAGS=-W -Wall -O4 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -mpopcnt -mssse3
+CFLAGS=-W -Wall -O4 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE
 endif
 INSTALLDIR=/usr/local/bin
 GLIBS=-lm -lpthread
diff --git a/ksw.c b/ksw.c
index 15dd0f2..22641ed 100644
--- a/ksw.c
+++ b/ksw.c
@@ -25,7 +25,7 @@
 
 #include <stdlib.h>
 #include <stdint.h>
-#include <emmintrin.h>
+#include "sse2neon.h"
 #include "ksw.h"
 
 #ifdef USE_MALLOC_WRAPPERS
