diff --git a/CMakeLists.txt b/CMakeLists.txt
index e98d74d..de8740e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,7 @@ macro(add_library NAME)
     endif()
 endmacro()
 find_package(hip)
-
+set( Tensile_TEST_LOCAL_PATH "" CACHE PATH "Use local Tensile directory instead of fetching a GitHub branch" )
 if(CMAKE_CXX_COMPILER MATCHES ".*/hcc$")
     set(TENSILE_USE_LLVM ON CACHE BOOL "Use LLVM for parsing config files.")
     set(TENSILE_USE_MSGPACK OFF CACHE BOOL "Use msgpack for parsing config files.")
@@ -79,11 +79,30 @@ endif()
 #     set(MIOPEN_TENSILE_SRC dev)
 # endif()
 # Use the virtual-env setup and download package from specified repo:
-virtualenv_install(wheel)
-virtualenv_install("git+https://github.com/ROCmSoftwarePlatform/Tensile.git@${MIOPEN_TENSILE_TAG}")
-list(APPEND CMAKE_PREFIX_PATH ${VIRTUALENV_HOME_DIR})
 
-find_package(Tensile REQUIRED COMPONENTS HIP)
+option(BUILD_USING_LOCAL_TENSILE "Build as a shared library" ON )
+
+if (WIN32)
+  set( Tensile_ROOT "${CMAKE_BINARY_DIR}/virtualenv/Lib/site-packages/Tensile" )
+endif()
+
+include(virtualenv)
+if (BUILD_USING_LOCAL_TENSILE)
+  virtualenv_install(${Tensile_TEST_LOCAL_PATH})
+  message (STATUS "using local Tensile from ${Tensile_TEST_LOCAL_PATH}, copied to ${Tensile_ROOT}")
+else()
+  # Use the virtual-env setup and download package from specified repot:
+  set( tensile_fork "ROCmSoftwarePlatform" CACHE STRING "Tensile fork to use" )
+  virtualenv_install("git+https://github.com/${tensile_fork}/Tensile.git@${MIOPEN_TENSILE_TAG}")
+  message (STATUS "using GIT Tensile fork=${tensile_fork} from branch=${MIOPEN_TENSILE_TAG}")
+endif()
+message(STATUS "Adding ${VIRTUALENV_HOME_DIR} to CMAKE_PREFIX_PATH")
+list(APPEND CMAKE_PREFIX_PATH ${VIRTUALENV_HOME_DIR})
+if (TENSILE_VERSION)
+  find_package(Tensile ${TENSILE_VERSION} EXACT REQUIRED HIP LLVM PATHS "${INSTALLED_TENSILE_PATH}")
+else()
+  find_package(Tensile 4.28.0 EXACT REQUIRED HIP LLVM PATHS "${INSTALLED_TENSILE_PATH}")
+endif()
 
 set_target_properties( TensileHost PROPERTIES POSITION_INDEPENDENT_CODE ON )
 
