diff --git a/VTK/Common/Math/vtkFFT.txx b/VTK/Common/Math/vtkFFT.txx
index 043812a2f32..8c8410aeee9 100644
--- a/VTK/Common/Math/vtkFFT.txx
+++ b/VTK/Common/Math/vtkFFT.txx
@@ -327,6 +327,6 @@ vtkSmartPointer<vtkFFT::vtkScalarNumberArray> vtkFFT::Csd(vtkScalarNumberArray*
   vtkSMPTools::Fill(averageRange.begin(), averageRange.end(), 0.0);
 
   auto resRange = vtk::DataArrayTupleRange(result);
-  using ConstTupleRef = decltype(resRange)::ConstTupleReferenceType;
+  using ConstTupleRef = typename decltype(resRange)::ConstTupleReferenceType;
   for (unsigned int i = 0; i < shape[0]; ++i)
   {
     auto begin = resRange.cbegin() + i * shape[1];