https://github.com/apache/incubator-mxnet/pull/17645
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 437d01668..1049a53b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -662,7 +662,7 @@ add_subdirectory("3rdparty/mshadow")
 
 set(MXNET_INSTALL_TARGETS mxnet)
 if(UNIX)
-  string(APPEND CMAKE_CUDA_FLAGS "${CUDA_ARCH_FLAGS_SPACES}")
+  string(APPEND CMAKE_CUDA_FLAGS " ${CUDA_ARCH_FLAGS_SPACES}")
   # Create dummy file since we want an empty shared library before linking
   set(DUMMY_SOURCE ${CMAKE_BINARY_DIR}/dummy.c)
   file(WRITE ${DUMMY_SOURCE} "")
