diff --git a/Makefile b/Makefile
index 4746cc434..335c79b64 100644
--- a/Makefile
+++ b/Makefile
@@ -518,7 +518,7 @@ ifeq ($(USE_CUDA), 1)
 	endif
 	# Make sure to add stubs as fallback in order to be able to build
 	# without full CUDA install (especially if run without nvidia-docker)
-	LDFLAGS += -L/usr/local/cuda/lib64/stubs
+	LDFLAGS += -L$(USE_CUDA_PATH)/lib64/stubs
 	ifeq ($(USE_NCCL), 1)
 		ifneq ($(USE_NCCL_PATH), NONE)
 			CFLAGS += -I$(USE_NCCL_PATH)/include
