diff --git a/cmake/FindHIP.cmake b/cmake/FindHIP.cmake
index f86edd2..24492ad 100644
--- a/cmake/FindHIP.cmake
+++ b/cmake/FindHIP.cmake
@@ -22,7 +22,7 @@ if(NOT DEFINED HIP_PATH)
       set(HIP_PATH $ENV{HIP_PATH} CACHE PATH "Path to where HIP has been installed")
   endif()
 endif()
-include(${HIP_PATH}/cmake/FindHIP.cmake)
+include(${HIP_PATH}/lib/cmake/hip/FindHIP.cmake)
 
 if(NOT HIP_INCLUDE_DIRS)
   list(APPEND HIP_INCLUDE_DIRS
