--- spack-src/install/configure	2024-08-23 15:53:49.149059054 +0200
+++ spack-src/install/configure.patched	2024-08-23 15:53:41.324161047 +0200
@@ -7756,7 +7756,7 @@
              if test $with_hdf5_libs -eq 1; then
                 hdf5_libs=$with_hdf5_libline
              else
-                hdf5_libs=`$with_hdf5_path/bin/h5pfc -show | awk -F'-L' '{$1=""; for (i=2; i<=NF;i++) $i="-L"$i; print $0}'`
+                hdf5_libs=`$with_hdf5_path/bin/h5pfc -show | awk -F'-L' '{$1=""; for (i=2; i<=NF;i++) $i="-L"$i; print $0}' | xargs`
              fi
          elif command -v h5pfc >/dev/null; then
              if test $with_hdf5_libs -eq 1; then
