--- spack-src/PHonon/Gamma/dgradcorr.f90.org	2021-01-27 16:41:04.692204340 +0900
+++ spack-src/PHonon/Gamma/dgradcorr.f90	2021-01-27 16:44:03.877839298 +0900
@@ -22,7 +22,7 @@
 
   REAL(DP), INTENT(IN) :: rho (dfft%nnr, nspin), grho (3, dfft%nnr, nspin), &
        g (3, dfft%ngm)
-  REAL(DP), INTENT(OUT):: drho (dfft%nnr,nspin),&
+  REAL(DP), INTENT(IN):: drho (dfft%nnr,nspin),&
        dvxc_rr(dfft%nnr, nspin, nspin), dvxc_sr (dfft%nnr, nspin, nspin), &
        dvxc_ss (dfft%nnr,nspin, nspin), dvxc_s (dfft%nnr, nspin, nspin)
   REAL(DP), INTENT(INOUT) ::  dvxc (dfft%nnr, nspin)
