diff -Naur config.c config.c
--- config.c	2015-07-15 14:31:07.000000000 -0500
+++ config.c	2015-07-17 10:33:34.966072446 -0500
@@ -1153,7 +1153,15 @@
 {
 	static const char *system_wide;
 	if (!system_wide)
-		system_wide = system_path(ETC_GITCONFIG);
+		/* system_wide = system_path(ETC_GITCONFIG); */
+		system_wide = ETC_GITCONFIG;
+		if (!is_absolute_path(system_wide)) {
+			/* interpret path relative to exec-dir */
+			const char *exec_path = git_exec_path();
+			system_wide = prefix_path(exec_path, strlen(exec_path),
+						system_wide);
+		}
+
 	return system_wide;
 }

diff -Naur attr.c attr.c
--- attr.c	2015-07-15 14:31:07.000000000 -0500
+++ attr.c	2015-07-17 10:33:34.966072446 -0500
@@ -479,7 +479,15 @@
 {
 	static const char *system_wide;
 	if (!system_wide)
-		system_wide = system_path(ETC_GITATTRIBUTES);
+		/* system_wide = system_path(ETC_GITATTRIBUTES); */
+		system_wide = ETC_GITATTRIBUTES;
+		if (!is_absolute_path(system_wide)) {
+			/* interpret path relative to exec-dir */
+			const char *exec_path = git_exec_path();
+			system_wide = prefix_path(exec_path, strlen(exec_path),
+						system_wide);
+		}
+
 	return system_wide;
 }
 
