--- mysql-server-mysql-8.0.29/sql-common/client.cc	2022-04-08 09:53:29.000000000 -0500
+++ spack-src/sql-common/client.cc	2022-09-17 17:57:48.656098217 -0500
@@ -118,6 +118,11 @@
 #include <mysql/client_plugin.h>
 #include <openssl/x509v3.h>
 
+/* SSL_SESSION_is_resumable is openssl 1.1.1+ */
+#if OPENSSL_VERSION_NUMBER < 0x10101000L
+#define SSL_SESSION_is_resumable(x) true
+#endif
+
 #include <new>
 
 #include "../libmysql/init_commands_array.h"
@@ -187,10 +192,6 @@
     {&key_memory_MYSQL_ssl_session_data, "MYSQL_SSL_session", 0, 0,
      "Saved SSL sessions"}};
 
-/* SSL_SESSION_is_resumable is openssl 1.1.1+ */
-#if OPENSSL_VERSION_NUMBER < 0x10101000L
-#define SSL_SESSION_is_resumable(x) true
-#endif
 
 void init_client_psi_keys(void) {
   const char *category = "client";
