diff --git a/tensorflow/tensorflow.bzl b/tensorflow/tensorflow.bzl
index 740f24ec4a..7b8300f678 100644
--- a/tensorflow/tensorflow.bzl
+++ b/tensorflow/tensorflow.bzl
@@ -1604,6 +1604,7 @@ def _py_wrap_cc_impl(ctx):
         outputs = outputs,
         mnemonic = "PythonSwig",
         progress_message = "SWIGing " + src.path,
+        use_default_shell_env = True,
     )
     return struct(files = depset(outputs))
 
