--- a/tensorflow/contrib/__init__.py	2020-05-05 18:08:09.361724827 -0500
+++ b/tensorflow/contrib/__init__.py	2020-05-05 18:08:46.345699058 -0500
@@ -20,7 +20,10 @@
 
 # Add projects here, they will show up under tf.contrib.
 from tensorflow.contrib import bayesflow
-from tensorflow.contrib import cloud
+try:
+    from tensorflow.contrib import cloud
+except ImportError:
+    pass
 from tensorflow.contrib import cluster_resolver
 from tensorflow.contrib import compiler
 from tensorflow.contrib import copy_graph
