--- a/tensorflow/contrib/__init__.py	2020-05-05 18:14:30.701463150 -0500
+++ b/tensorflow/contrib/__init__.py	2020-05-05 18:15:12.392435370 -0500
@@ -26,7 +26,10 @@
 from tensorflow.contrib import bayesflow
 from tensorflow.contrib import checkpoint
 if os.name != "nt":
-  from tensorflow.contrib import cloud
+    try:
+        from tensorflow.contrib import cloud
+    except ImportError:
+        pass
 from tensorflow.contrib import cluster_resolver
 from tensorflow.contrib import coder
 from tensorflow.contrib import compiler
