diff --git a/tensorflow/core/kernels/mlir_generated/build_defs.bzl b/tensorflow/core/kernels/mlir_generated/build_defs.bzl
index 1cc3d335577..327e6cda821 100644
--- a/tensorflow/core/kernels/mlir_generated/build_defs.bzl
+++ b/tensorflow/core/kernels/mlir_generated/build_defs.bzl
@@ -355,7 +355,7 @@ def _gen_kernel_library(
                 extra_args = extra_args,
                 host_triple = host_triple,
                 gpu_archs = gpu_archs,
-                jit = jit,
+                jit = True,
                 max_supported_rank = max_supported_rank,
                 mlir_op = "{op}_{name}_{platform}_{type}_{output_type}.mlir".format(
                     op = op,
@@ -366,7 +366,7 @@ def _gen_kernel_library(
                 ),
                 tile_size = typed_tile_size,
                 unroll_factors = typed_unroll_factors,
-                jit_i64_indexed_for_large_tensors = jit_i64_indexed_for_large_tensors,
+                jit_i64_indexed_for_large_tensors = False,
             )
 
             # We have to use a sh_test instead of build_test because it doesn't properly find the dependent targets.
