--- tensorflow-0.10.0/tensorflow/workspace.bzl       2016-09-08 23:49:36.000000000 +0200
+++ tensorflow-0.10.0/tensorflow/workspace.bzl       2017-01-31 09:11:07.391869277 +0100
@@ -198,7 +198,7 @@ def tf_workspace(path_prefix = "", tf_re

   native.new_http_archive(
     name = "zlib_archive",
-    url = "http://zlib.net/zlib-1.2.8.tar.gz",
-    sha256 = "36658cb768a54c1d4dec43c3116c27ed893e88b02ecfcb44f2166f9c0b7f2a0d",
+    url = "http://zlib.net/zlib-1.2.11.tar.gz",
+    sha256 = "c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1",
     build_file = path_prefix + "zlib.BUILD",
   )

