diff -ru a/tensorflow/core/kernels/mlir_generated/build_defs.bzl b/tensorflow/core/kernels/mlir_generated/build_defs.bzl
--- a/tensorflow/core/kernels/mlir_generated/build_defs.bzl	2021-01-21 00:25:54.000000000 +0000
+++ b/tensorflow/core/kernels/mlir_generated/build_defs.bzl	2021-10-27 22:39:26.656168382 +0000
@@ -61,6 +61,7 @@
                 "--input=%s" % ctx.file.mlir_op.path,
                 "--output=%s" % gpu_bin.path,
             ],
+            use_default_shell_env = True,
             mnemonic = "compile",
         )
         gpu_bins.append(gpu_bin)
diff -ru a/third_party/flatbuffers/build_defs.bzl b/third_party/flatbuffers/build_defs.bzl
--- a/third_party/flatbuffers/build_defs.bzl	2021-01-21 00:25:54.000000000 +0000
+++ b/third_party/flatbuffers/build_defs.bzl	2021-10-27 22:35:51.563534770 +0000
@@ -388,6 +388,7 @@
             ctx.attr.deps[0].files.to_list()[0].path,
             ctx.outputs.out.path,
         ),
+            use_default_shell_env = True,
     )
 
 _concat_flatbuffer_py_srcs = rule(
diff -ru a/third_party/nccl/build_defs.bzl.tpl b/third_party/nccl/build_defs.bzl.tpl
--- a/third_party/nccl/build_defs.bzl.tpl	2021-01-21 00:25:54.000000000 +0000
+++ b/third_party/nccl/build_defs.bzl.tpl	2021-10-27 22:42:23.841660961 +0000
@@ -97,6 +97,7 @@
                 "--output-file=%s" % cubin.path,
             ] + [file.path for file in inputs],
             mnemonic = "nvlink",
+	    use_default_shell_env = True,
         )
         cubins.append(cubin)
         images.append("--image=profile=%s,file=%s" % (arch, cubin.path))
@@ -122,6 +123,7 @@
         arguments = arguments_list + images,
         tools = [bin2c],
         mnemonic = "fatbinary",
+	use_default_shell_env = True,
     )
 
     # Generate the source file #including the headers generated above.
@@ -237,6 +239,7 @@
         inputs = ctx.files.srcs,  # + ctx.files._crosstool,
         outputs = [ctx.outputs.out],
         command = "echo -e \"%s\" | %s -M" % (mri_script, cc_toolchain.ar_executable),
+	use_default_shell_env = True,
     )
 
 _merge_archive = rule(
diff -ru a/third_party/systemlibs/grpc.bazel.generate_cc.bzl b/third_party/systemlibs/grpc.bazel.generate_cc.bzl
--- a/third_party/systemlibs/grpc.bazel.generate_cc.bzl	2021-01-21 00:25:54.000000000 +0000
+++ b/third_party/systemlibs/grpc.bazel.generate_cc.bzl	2021-10-27 22:43:22.662824642 +0000
@@ -140,6 +140,7 @@
         outputs = out_files,
         executable = ctx.executable._protoc,
         arguments = arguments,
+        use_default_shell_env = True,
     )
 
     return struct(files = depset(out_files))
