diff --git a/rvslib/CMakeLists.txt b/rvslib/CMakeLists.txt
index d3ea8af..55b26c3 100644
--- a/rvslib/CMakeLists.txt
+++ b/rvslib/CMakeLists.txt
@@ -114,9 +114,10 @@ else()
   set(ASAN_LIB_PATH "$ENV{LD_LIBRARY_PATH}")
 endif()
 
+find_path(DRM_INCLUDE_DIR libdrm/drm.h PATHS /usr/include /usr/local/include)
 ## define include directories
 include_directories(./ ../ ../rvs
-  ${ROCM_SMI_INC_DIR} ${ROCR_INC_DIR} ${ROCBLAS_INC_DIR} ${HIP_INC_DIR} ${HIPBLASLT_INC_DIR} ${YAML_CPP_INCLUDE_DIR} ${HIPRAND_INC_DIR} ${ROCRAND_INC_DIR} ${HIPBLAS-COMMON_INCLUDE_DIR})
+  ${ROCM_SMI_INC_DIR} ${ROCR_INC_DIR} ${ROCBLAS_INC_DIR} ${HIP_INC_DIR} ${HIPBLASLT_INC_DIR} ${YAML_CPP_INCLUDE_DIR} ${HIPRAND_INC_DIR} ${ROCRAND_INC_DIR} ${HIPBLAS-COMMON_INCLUDE_DIR} ${DRM_INCLUDE_DIR})
 
 link_directories(${ASAN_LIB_PATH} ${ROCM_SMI_LIB_DIR})
 
