--- include/Rivet/Tools/JetUtils.hh
+++ include/Rivet/Tools/JetUtils.hh
@@ -47,4 +47,5 @@
   struct BoolJetFunctor {
     virtual bool operator()(const Jet& p) const = 0;
+    virtual ~BoolJetFunctor() {}
   };

--- include/Rivet/Tools/ParticleBaseUtils.hh
+++ include/Rivet/Tools/ParticleBaseUtils.hh
@@ -23,4 +23,5 @@
   struct BoolParticleBaseFunctor {
     virtual bool operator()(const ParticleBase& p) const = 0;
+    virtual ~BoolParticleBaseFunctor() {}
   };
 
@@ -397,4 +398,5 @@
   struct DoubleParticleBaseFunctor {
     virtual double operator()(const ParticleBase& p) const = 0;
+    virtual ~DoubleParticleBaseFunctor() {}
   };

--- include/Rivet/Tools/ParticleUtils.hh
+++ include/Rivet/Tools/ParticleUtils.hh
@@ -486,4 +486,5 @@
   struct BoolParticleFunctor {
     virtual bool operator()(const Particle& p) const = 0;
+    virtual ~BoolParticleFunctor() {}
   };

--- analyses/Makefile.in.orig	2018-06-28 09:22:46.722022129 +0200
+++ analyses/Makefile.in	2018-06-28 10:11:59.772373529 +0200
@@ -523,21 +523,21 @@
 @ENABLE_ANALYSES_TRUE@%.so:
 @ENABLE_ANALYSES_TRUE@	@+echo && RIVET_BUILDPLUGIN_BEFORE_INSTALL=1 bash $(top_builddir)/bin/rivet-buildplugin -j2 $@ $^ -I$(top_builddir)/include
 
-@ENABLE_ANALYSES_TRUE@RivetALICEAnalyses.so: $(shell ls $(srcdir)/pluginALICE/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetATLASAnalyses.so: $(shell ls $(srcdir)/pluginATLAS/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetCDFAnalyses.so  : $(shell ls $(srcdir)/pluginCDF/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetCMSAnalyses.so  : $(shell ls $(srcdir)/pluginCMS/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetD0Analyses.so   : $(shell ls $(srcdir)/pluginD0/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetHERAAnalyses.so : $(shell ls $(srcdir)/pluginHERA/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetLEPAnalyses.so  : $(shell ls $(srcdir)/pluginLEP/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetLHCbAnalyses.so : $(shell ls $(srcdir)/pluginLHCb/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetLHCfAnalyses.so : $(shell ls $(srcdir)/pluginLHCf/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetMCAnalyses.so   : $(shell ls $(srcdir)/pluginMC/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetMiscAnalyses.so : $(shell ls $(srcdir)/pluginMisc/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetPetraAnalyses.so: $(shell ls $(srcdir)/pluginPetra/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetRHICAnalyses.so : $(shell ls $(srcdir)/pluginRHIC/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetSPSAnalyses.so  : $(shell ls $(srcdir)/pluginSPS/*.cc)
-@ENABLE_ANALYSES_TRUE@RivetTOTEMAnalyses.so: $(shell ls $(srcdir)/pluginTOTEM/*.cc)
+@ENABLE_ANALYSES_TRUE@RivetALICEAnalyses.so: $(shell find $(srcdir)/pluginALICE/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetATLASAnalyses.so: $(shell find $(srcdir)/pluginATLAS/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetCDFAnalyses.so  : $(shell find $(srcdir)/pluginCDF/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetCMSAnalyses.so  : $(shell find $(srcdir)/pluginCMS/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetD0Analyses.so   : $(shell find $(srcdir)/pluginD0/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetHERAAnalyses.so : $(shell find $(srcdir)/pluginHERA/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetLEPAnalyses.so  : $(shell find $(srcdir)/pluginLEP/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetLHCbAnalyses.so : $(shell find $(srcdir)/pluginLHCb/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetLHCfAnalyses.so : $(shell find $(srcdir)/pluginLHCf/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetMCAnalyses.so   : $(shell find $(srcdir)/pluginMC/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetMiscAnalyses.so : $(shell find $(srcdir)/pluginMisc/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetPetraAnalyses.so: $(shell find $(srcdir)/pluginPetra/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetRHICAnalyses.so : $(shell find $(srcdir)/pluginRHIC/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetSPSAnalyses.so  : $(shell find $(srcdir)/pluginSPS/ -name '*.cc' -not -name 'tmp*')
+@ENABLE_ANALYSES_TRUE@RivetTOTEMAnalyses.so: $(shell find $(srcdir)/pluginTOTEM/ -name '*.cc' -not -name 'tmp*')
 
 @ENABLE_ANALYSES_TRUE@all-local: $(PLUGIN_LIBS) $(PLUGIN_DATAFILES)
 @ENABLE_ANALYSES_TRUE@	mkdir -p $(builddir)/data
--- analyses/pluginATLAS/ATLAS_2016_CONF_2016_037.cc.orig	2018-07-02 10:43:08.656094368 +0200
+++ analyses/pluginATLAS/ATLAS_2016_CONF_2016_037.cc	2018-07-02 10:44:04.920361160 +0200
@@ -2,6 +2,7 @@
 #include "Rivet/Analysis.hh"
 #include "Rivet/Projections/FinalState.hh"
 #include "Rivet/Projections/PromptFinalState.hh"
+#include "Rivet/Projections/ChargedFinalState.hh"
 #include "Rivet/Projections/FastJets.hh"
 #include "Rivet/Projections/Sphericity.hh"
 #include "Rivet/Projections/SmearedParticles.hh"
--- include/Rivet/AnalysisHandler.hh.orig	2018-07-10 10:39:21.719532209 +0200
+++ include/Rivet/AnalysisHandler.hh	2018-07-10 10:39:29.225608530 +0200
@@ -17,7 +17,7 @@
 
   // Needed to make smart pointers compare equivalent in the STL set
   struct CmpAnaHandle {
-    bool operator() (const AnaHandle& a, const AnaHandle& b) {
+    bool operator() (const AnaHandle& a, const AnaHandle& b) const {
       return a.get() < b.get();
     }
   };
