--- src/Tools/Makefile.in.orig	2013-05-15 14:02:25.000000000 +0200
+++ src/Tools/Makefile.in	2013-05-15 14:03:14.000000000 +0200
@@ -79,7 +79,7 @@
 	libRivetTools_la-ParticleIdUtils.lo \
 	libRivetTools_la-tinyxml.lo libRivetTools_la-tinyxmlerror.lo \
 	libRivetTools_la-tinyxmlparser.lo \
-	libRivetTools_la-BinnedHistogram.lo \
+#	libRivetTools_la-BinnedHistogram.lo \
 	libRivetTools_la-mt2_bisect.lo libRivetTools_la-RivetMT2.lo \
 	libRivetTools_la-BinnedHistogram.lo
 libRivetTools_la_OBJECTS = $(am_libRivetTools_la_OBJECTS)
@@ -371,7 +371,7 @@
   TinyXML/tinyxml.cpp \
   TinyXML/tinyxmlerror.cpp \
   TinyXML/tinyxmlparser.cpp \
-  BinnedHistogram.cc \
+#  BinnedHistogram.cc \
   mt2_bisect.cc \
   RivetMT2.cc \
   BinnedHistogram.cc
--- include/LWH/Axis.h.orig	2013-05-14 18:30:02.000000000 +0200
+++ include/LWH/Axis.h	2013-05-14 18:31:35.000000000 +0200

@@ -115,7 +115,7 @@
    *
    */
   int coordToIndex(double coord) const {
-    assert( ! isnan(coord) );
+    assert( ! std::isnan(coord) );
     if ( coord >= upper ) return OVERFLOW_BIN;
     else if ( coord < lower ) return UNDERFLOW_BIN;
     else return int((coord - lower)/binWidth(0));

--- src/Tools/Makefile.am.orig  2013-05-15 13:14:11.000000000 +0200
+++ src/Tools/Makefile.am       2013-05-15 13:16:53.000000000 +0200
@@ -14,7 +14,7 @@
   TinyXML/tinyxml.cpp \
   TinyXML/tinyxmlerror.cpp \
   TinyXML/tinyxmlparser.cpp \
-  BinnedHistogram.cc \
+#  BinnedHistogram.cc \
   mt2_bisect.cc \
   RivetMT2.cc \
   BinnedHistogram.cc
