--- configure.orig      2016-09-30 14:12:42.317369715 +0200
+++ configure   2016-09-30 14:17:21.690229397 +0200
@@ -19644,11 +19644,15 @@
         cat >conftest.py <<_ACEOF

 import sys, string
+# Python 2 and 3 compatible
+from future.builtins import map
+# Python 2 and 3: forward-compatible
+from future.builtins import range
 # split strings by '.' and convert to numeric.  Append some zeros
 # because we need at least 4 digits for the hex conversion.
-minver = map(int, string.split('2.5', '.')) + [0, 0, 0]
+minver = list(map(int, '2.5'.split('.'))) + [0, 0, 0]
 minverhex = 0
-for i in xrange(0, 4): minverhex = (minverhex << 8) + minver[i]
+for i in range(0, 4): minverhex = (minverhex << 8) + minver[i]
 if sys.hexversion >= minverhex:
     sys.exit( 0 )
 else:
@@ -19676,15 +19680,15 @@
     fi


-  PYTHON_VERSION=`$PYTHON -c "import sys; print '.'.join(map(str, sys.version_info[:2]));"`
+  PYTHON_VERSION=`$PYTHON -c "from __future__ import print_function; import sys; print('.'.join(map(str, sys.version_info[:2])));"`

-  RIVET_PYTHONPATH=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_python_lib(prefix='$prefix', plat_specific=True);"`
+  RIVET_PYTHONPATH=`$PYTHON -c "from __future__ import print_function; import distutils.sysconfig; print(distutils.sysconfig.get_python_lib(prefix='$prefix', plat_specific=True));"`

   ## Test for Python header
   if test -x "$PYTHON"; then
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Python include path" >&5
 $as_echo_n "checking for Python include path... " >&6; }
-    python_incpath=`$PYTHON -c "import distutils.sysconfig; print distutils.sysconfig.get_python_inc();"`
+    python_incpath=`$PYTHON -c "from __future__ import print_function; import distutils.sysconfig; print(distutils.sysconfig.get_python_inc());"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $python_incpath" >&5
 $as_echo "$python_incpath" >&6; }
     python_header="$python_incpath/Python.h"
