--- configure.ac.orig	2019-01-23 11:02:06.206570992 +0100
+++ configure.ac	2019-01-23 11:02:26.704761899 +0100
@@ -151,12 +151,12 @@
 
 
 # ## GNU Scientific Library
-# AC_SEARCH_GSL
-# AC_CEDAR_HEADERS([gsl], , , [AC_MSG_ERROR([GSL (GNU Scientific Library) is required])])
-# oldCPPFLAGS=$CPPFLAGS
-# CPPFLAGS="$CPPFLAGS -I$GSLINCPATH"
-# AC_CHECK_HEADER([gsl/gsl_vector.h], [], [AC_MSG_ERROR([GSL vectors not found.])])
-# CPPFLAGS=$oldCPPFLAGS
+AC_SEARCH_GSL
+AC_CEDAR_HEADERS([gsl], , , [AC_MSG_ERROR([GSL (GNU Scientific Library) is required])])
+oldCPPFLAGS=$CPPFLAGS
+CPPFLAGS="$CPPFLAGS -I$GSLINCPATH"
+AC_CHECK_HEADER([gsl/gsl_vector.h], [], [AC_MSG_ERROR([GSL vectors not found.])])
+CPPFLAGS=$oldCPPFLAGS
 
 
 ## Disable build/install of standard analyses
@@ -256,7 +256,7 @@
 ## Set default build flags
 AM_CPPFLAGS="-I\$(top_srcdir)/include -I\$(top_builddir)/include"
 #AM_CPPFLAGS="$AM_CPPFLAGS -I\$(top_srcdir)/include/eigen3"
-#AM_CPPFLAGS="$AM_CPPFLAGS \$(GSL_CPPFLAGS)"
+AM_CPPFLAGS="$AM_CPPFLAGS \$(GSL_CPPFLAGS)"
 dnl AM_CPPFLAGS="$AM_CPPFLAGS \$(BOOST_CPPFLAGS)"
 AM_CPPFLAGS="$AM_CPPFLAGS -I\$(YODAINCPATH)"
 AM_CPPFLAGS="$AM_CPPFLAGS -I\$(HEPMCINCPATH)"
--- bin/rivet-config.in.orig	2019-01-23 11:08:04.608907832 +0100
+++ bin/rivet-config.in	2019-01-23 11:08:25.058098155 +0100
@@ -82,8 +82,8 @@
     test -n "$iyoda" && OUT="$OUT -I${iyoda}"
     ifastjet="@FASTJETINCPATH@"
     test -n "$ifastjet" && OUT="$OUT -I${ifastjet}"
-    # igsl="@GSLINCPATH@"
-    # test -n "$igsl" && OUT="$OUT -I${igsl}"
+    igsl="@GSLINCPATH@"
+    test -n "$igsl" && OUT="$OUT -I${igsl}"
     # iboost="@BOOST_CPPFLAGS@"
     # test -n "$iboost" && OUT="$OUT ${iboost}"
 fi
@@ -98,8 +98,8 @@
     test -n "$lyoda" && OUT="$OUT -L${lyoda} -lYODA"
     lfastjet="@FASTJETCONFIGLIBADD@"
     test -n "$lfastjet" && OUT="$OUT ${lfastjet}"
-    # lgsl="@GSLLIBPATH@"
-    # test -n "$lgsl" && OUT="$OUT -L${lgsl}"
+    lgsl="@GSLLIBPATH@"
+    test -n "$lgsl" && OUT="$OUT -L${lgsl}"
 fi
 
 tmp=$( echo "$*" | egrep -- '--\<libs\>|--\<ldadd\>')
--- bin/rivet-buildplugin.in.orig	2019-01-23 11:10:07.804054317 +0100
+++ bin/rivet-buildplugin.in	2019-01-23 11:10:34.370301517 +0100
@@ -169,8 +169,8 @@
 test -n "$iyoda" && mycppflags="$mycppflags -I${iyoda}"
 ifastjet="@FASTJETINCPATH@"
 test -n "$ifastjet" && mycppflags="$mycppflags -I${ifastjet}"
-# igsl="@GSLINCPATH@"
-# test -n "$igsl" && mycppflags="$mycppflags -I${igsl}"
+igsl="@GSLINCPATH@"
+test -n "$igsl" && mycppflags="$mycppflags -I${igsl}"
 # iboost="@BOOST_CPPFLAGS@"
 # test -n "$iboost" && mycppflags="$mycppflags ${iboost}"
 
--- pyext/setup.py.in.orig	2019-01-23 11:12:04.694141901 +0100
+++ pyext/setup.py.in	2019-01-23 11:12:44.531512512 +0100
@@ -29,11 +29,11 @@
 
 
 ## Be careful with extracting the GSL path from the flags string
-# import re
-# re_libdirflag = re.compile(r".*-L\s*(\S+).*")
-# re_match = re_libdirflag.search("@GSL_LDFLAGS@")
-# if re_match:
-    # lookupdirs.append( re_match.group(1) )
+import re
+re_libdirflag = re.compile(r".*-L\s*(\S+).*")
+re_match = re_libdirflag.search("@GSL_LDFLAGS@")
+if re_match:
+    lookupdirs.append( re_match.group(1) )
 
 ## A helper function
 def ext(name, depends=[], statics=[]):
@@ -47,8 +47,8 @@
         language="c++",
 #        depends=depends,
         include_dirs=[incdir1, incdir2, incdir3, incdir4],
-        # extra_compile_args="-I@prefix@/include @PYEXT_CXXFLAGS@ @HEPMCCPPFLAGS@ @FASTJETCPPFLAGS@ @YODACPPFLAGS@ @GSLCPPFLAGS@".split(),
-        extra_compile_args="-I@prefix@/include @PYEXT_CXXFLAGS@ @HEPMCCPPFLAGS@ @FASTJETCPPFLAGS@ @YODACPPFLAGS@".split(),
+        extra_compile_args="-I@prefix@/include @PYEXT_CXXFLAGS@ @HEPMCCPPFLAGS@ @FASTJETCPPFLAGS@ @YODACPPFLAGS@ @GSLCPPFLAGS@".split(),
+        # extra_compile_args="-I@prefix@/include @PYEXT_CXXFLAGS@ @HEPMCCPPFLAGS@ @FASTJETCPPFLAGS@ @YODACPPFLAGS@".split(),
         extra_link_args=BASE_LINK_ARGS,
         library_dirs=lookupdirs,
         runtime_library_dirs=lookupdirs[1:],
