--- bin/rivet-findid	2015-06-23 09:40:38.000000001 +0200
+++ bin/rivet-findid	2015-06-23 09:40:45.000000001 +0200
@@ -160,7 +160,7 @@
                     if k=='code':
                         entries[v] = c.text
             if entries.get('9') == 'SPIRESTeX':
-                result['bibtex'] = entries['z']
+                result['bibtex'] = entries['a']
 
         if i.get('tag') == '037':
             entries = {}
