--- doc/Makefile.in.orig	2018-10-17 18:29:15.568630034 +0200
+++ doc/Makefile.in	2018-10-17 18:31:45.829375147 +0200
@@ -594,15 +594,15 @@
 .PRECIOUS: Makefile
 
 @ENABLE_PYEXT_TRUE@analyses.html: $(top_srcdir)/analyses $(srcdir)/mk-analysis-html
-@ENABLE_PYEXT_TRUE@	LD_LIBRARY_PATH=$(top_builddir)/src/.libs:$(LD_LIBRARY_PATH) \
-@ENABLE_PYEXT_TRUE@    DYLD_LIBRARY_PATH=$(top_builddir)/src/.libs:$(DYLD_LIBRARY_PATH) \
+@ENABLE_PYEXT_TRUE@	LD_LIBRARY_PATH=$(top_builddir)/src/.libs:$(HEPMCLIBPATH):$(FASTJETLIBPATH):$(YODALIBPATH):$(RIVETLIBPATH):$(LD_LIBRARY_PATH) \
+@ENABLE_PYEXT_TRUE@    DYLD_LIBRARY_PATH=$(top_builddir)/src/.libs:$(HEPMCLIBPATH):$(FASTJETLIBPATH):$(YODALIBPATH):$(RIVETLIBPATH):$(DYLD_LIBRARY_PATH) \
 @ENABLE_PYEXT_TRUE@    PYTHONPATH=$(YODA_PYTHONPATH):$(RIVET_PYTHONPATH):$(PYTHONPATH) \
-@ENABLE_PYEXT_TRUE@    $(srcdir)/mk-analysis-html
+@ENABLE_PYEXT_TRUE@    $(PYTHON) $(srcdir)/mk-analysis-html
 @ENABLE_PYEXT_TRUE@analyses.json: $(top_srcdir)/analyses $(srcdir)/mk-analysis-json
-@ENABLE_PYEXT_TRUE@	LD_LIBRARY_PATH=$(top_builddir)/src/.libs:$(LD_LIBRARY_PATH) \
-@ENABLE_PYEXT_TRUE@    DYLD_LIBRARY_PATH=$(top_builddir)/src/.libs:$(DYLD_LIBRARY_PATH) \
+@ENABLE_PYEXT_TRUE@	LD_LIBRARY_PATH=$(top_builddir)/src/.libs:$(HEPMCLIBPATH):$(FASTJETLIBPATH):$(YODALIBPATH):$(RIVETLIBPATH):$(LD_LIBRARY_PATH) \
+@ENABLE_PYEXT_TRUE@    DYLD_LIBRARY_PATH=$(top_builddir)/src/.libs:$(HEPMCLIBPATH):$(FASTJETLIBPATH):$(YODALIBPATH):$(RIVETLIBPATH):$(DYLD_LIBRARY_PATH) \
 @ENABLE_PYEXT_TRUE@    PYTHONPATH=$(YODA_PYTHONPATH):$(RIVET_PYTHONPATH):$(PYTHONPATH) \
-@ENABLE_PYEXT_TRUE@    $(srcdir)/mk-analysis-json
+@ENABLE_PYEXT_TRUE@    $(PYTHON) $(srcdir)/mk-analysis-json
 
 @WITH_ASCIIDOC_TRUE@compare-histos.html: compare-histos.txt
 @WITH_ASCIIDOC_TRUE@	asciidoc -a toc compare-histos.txt
