--- include/Rivet/Config/RivetCommon.hh.orig	2017-05-31 12:32:04.547062840 +0200
+++ include/Rivet/Config/RivetCommon.hh	2017-05-31 12:32:12.379054886 +0200
@@ -2,6 +2,7 @@
 #define RIVET_RivetCommon_HH
 
 // Convenience build-setup header for Rivet internal use
+#include <functional>
 
 // Automatic build info from autoconf
 #include "Rivet/Config/RivetConfig.hh"
--- include/Rivet/Run.hh.orig	2017-05-31 12:36:00.225823486 +0200
+++ include/Rivet/Run.hh	2017-05-31 12:36:18.653804770 +0200
@@ -4,6 +4,7 @@
 
 #include "Rivet/Tools/RivetSTL.hh"
 #include "Rivet/Tools/RivetHepMC.hh"
+#include <functional>
 
 namespace Rivet {
 
--- include/Rivet/Particle.hh.orig	2017-05-31 12:39:14.273626411 +0200
+++ include/Rivet/Particle.hh	2017-05-31 14:00:02.387272379 +0200
@@ -11,6 +11,8 @@
 // NOTE: Rivet/Tools/ParticleUtils.hh included at the end
 #include "fastjet/PseudoJet.hh"
 
+#include <functional>
+
 namespace Rivet {
 
 
--- include/Rivet/Tools/RivetSTL.hh.orig	2017-05-31 14:05:28.368975178 +0200
+++ include/Rivet/Tools/RivetSTL.hh	2017-05-31 14:05:44.432960512 +0200
@@ -1,6 +1,8 @@
 #ifndef RIVET_RivetSTL_HH
 #define RIVET_RivetSTL_HH
 
+#include <functional>
+
 #include <string>
 #include <vector>
 #include <set>

