--- bin/rivet-buildplugin.in.orig	2014-09-02 11:22:32.000000000 +0200
+++ bin/rivet-buildplugin.in	2014-09-02 11:26:31.000000000 +0200
@@ -62,6 +62,7 @@
 mycppflags=""
 prefix="@prefix@"
 irivet="@includedir@"
+exec_prefix=@exec_prefix@
 test -n "$irivet" && mycppflags="$mycppflags -I${irivet}"
 ihepmc="@HEPMCINCPATH@"
 test -n "$ihepmc" && mycppflags="$mycppflags -I${ihepmc}"
@@ -74,7 +75,7 @@
 
 ## Get Rivet system linker flags (duplicating that in rivet-config.in)
 myldflags=""
-lrivet="@libdir@"
+lrivet="${exec_prefix}/lib"
 test -n "$lrivet" && myldflags="$myldflags -L${lrivet}"
 lhepmc="@HEPMCLIBPATH@"
 test -n "$lhepmc" && myldflags="$myldflags -L${lhepmc}"
--- bin/rivet-findid	2015-06-23 09:15:28.000000001 +0200
+++ bin/rivet-findid	2015-06-23 09:15:38.000000001 +0200
@@ -170,7 +170,7 @@
                     if k=='code':
                         entries[v] = c.text
             if entries.get('9') == 'SPIRESTeX':
-                result['bibtex'] = entries['z']
+                result['bibtex'] = entries['a']
 
         if i.get('tag') == '037':
             entries = {}
