diff --git a/arch/configure_new.defaults b/arch/configure_new.defaults
index bc311eae..e054b49b 100644
--- a/arch/configure_new.defaults
+++ b/arch/configure_new.defaults
@@ -29,7 +29,7 @@ FCDEBUG         =       # -g $(FCNOOPT)
 FORMAT_FIXED    =       -w -f3
 FORMAT_FREE     =       -w -f4
 FCSUFFIX        =       
-BYTESWAPIO      =       #-FIX_BYTE_SWAP_IF_NECESSARY_FOR_BIG_ENDIAN
+BYTESWAPIO      =       #-FIX_BYTE_SWAP_IF_NECESSARY_FOR_BIG_ENDIAN -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w -Wf'-M noflunf -M nozdiv' $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =      -I/SX/usr/include/module/dwdadW64/
@@ -73,7 +73,7 @@ FCDEBUG         =       # -g $(FCNOOPT)  # -fbacktrace -ggdb -fcheck=bounds,do,m
 FORMAT_FIXED    =       -ffixed-form
 FORMAT_FREE     =       -ffree-form -ffree-line-length-none
 FCSUFFIX        =       
-BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
+BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4 -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =
@@ -116,7 +116,7 @@ FCDEBUG         =       # -g $(FCNOOPT)
 FORMAT_FIXED    =       -ffixed-form
 FORMAT_FREE     =       -ffree-form -ffree-line-length-huge
 FCSUFFIX        =
-BYTESWAPIO      =       -fendian=big
+BYTESWAPIO      =       -fendian=big -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -Wno=101,139,155,158 $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -fmod=$(WRF_SRC_ROOT_DIR)/main
@@ -159,7 +159,7 @@ FCDEBUG         =       # -g $(FCNOOPT)  # -C -Ktrap=fp -traceback
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
@@ -202,7 +202,7 @@ FCDEBUG         =       # -g $(FCNOOPT) # -C -Ktrap=fp -traceback
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
@@ -245,7 +245,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -Kieee -acc -ta=nvidia,cuda5.0,cc35 -Mcu
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS      =       -w $(FCDEBUG) $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
 TRADFLAG        =      CONFIGURE_TRADFLAG
@@ -320,7 +320,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
@@ -367,7 +367,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -w -openmp -auto -ftz -fno-alias -fp-model fast=1 -no-prec-div -no-prec-sqrt $(FORMAT_FREE) $(BYTESWAPIO) -auto -align array64byte #-vec-report6
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
@@ -414,7 +414,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -w $(OMP) -auto -ftz -fno-alias -fp-model fast=1 -no-prec-div -no-prec-sqrt $(FORMAT_FREE) $(BYTESWAPIO) -auto -align array64byte #-vec-report6
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
@@ -485,7 +485,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
@@ -534,7 +534,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
@@ -613,7 +613,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common
 #FCBASEOPTS_NO_G =       -w -ftz -align all -fno-alias -IPF-fp-relaxed $(FORMAT_FREE) $(BYTESWAPIO)
@@ -696,7 +696,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common
 #FCBASEOPTS_NO_G =       -w -ftz -align all -fno-alias -IPF-fp-relaxed $(FORMAT_FREE) $(BYTESWAPIO)
@@ -742,7 +742,7 @@ FCDEBUG         =       # -g $(FCNOOPT)  # -C -trapuv
 FORMAT_FIXED    =       -fixedform
 FORMAT_FREE     =       -freeform
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w -fno-second-underscore $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
@@ -785,7 +785,7 @@ FCDEBUG         =       # -g $(FCNOOPT) # -ggdb -fbacktrace -fcheck=bounds,do,me
 FORMAT_FIXED    =       -ffixed-form
 FORMAT_FREE     =       -ffree-form -ffree-line-length-none
 FCSUFFIX        =       
-BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
+BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4 -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     
@@ -828,7 +828,7 @@ FCDEBUG         =       # -g $(FCNOOPT) # -C -Ktrap=fp -traceback
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =       
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG) -Mnomod
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
@@ -872,7 +872,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 # added -fno-common at suggestion of R. Dubtsov as workaround for failing to link program_name
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common
@@ -918,7 +918,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -fp-model precise -w -ftz -align all -fno-alias -fno-common $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
@@ -962,7 +962,7 @@ FCDEBUG         =       # -g $(FCNOOPT)
 FORMAT_FIXED    =       -ffixed-form
 FORMAT_FREE     =       -ffree-form -ffree-line-length-huge
 FCSUFFIX        =
-BYTESWAPIO      =       -fendian=big
+BYTESWAPIO      =       -fendian=big -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -Wno=101,139,155,158 $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 LIB_LOCAL       =    -L/usr/lib -lSystemStubs
@@ -1006,7 +1006,7 @@ FCDEBUG         =       # -g $(FCNOOPT) # -fbacktrace -ggdb -fcheck=bounds,do,me
 FORMAT_FIXED    =       -ffixed-form
 FORMAT_FREE     =       -ffree-form -ffree-line-length-none
 FCSUFFIX        =       
-BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
+BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4 -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =
@@ -1049,7 +1049,7 @@ FCDEBUG         =       # -g $(FCNOOPT) # -fbacktrace -ggdb -fcheck=bounds,do,me
 FORMAT_FIXED    =       -ffixed-form
 FORMAT_FREE     =       -ffree-form -ffree-line-length-none
 FCSUFFIX        =       
-BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
+BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4 -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =
@@ -1094,7 +1094,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -qfullpath
 FORMAT_FIXED    =       -qfixed
 FORMAT_FREE     =       -qfree=f90
 FCSUFFIX        =       -qsuffix=f=f90
-BYTESWAPIO      =       
+BYTESWAPIO      =        -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -qsave -qmaxmem=32767 -qspillsize=32767 -w
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -fmod=$(WRF_SRC_ROOT_DIR)/main
@@ -1142,7 +1142,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -qfullpath
 FORMAT_FIXED    =       -qfixed
 FORMAT_FREE     =       -qfree=f90
 FCSUFFIX        =       -qsuffix=f=f90
-BYTESWAPIO      =       
+BYTESWAPIO      =        -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w -qspill=81920 -qmaxmem=-1 $(FORMAT_FREE) $(BYTESWAPIO)  #-qflttrap=zerodivide:invalid:enable -qsigtrap -C # -qinitauto=7FF7FFFF
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     
@@ -1191,7 +1191,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -qfullpath
 FORMAT_FIXED    =       -qfixed
 FORMAT_FREE     =       -qfree=f90
 FCSUFFIX        =       -qsuffix=f=f90
-BYTESWAPIO      =       
+BYTESWAPIO      =        -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w -qspill=81920 -qmaxmem=-1 $(FORMAT_FREE) $(BYTESWAPIO)  #-qflttrap=zerodivide:invalid:enable -qsigtrap -C # -qinitauto=7FF7FFFF
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     
@@ -1255,7 +1255,7 @@ FORMAT_FREE     =       -Mfree
 #FORMAT_FIXED    =       -fixedform
 #FORMAT_FREE     =       -freeform
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OPTERON_TYPE) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =    
@@ -1305,7 +1305,7 @@ FCDEBUG         =       # -g -O0 # -K trap=fp -R bc
 FORMAT_FIXED    =       -f fixed
 FORMAT_FREE     =       -f free
 FCSUFFIX        =
-BYTESWAPIO      =       -h byteswapio
+BYTESWAPIO      =       -h byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -N1023 $(FORMAT_FREE) $(BYTESWAPIO) #-ra
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     
@@ -1352,7 +1352,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 #add -fp-model precise in FCBASEOPTS_NO_G to improve the accuracy of WRFPLUS check_AD test, suggested by Thomas Schwitalla.
 FCBASEOPTS_NO_G =       -w -ftz -fno-alias -align all $(FORMAT_FREE) $(BYTESWAPIO) #-fp-model precise #-vec-report6
@@ -1402,7 +1402,7 @@ FCDEBUG         =       # $(FCNOOPT) -qfullpath
 FORMAT_FIXED    =       -qfixed
 FORMAT_FREE     =       -qfree=f90
 FCSUFFIX        =       -qsuffix=f=f90
-BYTESWAPIO      =       
+BYTESWAPIO      =        -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w -qspill=20000 -qmaxmem=64000 $(FORMAT_FREE) $(BYTESWAPIO) $(MPI_INC) #-qflttrap=zerodivide:invalid:enable -qsigtrap
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     
@@ -1451,7 +1451,7 @@ FCDEBUG         =       # $(FCNOOPT) -g -qfullpath
 FORMAT_FIXED    =       -qfixed
 FORMAT_FREE     =       -qfree=f90
 FCSUFFIX        =       -qsuffix=f=f90
-BYTESWAPIO      =       
+BYTESWAPIO      =        -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w -qspill=20000 -qmaxmem=64000 $(FORMAT_FREE) $(BYTESWAPIO) #-qflttrap=zerodivide:invalid:enable -qsigtrap
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     
@@ -1496,7 +1496,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -qfullpath
 FORMAT_FIXED    =       -qfixed
 FORMAT_FREE     =       -qfree=f90
 FCSUFFIX        =       -qsuffix=f=f90
-BYTESWAPIO      =       
+BYTESWAPIO      =        -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w -qspill=20000 -qmaxmem=32767 $(FORMAT_FREE) $(BYTESWAPIO) #-qflttrap=zerodivide:invalid:enable -qsigtrap
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     
@@ -1540,7 +1540,7 @@ FCDEBUG         =       # -g $(FCNOOPT)  # -C -Ktrap=fp -traceback
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
@@ -1583,7 +1583,7 @@ FCDEBUG         =       # -g $(FCNOOPT) # -C -Ktrap=fp
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio 
+BYTESWAPIO      =       -byteswapio  -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG=       -module $(WRF_SRC_ROOT_DIR)/main
@@ -1640,7 +1640,7 @@ FCDEBUG         =       # -g $(FCNOOPT)  # -C -Ktrap=fp -traceback
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
@@ -1683,7 +1683,7 @@ FCDEBUG         =       # -g $(FCNOOPT) # -C -Ktrap=fp -traceback
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =       
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG) -Mnomod
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
@@ -1727,7 +1727,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 # added -fno-common at suggestion of R. Dubtsov as workaround for failing to link program_name
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) #-xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common
@@ -1772,7 +1772,7 @@ FCDEBUG         =       # -g $(FCNOOPT) # -fbacktrace -ggdb -fcheck=bounds,do,me
 FORMAT_FIXED    =       -ffixed-form
 FORMAT_FREE     =       -ffree-form -ffree-line-length-none
 FCSUFFIX        =       
-BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4
+BYTESWAPIO      =       -fconvert=big-endian -frecord-marker=4 -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =
@@ -1815,7 +1815,7 @@ FCDEBUG         =       # -g $(FCNOOPT)  # -C -Ktrap=fp -traceback
 FORMAT_FIXED    =       -Mfixed
 FORMAT_FREE     =       -Mfree
 FCSUFFIX        =
-BYTESWAPIO      =       -byteswapio
+BYTESWAPIO      =       -byteswapio -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -w $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =     -module $(WRF_SRC_ROOT_DIR)/main
@@ -1858,7 +1858,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) -xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common -xCORE-AVX2
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
@@ -1902,7 +1902,7 @@ FCDEBUG         =       # -g $(FCNOOPT) -traceback # -fpe0 -check noarg_temp_cre
 FORMAT_FIXED    =       -FI
 FORMAT_FREE     =       -FR
 FCSUFFIX        =
-BYTESWAPIO      =       -convert big_endian
+BYTESWAPIO      =       -convert big_endian -fallow-argument-mismatch -fallow-invalid-boz
 RECORDLENGTH    =       -assume byterecl
 FCBASEOPTS_NO_G =       -ip -fp-model precise -w -ftz -align all -fno-alias $(FORMAT_FREE) $(BYTESWAPIO) -xHost -fp-model fast=2 -no-heap-arrays -no-prec-div -no-prec-sqrt -fno-common -xMIC-AVX512
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
@@ -1948,7 +1948,7 @@ FCDEBUG         =       # -g $(FCNOOPT)
 FORMAT_FIXED    =       -Fixed
 FORMAT_FREE     =       -Free
 FCSUFFIX        =
-BYTESWAPIO      =
+BYTESWAPIO      = -fallow-argument-mismatch -fallow-invalid-boz
 FCBASEOPTS_NO_G =       -Kautoobjstack,ocl -fw $(FORMAT_FREE) $(BYTESWAPIO) $(OMP)
 FCBASEOPTS      =       $(FCBASEOPTS_NO_G) $(FCDEBUG)
 MODULE_SRCH_FLAG =
