From bda3adb5351757a0bbf2ba8aea646b6dbe42a362 Mon Sep 17 00:00:00 2001
From: Renjith Ravindran <Renjith.RavindranKannath@amd.com>
Date: Tue, 4 Oct 2022 16:15:51 -0700
Subject: [PATCH] Tensile use openmp from rocm-openmp-extras

---
 Tensile/Source/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Tensile/Source/CMakeLists.txt b/Tensile/Source/CMakeLists.txt
index 7d7e6c4..bc8e8ec 100644
--- a/Tensile/Source/CMakeLists.txt
+++ b/Tensile/Source/CMakeLists.txt
@@ -67,7 +67,7 @@ if(TENSILE_USE_OPENMP)
 
     if(TENSILE_USE_HIP)
         target_compile_options(custom_openmp_cxx INTERFACE "-fopenmp")
-        target_link_options(custom_openmp_cxx INTERFACE "-fopenmp")
+        target_link_libraries(custom_openmp_cxx INTERFACE ${ROCM_OPENMP_EXTRAS_PATH}/lib/libomp.so)
     else ()
         find_package(OpenMP REQUIRED)
         target_link_libraries(custom_openmp_cxx INTERFACE OpenMP::OpenMP_CXX)
-- 
2.25.1

