--- cpp/oskar/CMakeLists.txt.org	2022-05-30 10:26:52.000000000 +0800
+++ cpp/oskar/CMakeLists.txt	2022-05-30 10:27:12.000000000 +0800
@@ -25,7 +25,7 @@

 #------------------------------------------------------------------------------
 # Link against HDF5 and OpenMP
-target_link_libraries(oskar ${HDF5_LIBRARIES} ${HDF5_CXX_LIBRARIES} ${OpenMP_CXX_FLAGS})
+target_link_libraries(oskar ${HDF5_LIBRARIES} ${HDF5_CXX_LIBRARIES} ${HDF5_HL_LIBRARIES} ${OpenMP_CXX_FLAGS})

 #------------------------------------------------------------------------------
 # install libeverybeam-oskar.so

