--- OpenFOAM-v1612+.orig/etc/config.sh/mpi	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-v1612+/etc/config.sh/mpi	2017-03-29 13:55:57.507980699 +0200
@@ -75,8 +75,15 @@
     _foamAddMan     $MPI_ARCH_PATH/share/man
     ;;
 
+USERMPI)
+    # Use an arbitrary, user-specified mpi implementation
+    export FOAM_MPI=mpi-user
+    _foamSource `$WM_PROJECT_DIR/bin/foamEtcFile config.sh/mpi-user`
+    ;;
+
 SYSTEMMPI)
     export FOAM_MPI=mpi-system
+    _foamSource `$WM_PROJECT_DIR/bin/foamEtcFile config.sh/mpi-system`
 
     if [ -z "$MPI_ROOT" ]
     then
--- OpenFOAM-v1612+.orig/etc/config.csh/mpi	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-v1612+/etc/config.csh/mpi	2017-03-29 13:56:36.347835938 +0200
@@ -71,8 +71,15 @@
     _foamAddMan     $MPI_ARCH_PATH/share/man
     breaksw
 
+case USERMPI:
+    # Use an arbitrary, user-specified mpi implementation
+    setenv FOAM_MPI mpi-user
+    _foamSource `$WM_PROJECT_DIR/bin/foamEtcFile config.csh/mpi-user`
+    breaksw
+
 case SYSTEMMPI:
     setenv FOAM_MPI mpi-system
+    _foamSource `$WM_PROJECT_DIR/bin/foamEtcFile config.csh/mpi-system`
 
     if ( ! ($?MPI_ROOT) ) then
         echo
