--- OpenFOAM-v1612+.orig/applications/utilities/mesh/manipulation/renumberMesh/Allwmake	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-v1612+/applications/utilities/mesh/manipulation/renumberMesh/Allwmake	2017-03-28 11:13:35.222727218 +0200
@@ -4,20 +4,35 @@
 # Parse arguments for compilation (at least for error catching)
 . $WM_PROJECT_DIR/wmake/scripts/AllwmakeParseArguments
 
-export COMPILE_FLAGS=''
-export LINK_FLAGS=''
+unset COMP_FLAGS LINK_FLAGS
 
 if [ -f "${FOAM_LIBBIN}/libSloanRenumber.so" ]
 then
-    echo "Found libSloanRenumber.so  --  enabling Sloan renumbering support."
+    echo "    found libSloanRenumber  --  enabling sloan renumbering support."
     export LINK_FLAGS="${LINK_FLAGS} -lSloanRenumber"
 fi
 
-if [ -f "${ZOLTAN_ARCH_PATH}/lib/libzoltan.a" -a -f "${FOAM_LIBBIN}/libzoltanRenumber.so" ]
+if [ -f "${FOAM_LIBBIN}/libzoltanRenumber.so" ]
 then
-    echo "Found libzoltanRenumber.so  --  enabling zoltan renumbering support."
-    export COMPILE_FLAGS="-DFOAM_USE_ZOLTAN"
-    export LINK_FLAGS="${LINK_FLAGS} -lzoltanRenumber -L${ZOLTAN_ARCH_PATH}/lib -lzoltan"
+    if [ -z "$ZOLTAN_ARCH_PATH" ]
+    then
+        # Optional: get ZOLTAN_ARCH_PATH
+        if settings=$($WM_PROJECT_DIR/bin/foamEtcFile config.sh/zoltan)
+        then
+            . $settings
+        fi
+    fi
+
+    for libdir in lib "lib${WM_COMPILER_LIB_ARCH}"
+    do
+        if [ -f "$ZOLTAN_ARCH_PATH/$libdir/libzoltan.a" ]
+        then
+            echo "    found libzoltanRenumber  --  enabling zoltan renumbering support."
+            export COMP_FLAGS="-DFOAM_USE_ZOLTAN"
+            export LINK_FLAGS="${LINK_FLAGS} -lzoltanRenumber -L$ZOLTAN_ARCH_PATH/$libdir -lzoltan"
+            break
+        fi
+    done
 fi
 
 wmake $targetType
--- OpenFOAM-v1612+.orig/src/renumber/Allwmake	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-v1612+/src/renumber/Allwmake	2017-03-28 11:10:22.195543610 +0200
@@ -5,14 +5,11 @@
 targetType=libso
 . $WM_PROJECT_DIR/wmake/scripts/AllwmakeParseArguments
 
-## Get ZOLTAN_ARCH_PATH
-#if settings=$($WM_PROJECT_DIR/bin/foamEtcFile config.sh/zoltan)
-#then
-#    . $settings
-#    echo "using ZOLTAN_ARCH_PATH=$ZOLTAN_ARCH_PATH"
-#else
-#    echo "Error: no config.sh/zoltan settings"
-#fi
+# Optional: get ZOLTAN_ARCH_PATH
+if settings=$($WM_PROJECT_DIR/bin/foamEtcFile config.sh/zoltan)
+then
+    . $settings
+fi
 
 wmake $targetType renumberMethods
 
--- OpenFOAM-v1612+.orig/src/renumber/zoltanRenumber/Make/options	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-v1612+/src/renumber/zoltanRenumber/Make/options	2017-03-28 11:50:46.484343848 +0200
@@ -4,10 +4,13 @@
 EXE_INC = \
     /* -DFULLDEBUG -g -O0 */ \
     $(PFLAGS) $(PINC) \
+    ${c++LESSWARN} \
     -I$(FOAM_SRC)/renumber/renumberMethods/lnInclude \
     -I$(ZOLTAN_ARCH_PATH)/include/ \
     -I$(LIB_SRC)/meshTools/lnInclude
 
 LIB_LIBS = \
-    /* -L$(ZOLTAN_ARCH_PATH)/lib -lzoltan */ \
+    -L$(ZOLTAN_ARCH_PATH)/lib \
+    -L$(ZOLTAN_ARCH_PATH)/lib$(WM_COMPILER_LIB_ARCH) \
+    -lzoltan \
     -lmeshTools
