--- OpenFOAM-v1612+.orig/src/parallel/decompose/Allwmake	2017-03-21 16:34:44.599021283 +0100
+++ OpenFOAM-v1612+/src/parallel/decompose/Allwmake	2017-03-21 16:28:57.243969660 +0100
@@ -36,6 +36,7 @@
 
     # Library
     [ -r $FOAM_EXT_LIBBIN/libmetis.so ] || \
+    [ -r $METIS_ARCH_PATH/lib/libmetis.so ] || \
     [ -r $METIS_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/libmetis.so ] || \
     [ "${METIS_ARCH_PATH##*-}" = system ] || {
         echo "$warning (missing library)"
@@ -90,6 +91,7 @@
 
     # Library
     [ -r $FOAM_EXT_LIBBIN/libscotch.so ] || \
+    [ -r $SCOTCH_ARCH_PATH/lib/libscotch.so ] || \
     [ -r $SCOTCH_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/libscotch.so ] || \
     [ "${SCOTCH_ARCH_PATH##*-}" = system ] || {
         echo "$warning (missing library)"
--- OpenFOAM-v1612+.orig/src/parallel/decompose/metisDecomp/Make/options	2017-03-21 16:34:25.383075328 +0100
+++ OpenFOAM-v1612+/src/parallel/decompose/metisDecomp/Make/options	2017-03-21 16:30:15.727758338 +0100
@@ -8,6 +8,7 @@
  * to support central, non-thirdparty installations
  */
 LIB_LIBS = \
+    -L$(METIS_ARCH_PATH)/lib \
     -L$(METIS_ARCH_PATH)/lib$(WM_COMPILER_LIB_ARCH) \
     -L$(FOAM_EXT_LIBBIN) \
     -lmetis
--- OpenFOAM-v1612+.orig/src/parallel/decompose/ptscotchDecomp/Make/options	2017-03-21 16:34:34.607049385 +0100
+++ OpenFOAM-v1612+/src/parallel/decompose/ptscotchDecomp/Make/options	2017-03-21 16:30:00.479799399 +0100
@@ -16,6 +16,7 @@
  * to support central, non-thirdparty installations
  */
 LIB_LIBS = \
+    -L$(SCOTCH_ARCH_PATH)/lib \
     -L$(SCOTCH_ARCH_PATH)/lib$(WM_COMPILER_LIB_ARCH) \
     -L$(FOAM_EXT_LIBBIN) \
     -L$(FOAM_EXT_LIBBIN)/$(FOAM_MPI) \
--- OpenFOAM-v1612+.orig/src/parallel/decompose/scotchDecomp/Make/options	2017-03-21 16:34:39.159036582 +0100
+++ OpenFOAM-v1612+/src/parallel/decompose/scotchDecomp/Make/options	2017-03-21 16:29:46.719836452 +0100
@@ -16,6 +16,7 @@
  * to support central, non-thirdparty installations
  */
 LIB_LIBS = \
+    -L$(SCOTCH_ARCH_PATH)/lib \
     -L$(SCOTCH_ARCH_PATH)/lib$(WM_COMPILER_LIB_ARCH) \
     -L$(FOAM_EXT_LIBBIN) \
     -lscotch \
