--- OpenFOAM-plus.orig/etc/config.sh/settings	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-plus/etc/config.sh/settings	2017-04-04 12:21:31.295498985 +0200
@@ -157,7 +157,7 @@
 export FOAM_EXT_LIBBIN=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/lib
 
 # Site-specific directory
-siteDir="${WM_PROJECT_SITE:-$WM_PROJECT_INST_DIR/site}"
+siteDir="${WM_PROJECT_SITE:-$WM_PROJECT_DIR/site}"  #SPACK: not in parent directory
 
 # Shared site executables/libraries
 # Similar naming convention as ~OpenFOAM expansion
--- OpenFOAM-plus.orig/etc/config.csh/settings	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-plus/etc/config.csh/settings	2017-03-23 12:23:52.737891912 +0100
@@ -156,7 +156,7 @@
 if ( $?WM_PROJECT_SITE ) then
     set siteDir=$WM_PROJECT_SITE
 else
-    set siteDir=$WM_PROJECT_INST_DIR/site
+    set siteDir=$WM_PROJECT_DIR/site  #SPACK: not in parent directory
 endif
 
 # Shared site executables/libraries
