--- OpenFOAM-v1612+.orig/Allwmake	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-v1612+/Allwmake	2017-03-29 09:08:15.503865203 +0200
@@ -17,6 +17,14 @@
     exit 1
 }
 
+#------------------------------------------------------------------------------
+echo "========================================"
+date "+%Y-%m-%d %H:%M:%S %z" 2>/dev/null || echo "date is unknown"
+echo "Starting ${WM_PROJECT_DIR##*/} ${0##*}"
+echo "  $WM_COMPILER $WM_COMPILER_TYPE compiler"
+echo "  ${WM_OPTIONS}, with ${WM_MPLIB} ${FOAM_MPI}"
+echo
+
 # Compile wmake support applications
 (cd wmake/src && make)
 
