diff -uw OpenFOAM-plus.orig/etc/config.sh/settings OpenFOAM-plus/etc/config.sh/settings
--- OpenFOAM-plus.orig/etc/config.sh/settings	2017-04-04 17:34:29.875873400 +0200
+++ OpenFOAM-plus/etc/config.sh/settings	2017-04-04 17:38:40.174992466 +0200
@@ -154,10 +154,10 @@
 export FOAM_LIBBIN=$WM_PROJECT_DIR/platforms/$WM_OPTIONS/lib
 
 # External (ThirdParty) libraries
-export FOAM_EXT_LIBBIN=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/lib
+unset FOAM_EXT_LIBBIN  #SPACK: none
 
 # Site-specific directory
-siteDir="${WM_PROJECT_SITE:-$WM_PROJECT_INST_DIR/site}"
+siteDir="${WM_PROJECT_SITE:-$WM_PROJECT_DIR/site}"  #SPACK: not in parent directory
 
 # Shared site executables/libraries
 # Similar naming convention as ~OpenFOAM expansion
diff -uw OpenFOAM-plus.orig/etc/config.csh/settings OpenFOAM-plus/etc/config.csh/settings
--- OpenFOAM-plus.orig/etc/config.csh/settings	2017-04-04 17:34:28.255879107 +0200
+++ OpenFOAM-plus/etc/config.csh/settings	2017-04-04 17:39:22.214844670 +0200
@@ -151,13 +151,13 @@
 setenv FOAM_LIBBIN $WM_PROJECT_DIR/platforms/$WM_OPTIONS/lib
 
 # External (ThirdParty) libraries
-setenv FOAM_EXT_LIBBIN $WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/lib
+unsetenv FOAM_EXT_LIBBIN  #SPACK: none
 
 # Site-specific directory
 if ( $?WM_PROJECT_SITE ) then
     set siteDir=$WM_PROJECT_SITE
 else
-    set siteDir=$WM_PROJECT_INST_DIR/site
+    set siteDir=$WM_PROJECT_DIR/site  #SPACK: not in parent directory
 endif
 
 # Shared site executables/libraries
