diff -uw OpenFOAM-v1612+.orig/etc/config.sh/settings OpenFOAM-v1612+/etc/config.sh/settings
--- OpenFOAM-v1612+.orig/etc/config.sh/settings	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-v1612+/etc/config.sh/settings	2017-03-23 12:22:52.002101020 +0100
@@ -141,7 +141,7 @@
 #------------------------------------------------------------------------------
 
 # Location of the jobControl directory
-export FOAM_JOB_DIR=$WM_PROJECT_INST_DIR/jobControl
+export FOAM_JOB_DIR=$HOME/.OpenFOAM/jobControl  #SPACK: non-central location
 
 # wmake configuration
 export WM_DIR=$WM_PROJECT_DIR/wmake
@@ -157,7 +157,7 @@
 export FOAM_EXT_LIBBIN=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/lib
 
 # Site-specific directory
-siteDir="${WM_PROJECT_SITE:-$WM_PROJECT_INST_DIR/site}"
+siteDir="${WM_PROJECT_SITE:-$WM_PROJECT_DIR/site}"  #SPACK: not in parent directory
 
 # Shared site executables/libraries
 # Similar naming convention as ~OpenFOAM expansion
diff -uw OpenFOAM-v1612+.orig/etc/config.csh/settings OpenFOAM-v1612+/etc/config.csh/settings
--- OpenFOAM-v1612+.orig/etc/config.csh/settings	2016-12-23 15:22:59.000000000 +0100
+++ OpenFOAM-v1612+/etc/config.csh/settings	2017-03-23 12:23:52.737891912 +0100
@@ -137,7 +137,7 @@
 #------------------------------------------------------------------------------
 
 # Location of the jobControl directory
-setenv FOAM_JOB_DIR $WM_PROJECT_INST_DIR/jobControl
+setenv FOAM_JOB_DIR=$HOME/.OpenFOAM/jobControl  #SPACK: non-central location
 
 # wmake configuration
 setenv WM_DIR $WM_PROJECT_DIR/wmake
@@ -156,7 +156,7 @@
 if ( $?WM_PROJECT_SITE ) then
     set siteDir=$WM_PROJECT_SITE
 else
-    set siteDir=$WM_PROJECT_INST_DIR/site
+    set siteDir=$WM_PROJECT_DIR/site  #SPACK: not in parent directory
 endif
 
 # Shared site executables/libraries
