diff --git a/cmake_modules/env.cmake b/cmake_modules/env.cmake
index 6647578..70b411e 100644
--- a/cmake_modules/env.cmake
+++ b/cmake_modules/env.cmake
@@ -75,5 +75,5 @@ endif ()
 
 find_library ( FIND_AQL_PROFILE_LIB "libhsa-amd-aqlprofile64.so" HINTS ${CMAKE_INSTALL_PREFIX} PATHS ${ROCM_ROOT_DIR})
 if (  NOT FIND_AQL_PROFILE_LIB )
-  message ( FATAL_ERROR "AQL_PROFILE not installed. Please install AQL_PROFILE" )
+  message ( "AQL_PROFILE not installed. Please install AQL_PROFILE" )
 endif()
diff --git a/src/api/CMakeLists.txt b/src/api/CMakeLists.txt
index 822ee41..2956ceb 100644
--- a/src/api/CMakeLists.txt
+++ b/src/api/CMakeLists.txt
@@ -40,7 +40,7 @@ get_filename_component(HSA_RUNTIME_INC_PATH ${HSA_H} DIRECTORY)
 find_library(AQLPROFILE_LIB "libhsa-amd-aqlprofile64.so" HINTS ${CMAKE_PREFIX_PATH} PATHS ${ROCM_PATH} PATH_SUFFIXES lib)
 
 if(NOT AQLPROFILE_LIB)
-  message(FATAL_ERROR "AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
+  message("AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
 endif()
 
 # ############################################################################################################################################
@@ -224,7 +224,7 @@ target_compile_definitions(${ROCPROFILER_TARGET}
 target_include_directories(${ROCPROFILER_TARGET}
   PUBLIC
   ${ROCM_PATH}/include
-  ${HIP_INCLUDE_DIRECTORIES} ${HSA_RUNTIME_INCLUDE_DIRECTORIES}
+  ${HIP_INCLUDE_DIRECTORIES} ${HSA_RUNTIME_INCLUDE_DIRECTORIES} ${PCIACCESS_INCLUDE_DIR}
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/tools>
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/inc>
@@ -241,7 +241,12 @@ if(ASAN)
   target_link_libraries(${ROCPROFILER_TARGET} PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 Threads::Threads atomic asan dl c stdc++ stdc++fs amd_comgr ${PCIACCESS_LIBRARIES})
 else()
   target_link_options(${ROCPROFILER_TARGET} PRIVATE -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/exportmap -Wl,--no-undefined)
-  target_link_libraries(${ROCPROFILER_TARGET} PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 Threads::Threads atomic dl c stdc++ stdc++fs amd_comgr ${PCIACCESS_LIBRARIES})
+  if(NOT AQLPROFILE_LIB)
+    message("AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
+    target_link_libraries(${ROCPROFILER_TARGET} PRIVATE  hsa-runtime64::hsa-runtime64 Threads::Threads atomic dl c stdc++ stdc++fs amd_comgr ${PCIACCESS_LIBRARIES})
+  else()
+    target_link_libraries(${ROCPROFILER_TARGET} PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 Threads::Threads atomic dl c stdc++ stdc++fs amd_comgr ${PCIACCESS_LIBRARIES})
+  endif()
 endif()
 
 ## Install libraries: Non versioned lib file in dev package
diff --git a/src/tools/rocprofv2/CMakeLists.txt b/src/tools/rocprofv2/CMakeLists.txt
index b2dc968..f8bbe4a 100644
--- a/src/tools/rocprofv2/CMakeLists.txt
+++ b/src/tools/rocprofv2/CMakeLists.txt
@@ -6,7 +6,7 @@ get_property(HSA_RUNTIME_INCLUDE_DIRECTORIES TARGET hsa-runtime64::hsa-runtime64
 
 find_library(AQLPROFILE_LIB "libhsa-amd-aqlprofile64.so" HINTS ${CMAKE_PREFIX_PATH} PATHS ${ROCM_PATH} PATH_SUFFIXES lib)
 if(NOT AQLPROFILE_LIB)
-  message(FATAL_ERROR "AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
+  message("AQL_PROFILE not installed. Please install hsa-amd-aqlprofile!")
 endif()
 
 file(GLOB ROCPROFV2_SRC_FILES ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
@@ -23,7 +23,11 @@ target_include_directories(rocprofv2
   PRIVATE
     ${PROJECT_SOURCE_DIR}
     ${PROJECT_SOURCE_DIR}/inc)
-target_link_libraries(rocprofv2 PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 stdc++fs Threads::Threads atomic -ldl)
+if(NOT AQLPROFILE_LIB)
+target_link_libraries(rocprofv2 PRIVATE hsa-runtime64::hsa-runtime64 stdc++fs Threads::Threads atomic -ldl)
+else()
+ target_link_libraries(rocprofv2 PRIVATE ${AQLPROFILE_LIB} hsa-runtime64::hsa-runtime64 stdc++fs Threads::Threads atomic -ldl)
+endif()
 # install(TARGETS rocprofv2 RUNTIME
 #   DESTINATION ${CMAKE_INSTALL_BINDIR}
-#   COMPONENT runtime)
\ No newline at end of file
+#   COMPONENT runtime)
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 625dcd7..efa2011 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -118,7 +118,7 @@ execute_process ( COMMAND sh -xc "mkdir -p ${PROJECT_BINARY_DIR}/RESULTS" )
 set ( TEST_LIB "rocprof-tool" )
 set ( TEST_LIB_SRC ${TEST_DIR}/tool/tool.cpp ${UTIL_SRC} )
 add_library ( ${TEST_LIB} SHARED ${TEST_LIB_SRC} )
-target_include_directories ( ${TEST_LIB} PRIVATE ${TEST_DIR} ${ROOT_DIR} )
+target_include_directories ( ${TEST_LIB} PRIVATE ${TEST_DIR} ${ROOT_DIR} ${PCIACCESS_INCLUDE_DIR})
 target_link_libraries ( ${TEST_LIB} ${ROCPROFILER_TARGET} hsa-runtime64::hsa-runtime64 Threads::Threads dl )
 
 ## Build memory test bench
diff --git a/tests/unittests/core/CMakeLists.txt b/tests/unittests/core/CMakeLists.txt
index 5efc625..348650a 100644
--- a/tests/unittests/core/CMakeLists.txt
+++ b/tests/unittests/core/CMakeLists.txt
@@ -97,10 +97,16 @@ target_compile_definitions(runCoreUnitTests
   PUBLIC AMD_INTERNAL_BUILD
   PRIVATE PROF_API_IMPL HIP_PROF_HIP_API_STRING=1 __HIP_PLATFORM_AMD__=1)
 
-# Link test executable against gtest & gtest_main
-target_link_libraries(runCoreUnitTests PRIVATE ${ROCPROFILER_TARGET} ${AQLPROFILE_LIB}
-  hsa-runtime64::hsa-runtime64 c stdc++
-  GTest::gtest GTest::gtest_main stdc++fs dl ${PCIACCESS_LIBRARIES})
+ # Link test executable against gtest & gtest_main
+if(NOT AQLPROFILE_LIB )
+  target_link_libraries(runCoreUnitTests PRIVATE ${ROCPROFILER_TARGET}
+    hsa-runtime64::hsa-runtime64 c stdc++
+    GTest::gtest GTest::gtest_main stdc++fs dl ${PCIACCESS_LIBRARIES})
+else()
+  target_link_libraries(runCoreUnitTests PRIVATE ${ROCPROFILER_TARGET} ${AQLPROFILE_LIB}
+    hsa-runtime64::hsa-runtime64 c stdc++
+    GTest::gtest GTest::gtest_main stdc++fs dl ${PCIACCESS_LIBRARIES})
+endif()
 
 add_dependencies(tests runCoreUnitTests)
 install(TARGETS runCoreUnitTests RUNTIME DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/tests COMPONENT tests)
diff --git a/tests/unittests/profiler/CMakeLists.txt b/tests/unittests/profiler/CMakeLists.txt
index 7e08ace..35cb945 100644
--- a/tests/unittests/profiler/CMakeLists.txt
+++ b/tests/unittests/profiler/CMakeLists.txt
@@ -84,10 +84,17 @@ target_compile_definitions(runProfilerUnitTests
   PUBLIC AMD_INTERNAL_BUILD
   PRIVATE PROF_API_IMPL HIP_PROF_HIP_API_STRING=1 __HIP_PLATFORM_AMD__=1)
 
-target_link_libraries(runProfilerUnitTests PRIVATE rocprofiler_tool ${AQLPROFILE_LIB}
-  hsa-runtime64::hsa-runtime64
-  GTest::gtest GTest::gtest_main stdc++fs
-  ${PCIACCESS_LIBRARIES})
+if(NOT AQLPROFILE_LIB )
+  target_link_libraries(runProfilerUnitTests PRIVATE rocprofiler_tool
+    hsa-runtime64::hsa-runtime64
+    GTest::gtest GTest::gtest_main stdc++fs
+    ${PCIACCESS_LIBRARIES})
+else()
+  target_link_libraries(runProfilerUnitTests PRIVATE rocprofiler_tool ${AQLPROFILE_LIB}
+    hsa-runtime64::hsa-runtime64
+    GTest::gtest GTest::gtest_main stdc++fs
+    ${PCIACCESS_LIBRARIES})
+endif()
 
 add_dependencies(tests runProfilerUnitTests)
 install(TARGETS runProfilerUnitTests RUNTIME DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/tests COMPONENT tests)
