From e7dd0146c83527e4977ecf51b22c9cb6939d74f2 Mon Sep 17 00:00:00 2001
From: Afzal Patel <Afzal.Patel@amd.com>
Date: Fri, 2 Feb 2024 13:50:58 +0000
Subject: [PATCH] Disable tests and add fPIC compile option

---
 CMakeLists.txt | 26 ++++----------------------
 1 file changed, 4 insertions(+), 22 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8f5da7..5c2537a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,7 +44,7 @@ endif()

 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-add_compile_options(-Wall)
+add_compile_options(-Wall -fPIC)

 set(THREADS_PREFER_PTHREAD_FLAG ON)

@@ -162,17 +162,17 @@ add_subdirectory(src)

 if(${LIBRARY_TYPE} STREQUAL SHARED)
     # Build samples
-    add_subdirectory(samples)
+    # add_subdirectory(samples)

     # Build tests
-    add_subdirectory(tests-v2)
+    # add_subdirectory(tests-v2)
 endif()

 # Build Plugins
 add_subdirectory(plugin)

 # Build tests
-add_subdirectory(${TEST_DIR} ${PROJECT_BINARY_DIR}/test)
+# add_subdirectory(${TEST_DIR} ${PROJECT_BINARY_DIR}/test)

 # Installation and packaging
 set(DEST_NAME ${ROCPROFILER_NAME})
@@ -236,24 +236,6 @@ install(
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/${ROCPROFILER_NAME}
     COMPONENT runtime)

-# librocprof-tool.so
-install(
-    FILES ${PROJECT_BINARY_DIR}/test/librocprof-tool.so
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/${ROCPROFILER_NAME}
-    COMPONENT runtime)
-
-install(
-    FILES ${PROJECT_BINARY_DIR}/test/librocprof-tool.so
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/${ROCPROFILER_NAME}
-    COMPONENT asan)
-
-install(
-    FILES ${PROJECT_BINARY_DIR}/test/rocprof-ctrl
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/${ROCPROFILER_NAME}
-    PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ
-                WORLD_EXECUTE
-    COMPONENT runtime)
-
 # File reorg backward compatibility for non ASAN packaging
 if(NOT ENABLE_ASAN_PACKAGING)
     # File reorg Backward compatibility
