diff --git a/test/ctrl/test_kernel.h b/test/ctrl/test_kernel.h
index 2ee6f9c..ca8d6e8 100644
--- a/test/ctrl/test_kernel.h
+++ b/test/ctrl/test_kernel.h
@@ -23,8 +23,9 @@ THE SOFTWARE.
 #ifndef TEST_CTRL_TEST_KERNEL_H_
 #define TEST_CTRL_TEST_KERNEL_H_
 
-#include <string.h>
-#include <stdint.h>
+#include <string>
+#include <cstdlib>
+#include <cstdint>
 #include <map>
 
 // Class implements kernel test
