diff --git a/setuptools/_distutils/unixccompiler.py b/setuptools/_distutils/unixccompiler.py
--- a/setuptools/_distutils/unixccompiler.py
+++ b/setuptools/_distutils/unixccompiler.py
@@ -257,7 +257,7 @@ class UnixCCompiler(CCompiler):
                 # No idea how --enable-new-dtags would be passed on to
                 # ld if this system was using GNU ld.  Don't know if a
                 # system like this even exists.
-                return "-R" + dir
+                return "-Wl,-rpath," + dir
 
     def library_option(self, lib):
         return "-l" + lib

