diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 120a899..05991f1 100755
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -41,7 +41,7 @@ SET(headers
   ${CMAKE_SOURCE_DIR}/include/podio/PythonEventStore.h
   )
 PODIO_GENERATE_DICTIONARY(podioDict ${headers} SELECTION selection.xml
-  OPTIONS --library ${CMAKE_SHARED_LIBRARY_PREFIX}podio${CMAKE_SHARED_LIBRARY_SUFFIX}
+  OPTIONS --library ${CMAKE_SHARED_LIBRARY_PREFIX}podioDict${CMAKE_SHARED_LIBRARY_SUFFIX}
   )
 # prevent generating dictionary twice
 set_target_properties(podioDict-dictgen PROPERTIES EXCLUDE_FROM_ALL TRUE)
@@ -58,5 +58,5 @@ install(TARGETS podio podioDict podioRootIO
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/podio DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
 install(FILES
   ${CMAKE_CURRENT_BINARY_DIR}/podioDictDict.rootmap
-  ${CMAKE_CURRENT_BINARY_DIR}/libpodio_rdict.pcm
+  ${CMAKE_CURRENT_BINARY_DIR}/libpodioDict_rdict.pcm
   DESTINATION "${CMAKE_INSTALL_LIBDIR}")
