Index: src/config/makefile.h
===================================================================
--- src/config/makefile.h	(revision 27729)
+++ src/config/makefile.h	(revision 27844)
@@ -2257,11 +2258,7 @@
   DEFINES += -DFDIST
 endif
 
-_TOOLS_BUILD= $(shell [ -e ${NWCHEM_TOP}/src/tools/build/config.h ] && cat ${NWCHEM_TOP}/src/tools/build/config.h | awk ' /HAVE_SQRT/ {print "Y"}')
-
-ifeq ($(_TOOLS_BUILD),Y)
 _USE_SCALAPACK = $(shell cat ${NWCHEM_TOP}/src/tools/build/config.h | awk ' /HAVE_SCALAPACK\ 1/ {print "Y"}')
-endif
 
 ifeq ($(_USE_SCALAPACK),Y)
   DEFINES += -DSCALAPACK
@@ -2286,8 +2283,8 @@
 	      -brename:.pdgetrf_,.pdgetrf \
 	      -brename:.pdgetrs_,.pdgetrs 
 endif
-  CORE_LIBS += $(ELPA) $(SCALAPACK) $(PBLAS) $(BLACS)
 endif
+  CORE_LIBS += $(ELPA) $(SCALAPACK)
 
 ifdef USE_64TO32
       CORE_LIBS +=  -l64to32
@@ -2436,18 +2433,11 @@
  DEFINES += -DUSE_F90_ALLOCATABLE
 endif
 
-ifeq ($(_TOOLS_BUILD),Y)
 # lower level libs used by communication libraries 
 COMM_LIBS=  $(shell grep ARMCI_NETWORK_LIBS\ = ${NWCHEM_TOP}/src/tools/build/Makefile | cut -b 22-)
 COMM_LIBS +=  $(shell grep ARMCI_NETWORK_LDFLAGS\ = ${NWCHEM_TOP}/src/tools/build/Makefile | cut -b 24-)
 #comex bit
-HAVE_COMEX = $(shell [ -e ${NWCHEM_TOP}/src/tools/build/comex/config.h ] && cat ${NWCHEM_TOP}/src/tools/build/comex/config.h| grep COMEX_NETWORK| awk ' / 1/ {print "Y"}')
-ifeq ($(HAVE_COMEX),Y)
-COMM_LIBS +=  $(shell grep LIBS\ = ${NWCHEM_TOP}/src/tools/build/comex/Makefile|grep -v _LIBS| cut -b 8-)
-#we often need pthread, let's add it
-COMM_LIBS += -lpthread
-endif
-endif
+COMM_LIBS +=  $(shell [ -e ${NWCHEM_TOP}/src/tools/build/comex/config.h ] && grep LIBS\ = ${NWCHEM_TOP}/src/tools/build/comex/Makefile|grep -v _LIBS| cut -b 8-) -lpthread
 ifdef COMM_LIBS 
  CORE_LIBS += $(COMM_LIBS) 
 endif 
