Index: src/nwdft/scf_dft/dft_scf.F
===================================================================
--- src/nwdft/scf_dft/dft_scf.F	(revision 28116)
+++ src/nwdft/scf_dft/dft_scf.F	(revision 28117)
@@ -1884,6 +1884,13 @@
          if (abs(Edisp).gt.0.0d0) then
           write(LuOut,224)Edisp
          endif
+         if (cosmo_on.and.cosmo_phase.eq.2) then
+          if (do_cosmo_smd) then
+           write(LuOut,225) ecosmo+gcds
+          else
+           write(LuOut,225) ecosmo
+          end if
+         endif
          if (do_zora) write(luout,2221) ener_scal
          write(luout,2222) rho_n
          write(luout,2223) dft_time
@@ -2457,6 +2464,7 @@
      &     '       Correlation energy =', f22.12/
      &     ' Nuclear repulsion energy =', f22.12/)
  224  format('    Dispersion correction =', f22.12/)
+ 225  format('             COSMO energy =', f22.12/)
 c
  2221 format('       Scaling correction =', f22.12/)
  2222 format(' Numeric. integr. density =', f22.12/)
