Index: src/dplot/dplot_input.F
===================================================================
--- src/dplot/dplot_input.F	(revision 27986)
+++ src/dplot/dplot_input.F	(revision 27987)
@@ -63,6 +63,7 @@
       iroot     = 1
       ltransden = .true.
       ldiffden = .false.
+      tol_rho = 1d-40
 c
 c     try to get a scf movecs
 c
@@ -263,10 +264,10 @@
       goto 10
 c
  1998 continue
-      tol_rho = 1d-15
       If (.not. inp_f(tol_rho))
      &  Call ErrQuit('DPlot_Input: failed to read tol_rho',0,
      &     INPUT_ERR)
+      tol_rho=max(1d-99,tol_rho)
       goto 10
 c
  1999 continue
Index: src/dplot/dplot_dump.F
===================================================================
--- src/dplot/dplot_dump.F	(revision 27986)
+++ src/dplot/dplot_dump.F	(revision 27987)
@@ -90,7 +90,7 @@
      .              No_Of_Spacings(3))
 99498          format(6E13.5)
             enddo
-            else
+         else
             Do i = 1, nGrid
                Write(Out_Unit,'(f15.10)')values(i)
             End Do
@@ -107,6 +107,7 @@
             End Do
             AppCh = Sum*Volume
             Write(LuOut,*)
+            Write(LuOut,'(a,e30.5)')'  Tol_rho              = ',tol_rho
             Write(LuOut,'(a,f30.5)')'  Sum of elements      = ',sum
             Write(LuOut,'(a,f30.5)')'  Integration volume   = ',volume
             Write(LuOut,'(a,f30.5)')'  Integrated Charge    = ',AppCh
