Index: src/tools/ga-5-4/gaf2c/gaf2c.c
===================================================================
--- src/tools/ga-5-4/gaf2c/gaf2c.c	(revision 10630)
+++ src/tools/ga-5-4/gaf2c/gaf2c.c	(revision 10631)
@@ -106,6 +106,7 @@
     }
     *argc = iargc;
     *argv = iargv;
+    iargv[iargc] = 0;
 }
 
 
Index: src/tools/ga-5-4/tcgmsg/fapi.c
===================================================================
--- src/tools/ga-5-4/tcgmsg/fapi.c	(revision 10630)
+++ src/tools/ga-5-4/tcgmsg/fapi.c	(revision 10631)
@@ -197,6 +197,7 @@
         argv[i] = strdup(arg);
     }
 
+    argv[argc] = 0;
     tcgi_pbegin(argc, argv);
     free(argv);
 }
