Index: src/util/util_mpinap.c
===================================================================
--- src/util/util_mpinap.c	(revision 28079)
+++ src/util/util_mpinap.c	(revision 28083)
@@ -17,7 +17,7 @@
 #ifdef MPI
   MPI_Comm_rank(MPI_COMM_WORLD,&myid);
 #else
-  myid=ga_nodeid_();
+  myid=GA_Nodeid();
 #endif
   sleeptime=(myid+1)/((long) *factor);
 #ifdef DEBUG
Index: src/util/util_getppn.c
===================================================================
--- src/util/util_getppn.c	(revision 28079)
+++ src/util/util_getppn.c	(revision 28083)
@@ -8,6 +8,7 @@
 #include <unistd.h>
 #include <mpi.h>
 #include "ga.h"
+#include "ga-mpi.h"
 #include "typesf2c.h"
 
 #if defined(__bgq__)
