Index: src/symmetry/sym_abelian.F
===================================================================
--- src/symmetry/sym_abelian.F	(revision 27901)
+++ src/symmetry/sym_abelian.F	(revision 27902)
@@ -10,9 +10,11 @@
 c
       character*8 group
       integer nab, ind
-      parameter (nab = 8)
+      parameter (nab = 18)
       character*4 ab(nab)
-      data ab/ 'C1','Cs','Ci','C2','D2','C2v','C2h','D2h'/
+      data ab/ 'C1','Cs','Ci','C2','D2','C2v','C2h','D2h',
+     C     'C3','C4','C5','C6','C7','C8',
+     C     'C3h','C4h','C5h','C6h'/
 c
       call sym_group_name(geom, group)
 c
