Index: src/NWints/texas/assemblx.f
===================================================================
--- src/NWints/texas/assemblx.f	(revision 28366)
+++ src/NWints/texas/assemblx.f	(working copy)
@@ -133,7 +133,9 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 C
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
 c
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
@@ -258,7 +260,9 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension aax(nbls1),bbx(nbls1),ccx(nbls1)
@@ -346,7 +350,9 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension aax(nbls1),bbx(nbls1),ccx(nbls1)
@@ -428,7 +434,9 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 C
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
 c
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
@@ -626,7 +634,9 @@
       character*11 scftype
       character*8 where
       common /runtype/ scftype,where
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       COMMON/SHELL/LSHELLT,LSHELIJ,LSHELKL,LHELP,LCAS2(4),LCAS3(4)
       common /lcases/ lcase
       common/obarai/
@@ -913,7 +923,9 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 C
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
 C
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
@@ -972,7 +984,9 @@
       implicit real*8 (a-h,o-z)
       logical firstc
 c
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
 c
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
@@ -1045,7 +1059,9 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 C
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
 c
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
@@ -1131,7 +1147,9 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 C
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
 c
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
@@ -1217,7 +1235,9 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 C
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
 C
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2), bf3l(nbls,lt5,lt6)
@@ -1385,7 +1405,9 @@
       character*11 scftype
       character*8 where
       common /runtype/ scftype,where
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       COMMON/SHELL/LSHELLT,LSHELIJ,LSHELKL,LHELP,LCAS2(4),LCAS3(4)
       common /lcases/ lcase
       common/obarai/
@@ -1659,7 +1681,9 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension bfij1(nbls,lt3,lt4)
@@ -1707,7 +1731,9 @@
      *                    bfij3,lt3,lt4, factij, indx, ij3b,kl3b)
       implicit real*8 (a-h,o-z)
       logical firstc
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension bfij3(nbls,lt3,lt4)
@@ -1762,7 +1788,9 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension bf2l1(nbls,lt3,lt4)
@@ -1829,7 +1857,9 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension bf3l(nbls,lt5,lt6)
@@ -1895,7 +1925,9 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2), bf3l(nbls,lt5,lt6)
 cccc  dimension facti(*), factkl(*)
@@ -2018,7 +2050,9 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension aax(nbls1),bbx(nbls1),ccx(nbls1)
@@ -2110,7 +2144,9 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension aax(nbls1),bbx(nbls1),ccx(nbls1)
@@ -2196,7 +2232,9 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 C
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
 c
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
Index: src/NWints/texas/derivat.f
===================================================================
--- src/NWints/texas/derivat.f	(revision 28366)
+++ src/NWints/texas/derivat.f	(working copy)
@@ -16,7 +16,9 @@
 c
       implicit real*8 (a-h,o-z)
 c
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter (lpar1=34)
+      common /logic4/ nfu(lpar1)
       common /big/ bl(1)
       COMMON/SHELL/LSHELLT,LSHELIJ,LSHELKL,LHELP,LCAS2(4),LCAS3(4)
       common /lcases/ lcase
@@ -289,9 +291,15 @@
      *                    nqij,nqkl, deriv,  xab,xcd, xyab,xycd)
       implicit real*8 (a-h,o-z)
 c
-      common /logic4/ nfu(1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
       dimension buf2(nbls,lnijr,lnklr,ngcd)
       dimension deriv(6,nbls,lnij,lnkl,ngcd)
@@ -374,7 +382,9 @@
 c
       implicit real*8 (a-h,o-z)
 c
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       COMMON/SHELL/LSHELLT,LSHELIJ,LSHELKL,LHELP,LCAS2(4),LCAS3(4)
       common /lcases/ lcase
       common/obarai/
@@ -705,10 +715,15 @@
 c second-der. That's why dimension for buf2(ndim,*,*,*,*) has ndim=4
 c for first- and ndim=10 for second-derivatives.
 c
-      common /logic4/ nfu(1)
-      common /logic9/ nia(3,1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
 cccc  dimension buf2(4,nbls,lnijr,lnklr,ngcd) OR buf2(10,etc.)
 c2002 dimension buf2(ndim,nbls,lnijr,lnklr,ngcd)
@@ -862,7 +877,9 @@
 c
       implicit real*8 (a-h,o-z)
 c
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter(lpar1=34)
+      common /logic4/ nfu(lpar1)
       COMMON/SHELL/LSHELLT,LSHELIJ,LSHELKL,LHELP,LCAS2(4),LCAS3(4)
       common /lcases/ lcase
       common/obarai/
@@ -1131,10 +1148,15 @@
      *                     nqij,nqkl,der2,xab)
       implicit real*8 (a-h,o-z)
 c
-      common /logic4/ nfu(1)
-      common /logic9/ nia(3,1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
 c2002 dimension buf2(10,nbls,lnijr,lnklr,ngcd)
       dimension buf2(nbls,lnijr,lnklr,ngcd,10)
@@ -1386,10 +1408,15 @@
      *                    nqij,nqkl,
      *                    nder_aa,der2)
       implicit real*8 (a-h,o-z)
-      common /logic4/ nfu(1)
-      common /logic9/ nia(3,1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
       dimension buf2(nbls,lnijr,lnklr,ngcd,10)
       dimension der2(45,nbls,lnij,lnkl,ngcd)
@@ -1462,10 +1489,15 @@
      *                    nqij,nqkl,
      *                    nder_cc,der2)
       implicit real*8 (a-h,o-z)
-      common /logic4/ nfu(1)
-      common /logic9/ nia(3,1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
 c2002 dimension buf2(10,nbls,lnijr,lnklr,ngcd)
       dimension buf2(nbls,lnijr,lnklr,ngcd,10)
@@ -1533,10 +1565,15 @@
      *                    nqij,nqkl,
      *                    nder_bb,der2,xab)
       implicit real*8 (a-h,o-z)
-      common /logic4/ nfu(1)
-      common /logic9/ nia(3,1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
 c2002 dimension buf2(10,nbls,lnijr,lnklr,ngcd)
       dimension buf2(nbls,lnijr,lnklr,ngcd,10)
@@ -1592,10 +1629,15 @@
      *                    nqij,nqkl,
      *                    nder_ab,der2,xab)
       implicit real*8 (a-h,o-z)
-      common /logic4/ nfu(1)
-      common /logic9/ nia(3,1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
 c2002 dimension buf2(10,nbls,lnijr,lnklr,ngcd)
       dimension buf2(nbls,lnijr,lnklr,ngcd,10)
@@ -1668,10 +1710,15 @@
      *                    nqij,nqkl,
      *                    nder_ac,der2)
       implicit real*8 (a-h,o-z)
-      common /logic4/ nfu(1)
-      common /logic9/ nia(3,1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
 c2002 dimension buf2(10,nbls,lnijr,lnklr,ngcd)
       dimension buf2(nbls,lnijr,lnklr,ngcd,10)
@@ -1742,10 +1789,15 @@
      *                    nqij,nqkl,
      *                    nder_bc,der2,xab)
       implicit real*8 (a-h,o-z)
-      common /logic4/ nfu(1)
-      common /logic9/ nia(3,1)
-      common /logic10/ nmxyz(3,1)
-      common /logic11/ npxyz(3,1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
 c2002 dimension buf2(10,nbls,lnijr,lnklr,ngcd)
       dimension buf2(nbls,lnijr,lnklr,ngcd,10)
Index: src/NWints/texas/gencon.f
===================================================================
--- src/NWints/texas/gencon.f	(revision 28366)
+++ src/NWints/texas/gencon.f	(working copy)
@@ -388,7 +388,15 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,mmax,
      * nqi,nqj,nqk,nql,nsij,nskl,
      * nqij,nqij1,nsij1,nqkl,nqkl1,nskl1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension buf2(nbls,lt1,lt2,ngcd)
@@ -466,7 +474,15 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 c
-      common /logic4/ nfu(1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
@@ -579,7 +595,15 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,mmax,
      * nqi,nqj,nqk,nql,nsij,nskl,
      * nqij,nqij1,nsij1,nqkl,nqkl1,nskl1,ijbeg,klbeg
-      common /logic4/ nfu(1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
       dimension indx(*)
       dimension xt1(nbls1,lt1,lt2)
       dimension gcoef(nbls,ngcd)
Index: src/NWints/texas/shells.f
===================================================================
--- src/NWints/texas/shells.f	(revision 28366)
+++ src/NWints/texas/shells.f	(working copy)
@@ -5,7 +5,12 @@
       common /contr/ ngci,ngcj,ngck,ngcl,lci,lcj,lck,lcl,lcij,lckl
       common /lengt/ ilen,jlen,klen,llen, ilen1,jlen1,klen1,llen1
       common /gcont/ ngci1,ngcj1,ngck1,ngcl1,ngcd
-      common /logic2/ len(1)
+      integer lpar1,lpar4,lpar5
+      parameter(lpar1=34,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
       dimension inx(12,*)
 c
 c This subroutine sets up TYPE and LENGTH of shells and
@@ -93,10 +98,12 @@
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
 C
-      common /logic1/ ndege(1)
-      common /logic2/ len(1)
-      common /logic3/ lensm(1)
-      common /logic4/ nfu(1)
+      integer lpar1,lpar4,lpar5
+      parameter(lpar1=34,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
 c
       COMMON/SHELL/LSHELLT,LSHELIJ,LSHELKL,LHELP,LCAS2(4),LCAS3(4)
       common /lcases/ lcase
@@ -237,7 +244,15 @@
      * lni,lnj,lnk,lnl,lnij,lnkl,lnijkl,MMAX,
      * NQI,NQJ,NQK,NQL,NSIJ,NSKL,
      * NQIJ,NQIJ1,NSIJ1,NQKL,NQKL1,NSKL1,ijbeg,klbeg
-      common /logic3/ lensm(1)
+      integer lpar1,lpar2,lpar3,lpar4,lpar5
+      parameter(lpar1=34,lpar2=6545,lpar3=4060,lpar4=10,lpar5=33)
+      common /logic1/ ndege(lpar4)
+      common /logic2/ len(lpar4)
+      common /logic3/ lensm(lpar5)
+      common /logic4/ nfu(lpar1)
+      common /logic9/ nia(3,lpar2)
+      common /logic10/ nmxyz(3,lpar2)
+      common /logic11/ npxyz(3,lpar3)
 c
 C************************************************************
 c
Index: src/NWints/texas/zeroint.f
===================================================================
--- src/NWints/texas/zeroint.f	(revision 28366)
+++ src/NWints/texas/zeroint.f	(working copy)
@@ -12,7 +12,9 @@
       character*11 scftype
       character*8 where
       common /runtype/ scftype,where
-      common /logic4/ nfu(1)
+      integer lpar1
+      parameter (lpar1=34)
+      common /logic4/ nfu(lpar1)
       COMMON/SHELL/LSHELLT,LSHELIJ,LSHELKL,LHELP,LCAS2(4),LCAS3(4)
       common /lcases/ lcase
       common/obarai/
