Index: src/util/GNUmakefile
===================================================================
--- src/util/GNUmakefile	(revision 27774)
+++ src/util/GNUmakefile	(revision 27782)
@@ -234,7 +234,7 @@
 
     USES_BLAS = util.fh ga_it_lsolve.F ga_maxelt.F ga_mix.F ga_iter_diag.F \
                 ga_orthog.F dabsmax.F ga_normf.F corr_mk_ref.F ga_it2.F ga_lkain_ext.F util_file_name.F dgefa.f util_patch_test.F stpr_sjacobi.F util_dgeev.F \
-		util_test_cholesky.F
+		util_test_cholesky.F dfill.f ga_lkain_2cpl3_ext.F ga_it2.F
 
 ifdef SPEECH
    LIB_DEFINES += -DSPEECH
@@ -254,6 +254,7 @@
 ifeq ($(TARGET),$(findstring $(TARGET),BGL BGP BGQ))
 DEFINES += -DNEED_LOC
 LIB_DEFINES += -DNO_UTIL_TESTS
+LIB_DEFINES += -I/bgsys/drivers/ppcfloor/firmware/include -I/bgsys/drivers/ppcfloor/spi/include/kernel
 endif
 
 ifdef SLURM
