Index: src/util/util_getppn.c
===================================================================
--- src/util/util_getppn.c	(revision 27443)
+++ src/util/util_getppn.c	(working copy)
@@ -32,7 +33,9 @@
 void FATR util_getppn_(Integer *ppn_out){
 
 #if defined(__bgq__)
-    *ppn_out = Kernel_ProcessCount();
+  *ppn_out = (Integer) Kernel_ProcessCount();
+  return;
+  if(0) {
 #elif MPI_VERSION >= 3
 
     int err;
