Index: src/64to32blas/xgesvd.F
===================================================================
--- src/64to32blas/xgesvd.F	(revision 0)
+++ src/64to32blas/xgesvd.F	(revision 28050)
@@ -0,0 +1,25 @@
+      SUBROUTINE XGESVD( JOBU, JOBVT, M, N, A, LDA, S, U, LDU, 
+     $                   VT, LDVT,  WORK, LWORK, RWORK, INFO )
+* $Id: ygesvd.F 19697 2010-10-29 16:57:34Z d3y133 $
+      implicit none
+#include "y64.fh"
+      CHARACTER          JOBU, JOBVT
+      INTEGER            INFO, LDA, LDU, LDVT, LWORK, M, N
+      DOUBLE PRECISION   A( LDA, * ), S( * ), U( LDU, * ),
+     $                   VT( LDVT, * ), WORK( * ), RWORK(*)
+c
+      INTGR4            INFO4, LDA4, LDU4, LDVT4, LWORK4, M4, N4
+c
+      lda4=lda
+      ldu4=ldu
+      ldvt4=ldvt
+      m4=m
+      n4=n
+      lwork4=lwork
+c
+      call ZGESVD( JOBU, JOBVT, M4, N4, A, LDA4, S, U, LDU4, 
+     $                   VT, LDVT4,  WORK, LWORK4, RWORK, INFO4 )
+      info=info4
+
+      RETURN
+      END
Index: src/64to32blas/GNUmakefile
===================================================================
--- src/64to32blas/GNUmakefile	(revision 28049)
+++ src/64to32blas/GNUmakefile	(revision 28050)
@@ -10,7 +10,7 @@
                  ypotri.o ypotrf.o ysygv.o ygeev.o ygeevx.o \
                  ifily.o\
                  xscal.o xaxpy.o xgemm.o xheev.o xcopy.o xdotc.o \
-                 ixamax.o
+                 ixamax.o xgesvd.o
 
         ifeq ($(BLAS_SIZE),8)
 	LIB_DEFINES += -DUSE_INTEGER8
Index: src/config/data.64_to_32
===================================================================
--- src/config/data.64_to_32	(revision 28049)
+++ src/config/data.64_to_32	(revision 28050)
@@ -50,6 +50,7 @@
 zdotc      xdotc         
 zdscal     xsscal        
 zgemm      xgemm         
+zgesvd     xgesvd
 zgemv      xgemv         
 zgerc      xgerc         
 zhemm      xhemm         
