--- a/src/interface/makefile
+++ b/src/interface/makefile
@@ -46,10 +46,10 @@ lammps-nnp:
 	tar -xzf $(LAMMPS_VERSION).tar.gz && mv lammps-$(LAMMPS_VERSION) lammps-nnp
 	ln -s $(PROJECT_DIR)/../../ lammps-nnp/lib/nnp
 	cp -r ./LAMMPS/src/USER-NNP lammps-nnp/src/
-	@sed -i.bak "s/^CC .*$$/CC      =       $(PROJECT_MPICC)/" lammps-nnp/src/MAKE/Makefile.mpi
-	@sed -i.bak "s/^CCFLAGS .*$$/CCFLAGS =       $(PROJECT_CFLAGS) $(PROJECT_CFLAGS_MPI)/" lammps-nnp/src/MAKE/Makefile.mpi
-	@sed -i.bak "s/^LINK .*$$/LINK =          $(PROJECT_MPICC)/" lammps-nnp/src/MAKE/Makefile.mpi
-	@sed -i.bak "s/^LINKFLAGS .*$$/LINKFLAGS =     $(PROJECT_CFLAGS) $(PROJECT_CFLAGS_MPI)/" lammps-nnp/src/MAKE/Makefile.mpi
+	@sed -i.bak "s|^CC .*$$|CC      =       $(PROJECT_MPICC)|" lammps-nnp/src/MAKE/Makefile.mpi
+	@sed -i.bak "s|^CCFLAGS .*$$|CCFLAGS =       $(PROJECT_CFLAGS) $(PROJECT_CFLAGS_MPI)|" lammps-nnp/src/MAKE/Makefile.mpi
+	@sed -i.bak "s|^LINK .*$$|LINK =          $(PROJECT_MPICC)|" lammps-nnp/src/MAKE/Makefile.mpi
+	@sed -i.bak "s|^LINKFLAGS .*$$|LINKFLAGS =     $(PROJECT_CFLAGS) $(PROJECT_CFLAGS_MPI)|" lammps-nnp/src/MAKE/Makefile.mpi
 	if [ "$(MODE)" = "test" ]; then \
 		sed -i.bak "/^CCFLAGS =/   s/$$/ $(LAMMPS_DEBUG) $(PROJECT_TEST)/" lammps-nnp/src/MAKE/Makefile.mpi; \
 		sed -i.bak "/^LINKFLAGS =/ s/$$/ $(LAMMPS_DEBUG) $(PROJECT_TEST)/" lammps-nnp/src/MAKE/Makefile.mpi; \
