diff --git a/src/mpi/msmpi/dll/msmpi.vcxproj b/src/mpi/msmpi/dll/msmpi.vcxproj
index 255b9f5..9eb7dff 100644
--- a/src/mpi/msmpi/dll/msmpi.vcxproj
+++ b/src/mpi/msmpi/dll/msmpi.vcxproj
@@ -45,6 +45,7 @@
         $(PUBLIC_SDK_LIB)\Iphlpapi.lib;
         $(PUBLIC_SDK_LIB)\authz.lib;
         $(PUBLIC_SDK_LIB)\ntdsapi.lib;
+        $(ND_DIR)\ndutil.lib;
         $(OutDir)\..\mpichannels\mpichannels.lib;
         $(OutDir)\..\mpiio\mpiio.lib;
         $(OutDir)\..\mpicommon\mpicommon.lib;
@@ -57,6 +58,10 @@
         $(OutDir)\..\mpi_debugger\mpi_debugger.lib;
         $(CRT_Libs);
       </AdditionalDependencies>
+      <AdditionalLibraryDirectories>
+        $(SPACK_IFORT)\lib;
+        $(ND_DIR);
+      </AdditionalLibraryDirectories>
       <ModuleDefinitionFile>.\msmpi.def</ModuleDefinitionFile>
     </Link>
   </ItemDefinitionGroup>
diff --git a/src/mpi/msmpi/fortran/lib/mpifort.vcxproj b/src/mpi/msmpi/fortran/lib/mpifort.vcxproj
index 24bd29d..319153d 100644
--- a/src/mpi/msmpi/fortran/lib/mpifort.vcxproj
+++ b/src/mpi/msmpi/fortran/lib/mpifort.vcxproj
@@ -8,12 +8,12 @@
   </PropertyGroup>
   <Target Name="CompileFortran" AfterTargets="ClCompile" Inputs="@(ForCompile)" Outputs="@(ForCompile->'$(O)%(FileName).obj')">
     <PropertyGroup Condition="'$(BuildArchitecture)'=='i386'">
-      <Fort_Flags>-fno-underscoring -D_X86_=1 -Di386=1 -march=x86-64 -m32</Fort_Flags>
+      <Fort_Flags>/D _X86_=1 /D i386=1 -march=x86-64 -m32</Fort_Flags>
     </PropertyGroup>
     <PropertyGroup Condition="'$(BuildArchitecture)'=='amd64'">
-      <Fort_Flags>-fno-underscoring -D_WIN64 -D_AMD64_ -DAMD64</Fort_Flags>
+      <Fort_Flags>/D _WIN64=1 /D _AMD64_=1 /D AMD64=1</Fort_Flags>
     </PropertyGroup>
-    <Exec Command="$(GFORTRAN_BIN)\gfortran.exe -I$(MPI_INC_ROOT) -c %(ForCompile.Identity) $(Fort_Flags)  -o $(O)\%(ForCompile.FileName).obj" />
+    <Exec Command="$(IFORT_BIN)\ifx.exe /I$(MPI_INC_ROOT) /c %(ForCompile.Identity) $(Fort_Flags) /names:lowercase /assume:nounderscore /o $(O)\%(ForCompile.FileName).obj" />
     <ItemGroup>
       <Lib Condition="'$(ConfigurationType)'=='StaticLibrary'" Include="@(ForCompile->'$(O)\%(Filename).obj')" />
     </ItemGroup>
