diff --git a/client/src/pmpi_wrappers.c b/client/src/pmpi_wrappers.c
index ac86d39..38755f5 100644
--- a/client/src/pmpi_wrappers.c
+++ b/client/src/pmpi_wrappers.c
@@ -14,7 +14,6 @@
 
 #include "pmpi_wrappers.h"
 #include "unifyfs.h"
-#include <mpi.h>
 #include <stdio.h>
 
 int unifyfs_mpi_init(int* argc, char*** argv)
diff --git a/client/src/pmpi_wrappers.h b/client/src/pmpi_wrappers.h
index b90057c..995e44b 100644
--- a/client/src/pmpi_wrappers.h
+++ b/client/src/pmpi_wrappers.h
@@ -15,6 +15,8 @@
 #ifndef UNIFYFS_PMPI_WRAPPERS_H
 #define UNIFYFS_PMPI_WRAPPERS_H
 
+#include <mpi.h>
+
 /* MPI_Init PMPI wrapper */
 int unifyfs_mpi_init(int* argc, char*** argv);
 int MPI_Init(int* argc, char*** argv);
