--- spack-src/client/src/unifyfs-sysio.c.bak	2020-12-03 09:04:41.197202000 +0900
+++ spack-src/client/src/unifyfs-sysio.c	2021-03-04 10:06:32.626989637 +0900
@@ -245,7 +245,7 @@
          * that is big enough */
         buf = (char*) malloc(len);
         if (buf != NULL) {
-            strncpy(buf, unifyfs_cwd, len);
+            memcpy(buf, unifyfs_cwd, strlen(unifyfs_cwd));
         } else {
             errno = ENOMEM;
         }
