diff --git a/sysdep/Makedef.FUGAKU b/sysdep/Makedef.FUGAKU
index 37d41388a..c08707165 100644
--- a/sysdep/Makedef.FUGAKU
+++ b/sysdep/Makedef.FUGAKU
@@ -7,8 +7,8 @@
 ##### Fortran setting
 #FFLAGS is set in Mkinclude. FFLAGS_DEBUG is used if SCALE_DEBUG=T
 
-FC  = mpifrtpx
-SFC = frtpx
+FC  = mpifrt
+SFC = frt
 MODDIROPT = -M
 
 FFLAGS_FAST  = -Kfast,parallel,ocl,preex,array_private,noalias=s,mfunc=2 \
@@ -34,9 +34,9 @@ FFLAGS_DEBUG = -O0                                                       \
 ##### C setting
 #CFLAGS is set in Mkinclude. CFLAGS_DEBUG is used if SCALE_DEBUG=T
 
-CC = mpifccpx
+CC = mpifcc
 
-CFLAGS_FAST  = -Kfast,parallel,ocl,preex,array_private,region_extension,restp=all -Ksimd
+CFLAGS_FAST  = -Kfast,ocl,preex -Ksimd
 CFLAGS_DEBUG = -O0
 
 
diff --git a/sysdep/Makedef.FX100 b/sysdep/Makedef.FX100
index 30077bd21..a26d29928 100644
--- a/sysdep/Makedef.FX100
+++ b/sysdep/Makedef.FX100
@@ -7,8 +7,8 @@
 ##### Fortran setting
 #FFLAGS is set in Mkinclude. FFLAGS_DEBUG is used if SCALE_DEBUG=T
 
-FC  = mpifrtpx
-SFC = frtpx
+FC  = mpifrt
+SFC = frt
 MODDIROPT = -M
 
 FFLAGS_FAST  = -Kfast,parallel,ocl,preex,array_private,noalias=s,mfunc=2 \
@@ -35,7 +35,7 @@ FFLAGS_DEBUG = -O0                                                       \
 ##### C setting
 #CFLAGS is set in Mkinclude. CFLAGS_DEBUG is used if SCALE_DEBUG=T
 
-CC = mpifccpx
+CC = mpifcc
 
 CFLAGS_FAST  = -Kfast,parallel,ocl,preex,array_private,region_extension,restp=all -Ksimd
 CFLAGS_DEBUG = -O0
