diff --git a/GaudiExamples/CMakeLists.txt b/GaudiExamples/CMakeLists.txt
index ef6f9fcff..672b76d26 100644
--- a/GaudiExamples/CMakeLists.txt
+++ b/GaudiExamples/CMakeLists.txt
@@ -0,0 +1,7 @@
+# GaudiExamples subdirectory
+if(NOT BUILD_TESTING)
+  # Ignore examples if not building tests
+  # see https://gitlab.cern.ch/atlas/atlasexternals/-/merge_requests/664#note_3395313
+  return()
+endif()
+

