diff --git a/cmake/GaudiDependencies.cmake b/cmake/GaudiDependencies.cmake
index 7fc224cef..0323f8e5c 100644
--- a/cmake/GaudiDependencies.cmake
+++ b/cmake/GaudiDependencies.cmake
@@ -97,7 +97,7 @@ foreach(component IN ITEMS system filesystem regex thread python unit_test_frame
 endforeach()
 
 set(Python_FIND_STRATEGY VERSION) # Find latest version available
-find_package(Python 2.7.15 ${__quiet} COMPONENTS Interpreter Development)
+find_package(Python ${__quiet} COMPONENTS Interpreter Development)
 set_package_properties(Python PROPERTIES TYPE REQUIRED)
 
 find_package(ROOT 6.18 ${__quiet} CONFIG COMPONENTS Core RIO Hist Thread Matrix
`
