--- a/scripts/bootstrap/bootstrap.sh	1980-01-01 00:00:00.000000000 +0100
+++ b/scripts/bootstrap/bootstrap.sh	2022-12-16 15:01:07.619295000 +0100
@@ -33,7 +33,7 @@
 
 _BAZEL_ARGS="--spawn_strategy=standalone \
       --nojava_header_compilation \
-      --strategy=Javac=worker --worker_quit_after_build --ignore_unsupported_sandboxing \
+      --strategy=Javac=local --ignore_unsupported_sandboxing \
       --compilation_mode=opt \
       --distdir=derived/distdir \
       --extra_toolchains=//scripts/bootstrap:bootstrap_toolchain_definition \
