diff --git a/src/lib_proj.cmake b/src/lib_proj.cmake
index cf30734e5779b102dac3e72ac13962687f6e376e..b2f42c13bc43b6cf5475965c2ce0077daefc47e2 100644
--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -420,7 +420,7 @@ target_link_libraries(${PROJ_CORE_TARGET} ${SQLITE3_LIBRARY})
 if(TIFF_ENABLED)
   target_compile_definitions(${PROJ_CORE_TARGET} PRIVATE -DTIFF_ENABLED)
   target_include_directories(${PROJ_CORE_TARGET} PRIVATE ${TIFF_INCLUDE_DIR})
-  target_link_libraries(${PROJ_CORE_TARGET} ${TIFF_LIBRARY})
+  target_link_libraries(${PROJ_CORE_TARGET} ${TIFF_LIBRARIES})
 endif()
 
 if(CURL_ENABLED)
