diff --git a/src/lib_proj.cmake b/src/lib_proj.cmake
index f2de1c10..a5697608 100644
--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -405,7 +405,7 @@ endif()
 if(TIFF_ENABLED)
   target_compile_definitions(proj PRIVATE -DTIFF_ENABLED)
   target_include_directories(proj PRIVATE ${TIFF_INCLUDE_DIR})
-  target_link_libraries(proj PRIVATE ${TIFF_LIBRARY})
+  target_link_libraries(proj PRIVATE ${TIFF_LIBRARIES})
 endif()
 
 if(CURL_ENABLED)
