--- spack-src/src/lib_proj.cmake.org	2024-02-20 15:43:59.243168600 -0500
+++ spack-src/src/lib_proj.cmake	2024-02-20 15:44:22.114124000 -0500
@@ -403,8 +403,7 @@
 
 if(TIFF_ENABLED)
   target_compile_definitions(proj PRIVATE -DTIFF_ENABLED)
-  target_include_directories(proj PRIVATE ${TIFF_INCLUDE_DIR})
-  target_link_libraries(proj PRIVATE ${TIFF_LIBRARY})
+  target_link_libraries(proj PRIVATE TIFF::tiff)
 endif()
 
 if(CURL_ENABLED)
