--- spack-src/src/CMakeLists.txt.org	2016-09-01 06:46:51.000000000 +0900
+++ spack-src/src/CMakeLists.txt	2020-11-12 11:13:01.345096722 +0900
@@ -220,8 +220,8 @@
 	endif( )
 
 	if( BUILD64 )
-		set( CMAKE_CXX_FLAGS "-m64 ${CMAKE_CXX_FLAGS}" )
-		set( CMAKE_C_FLAGS "-m64 ${CMAKE_C_FLAGS}" )
+		set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" )
+		set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS}" )
 	else( )
 		set( CMAKE_CXX_FLAGS "-m32 ${CMAKE_CXX_FLAGS}" )
 		set( CMAKE_C_FLAGS "-m32 ${CMAKE_C_FLAGS}" )
