diff --git a/src/control/lib/spdk/include/nvme_control_common.h b/src/control/lib/spdk/include/nvme_control_common.h
index 69bccddd4..de83689c9 100644
--- a/src/control/lib/spdk/include/nvme_control_common.h
+++ b/src/control/lib/spdk/include/nvme_control_common.h
@@ -122,7 +122,7 @@ struct dev_health_entry {
 	int					 inflight;
 };
 
-struct ctrlr_entry	*g_controllers;
+extern struct ctrlr_entry	*g_controllers;
 
 /**
  * Attach call back function to report a device that has been
diff --git a/src/control/lib/spdk/src/nvme_control_common.c b/src/control/lib/spdk/src/nvme_control_common.c
index 330b06f56..020a1b7e1 100644
--- a/src/control/lib/spdk/src/nvme_control_common.c
+++ b/src/control/lib/spdk/src/nvme_control_common.c
@@ -27,6 +27,8 @@
 
 #include "nvme_control_common.h"
 
+struct ctrlr_entry       *g_controllers;
+
 static bool
 probe_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
 	 struct spdk_nvme_ctrlr_opts *opts)
diff --git a/src/vos/vos_common.c b/src/vos/vos_common.c
index da32ab408..cf2e1adad 100644
--- a/src/vos/vos_common.c
+++ b/src/vos/vos_common.c
@@ -35,6 +35,8 @@
 #include <daos/lru.h>
 #include <daos/btree_class.h>
 
+struct bio_xs_context            *vsa_xsctxt_inst;
+
 static pthread_mutex_t	mutex = PTHREAD_MUTEX_INITIALIZER;
 
 static bool vsa_nvme_init;
diff --git a/src/vos/vos_internal.h b/src/vos/vos_internal.h
index 1585c398c..c3a871079 100644
--- a/src/vos/vos_internal.h
+++ b/src/vos/vos_internal.h
@@ -212,7 +212,7 @@ struct vos_imem_strts {
 	struct d_hash_table	*vis_cont_hhash;
 };
 /* in-memory structures standalone instance */
-struct bio_xs_context		*vsa_xsctxt_inst;
+extern struct bio_xs_context		*vsa_xsctxt_inst;
 extern int vos_evt_feats;
 
 static inline struct bio_xs_context *
