diff --git a/SConstruct b/SConstruct
index 79e9af6e3..7ce1a60df 100644
--- a/SConstruct
+++ b/SConstruct
@@ -91,7 +91,7 @@ def preload_prereqs(prereqs):
     prereqs.define('readline', libs=['readline', 'history'],
                    package='readline')
     reqs = ['cart', 'argobots', 'pmdk', 'cmocka', 'ofi', 'hwloc',
-            'uuid', 'crypto', 'fuse', 'protobufc']
+            'uuid', 'crypto', 'fuse', 'protobufc', 'ompi', 'go']
     if not is_platform_arm():
         reqs.extend(['spdk', 'isal'])
     prereqs.load_definitions(prebuild=reqs)
diff --git a/src/control/SConscript b/src/control/SConscript
index 5b75a9c04..0514eeb4a 100644
--- a/src/control/SConscript
+++ b/src/control/SConscript
@@ -66,9 +66,13 @@ def install_go_bin(denv, gosrc, gopath, libs, name, install_name):
     src = [mod_src]
     if libs is not None:
         src.extend(libs)
+    if denv.subst("$GO_PREFIX"):
+        gobin = join(denv.subst("$GO_PREFIX"), "bin", "go")
+    else:
+        gobin = "go"
     denv.Command(bin_path, src,
                  cmd_build(path, gosrc,
-                           "go install %s" % install_src))
+                           "%s install %s" % (gobin, install_src)))
     denv.InstallAs(install_bin, bin_path)
 #pylint: enable=too-many-arguments

@@ -95,7 +99,10 @@ def scons():
     def check_go_version(context):
         """Check GO Version"""
         context.Display('Checking for Go compiler in $PATH... ')
-        go_path = context.env.WhereIs(GO_COMPILER)
+        if denv.subst("$GO_PREFIX"):
+            go_path = join(denv.subst("$GO_PREFIX"), "bin", "go")
+        else:
+            go_path = context.env.WhereIs(GO_COMPILER)
         if go_path:
             context.Display(go_path + '\n')
         else:
@@ -149,7 +156,7 @@ def scons():
     AlwaysBuild([agentbin, dmgbin, adminbin, drpcbin])

     senv = denv.Clone()
-    prereqs.require(senv, 'pmdk', 'spdk', 'ofi', 'hwloc')
+    prereqs.require(senv, 'pmdk', 'spdk', 'ofi', 'hwloc', 'go')

     gospdkpath = join(gosrc, "lib", "spdk")
 
diff --git a/scons_local/components/__init__.py b/scons_local/components/__init__.py
index b85de4c..2877366 100644
--- a/scons_local/components/__init__.py
+++ b/scons_local/components/__init__.py
@@ -143,6 +143,8 @@ def define_common(reqs):
     reqs.define('crypto', libs=['crypto'], headers=['openssl/md5.h'],
                 package='openssl-devel')
 
+    reqs.define('go', package='golang')
+
     if reqs.get_env('PLATFORM') == 'darwin':
         reqs.define('uuid', headers=['uuid/uuid.h'])
     else:
