diff --git a/site_scons/components/__init__.py b/site_scons/components/__init__.py
index 72f1a1473f..0be6b721ad 100644
--- a/site_scons/components/__init__.py
+++ b/site_scons/components/__init__.py
@@ -325,7 +325,7 @@ def define_components(reqs):
                           '"$SPDK_PREFIX/include/dpdk"',
                           'mkdir -p "$SPDK_PREFIX/share/spdk"',
                           'cp -r include scripts "$SPDK_PREFIX/share/spdk"'],
-                headers=['spdk/nvme.h', 'dpdk/rte_eal.h'],
+                headers=['spdk/nvme.h'],
                 extra_include_path=['/usr/include/dpdk',
                                     '$SPDK_PREFIX/include/dpdk',
                                     # debian dpdk rpm puts rte_config.h here
