From 7263c8f796297281aa433a16e8f34600107b5a6e Mon Sep 17 00:00:00 2001
From: Ben Boeckel <ben.boeckel@kitware.com>
Date: Sat, 7 Dec 2019 14:30:44 -0500
Subject: [PATCH] vtkWrapPythonMethod: avoid more clang-tidy warnings

---
 Wrapping/Tools/vtkWrapPythonMethod.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Wrapping/Tools/vtkWrapPythonMethod.c b/Wrapping/Tools/vtkWrapPythonMethod.c
index 769ed0699c1..2b02defc20d 100644
--- a/Wrapping/Tools/vtkWrapPythonMethod.c
+++ b/Wrapping/Tools/vtkWrapPythonMethod.c
@@ -603,7 +603,7 @@ void vtkWrapPython_ReturnValue(FILE* fp, ClassInfo* data, ValueInfo* val, int st
       fprintf(fp,
         "      if (result && PyVTKObject_Check(result))\n"
         "      {\n"
-        "        PyVTKObject_GetObject(result)->UnRegister(0);\n"
+        "        PyVTKObject_GetObject(result)->UnRegister(nullptr);\n"
         "        PyVTKObject_SetFlag(result, VTK_PYTHON_IGNORE_UNREGISTER, 1);\n"
         "      }\n");
     }
@@ -632,7 +632,7 @@ void vtkWrapPython_ReturnValue(FILE* fp, ClassInfo* data, ValueInfo* val, int st
   }
   else if (vtkWrap_IsArray(val))
   {
-    fprintf(fp, "      result = %sBuildTuple(tempr, sizer);\n", prefix);
+    fprintf(fp, "      result = vtkPythonArgs::BuildTuple(tempr, sizer);\n");
   }
   else if (vtkWrap_IsStdVector(val))
   {
-- 
GitLab

