--- spack-src/Modules/FortranCInterface/Verify/CMakeLists.txt.org	2020-06-05 15:54:59.559043595 +0900
+++ spack-src/Modules/FortranCInterface/Verify/CMakeLists.txt	2020-06-05 15:58:28.150062948 +0900
@@ -4,6 +4,7 @@
 cmake_minimum_required(VERSION ${CMAKE_VERSION})
 project(VerifyFortranC C Fortran)
 
+set (CMAKE_EXE_LINKER_FLAGS "--linkfortran")
 option(VERIFY_CXX "Whether to verify C++ and Fortran" OFF)
 if(VERIFY_CXX)
   enable_language(CXX)
