--- a/Makefile	2021-03-01 08:17:01.000000000 -0600
+++ b/Makefile	2021-04-11 15:20:15.240729080 -0500
@@ -315,7 +315,7 @@
 ifeq ($(BUILDTYPE), MacOSX)
 CUDA_L := -L$(CUDA_BASE)/lib -lcufft -lcudart -lcublas -m64 -lstdc++
 else
-CUDA_L := -L$(CUDA_BASE)/lib -lcufft -lcudart -lcublas -lstdc++ -Wl,-rpath $(CUDA_BASE)/lib
+CUDA_L := -L$(CUDA_BASE)/lib -lcufft -lcudart -lcublas -lstdc++
 endif
 else
 CUDA_H :=
@@ -355,14 +355,13 @@
 CPPFLAGS += -DUSE_ACML
 else
 BLAS_H := -I$(BLAS_BASE)/include
-ifeq ($(BUILDTYPE), MacOSX)
-BLAS_L := -L$(BLAS_BASE)/lib -lopenblas
+ifeq ($(OPENBLAS),1)
+BLAS_L := -lopenblas
 else
 ifeq ($(NOLAPACKE),1)
-BLAS_L := -L$(BLAS_BASE)/lib -llapack -lblas
-CPPFLAGS += -Isrc/lapacke
+BLAS_L := -llapack -lcblas
 else
-BLAS_L := -L$(BLAS_BASE)/lib -llapacke -lblas
+BLAS_L := -llapacke -lcblas
 endif
 endif
 endif
