--- a/src/htslib-1.7/Makefile.Rhtslib	2020-02-08 17:12:30.000000000 -0600
+++ b/src/htslib-1.7/Makefile.Rhtslib	2020-02-08 19:08:45.969675431 -0600
@@ -37,13 +37,13 @@
 # Default libraries to link if configure is not used
 htslib_default_libs = -lz -lm -lbz2 -llzma
 
-CPPFLAGS =
+CPPFLAGS += $(BZIP2_INCLUDE) $(XZ_INCLUDE)
 # TODO: probably update cram code to make it compile cleanly with -Wc++-compat
 # For testing strict C99 support add -std=c99 -D_XOPEN_SOURCE=600
 #CFLAGS   = -g -Wall -O2 -pedantic -std=c99 -D_XOPEN_SOURCE=600 -D__FUNCTION__=__func__
-CFLAGS   = -g -Wall -O2
+CFLAGS   += -g -Wall -O2
 EXTRA_CFLAGS_PIC = -fpic
-LDFLAGS  =
+LDFLAGS  += $(BZIP2_LIB) $(XZ_LIB)
 LIBS     = $(htslib_default_libs)
 
 prefix      = /usr/local
