diff -Naur papi-7.0.1-orig/src/components/sde/tests/Makefile papi-7.0.1/src/components/sde/tests/Makefile
--- papi-7.0.1-orig/src/components/sde/tests/Makefile	2023-06-19 14:27:44.224943877 -0400
+++ papi-7.0.1/src/components/sde/tests/Makefile	2023-06-19 14:46:18.462536297 -0400
@@ -1,11 +1,13 @@
 NAME=sde
 include ../../Makefile_comp_tests.target
 INCLUDE += -I$(datadir)/sde_lib -I..
+
+intel_compilers := ifort ifx
 ifeq ($(notdir $(F77)),gfortran)
 	FFLAGS +=-ffree-form -ffree-line-length-none
 else ifeq ($(notdir $(F77)),flang)
 	FFLAGS +=-ffree-form
-else
+else ifneq ($(findstring $(notdir $(F77)),$(intel_compilers)),)
 	FFLAGS +=-free
 endif
 FFLAGS +=-g
diff -Naur papi-7.0.1-orig/src/components/sysdetect/tests/Makefile papi-7.0.1/src/components/sysdetect/tests/Makefile
--- papi-7.0.1-orig/src/components/sysdetect/tests/Makefile	2023-06-19 14:27:44.220943888 -0400
+++ papi-7.0.1/src/components/sysdetect/tests/Makefile	2023-06-19 14:46:20.678540592 -0400
@@ -19,11 +19,12 @@
     FTESTS =
 endif
 
+intel_compilers := ifort ifx
 ifeq ($(notdir $(F77)),gfortran)
     FFLAGS +=-ffree-form -ffree-line-length-none
 else ifeq ($(notdir $(F77)),flang)
     FFLAGS +=-ffree-form
-else
+else ifneq ($(findstring $(notdir $(F77)),$(intel_compilers)),)
     FFLAGS +=-free
 endif
 
