--- a/src/build/env.py	2023-01-11 14:09:08.000000000 -0600
+++ b/src/build/env.py	2023-02-03 12:04:04.000000000 -0600
@@ -269,6 +269,7 @@
         with warnings.catch_warnings():
             if sys.version_info[:3] == (3, 11, 0):
                 warnings.filterwarnings('ignore', 'check_home argument is deprecated and ignored.', DeprecationWarning)
+            os.environ.pop('PYTHONPATH', None)
             venv.EnvBuilder(with_pip=True, symlinks=symlinks).create(path)
     except subprocess.CalledProcessError as exc:
         raise build.FailedProcessError(exc, 'Failed to create venv. Maybe try installing virtualenv.') from None
