--- /usr/workspace/wsrzc/pozulp1/sp/pozulpspack/spack/var/spack/stage/hdf5-1.8.10-q7gnahmiajrcbk6lyp3fwaixcofqjhwf/hdf5-1.8.10/tools/h5dump/h5dump_ddl.c	2012-10-11 10:31:22.000000000 -0700
+++ hdf5-1.8.10/tools/h5dump/h5dump_ddl.c	2018-03-02 18:12:28.896508953 -0800
@@ -1341,8 +1341,8 @@
     string_dataformat.do_escape = display_escape;
     outputformat = &string_dataformat;
 
-    //attr_name = attr + j + 1;
-	// need to replace escape characters
+    /* attr_name = attr + j + 1; */
+	/* need to replace escape characters */
 	attr_name = h5tools_str_replace(attr + j + 1, "\\/", "/");
 
 
