--- /usr/workspace/wsrzc/pozulp1/sp/pozulpspack/spack/var/spack/stage/hdf5-1.8.10-q7gnahmiajrcbk6lyp3fwaixcofqjhwf/hdf5-1.8.10/test/th5s.c	2012-10-11 10:30:44.000000000 -0700
+++ hdf5-1.8.10p1/test/th5s.c	2013-02-14 11:04:26.000000000 -0800
@@ -730,8 +730,8 @@
         ret = H5Pset_chunk(plist_id, SPACE1_RANK, chunk_dims);
         CHECK(ret, FAIL, "H5Pset_chunk");
 
-        // ret = H5Pset_alloc_time(plist_id, alloc_time);
-        // CHECK(ret, FAIL, "H5Pset_alloc_time");
+        /* ret = H5Pset_alloc_time(plist_id, alloc_time); */
+        /* CHECK(ret, FAIL, "H5Pset_alloc_time"); */
 
         dset1 = H5Dcreate2(fid1, BASICDATASET1, H5T_NATIVE_INT, sid_chunk, H5P_DEFAULT, plist_id, H5P_DEFAULT);
         CHECK(dset1, FAIL, "H5Dcreate2");
