From 4f324ddc12f358de47ea0ef1872c61b1dc4c46e2 Mon Sep 17 00:00:00 2001
From: Harmen Stoppels <harmenstoppels@gmail.com>
Date: Mon, 10 Aug 2020 15:00:06 +0200
Subject: [PATCH] Fix detection of HIP_CLANG_ROOT

---
 hip-config.cmake.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/hip-config.cmake.in b/hip-config.cmake.in
index 859e2fa0..4baafbde 100644
--- a/hip-config.cmake.in
+++ b/hip-config.cmake.in
@@ -58,8 +58,9 @@ if(HIP_COMPILER STREQUAL "clang")
     execute_process(COMMAND ${HIP_CXX_COMPILER} --version
                     OUTPUT_STRIP_TRAILING_WHITESPACE
                     OUTPUT_VARIABLE HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT)
-    if(HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT MATCHES "InstalledDir:[\t\r\n][\t\r\n]*([^\t\r\n])")
-      set(HIP_CLANG_ROOT ${CMAKE_MATCH_1})
+    if(HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT MATCHES "InstalledDir: ([^\t\r\n]+)")
+      # InstalledDir points to the bin/ directory, so the root is the parent dir
+      get_filename_component(HIP_CLANG_ROOT ${CMAKE_MATCH_1} DIRECTORY)
     else()
       set(HIP_CLANG_ROOT /opt/rocm/llvm)
     endif()
-- 
2.25.1

