diff --git a/CMakeLists.txt b/CMakeLists.txt
index f97fb83..a059c42 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,7 +72,17 @@ list(GET VERSION_LIST 0 HIP_VERSION_MAJOR)
 list(GET VERSION_LIST 1 HIP_VERSION_MINOR)
 set(HIP_VERSION_GITDATE 0)
 
-find_package(Git)
+# only look for git when we have a git repo
+if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
+  find_package(Git)
+endif()
+
+# FIXME: Two different version strings used.
+
+set(HIP_PACKAGING_VERSION_PATCH "0")
+set(HIP_VERSION_GITDATE "0")
+set(HIP_VERSION_PATCH "0")
+
 
 # FIXME: Two different version strings used.
 if(GIT_FOUND)
@@ -118,9 +128,6 @@ if(GIT_FOUND)
   else()
     set(HIP_PACKAGING_VERSION_PATCH ${HIP_VERSION_GITDATE}.${HIP_VERSION_GITCOUNT}-${HIP_VERSION_GITHASH})
   endif()
-else()
-  # FIXME: Some parts depend on this being set.
-  set(HIP_PACKAGING_VERSION_PATCH "0")
 endif()
 
 ## Debian package specific variables
diff --git a/hip-config.cmake.in b/hip-config.cmake.in
index cbb4ffe..94e2ec1 100644
--- a/hip-config.cmake.in
+++ b/hip-config.cmake.in
@@ -265,9 +265,6 @@ if(HIP_COMPILER STREQUAL "clang")
   endif()
 
   # Add support for __fp16 and _Float16, explicitly link with compiler-rt
-  set_property(TARGET hip::host APPEND PROPERTY
-    INTERFACE_LINK_LIBRARIES "-L\"${HIP_CLANG_INCLUDE_PATH}/../lib/linux\" -lclang_rt.builtins-x86_64"
-  )
   set_property(TARGET hip::device APPEND PROPERTY
     INTERFACE_LINK_LIBRARIES "-L\"${HIP_CLANG_INCLUDE_PATH}/../lib/linux\" -lclang_rt.builtins-x86_64"
   )
