--- a/bld.cfg	2017-12-23 11:41:14.000000000 +1100
+++ b/bld.cfg	2017-12-23 15:34:53.914024629 +1100
@@ -24,9 +24,6 @@
 
 search_src           true
 src::zzz .
-src::date $PWD/extern/boost/src/date_time
-src::blitz $PWD/extern/blitz/src
-src::netcdf $PWD/extern/netcdf4
 bld::lib xios
 bld::target libxios.a
 #bld::target generate_fortran_interface.exe 
@@ -37,11 +34,11 @@
 bld::tool::fc        %FCOMPILER
 bld::tool::fpp       %FPP
 bld::tool::cpp       %CPP
-bld::tool::cppflags  %CBASE_INC -I${PWD}/extern/src_netcdf -I${PWD}/extern/boost/include -I${PWD}/extern/rapidxml/include -I${PWD}/extern/blitz/include 
-bld::tool::fppflags  %BASE_INC -I${PWD}/extern/boost/include -I${PWD}/extern/rapidxml/include 
+bld::tool::cppflags  %CBASE_INC -I${PWD}/extern/rapidxml/include
+bld::tool::fppflags  %BASE_INC -I${PWD}/extern/rapidxml/include 
 bld::tool::ld        %LINKER
 bld::tool::ldflags   %LD_FLAGS 
-bld::tool::cflags    %CFLAGS %CBASE_INC -I${PWD}/extern/src_netcdf -I${PWD}/extern/boost/include -I${PWD}/extern/rapidxml/include -I${PWD}/extern/blitz/include 
+bld::tool::cflags    %CFLAGS %CBASE_INC -I${PWD}/extern/rapidxml/include
 bld::tool::fflags    %FFLAGS %FBASE_INC 
 bld::tool::cppkeys   %CPP_KEY
 bld::tool::fppkeys   %CPP_KEY
