--- a/dataflowAPI/h/stackanalysis.h	2016-06-29 14:54:14.000000000 -0700
+++ b/dataflowAPI/h/stackanalysis.h	2016-08-02 09:50:13.619079000 -0700
@@ -331,7 +331,7 @@
 
    // To build intervals, we must replay the effect of each instruction.
    // To avoid sucking enormous time, we keep those transfer functions around...
-   typedef std::map<ParseAPI::Block *, std::map<Offset, TransferFuncs>>
+   typedef std::map<ParseAPI::Block *, std::map<Offset, TransferFuncs> >
       InstructionEffects;
 
    DATAFLOW_EXPORT StackAnalysis();
