--- plugins/D0RunIICone/ProtoJet.hpp	2025-04-10 11:37:59
+++ plugins/D0RunIICone/ProtoJet.hpp	2025-04-10 11:38:18
@@ -195,7 +195,7 @@
 
 template<class Item>
 void ProtoJet<Item>::print(std::ostream& os) const {
-  os<<"y phi Et = ("<<_y<<", "<<_phi<<", "<<this->_Et<<")"<<std::endl;
+  os<<"y phi = ("<<_y<<", "<<_phi<<")"<<std::endl;
   os<< " members= " << std::endl;
   typename std::list<const Item*>::const_iterator i;
   for(i = _LItems.begin(); i != _LItems.end(); ++i)

