diff --git a/Addons/CMakeLists.txt b/Addons/CMakeLists.txt
index 1279ada..0b6d9be 100644
--- a/Addons/CMakeLists.txt
+++ b/Addons/CMakeLists.txt
@@ -49,19 +49,7 @@ endif()
 
 # The Tasmanian MPI capabilities are templated into the Addons
 if (Tasmanian_ENABLE_MPI)
-    target_link_libraries(Tasmanian_addons INTERFACE ${MPI_CXX_LIBRARIES})
-
-    if (DEFINED MPI_CXX_INCLUDE_PATH)
-        target_include_directories(Tasmanian_addons INTERFACE "${MPI_CXX_INCLUDE_PATH}")
-    endif()
-
-    if(DEFINED MPI_CXX_COMPILE_FLAGS)
-        target_compile_options(Tasmanian_addons INTERFACE "${MPI_CXX_COMPILE_FLAGS}")
-    endif()
-
-    if(DEFINED MPI_CXX_LINK_FLAGS)
-        set_target_properties(Tasmanian_addons PROPERTIES INTERFACE_LINK_OPTIONS "${MPI_CXX_LINK_FLAGS}")
-    endif()
+    target_link_libraries(Tasmanian_addons INTERFACE MPI::MPI_CXX)
 
     add_executable(Tasmanian_mpitester testMPI.cpp testMPI.hpp testMPIDream.hpp)
     set_target_properties(Tasmanian_mpitester PROPERTIES OUTPUT_NAME "mpitester")
