diff --git a/Config/TasmanianConfig.in.cmake b/Config/TasmanianConfig.in.cmake
index 8912e4c..df54aaf 100644
--- a/Config/TasmanianConfig.in.cmake
+++ b/Config/TasmanianConfig.in.cmake
@@ -7,6 +7,10 @@ cmake_minimum_required(VERSION 3.10)
 # but this doesn't seem to work, not sure if this is a "relocatable package" (low concern)
 include("@CMAKE_INSTALL_PREFIX@/lib/@CMAKE_PROJECT_NAME@/@CMAKE_PROJECT_NAME@.cmake")
 
+if (@Tasmanian_ENABLE_MPI@)
+    find_package(MPI REQUIRED)
+endif()
+
 add_executable(Tasmanian::tasgrid IMPORTED)
 set_property(TARGET Tasmanian::tasgrid PROPERTY IMPORTED_LOCATION "@CMAKE_INSTALL_PREFIX@/bin/tasgrid${CMAKE_EXECUTABLE_SUFFIX_CXX}")
 
