--- knem/driver/linux/check_kernel_headers.sh   2021-01-19 12:33:28.699210330 +0530
+++ knem_patched/driver/linux/check_kernel_headers.sh   2021-01-19 12:31:41.156216657 +0530
@@ -356,7 +356,7 @@
 # task_struct cpus_allowed replaced with cpus_mask in 5.3
 echo -n "  checking (in kernel headers) whether task_struct contains a cpus_mask field ... "
 if sed -ne '/^struct task_struct {/,/^};/p' ${LINUX_HDR}/include/linux/sched.h \
-  | egrep "cpumask_t[  ]+cpus_mask;" > /dev/null ; then
+  | egrep "cpumask_t[  ]+cpus_mask" > /dev/null ; then
   echo "#define KNEM_HAVE_TASK_CPUS_MASK 1" >> ${TMP_CHECKS_NAME}
   echo yes
 else
