--- a/m4/extern-inline.m4
+++ b/m4/extern-inline.m4
@@ -34,6 +34,7 @@
       ? defined __GNUC_STDC_INLINE__ && __GNUC_STDC_INLINE__ \
       : (199901L <= __STDC_VERSION__ \
          && !defined __HP_cc \
+         && !defined __PGI \
          && !(defined __SUNPRO_C && __STDC__))) \
      && !defined _GL_EXTERN_INLINE_STDHEADER_BUG)
 # define _GL_INLINE inline
--- a/configure.ac
+++ b/configure.ac
@@ -22,7 +22,7 @@
   [bug-m4@gnu.org])
 AC_CONFIG_AUX_DIR([build-aux])
 
-AM_INIT_AUTOMAKE([1.11.6 dist-bzip2 dist-xz color-tests parallel-tests
+AM_INIT_AUTOMAKE([dist-bzip2 dist-xz color-tests parallel-tests
 silent-rules subdir-objects gnu])
 
 m4_pattern_forbid([^M4_[A-Z]])

