--- spack-src/Ipopt/config.guess.org	2019-10-30 13:52:51.920159919 +0900
+++ spack-src/Ipopt/config.guess	2019-10-30 13:55:27.956506585 +0900
@@ -835,6 +835,9 @@
     arm*:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-gnu
 	exit ;;
+    aarch64:Linux:*:*)
+        echo ${UNAME_MACHINE}-unknown-linux-gnu
+        exit ;;
     avr32*:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-gnu
 	exit ;;
