*** ./phasta/phiotimer.cc	2017-05-03 12:26:25.000000000 -0700
--- ./phasta/phiotimer.cc.patched	2017-05-03 12:26:11.000000000 -0700
***************
*** 56,63 ****
  }
  #else
  void phastaio_time(phastaioTime* t) {
!   int err;
!   err = clock_gettime(CLOCK_MONOTONIC,t);
    PCU_ALWAYS_ASSERT(!err);
  }
  /*return elapsed time in micro seconds*/
--- 56,63 ----
  }
  #else
  void phastaio_time(phastaioTime* t) {
!   int err=0;
!   /*err = clock_gettime(CLOCK_MONOTONIC,t);*/
    PCU_ALWAYS_ASSERT(!err);
  }
  /*return elapsed time in micro seconds*/
