--- a/scipy/sparse/linalg/_propack/setup.py	2022-04-11 11:33:02.000000000 -0600
+++ b/scipy/sparse/linalg/_propack/setup.py	2022-04-11 11:34:15.000000000 -0600
@@ -61,6 +61,7 @@
             src += get_g77_abi_wrappers(lapack_opt)
 
         cmacros = [('_OPENMP',)]
+        cmacros += [('__STDC_NO_THREADS__',1)]
         if needs_g77_abi_wrapper(lapack_opt):
             cmacros += [('SCIPY_USE_G77_CDOTC_WRAP', 1)]
 
@@ -73,6 +74,7 @@
                                    libraries=[propack_lib],
                                    extra_info=lapack_opt,
                                    undef_macros=['_OPENMP'],
+                                   define_macros=[('__STDC_NO_THREADS__',1)],
                                    f2py_options=f2py_options,
                                    depends=['setup.py'] + src)
         ext._pre_build_hook = pre_build_hook
