From 7445b29108ea2f0b3f28f947a1ca627575173292 Mon Sep 17 00:00:00 2001
From: Harmen Stoppels <harmenstoppels@gmail.com>
Date: Fri, 7 Aug 2020 13:24:00 +0200
Subject: [PATCH] Remove relative link paths to external libraries

---
 src/runtime/core/CMakeLists.txt | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/runtime/core/CMakeLists.txt b/src/runtime/core/CMakeLists.txt
index 29abffb..f2cd4b4 100644
--- a/src/runtime/core/CMakeLists.txt
+++ b/src/runtime/core/CMakeLists.txt
@@ -109,12 +109,6 @@ target_link_libraries(
   ${amd_comgr_LIBRARIES}
   ${LIBELF_LIBRARIES}
   ${ROCM_LIBRARIES}
-  -L${ROCM_LIBRARIES_DIR}/../hsa/lib
-  -L${ROCM_LIBRARIES_DIR}
-  -Wl,--enable-new-dtags
-  -Wl,-rpath,\$ORIGIN
-  -Wl,-rpath,\$ORIGIN/../../hsa/lib
-  -Wl,-rpath,\$ORIGIN/../../lib
 )
 
 # set output dir for .h files
-- 
2.25.1

