--- a/CMakeLists.txt	2021-06-17 05:48:48.000000000 +0000
+++ b/CMakeLists.txt	2021-06-17 15:50:13.683664538 +0000
@@ -145,7 +145,7 @@
     endif()
 
     if (GPU_TARGET MATCHES Kepler)
-        set( GPU_TARGET "${GPU_TARGET} sm_30 sm_35" )
+        set( GPU_TARGET "${GPU_TARGET} sm_30 sm_35 sm_37" )
     endif()
 
     if (GPU_TARGET MATCHES Maxwell)
@@ -195,6 +195,15 @@
         message( STATUS "    compile for CUDA arch 3.5 (Kepler)" )
     endif()
 
+    if (GPU_TARGET MATCHES sm_37)
+        if (NOT MIN_ARCH)
+            set( MIN_ARCH 300 )
+        endif()
+        set( NV_SM ${NV_SM} -gencode arch=compute_37,code=sm_37 )
+        set( NV_COMP        -gencode arch=compute_37,code=compute_37 )
+        message( STATUS "    compile for CUDA arch 3.7 (Kepler)" )
+    endif()
+
     if (GPU_TARGET MATCHES sm_50)
         if (NOT MIN_ARCH)
             set( MIN_ARCH 500 )
