--- spack-src/CMakeLists.txt.org	2020-03-24 14:01:58.856142450 +0900
+++ spack-src/CMakeLists.txt	2020-03-24 14:01:05.715872685 +0900
@@ -204,7 +204,7 @@
         ${ONNX_DLLEXPORT_STR}${CMAKE_CURRENT_BINARY_DIR})
     if(BUILD_ONNX_PYTHON)
       list(APPEND PROTOC_ARGS --python_out
-                  ${ONNX_DLLEXPORT_STR}${CMAKE_CURRENT_BINARY_DIR})
+                  ${CMAKE_CURRENT_BINARY_DIR})
       if(ONNX_GEN_PB_TYPE_STUBS)
         # Haven't figured out how to generate mypy stubs on Windows yet
         if(NOT WIN32)
