diff -u -r -N a/src/blt/cmake/SetupCompilerOptions.cmake b/src/blt/cmake/SetupCompilerOptions.cmake
--- a/src/blt/cmake/SetupCompilerOptions.cmake	2020-08-26 11:58:24.000000000 +0900
+++ b/src/blt/cmake/SetupCompilerOptions.cmake	2020-08-26 13:31:04.000000000 +0900
@@ -206,7 +206,9 @@
 
 endif()
 
-
+set(CMAKE_Fortran_MODDIR_FLAG -M)
+set(CMAKE_Fortran_FLAGS_RELWITHDEBINFO "-O2 -g -DNDEBUG")
+set(CMAKE_SHARED_LIBRARY_LINK_CXX_FLAGS "--linkfortran")
 
 ################################
 # RPath Settings
