--- a/build_config/Darwin.gfortran.default/ESMC_Conf.h	2019-08-21 10:06:06.000000000 -0600
+++ b/build_config/Darwin.gfortran.default/ESMC_Conf.h	2019-08-21 10:03:41.000000000 -0600
@@ -31,6 +31,7 @@
 // Typedef to match the data type of the 'hidden' string length
 // argument that Fortran uses when passing CHARACTER strings.
 #if (__GNUC__ > 7)
+#include <cstddef>
 typedef size_t ESMCI_FortranStrLenArg;
 #else
 typedef int ESMCI_FortranStrLenArg;
