--- a/src/gaBaseTypes.cc	2008-07-25 14:43:25.000000000 -0600
+++ b/src/gaBaseTypes.cc	2023-09-13 14:59:50.000000000 -0600
@@ -31,6 +31,7 @@
 #include "gaBaseTypes.h"
 #include "gaTypeFactory.h"
 
+
 /** class declarations for gaStr and gaUrl are in gaBaseTypes.h so we
  * can't declare them again, we have to define each function
  * individually
--- a/src/gaBaseTypes.h	2008-07-25 14:43:25.000000000 -0600
+++ b/src/gaBaseTypes.h	2023-09-13 14:44:39.000000000 -0600
@@ -31,6 +31,7 @@
  * 
  */
 
+using namespace libdap;
 
 class gaStr: public Str {
 public:
--- a/src/gaReports.cc	2008-07-25 14:43:25.000000000 -0600
+++ b/src/gaReports.cc	2023-09-13 14:50:28.000000000 -0600
@@ -12,6 +12,7 @@
  */
 
 #include <iostream>
+#include <cstring>
 #include "gaReports.h"
 #include "gaUtils.h"
 #include "Grid.h"
--- a/src/gaTypeFactory.h	2008-07-25 14:43:25.000000000 -0600
+++ b/src/gaTypeFactory.h	2023-09-13 14:44:43.000000000 -0600
@@ -28,6 +28,8 @@
 
 #include <string>
 
+using namespace libdap;
+
 // Class declarations; Make sure to include the corresponding headers in the
 // implementation file.
 
--- a/src/gaUtils.h	2008-07-25 14:43:25.000000000 -0600
+++ b/src/gaUtils.h	2023-09-13 14:46:27.000000000 -0600
@@ -23,6 +23,8 @@
 #include "Sequence.h"
 #include "gadap.h"
 
+using namespace libdap;
+
 typedef struct varinfo {
   string name;
   string longname;
--- a/test/test.cc	2008-07-30 06:35:43.000000000 -0600
+++ b/test/test.cc	2023-09-13 15:12:48.000000000 -0600
@@ -9,6 +9,7 @@
 #include "gadap.h"
 #include <iostream>
 #include <string>
+#include <cstring>
 
 using namespace std;
 
