diff --git a/CMakeLists.txt b/CMakeLists.txt
index 58dcbd4..f0658eb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,7 +43,11 @@ endif()
 
 find_package(OpenMP REQUIRED)
 find_package(MPI)
-find_package(OpenCL 2.0)
+
+option(BRICK_USE_OPENCL "Use OpenCL targets" ON)
+if (BRICK_USE_OPENCL)
+    find_package(OpenCL 2.0)
+endif()
 
 option(BRICK_USE_MEMFD "Using memfd instead of shm_open, supported on Linux >= 3.17 with \"CONFIG_MEMFD_CREATE\"" OFF)
 if (BRICK_USE_MEMFD)
