diff --git a/server/src/Broker.cpp b/server/src/Broker.cpp
index 03d60f4..aab449f 100644
--- a/server/src/Broker.cpp
+++ b/server/src/Broker.cpp
@@ -108,14 +108,14 @@ MetaDataClient* Broker::addDataClient()
 	}
 	boolean isaac_jpeg_fill_input_buffer(j_decompress_ptr cinfo)
 	{
-		return true;
+		return TRUE;
 	}
 	void isaac_jpeg_skip_input_data(j_decompress_ptr cinfo,long num_bytes)
 	{
 	}
 	boolean isaac_jpeg_resync_to_restart(j_decompress_ptr cinfo, int desired)
 	{
-		return true;
+		return TRUE;
 	}
 	void isaac_jpeg_term_source(j_decompress_ptr cinfo)
 	{
diff --git a/server/src/URIImageConnector.cpp b/server/src/URIImageConnector.cpp
index 0b11800..e843aa4 100644
--- a/server/src/URIImageConnector.cpp
+++ b/server/src/URIImageConnector.cpp
@@ -40,7 +40,7 @@ void isaac_init_destination(j_compress_ptr cinfo)
 }
 boolean isaac_jpeg_empty_output_buffer(j_compress_ptr cinfo)
 {
-	return true;
+	return TRUE;
 }
 void isaac_jpeg_term_destination(j_compress_ptr cinfo)
 {
