diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4576e73..c73084c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,27 +5,35 @@ set(CMAKE_MACOSX_RPATH 1)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 set(CMAKE_CXX_STANDARD 17)
+#Path management
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+    set (CMAKE_INSTALL_PREFIX "$ENV{HOME}"
+        CACHE PATH "Default install prefix" FORCE)
+endif()
+
+INCLUDE_DIRECTORIES(   "${CMAKE_INSTALL_PREFIX}/include")
+LINK_DIRECTORIES(      "${CMAKE_INSTALL_PREFIX}/lib")
 
 #Elasticity library
 project(Elas)
 file(                  GLOB_RECURSE source_files sources/*)
 file(                  GLOB_RECURSE header_files sources/*.h)
 ADD_LIBRARY(           Elas SHARED ${source_files})
-find_library(          Commons NAMES Commons HINTS "$ENV{HOME}/lib")
-INCLUDE_DIRECTORIES(  "$ENV{HOME}/include")
-LINK_DIRECTORIES(     "$ENV{HOME}/lib")
+find_library(          Commons NAMES Commons HINTS "${CMAKE_INSTALL_PREFIX}/lib")
 target_link_libraries( Elas ${Commons})
+
 INSTALL(
   TARGETS Elas
-  LIBRARY  DESTINATION "$ENV{HOME}/lib"
-  RUNTIME  DESTINATION "$ENV{HOME}/lib")
-INSTALL(               FILES ${header_files} DESTINATION "$ENV{HOME}/include")
+  LIBRARY  DESTINATION "${CMAKE_INSTALL_PREFIX}/lib"
+  RUNTIME  DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+INSTALL(               FILES ${header_files} DESTINATION "${CMAKE_INSTALL_PREFIX}/include")
 
 #Executable
 project(main)
 add_executable(        elastic sources/elastic.c)
+
 target_link_libraries( elastic ${Commons} Elas)
-INSTALL(               TARGETS elastic RUNTIME DESTINATION "$ENV{HOME}/bin")
+INSTALL(               TARGETS elastic RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
 
 
  
