--- a/configure.ac
+++ b/configure.ac
@@ -23,6 +23,7 @@
 AC_PROG_RANLIB
 AC_PROG_INSTALL
 AC_PROG_LN_S
+AC_PROG_LIBTOOL
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
@@ -39,10 +40,7 @@
 dnl Checks for library functions.
 AC_CHECK_FUNCS(strtoul)
 
-dnl Set flags for compiler
-if test X"$GCC" = Xyes ; then
-	AC_SUBST(AM_CFLAGS,"-Wall -fomit-frame-pointer")
-fi
+AC_CHECK_LIB([m], [pow])
 
 AC_CONFIG_FILES([\
         Makefile \
--- a/examples/Makefile.am
+++ b/examples/Makefile.am
@@ -1,11 +1,11 @@
 ## Process this file with automake to produce Makefile.in
 # $Id$
 
-LDADD = $(top_builddir)/src/libprng.a -lm
+LDADD = $(top_builddir)/src/libprng.la -lm
 
 INCLUDES = -I$(top_srcdir)/src
 
-noinst_PROGRAMS = pairs tuples
+check_PROGRAMS = pairs tuples
 
 # clean backup files
 CLEANFILES = *~
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -3,9 +3,9 @@
 
 INCLUDES = 
 
-lib_LIBRARIES = libprng.a
+lib_LTLIBRARIES = libprng.la
 
-libprng_a_SOURCES = \
+libprng_la_SOURCES = \
  prng.c \
  dicg.c \
  eicg.c \
