diff --git a/packages/kokkos-kernels/src/sparse/KokkosSparse_spadd_handle.hpp b/packages/kokkos-kernels/src/sparse/KokkosSparse_spadd_handle.hpp
index 917b1038a61..d5e4d30653d 100644
--- a/packages/kokkos-kernels/src/sparse/KokkosSparse_spadd_handle.hpp
+++ b/packages/kokkos-kernels/src/sparse/KokkosSparse_spadd_handle.hpp
@@ -104,10 +104,6 @@ class SPADDHandle {
    */
   size_type get_c_nnz() { return this->result_nnz_size; }

-  void set_sort_option(int option) { this->sort_option = option; }
-
-  int get_sort_option() { return this->sort_option; }
-
   /**
    * \brief Default constructor.
    */
