--- old/Makefile
+++ new/Makefile
@@ -76,6 +76,9 @@

 .PHONY: clean tests install

+includedir = $(PREFIX)/include
+libdir = $(PREFIX)/lib
+
 clean:
 	/bin/rm -f *.o ctest ftest $(LIB) mpi.mod config.log config.status
 	cd tests ; $(MAKE) clean
@@ -83,7 +86,7 @@
 tests:
 	cd tests; make

-install: lib
+install: $(LIB)
 	$(MKINSTALLDIRS) $(libdir) $(includedir)
 	$(INSTALL) lib$(MODULE).a -m 644 $(libdir)
 	$(INSTALL) mpi.h -m 644 $(includedir)
--- old/Makefile.conf.in
+++ new/Makefile.conf.in
@@ -10,7 +10,8 @@
 LIBS            = @LIBS@
 CRULE           = .c.o
 F90RULE         = .F90.o
-
+MKINSTALLDIRS   = mkdir
 SHELL		= /bin/sh
-
+INSTALL         = install
 MODULE		= mpi-serial
+PREFIX          = @prefix@
