diff -r -u a/src/glew.c b/src/glew.c
--- a/src/glew.c	2020-03-15 04:53:59.000000000 -0700
+++ b/src/glew.c	2025-05-10 09:30:10.610259000 -0700
@@ -38,7 +38,15 @@
 
 #if defined(GLEW_OSMESA)
 #  define GLAPI extern
+#  ifndef APIENTRY
+#    define APIENTRY
+#    define GLEW_APIENTRY_DEFINED
+#  endif
 #  include <GL/osmesa.h>
+#  ifdef GLEW_APIENTRY_DEFINED
+#    undef APIENTRY
+#    undef GLEW_APIENTRY_DEFINED
+#  endif
 #elif defined(GLEW_EGL)
 #  include <GL/eglew.h>
 #elif defined(_WIN32)
diff -r -u a/src/glewinfo.c b/src/glewinfo.c
--- a/src/glewinfo.c	2020-03-15 04:53:59.000000000 -0700
+++ b/src/glewinfo.c	2025-05-10 09:45:02.853885000 -0700
@@ -38,7 +38,15 @@
 #include <GL/eglew.h>
 #elif defined(GLEW_OSMESA)
 #define GLAPI extern
+#ifndef APIENTRY
+#  define APIENTRY
+#  define GLEW_APIENTRY_DEFINED
+#endif
 #include <GL/osmesa.h>
+#ifdef GLEW_APIENTRY_DEFINED
+#  undef APIENTRY
+#  undef GLEW_APIENTRY_DEFINED
+#endif
 #elif defined(_WIN32)
 #include <GL/wglew.h>
 #elif !defined(__APPLE__) && !defined(__HAIKU__) || defined(GLEW_APPLE_GLX)
diff -r -u a/src/visualinfo.c b/src/visualinfo.c
--- a/src/visualinfo.c	2020-03-15 04:53:59.000000000 -0700
+++ b/src/visualinfo.c	2025-05-10 09:45:38.136185000 -0700
@@ -36,7 +36,15 @@
 #include <GL/glew.h>
 #if defined(GLEW_OSMESA)
 #define GLAPI extern
+#ifndef APIENTRY
+#  define APIENTRY
+#  define GLEW_APIENTRY_DEFINED
+#endif
 #include <GL/osmesa.h>
+#ifdef GLEW_APIENTRY_DEFINED
+#  undef APIENTRY
+#  undef GLEW_APIENTRY_DEFINED
+#endif
 #elif defined(GLEW_EGL)
 #include <GL/eglew.h>
 #elif defined(_WIN32)
