diff -ur a/install_components/Linux_ppc64le/21.5/compilers/bin/makelocalrc b/install_components/Linux_ppc64le/21.5/compilers/bin/makelocalrc
--- a/install_components/Linux_ppc64le/21.5/compilers/bin/makelocalrc	2021-05-29 16:31:07.000000000 -0600
+++ b/install_components/Linux_ppc64le/21.5/compilers/bin/makelocalrc	2021-07-01 08:37:34.136595453 -0600
@@ -31,8 +31,8 @@
 ##      [-ofeddir dir]
 ##
 
-# Just in case, add /usr/bin and /bin to PATH
-PATH=/usr/bin:/bin:$PATH
+# Just in case, append /usr/bin and /bin to PATH
+PATH=${PATH:+$PATH:}/usr/bin:/bin
 LANG=en_US.UTF-8
 
 # set TMPDIR if environment variable $TMPDIR is not set

