diff --git a/runtime/libpgmath/lib/common/ceil.c b/runtime/libpgmath/lib/common/ceil.c
index 3b66ca5c..341c5194 100644
--- a/runtime/libpgmath/lib/common/ceil.c
+++ b/runtime/libpgmath/lib/common/ceil.c
@@ -16,9 +16,11 @@
  */
 
 #include "mthdecls.h"
+/*
 #if     defined(__SSE4_1__) || defined(__AVX__)
 #include    <immintrin.h>
 #endif
+*/
 
 float
 __mth_i_ceil(float x)
diff --git a/runtime/libpgmath/lib/common/dceil.c b/runtime/libpgmath/lib/common/dceil.c
index 67482ef2..5dae82f9 100644
--- a/runtime/libpgmath/lib/common/dceil.c
+++ b/runtime/libpgmath/lib/common/dceil.c
@@ -16,6 +16,7 @@
  */
 
 #include "mthdecls.h"
+/*
 #if     defined(__SSE4_1__) || defined(__AVX__)
 #include    <immintrin.h>
 #endif
@@ -33,9 +34,12 @@ __mth_i_dceil_sse(double x)
   return _mm_cvtsd_f64(_mm_ceil_sd(_mm_set1_pd(x), _mm_set1_pd(x)));
 }
 #else
+*/
 double
 __mth_i_dceil(double x)
 {
   return ceil(x);
 }
+/*
 #endif
+*/
diff --git a/runtime/libpgmath/lib/common/dfloor.c b/runtime/libpgmath/lib/common/dfloor.c
index f947ef33..8b955897 100644
--- a/runtime/libpgmath/lib/common/dfloor.c
+++ b/runtime/libpgmath/lib/common/dfloor.c
@@ -16,6 +16,7 @@
  */
 
 #include "mthdecls.h"
+/*
 #if     defined(__SSE4_1__) || defined(__AVX__)
 #include    <immintrin.h>
 #endif
@@ -33,9 +34,12 @@ __mth_i_dfloor_sse(double x)
   return _mm_cvtsd_f64(_mm_floor_sd(_mm_set1_pd(x), _mm_set1_pd(x)));
 }
 #else
+*/
 double
 __mth_i_dfloor(double x)
 {
   return floor(x);
 }
+/*
 #endif
+*/
diff --git a/runtime/libpgmath/lib/common/floor.c b/runtime/libpgmath/lib/common/floor.c
index 7865bedc..c13872df 100644
--- a/runtime/libpgmath/lib/common/floor.c
+++ b/runtime/libpgmath/lib/common/floor.c
@@ -16,6 +16,7 @@
  */
 
 #include "mthdecls.h"
+/*
 #if     defined(__SSE4_1__) || defined(__AVX__)
 #include    <immintrin.h>
 #endif
@@ -33,9 +34,12 @@ __mth_i_floor_sse(float x)
   return _mm_cvtss_f32(_mm_floor_ss(_mm_set1_ps(x), _mm_set1_ps(x)));
 }
 #else
+*/
 float
 __mth_i_floor(float x)
 {
   return floorf(x);
 }
+/*
 #endif
+*/
