diff --git a/cmake/cudalibs_target.cmake b/cmake/cudalibs_target.cmake
index 29802dd3..3d4d761b 100644
--- a/cmake/cudalibs_target.cmake
+++ b/cmake/cudalibs_target.cmake
@@ -1,6 +1,5 @@
+find_package(CUDAToolkit REQUIRED)
 if (NOT TARGET sirius::cudalibs)
   add_library(sirius::cudalibs INTERFACE IMPORTED)
-  set_target_properties(sirius::cudalibs PROPERTIES
-                                         INTERFACE_INCLUDE_DIRECTORIES "${CUDA_INCLUDE_DIRS}"
-                                         INTERFACE_LINK_LIBRARIES "${CUDA_LIBRARIES};${CUDA_CUBLAS_LIBRARIES};${CUDA_CUFFT_LIBRARIES};${CUDA_cusolver_LIBRARY}")
+  target_link_libraries(sirius::cudalibs INTERFACE CUDA::cudart CUDA::cublas CUDA::cufft CUDA::cusolver)
 endif()
