--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,18 @@ ecbuild_add_option( FEATURE JPG
     DESCRIPTION "support for JPG decoding/encoding"
     DEFAULT ON
 )
+# Options related to JPG. The Jasper and OpenJPEG libraries
+ecbuild_add_option( FEATURE JPG_LIBJASPER
+    DESCRIPTION "Support for JPG decoding/encoding with the Jasper library"
+    CONDITION ENABLE_JPG
+    DEFAULT ON
+)
+ecbuild_add_option( FEATURE JPG_LIBOPENJPEG
+    DESCRIPTION "Support for JPG decoding/encoding with the OpenJPEG library"
+    CONDITION ENABLE_JPG
+    DEFAULT ON
+)
+
 
 ecbuild_add_option( FEATURE PNG
     DESCRIPTION "support for PNG decoding/encoding"
@@ -144,7 +156,7 @@ if( ENABLE_JPG )
 
     find_package( OpenJPEG )
 
-    if( JASPER_FOUND )
+    if( JASPER_FOUND AND ENABLE_JPG_LIBJASPER )
         list( APPEND ECCODES_TPLS Jasper )
         set( HAVE_JPEG 1 )
         set( HAVE_LIBJASPER 1 )
@@ -152,7 +164,7 @@ if( ENABLE_JPG )
         string(REGEX REPLACE "^([0-9]+)\\.[0-9]+\\.[0-9]+.*" "\\1" JASPER_VERSION_MAJOR "${JASPER_VERSION_STRING}")
     endif()
 
-    if( OPENJPEG_FOUND )
+    if( OPENJPEG_FOUND AND ENABLE_JPG_LIBOPENJPEG )
         list( APPEND ECCODES_TPLS OpenJPEG )
         set( HAVE_JPEG 1 )
         set( HAVE_LIBOPENJPEG 1 )
