diff --git a/configure.ac b/configure.ac
index ad17465..e020e96 100644
--- a/configure.ac
+++ b/configure.ac
@@ -198,11 +198,13 @@ AS_IF([test "x$ac_cv_path_NCCONF" != "x"],
       nc_has_logging=`$ac_cv_path_NCCONF --has-logging`
       nc_has_dap=`$ac_cv_path_NCCONF --has-dap`
       nc_has_pnetcdf=`$ac_cv_path_NCCONF --has-pnetcdf`
+      nc_libs=`$ac_cv_path_NCCONF --libs`
+      LIBS="$nc_libs $LIBS"
     ],
     [
       # See if these functions are in the library.
       AC_CHECK_HEADERS([netcdf.h], [], [AC_MSG_ERROR([netcdf.h could not be found. Please set CPPFLAGS.])])
-      AC_SEARCH_LIBS([nc_create], [netcdf], [], [])
+      AC_CHECK_LIB([nc_create], [netcdf], [], [])
       AC_CHECK_FUNCS([nc_def_opaque nccreate nc_set_log_level oc_open nc_use_parallel_enabled])
 
       # Whether we build/test some functionality depends on what we found in
