diff --git a/configure.ac b/configure.ac
index 906d3e2..9aee5d3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -177,7 +177,7 @@ AC_SYS_LARGEFILE
 
 # See if these functions are in the library.
 AC_CHECK_HEADERS([netcdf.h], [], [AC_MSG_ERROR([netcdf.h could not be found. Please set CPPFLAGS.])])
-AC_SEARCH_LIBS([nc_create], [netcdf], [], [])
+AC_CHECK_LIB([netcdf], [nc_create], [], [])
 AC_CHECK_FUNCS([nc_def_opaque nccreate nc_set_log_level oc_open nc_use_parallel_enabled])
 test "x$ac_cv_func_oc_open" = xyes && nc_has_dap=yes || nc_has_dap=no
 
