diff -ur spack-src.org/cpp/app/gen_mi/CMakeLists.txt spack-src/cpp/app/gen_mi/CMakeLists.txt
--- spack-src.org/cpp/app/gen_mi/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/gen_mi/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -29,7 +29,7 @@
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (gen_mi gfortran stdc++)
+    target_link_libraries (gen_mi stdc++)
   else()
     target_link_libraries (gen_mi ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/generate_quad/CMakeLists.txt spack-src/cpp/app/generate_quad/CMakeLists.txt
--- spack-src.org/cpp/app/generate_quad/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/generate_quad/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -29,7 +29,7 @@
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (generate_quad gfortran stdc++)
+    target_link_libraries (generate_quad stdc++)
   else()
     target_link_libraries (generate_quad ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/gkpSparse/CMakeLists.txt spack-src/cpp/app/gkpSparse/CMakeLists.txt
--- spack-src.org/cpp/app/gkpSparse/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/gkpSparse/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -30,7 +30,7 @@
 elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (gkpSparse gfortran stdc++)
+    target_link_libraries (gkpSparse stdc++)
   else()
     target_link_libraries (gkpSparse ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/gp_regr/CMakeLists.txt spack-src/cpp/app/gp_regr/CMakeLists.txt
--- spack-src.org/cpp/app/gp_regr/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/gp_regr/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -41,7 +41,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (gp_regr gfortran stdc++)
+    target_link_libraries (gp_regr stdc++)
   else()
     target_link_libraries (gp_regr ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/lr_eval/CMakeLists.txt spack-src/cpp/app/lr_eval/CMakeLists.txt
--- spack-src.org/cpp/app/lr_eval/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/lr_eval/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -42,7 +42,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (lr_eval gfortran stdc++)
+    target_link_libraries (lr_eval stdc++)
   else()
     target_link_libraries (lr_eval ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/lr_regr/CMakeLists.txt spack-src/cpp/app/lr_regr/CMakeLists.txt
--- spack-src.org/cpp/app/lr_regr/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/lr_regr/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -42,7 +42,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (lr_regr gfortran stdc++)
+    target_link_libraries (lr_regr stdc++)
   else()
     target_link_libraries (lr_regr ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/model_inf/CMakeLists.txt spack-src/cpp/app/model_inf/CMakeLists.txt
--- spack-src.org/cpp/app/model_inf/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/model_inf/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -40,7 +40,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (model_inf gfortran stdc++)
+    target_link_libraries (model_inf stdc++)
   else()
     target_link_libraries (model_inf ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/pce_eval/CMakeLists.txt spack-src/cpp/app/pce_eval/CMakeLists.txt
--- spack-src.org/cpp/app/pce_eval/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/pce_eval/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -37,7 +37,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (pce_eval gfortran stdc++)
+    target_link_libraries (pce_eval stdc++)
   else()
     target_link_libraries (pce_eval ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/pce_quad/CMakeLists.txt spack-src/cpp/app/pce_quad/CMakeLists.txt
--- spack-src.org/cpp/app/pce_quad/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/pce_quad/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -38,7 +38,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (pce_quad gfortran stdc++)
+    target_link_libraries (pce_quad stdc++)
   else()
     target_link_libraries (pce_quad ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/pce_resp/CMakeLists.txt spack-src/cpp/app/pce_resp/CMakeLists.txt
--- spack-src.org/cpp/app/pce_resp/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/pce_resp/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -37,7 +37,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (pce_resp gfortran stdc++)
+    target_link_libraries (pce_resp stdc++)
   else()
     target_link_libraries (pce_resp ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/pce_rv/CMakeLists.txt spack-src/cpp/app/pce_rv/CMakeLists.txt
--- spack-src.org/cpp/app/pce_rv/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/pce_rv/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -36,7 +36,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (pce_rv gfortran stdc++)
+    target_link_libraries (pce_rv stdc++)
   else()
     target_link_libraries (pce_rv ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/pce_sens/CMakeLists.txt spack-src/cpp/app/pce_sens/CMakeLists.txt
--- spack-src.org/cpp/app/pce_sens/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/pce_sens/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -37,7 +37,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (pce_sens gfortran stdc++)
+    target_link_libraries (pce_sens stdc++)
   else()
     target_link_libraries (pce_sens ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/pdf_cl/CMakeLists.txt spack-src/cpp/app/pdf_cl/CMakeLists.txt
--- spack-src.org/cpp/app/pdf_cl/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/pdf_cl/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -34,7 +34,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (pdf_cl gfortran stdc++)
+    target_link_libraries (pdf_cl stdc++)
   else()
     target_link_libraries (pdf_cl ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/regression/CMakeLists.txt spack-src/cpp/app/regression/CMakeLists.txt
--- spack-src.org/cpp/app/regression/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/regression/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -42,7 +42,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (regression gfortran stdc++)
+    target_link_libraries (regression stdc++)
   else()
     target_link_libraries (regression ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/app/sens/CMakeLists.txt spack-src/cpp/app/sens/CMakeLists.txt
--- spack-src.org/cpp/app/sens/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/app/sens/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -52,8 +52,8 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (sens gfortran stdc++)
-    target_link_libraries (trdSpls gfortran stdc++)
+    target_link_libraries (sens stdc++)
+    target_link_libraries (trdSpls stdc++)
   else()
     target_link_libraries (sens ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
     target_link_libraries (trdSpls ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
diff -ur spack-src.org/cpp/tests/Array1DMiscTest/CMakeLists.txt spack-src/cpp/tests/Array1DMiscTest/CMakeLists.txt
--- spack-src.org/cpp/tests/Array1DMiscTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/Array1DMiscTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -29,7 +29,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (Array1DMiscTest gfortran stdc++)
+    target_link_libraries (Array1DMiscTest stdc++)
   else()
     target_link_libraries (Array1DMiscTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/Array2DMiscTest/CMakeLists.txt spack-src/cpp/tests/Array2DMiscTest/CMakeLists.txt
--- spack-src.org/cpp/tests/Array2DMiscTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/Array2DMiscTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -29,7 +29,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (Array2DMiscTest gfortran stdc++)
+    target_link_libraries (Array2DMiscTest stdc++)
   else()
     target_link_libraries (Array2DMiscTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/ArrayDelColumn/CMakeLists.txt spack-src/cpp/tests/ArrayDelColumn/CMakeLists.txt
--- spack-src.org/cpp/tests/ArrayDelColumn/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/ArrayDelColumn/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -28,7 +28,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (ArrayDelColumn gfortran stdc++)
+    target_link_libraries (ArrayDelColumn stdc++)
   else()
     target_link_libraries (ArrayDelColumn ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/ArrayReadAndWrite/CMakeLists.txt spack-src/cpp/tests/ArrayReadAndWrite/CMakeLists.txt
--- spack-src.org/cpp/tests/ArrayReadAndWrite/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/ArrayReadAndWrite/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -28,7 +28,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (ArrayReadAndWrite gfortran expat stdc++)
+    target_link_libraries (ArrayReadAndWrite expat stdc++)
   else()
     target_link_libraries (ArrayReadAndWrite ${ClangLibPath}/libgfortran.dylib expat ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/ArraySortTest/CMakeLists.txt spack-src/cpp/tests/ArraySortTest/CMakeLists.txt
--- spack-src.org/cpp/tests/ArraySortTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/ArraySortTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -30,7 +30,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (ArraySortTest gfortran stdc++)
+    target_link_libraries (ArraySortTest stdc++)
   else()
     target_link_libraries (ArraySortTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/BCS1dTest/CMakeLists.txt spack-src/cpp/tests/BCS1dTest/CMakeLists.txt
--- spack-src.org/cpp/tests/BCS1dTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/BCS1dTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -31,7 +31,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (BCS1dTest gfortran stdc++)
+    target_link_libraries (BCS1dTest stdc++)
   else()
     target_link_libraries (BCS1dTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/BCS2dTest/CMakeLists.txt spack-src/cpp/tests/BCS2dTest/CMakeLists.txt
--- spack-src.org/cpp/tests/BCS2dTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/BCS2dTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -37,7 +37,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (BCS2dTest gfortran stdc++)
+    target_link_libraries (BCS2dTest stdc++)
   else()
     target_link_libraries (BCS2dTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/CorrTest/CMakeLists.txt spack-src/cpp/tests/CorrTest/CMakeLists.txt
--- spack-src.org/cpp/tests/CorrTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/CorrTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -48,7 +48,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (CorrTest gfortran stdc++)
+    target_link_libraries (CorrTest stdc++)
   else()
     target_link_libraries (CorrTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/Deriv1dTest/CMakeLists.txt spack-src/cpp/tests/Deriv1dTest/CMakeLists.txt
--- spack-src.org/cpp/tests/Deriv1dTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/Deriv1dTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -32,7 +32,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (Deriv1dTest gfortran stdc++)
+    target_link_libraries (Deriv1dTest stdc++)
   else()
     target_link_libraries (Deriv1dTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/GradHessianTest/CMakeLists.txt spack-src/cpp/tests/GradHessianTest/CMakeLists.txt
--- spack-src.org/cpp/tests/GradHessianTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/GradHessianTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -31,7 +31,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (GradHessianTest gfortran stdc++)
+    target_link_libraries (GradHessianTest stdc++)
   else()
     target_link_libraries (GradHessianTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/GradientPCETest/CMakeLists.txt spack-src/cpp/tests/GradientPCETest/CMakeLists.txt
--- spack-src.org/cpp/tests/GradientPCETest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/GradientPCETest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -31,7 +31,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (GradientPCETest gfortran stdc++)
+    target_link_libraries (GradientPCETest stdc++)
   else()
     target_link_libraries (GradientPCETest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/Hessian2dTest/CMakeLists.txt spack-src/cpp/tests/Hessian2dTest/CMakeLists.txt
--- spack-src.org/cpp/tests/Hessian2dTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/Hessian2dTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -31,7 +31,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (Hessian2dTest gfortran stdc++)
+    target_link_libraries (Hessian2dTest stdc++)
   else()
     target_link_libraries (Hessian2dTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/LowRankRegrTest/CMakeLists.txt spack-src/cpp/tests/LowRankRegrTest/CMakeLists.txt
--- spack-src.org/cpp/tests/LowRankRegrTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/LowRankRegrTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -25,7 +25,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (LowRankRegrTest gfortran stdc++)
+    target_link_libraries (LowRankRegrTest stdc++)
   else()
     target_link_libraries (LowRankRegrTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/MCMC2dTest/CMakeLists.txt spack-src/cpp/tests/MCMC2dTest/CMakeLists.txt
--- spack-src.org/cpp/tests/MCMC2dTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/MCMC2dTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -31,7 +31,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (MCMC2dTest gfortran stdc++)
+    target_link_libraries (MCMC2dTest stdc++)
   else()
     target_link_libraries (MCMC2dTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/MCMCNestedTest/CMakeLists.txt spack-src/cpp/tests/MCMCNestedTest/CMakeLists.txt
--- spack-src.org/cpp/tests/MCMCNestedTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/MCMCNestedTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -31,7 +31,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (MCMCNestedTest gfortran stdc++)
+    target_link_libraries (MCMCNestedTest stdc++)
   else()
     target_link_libraries (MCMCNestedTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/MCMCRandomTest/CMakeLists.txt spack-src/cpp/tests/MCMCRandomTest/CMakeLists.txt
--- spack-src.org/cpp/tests/MCMCRandomTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/MCMCRandomTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -31,7 +31,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (MCMCRandomTest gfortran stdc++)
+    target_link_libraries (MCMCRandomTest stdc++)
   else()
     target_link_libraries (MCMCRandomTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/MultiIndexTest/CMakeLists.txt spack-src/cpp/tests/MultiIndexTest/CMakeLists.txt
--- spack-src.org/cpp/tests/MultiIndexTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/MultiIndexTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -34,7 +34,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (MultiIndexTest gfortran stdc++)
+    target_link_libraries (MultiIndexTest stdc++)
   else()
     target_link_libraries (MultiIndexTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/PCE1dTest/CMakeLists.txt spack-src/cpp/tests/PCE1dTest/CMakeLists.txt
--- spack-src.org/cpp/tests/PCE1dTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/PCE1dTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -32,7 +32,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (PCE1dTest gfortran stdc++)
+    target_link_libraries (PCE1dTest stdc++)
   else()
     target_link_libraries (PCE1dTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/PCEImplTest/CMakeLists.txt spack-src/cpp/tests/PCEImplTest/CMakeLists.txt
--- spack-src.org/cpp/tests/PCEImplTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/PCEImplTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -32,7 +32,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (PCEImplTest gfortran stdc++)
+    target_link_libraries (PCEImplTest stdc++)
   else()
     target_link_libraries (PCEImplTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/PCELogTest/CMakeLists.txt spack-src/cpp/tests/PCELogTest/CMakeLists.txt
--- spack-src.org/cpp/tests/PCELogTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/PCELogTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -32,7 +32,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (PCELogTest gfortran stdc++)
+    target_link_libraries (PCELogTest stdc++)
   else()
     target_link_libraries (PCELogTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/QuadLUTest/CMakeLists.txt spack-src/cpp/tests/QuadLUTest/CMakeLists.txt
--- spack-src.org/cpp/tests/QuadLUTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/QuadLUTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -36,7 +36,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (QuadLUTest gfortran stdc++)
+    target_link_libraries (QuadLUTest stdc++)
   else()
     target_link_libraries (QuadLUTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/SecondDeriv1dTest/CMakeLists.txt spack-src/cpp/tests/SecondDeriv1dTest/CMakeLists.txt
--- spack-src.org/cpp/tests/SecondDeriv1dTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/SecondDeriv1dTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -31,7 +31,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (SecondDeriv1dTest gfortran stdc++)
+    target_link_libraries (SecondDeriv1dTest stdc++)
   else()
     target_link_libraries (SecondDeriv1dTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/cpp/tests/TMCMC2dTest/CMakeLists.txt spack-src/cpp/tests/TMCMC2dTest/CMakeLists.txt
--- spack-src.org/cpp/tests/TMCMC2dTest/CMakeLists.txt	2020-10-13 16:38:53.000000000 +0900
+++ spack-src/cpp/tests/TMCMC2dTest/CMakeLists.txt	2020-10-13 16:38:56.000000000 +0900
@@ -47,7 +47,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (TMCMC2dTest gfortran stdc++)
+    target_link_libraries (TMCMC2dTest stdc++)
   else()
     target_link_libraries (TMCMC2dTest ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/examples/kle_ex1/CMakeLists.txt spack-src/examples/kle_ex1/CMakeLists.txt
--- spack-src.org/examples/kle_ex1/CMakeLists.txt	2020-10-13 16:38:55.000000000 +0900
+++ spack-src/examples/kle_ex1/CMakeLists.txt	2020-10-13 17:15:16.000000000 +0900
@@ -73,9 +73,9 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (kl_1D.x  gfortran stdc++)
-    target_link_libraries (kl_2D.x  gfortran stdc++)
-    target_link_libraries (kl_2Du.x gfortran stdc++)
+    target_link_libraries (kl_1D.x  stdc++)
+    target_link_libraries (kl_2D.x  stdc++)
+    target_link_libraries (kl_2Du.x stdc++)
   else()
     target_link_libraries (kl_1D.x  ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
     target_link_libraries (kl_2D.x  ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
diff -ur spack-src.org/examples/line_infer/CMakeLists.txt spack-src/examples/line_infer/CMakeLists.txt
--- spack-src.org/examples/line_infer/CMakeLists.txt	2020-10-13 16:38:55.000000000 +0900
+++ spack-src/examples/line_infer/CMakeLists.txt	2020-10-13 17:16:41.000000000 +0900
@@ -56,7 +56,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (line_infer.x gfortran stdc++)
+    target_link_libraries (line_infer.x stdc++)
   else()
     target_link_libraries (line_infer.x ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/examples/muq/CMakeLists.txt spack-src/examples/muq/CMakeLists.txt
--- spack-src.org/examples/muq/CMakeLists.txt	2020-10-13 16:38:55.000000000 +0900
+++ spack-src/examples/muq/CMakeLists.txt	2020-10-13 17:18:04.000000000 +0900
@@ -61,7 +61,7 @@
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wno-maybe-uninitialized -Wno-sign-compare")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas -Wno-unused-variable -Wno-unused-local-typedefs")
 if ("${GnuLibPath}" STREQUAL "")
-  target_link_libraries (muqPCE2uqtk.x gfortran stdc++)
+  target_link_libraries (muqPCE2uqtk.x stdc++)
 else()
   target_link_libraries (muqPCE2uqtk.x  ${GnuLibPath}/libgfortran.a ${GnuLibPath}/libquadmath.a stdc++)
 endif()
diff -ur spack-src.org/examples/ops/CMakeLists.txt spack-src/examples/ops/CMakeLists.txt
--- spack-src.org/examples/ops/CMakeLists.txt	2020-10-13 16:38:55.000000000 +0900
+++ spack-src/examples/ops/CMakeLists.txt	2020-10-13 17:44:31.000000000 +0900
@@ -58,7 +58,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (Ops.x gfortran stdc++)
+    target_link_libraries (Ops.x stdc++)
   else()
     target_link_libraries (Ops.x ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
@@ -99,7 +99,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (LogComp.x gfortran stdc++)
+    target_link_libraries (LogComp.x stdc++)
   else()
     target_link_libraries (LogComp.x ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
diff -ur spack-src.org/examples/pce_bcs/CMakeLists.txt spack-src/examples/pce_bcs/CMakeLists.txt
--- spack-src.org/examples/pce_bcs/CMakeLists.txt	2020-10-13 16:38:55.000000000 +0900
+++ spack-src/examples/pce_bcs/CMakeLists.txt	2020-10-13 17:16:04.000000000 +0900
@@ -59,7 +59,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (pce_bcs.x gfortran stdc++)
+    target_link_libraries (pce_bcs.x stdc++)
   else()
     target_link_libraries (pce_bcs.x ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
   endif()
Only in spack-src/examples: spack-cc-uqtk-re5kp6x.in.log
Only in spack-src/examples: spack-cc-uqtk-re5kp6x.out.log
diff -ur spack-src.org/examples/surf_rxn/CMakeLists.txt spack-src/examples/surf_rxn/CMakeLists.txt
--- spack-src.org/examples/surf_rxn/CMakeLists.txt	2020-10-13 16:38:55.000000000 +0900
+++ spack-src/examples/surf_rxn/CMakeLists.txt	2020-10-13 17:17:04.000000000 +0900
@@ -99,10 +99,10 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (SurfRxnNISP.x    gfortran stdc++)
-    target_link_libraries (SurfRxnNISP_MC.x gfortran stdc++)
-    target_link_libraries (SurfRxnISP.x     gfortran stdc++)
-    target_link_libraries (SurfRxnDet.x     gfortran stdc++)
+    target_link_libraries (SurfRxnNISP.x    stdc++)
+    target_link_libraries (SurfRxnNISP_MC.x stdc++)
+    target_link_libraries (SurfRxnISP.x     stdc++)
+    target_link_libraries (SurfRxnDet.x     stdc++)
   else()
     target_link_libraries (SurfRxnNISP.x    ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
     target_link_libraries (SurfRxnNISP_MC.x ${ClangLibPath}/libgfortran.dylib ${ClangLibPath}/libquadmath.dylib ${ClangLibPath}/libstdc++.dylib)
diff -ur spack-src.org/examples/tmcmc_bimodal/CMakeLists.txt spack-src/examples/tmcmc_bimodal/CMakeLists.txt
--- spack-src.org/examples/tmcmc_bimodal/CMakeLists.txt	2020-10-13 16:38:55.000000000 +0900
+++ spack-src/examples/tmcmc_bimodal/CMakeLists.txt	2020-10-13 17:14:38.000000000 +0900
@@ -41,7 +41,7 @@
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   # using Clang
   if ("${ClangLibPath}" STREQUAL "")
-    target_link_libraries (tmcmc_bimodal.x gfortran stdc++)
+    target_link_libraries (tmcmc_bimodal.x stdc++)
   else()
     target_link_libraries (tmcmc_bimodal.x ${ClangLibPath}/libgfortran.dylib  ${ClangLibPath}/libstdc++.dylib)
   endif()
