diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1379294..6de15c3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -61,11 +61,6 @@ if(NOT ENABLE_SERIAL AND NOT ENABLE_MPI)
     "Please set ENABLE_SERIAL, ENABLE_MPI or both to ON")
 endif()
 
-
-if(ENABLE_CUDA AND BUILD_SHARED_LIBS)
-  message(FATAL_ERROR "Static libraries are required when building with CUDA")
-endif()
-
 if(ENABLE_CUDA)
     # ask politely a few times for SEPARABLE COMPILATION
     set (CMAKE_CUDA_SEPARABLE_COMPILATION ON CACHE BOOL "" )
diff --git a/src/cmake/thirdparty/SetupVTKm.cmake b/src/cmake/thirdparty/SetupVTKm.cmake
index 1bc5b56..e242738 100644
--- a/src/cmake/thirdparty/SetupVTKm.cmake
+++ b/src/cmake/thirdparty/SetupVTKm.cmake
@@ -64,6 +64,11 @@ if(ENABLE_CUDA AND NOT VTKm_ENABLE_CUDA)
    message(FATAL_ERROR "VTK-h CUDA support requires VTK-m with CUDA support (ENABLE_CUDA == TRUE, however VTKm_ENABLE_CUDA == FALSE")
 endif()
 
+if(ENABLE_CUDA AND BUILD_SHARED_LIBS)
+  if(VTKm_VERSION VERSION_LESS "1.7.0")
+    message(FATAL_ERROR "Cannot build shared libs with CUDA when VTKm is < v1.7.0")
+  endif()
+endif()
 
 set(VTKM_FOUND TRUE)
 
