diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86ef7b3..acde12e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,12 +32,10 @@ else()
 endif()
 
 # Pick up static and dynamic shared object files
-list( APPEND CMAKE_PREFIX_PATH /opt/rocm/llvm /opt/rocm/hip )
 
 # CMake modules
 list(APPEND CMAKE_MODULE_PATH
      ${CMAKE_CURRENT_SOURCE_DIR}/cmake
-     ${CMAKE_PREFIX_PATH}/cmake
 )
 
 # Set a default build type if none was specified
