--- opencv-3.3.0/cmake/OpenCVDetectPython.cmake	2019-04-23 23:53:05.054791077 +0200
+++ opencv-3.3.0/cmake/neu.cmake	2019-04-23 23:52:49.575317624 +0200
@@ -56,7 +56,7 @@
 
   if(PYTHONINTERP_FOUND)
     # Check if python major version is correct
-    if(${_preferred_version_major} EQUAL ${PYTHON_VERSION_MAJOR})
+    if("${_preferred_version_major}" STREQUAL "${PYTHON_VERSION_MAJOR}")
       # Copy outputs
       set(_found ${PYTHONINTERP_FOUND})
       set(_executable ${PYTHON_EXECUTABLE})
