From ba5e2b5dc9de106635c12ebe9260e2fc0212ff91 Mon Sep 17 00:00:00 2001
From: @vsoch <vsoch@noreply.users.github.com>
Date: Fri, 15 Oct 2021 05:17:47 +0000
Subject: [PATCH] fixing incorrect symbol

Signed-off-by: @vsoch <vsoch@noreply.users.github.com>
---
 src/abg-dwarf-reader.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/abg-dwarf-reader.cc b/src/abg-dwarf-reader.cc
index 1d6ad24c..8dd86c5a 100644
--- a/src/abg-dwarf-reader.cc
+++ b/src/abg-dwarf-reader.cc
@@ -11053,7 +11053,7 @@ dwarf_language_to_tu_language(size_t l)
       return translation_unit::LANG_Ada95;
     case DW_LANG_Fortran95:
       return translation_unit::LANG_Fortran95;
-    case DW_LANG_PL1:
+    case DW_LANG_PLI:
       return translation_unit::LANG_PL1;
     case DW_LANG_ObjC:
       return translation_unit::LANG_ObjC;
-- 
2.17.1


