--- a/CMakeLists.txt	2016-08-16 02:30:13.466078087 +0200
+++ b/CMakeLists.txt	2016-08-16 02:30:36.879586772 +0200
@@ -553,7 +553,7 @@
     set(ENV{HDF5_ROOT} "$ENV{HDF5_DIR}")
   endif()
   set(HDF5_PREFER_PARALLEL TRUE)
-  find_package(HDF5)
+  find_package(HDF5 COMPONENTS C)
   set_package_properties(HDF5 PROPERTIES TYPE OPTIONAL
     DESCRIPTION "Hierarchical Data Format 5 (HDF5)"
     URL "https://www.hdfgroup.org/HDF5")
