diff --git a/src/settings.js b/src/settings.js
index 265c13a47..94cf2e744 100644
--- a/src/settings.js
+++ b/src/settings.js
@@ -876,7 +876,7 @@ var FILESYSTEM = 1;
 // it is not used. For example, if your C code uses no files, but you include
 // some JS that does, you might need this.
 // [link]
-var FORCE_FILESYSTEM = 0;
+var FORCE_FILESYSTEM = 1;
 
 // Enables support for the NODERAWFS filesystem backend. This is a special
 // backend as it replaces all normal filesystem access with direct Node.js
@@ -889,7 +889,7 @@ var FORCE_FILESYSTEM = 0;
 // handles permissions and errors and so forth may be noticeable.  This has
 // mostly been tested on Linux so far.
 // [link]
-var NODERAWFS = 0;
+var NODERAWFS = 1;
 
 // This saves the compiled wasm module in a file with name
 //   $WASM_BINARY_NAME.$V8_VERSION.cached
