--- spack-src/src/AlignmentBuffer.cpp.bak	2017-06-27 02:01:12.000000000 +0900
+++ spack-src/src/AlignmentBuffer.cpp	2020-09-08 11:38:24.075464116 +0900
@@ -2365,7 +2365,7 @@
 		va_list args;
 
 		va_start(args, s);
-		if (args != 0)
+		if (va_arg(args, char *) != 0)
 			vfprintf(stderr, s, args);
 		else
 			fprintf(stderr, "%s", s);
--- spack-src/src/Logging.cpp.bak	2020-09-08 11:30:20.915422258 +0900
+++ spack-src/src/Logging.cpp	2020-09-08 11:32:33.125437676 +0900
@@ -94,7 +94,7 @@
 
 	if (color)
 		SetConsoleColor((ConsoleColor) (Message + (lvl * 2)));
-	if (args != 0)
+	if (va_arg(args, char *) != 0)
 		vfprintf(stderr, s, args);
 	else
 		fprintf(stderr, "%s", s);
