--- a/include/netCDF4.pxi
+++ b/include/netCDF4.pxi
@@ -4,8 +4,8 @@
     ctypedef long ptrdiff_t
 
 # hdf5 version info.
-cdef extern from "H5public.h":
-    ctypedef int herr_t
+ctypedef int herr_t
+cdef extern \
     int H5get_libversion( unsigned int *majnum, unsigned int *minnum, unsigned int *relnum ) nogil
  
 cdef extern from *:
--- a/include/netcdf-compat.h
+++ b/include/netcdf-compat.h
@@ -95,7 +95,7 @@
 #define HAS_CDF5_FORMAT 0
 #endif
 
-#if defined(NC_HAS_PARALLEL) && NC_HAS_PARALLEL
+#if 0
 #include <netcdf_par.h>
 #define HAS_PARALLEL_SUPPORT 1
 #else
--- a/setup.py
+++ b/setup.py
@@ -392,7 +392,7 @@
        (netcdf_lib_version > "4.4" and netcdf_lib_version < "4.5"):
         has_cdf5_format = True
 
-    has_parallel_support = check_has_parallel_support(inc_dirs)
+    has_parallel_support = False
     has_has_not = "has" if has_parallel_support else "does not have"
     print(f"netcdf lib {has_has_not} parallel functions")
 
