diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 6413c03..11325a4 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -54,6 +54,7 @@ IF(TARGET ROOT::ROOTDataFrame)
   add_test(NAME test_rdf COMMAND test_rdf)
   set_tests_properties(test_rdf PROPERTIES
     ENVIRONMENT LD_LIBRARY_PATH=$<TARGET_FILE_DIR:edm4hep>:$<TARGET_FILE_DIR:podio::podio>:$ENV{LD_LIBRARY_PATH}
+    DEPENDS write_events
     )
 endif()
 
