diff --git a/CMakeLists.txt b/CMakeLists.txt
index 81205c6..25ab7aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,28 +1,30 @@
 cmake_minimum_required(VERSION 2.8)
 set(CMAKE_BUILD_TYPE  "Release")
-INCLUDE_DIRECTORIES(  "$ENV{HOME}/include")
-LINK_DIRECTORIES(     "$ENV{HOME}/lib")
 SET(CMAKE_CXX_FLAGS   "-w -g -O3")
 set(CMAKE_MACOSX_RPATH 1)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
+#Path management
+if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+    set (CMAKE_INSTALL_PREFIX "$ENV{HOME}"
+        CACHE PATH "Default install prefix" FORCE)
+endif()
+
+INCLUDE_DIRECTORIES(  "${CMAKE_INSTALL_PREFIX}/include")
+LINK_DIRECTORIES(     "${CMAKE_INSTALL_PREFIX}/lib")
+
 #Advection library
 project(Advection)
 file(                  GLOB_RECURSE source_files sources/*)
 file(                  GLOB_RECURSE header_files sources/*.h)
 ADD_LIBRARY(           Advection SHARED ${source_files})
-find_library(          Commons NAMES Commons HINTS "$ENV{HOME}/lib")
+find_library(          Commons NAMES Commons HINTS "${CMAKE_INSTALL_PREFIX}/lib")
 target_link_libraries( Advection ${Commons})
-INSTALL(               TARGETS Advection LIBRARY  DESTINATION "$ENV{HOME}/lib")
-INSTALL(               FILES ${header_files} DESTINATION "$ENV{HOME}/include")
+INSTALL(               TARGETS Advection LIBRARY  DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+INSTALL(               FILES ${header_files} DESTINATION "${CMAKE_INSTALL_PREFIX}/include")
 
 #Executable
 project(main)
 add_executable(        advect sources/advect.c)
 target_link_libraries( advect Commons Advection)
-INSTALL(               TARGETS advect RUNTIME DESTINATION "$ENV{HOME}/bin")
-
- 
-
-
-
+INSTALL(               TARGETS advect RUNTIME DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
