From 5565f0bf0a8e7b8217ed1a943a4210fec303ec42 Mon Sep 17 00:00:00 2001
From: sreenivasa murthy kolam <sreenivasamurthy.kolam@amd.com>
Date: Thu, 24 Oct 2024 13:55:01 +0000
Subject: [PATCH] link with roctracer when building miopendriver

---
 driver/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/driver/CMakeLists.txt b/driver/CMakeLists.txt
index 8f19a90..6c701d6 100644
--- a/driver/CMakeLists.txt
+++ b/driver/CMakeLists.txt
@@ -64,6 +64,9 @@ endif()
 add_dependencies(MIOpenDriver generate_kernels)
 target_include_directories(MIOpenDriver PRIVATE ../src/kernels)
 target_link_libraries(MIOpenDriver MIOpen Threads::Threads roc::rocrand)
+if(MIOPEN_USE_ROCTRACER)
+    target_link_libraries(MIOpenDriver ${rocTracer})
+endif()
 if(NOT MIOPEN_EMBED_DB STREQUAL "")
 target_link_libraries(MIOpenDriver $<BUILD_INTERFACE:miopen_data> )
 endif()
-- 
2.39.3

