From bbfc08e034b80d8b8c6895cb74c38544ffa9a9b4 Mon Sep 17 00:00:00 2001
From: sreenivasa murthy kolam <sreenivasamurthy.kolam@amd.com>
Date: Thu, 24 Oct 2024 14:01:27 +0000
Subject: [PATCH] link with roctracer when building miopendriver for 6.1.0

---
 driver/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/driver/CMakeLists.txt b/driver/CMakeLists.txt
index 7d4fdbb..31de1ba 100644
--- a/driver/CMakeLists.txt
+++ b/driver/CMakeLists.txt
@@ -34,6 +34,9 @@ endif()
 add_dependencies(MIOpenDriver generate_kernels)
 target_include_directories(MIOpenDriver PRIVATE ../src/kernels)
 target_link_libraries(MIOpenDriver MIOpen Threads::Threads)
+if(MIOPEN_USE_ROCTRACER)
+    target_link_libraries(MIOpenDriver ${rocTracer})
+endif()
 if(NOT MIOPEN_EMBED_DB STREQUAL "")
 target_link_libraries(MIOpenDriver $<BUILD_INTERFACE:miopen_data> )
 endif()
-- 
2.39.3

