diff --git a/CMakeLists.txt b/CMakeLists.txt
index 985f59f..515e7e0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,7 +184,8 @@ string(REGEX REPLACE --cuda-gpu-arch=[a-z0-9]+ "" HIP_COMPILER_FLAGS "${HIP_COMP
 message(STATUS "Hip compiler flags: ${HIP_COMPILER_FLAGS}")
 
 add_definitions("-DHIP_COMPILER_FLAGS=${HIP_COMPILER_FLAGS}")
-
+add_definitions("--rocm-path=${HIP_PREFIX_PATH}")
+add_definitions("--rocm-device-lib-path=${DEVICELIBS_PREFIX_PATH}")
 
 
 # HIP
