From 1693afd9690b97fcceff09ffce765712e3c7361a Mon Sep 17 00:00:00 2001
From: Renjith Ravindran <Renjith.RavindranKannath@amd.com>
Date: Mon, 29 Apr 2024 08:01:47 +0000
Subject: [PATCH] Adding roctracer-dev include and library path

---
 src/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0741a60..84b6805 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -739,6 +739,9 @@ if(WIN32)
 endif()
 
 target_include_directories(MIOpen SYSTEM PUBLIC $<BUILD_INTERFACE:${HALF_INCLUDE_DIR}>)
+target_include_directories(MIOpen SYSTEM PUBLIC "${NLOHMANN_JSON_INCLUDE}")
+target_include_directories(MIOpen SYSTEM PUBLIC "${ROCTRACER_INCLUDE_DIR}")
+target_include_directories(MIOpen SYSTEM PUBLIC "${SQLITE_INCLUDE_DIR}")
 # Workaround : change in rocm-cmake was causing linking error so had to add ${CMAKE_DL_LIBS}
 #               We can remove ${CMAKE_DL_LIBS} once root cause is identified.
 target_link_libraries(MIOpen PRIVATE ${CMAKE_DL_LIBS} Threads::Threads BZip2::BZip2 ${MIOPEN_CK_LINK_FLAGS})
@@ -861,7 +864,7 @@ if(NOT WIN32 AND NOT APPLE)
 endif()
 
 if(MIOPEN_USE_ROCTRACER)
-    target_link_libraries(MIOpen PRIVATE roctx64)
+    target_link_libraries(MIOpen PRIVATE "${ROCTRACER_LIB_DIR}/libroctx64.so")
 endif()
 
 ############################################################
-- 
2.31.1

