From f60aa1ff89f8fb596b4a6a4c70aa7d557803db87 Mon Sep 17 00:00:00 2001
From: Tal Ben-Nun <talbn@llnl.gov>
Date: Wed, 26 Jul 2023 13:31:56 -0700
Subject: [PATCH] Fix CMake search for unzipper

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b45cb6d2b2..3d2d760443 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -470,7 +470,7 @@ endif()
 function(unpack_db db_bzip2_file)
     set(KERNELS_DIR "${CMAKE_SOURCE_DIR}/src/kernels")
     STRING(REPLACE ".bz2" "" db_file "${db_bzip2_file}")
-    find_program(UNZIPPER lbunzip2 bunzip2)
+    find_program(UNZIPPER NAMES lbunzip2 bunzip2)
     if(EXISTS "${db_file}")
         message(STATUS "WARNING: ${db_file} already exists and will be overwritten")
     endif()
