--- a/third_party/fbgemm/CMakeLists.txt        2023-01-11 14:15:27.245817069 +0100
+++ b/third_party/fbgemm/CMakeLists.txt       2023-01-11 15:33:02.127049131 +0100
@@ -144,7 +144,6 @@
 else(MSVC)
   string(APPEND CMAKE_CXX_FLAGS " -Wall")
   string(APPEND CMAKE_CXX_FLAGS " -Wextra")
-  string(APPEND CMAKE_CXX_FLAGS " -Werror")
   string(APPEND CMAKE_CXX_FLAGS " -Wno-deprecated-declarations")
   target_compile_options(fbgemm_avx2 PRIVATE
     "-m64" "-mavx2" "-mf16c" "-mfma")
@@ -160,8 +160,7 @@ else(MSVC)
     src/FbgemmI64.cc
     src/FbgemmI8Depthwise3DAvx2.cc
     src/FbgemmI8DepthwiseAvx2.cc
-    src/UtilsAvx2.cc
-    PROPERTIES COMPILE_FLAGS "-Wno-uninitialized")
+    src/UtilsAvx2.cc)
   set_source_files_properties(src/PackMatrix.cc
     PROPERTIES COMPILE_FLAGS "-Wno-infinite-recursion")
   # Workaround for https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80947
