diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index a96075245a..70980fbeef 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1002,7 +1002,11 @@ if(NOT TARGET fp16 AND NOT USE_SYSTEM_FP16)
     "${FP16_SOURCE_DIR}"
     "${CONFU_DEPENDENCIES_BINARY_DIR}/FP16")
 elseif(NOT TARGET fp16 AND USE_SYSTEM_FP16)
-  add_library(fp16 STATIC "/usr/include/fp16.h")
+  if (DEFINED ENV{FP16_PATH})
+    add_library(fp16 STATIC "$ENV{FP16_PATH}/include/fp16.h")
+  else()
+    add_library(fp16 STATIC "/usr/include/fp16.h")
+  endif()
   set_target_properties(fp16 PROPERTIES LINKER_LANGUAGE C)
 endif()
 list(APPEND Caffe2_DEPENDENCY_LIBS fp16)
