--- spack-src/cmake/Dependencies.cmake.orig     2022-03-07 14:40:07.000000000 +0900
+++ spack-src/cmake/Dependencies.cmake  2022-03-07 14:37:07.000000000 +0900
@@ -187,7 +187,12 @@
   set(BLAS_LIBRARIES ${vecLib_LINKER_LIBS})
 elseif(BLAS STREQUAL "Generic")
   # On Debian family, the CBLAS ABIs have been merged into libblas.so
-  find_library(BLAS_LIBRARIES blas)
+  if(ENV{GENERIC_BLAS_LIBRARIES} STREQUAL "")
+    set(GENERIC_BLAS "blas")
+  else()
+    set(GENERIC_BLAS $ENV{GENERIC_BLAS_LIBRARIES})
+  endif()
+  find_library(BLAS_LIBRARIES NAMES ${GENERIC_BLAS})
   message("-- Using BLAS: ${BLAS_LIBRARIES}")
   list(APPEND Caffe2_PUBLIC_DEPENDENCY_LIBS ${BLAS_LIBRARIES})
   set(GENERIC_BLAS_FOUND TRUE)
