diff --git a/src/LinearAlgebraServicesPKG/ksparse/alloc.c b/src/LinearAlgebraServicesPKG/ksparse/alloc.c
index 320878d7817273269e8805acaa9f6f7a252443f1..af40b7e800e5e8b573a69608aae49c324f2a0253 100644
--- a/src/LinearAlgebraServicesPKG/ksparse/alloc.c
+++ b/src/LinearAlgebraServicesPKG/ksparse/alloc.c
@@ -53,6 +53,7 @@ Copyright 1990 Regents of the University of California.  All rights reserved.
  */

 void bye_bye(i)
+    int i;
 {
     printf ("inv = %d\n",1/i);
 }

