--- a/Wrapping/Generators/Python/itk/support/types.py	2021-08-12 07:19:17.000000000 -0500
+++ b/Wrapping/Generators/Python/itk/support/types.py	2023-03-24 15:25:42.240153344 -0500
@@ -137,7 +137,7 @@
         _SS: "itkCType" = itkCType("signed short", "SS", np.int16)
         _SI: "itkCType" = itkCType("signed int", "SI", np.int32)
         _SLL: "itkCType" = itkCType("signed long long", "SLL", np.int64)
-        _B: "itkCType" = itkCType("bool", "B", np.bool)
+        _B: "itkCType" = itkCType("bool", "B", np.bool_)
         return _F, _D, _UC, _US, _UI, _UL, _SL, _LD, _ULL, _SC, _SS, _SI, _SLL, _B
 
 
