diff --git a/clients/CMakeLists.txt b/clients/CMakeLists.txt
index 7f14a55..a62fbd4 100644
--- a/clients/CMakeLists.txt
+++ b/clients/CMakeLists.txt
@@ -103,6 +103,7 @@ if( BUILD_CLIENTS_TESTS OR BUILD_CLIENTS_SAMPLES )
  # look for installed FFTW if we weren't asked to build it
  if( NOT BUILD_FFTW )
    find_package( FFTW 3.0 MODULE COMPONENTS FLOAT DOUBLE )
+   set( FFTW_INCLUDES ${FFTW_INCLUDE_DIRS} )
  endif()
 
  # also try to build FFTW if FFTW isn't present
