diff --git a/library/src/CMakeLists.txt b/library/src/CMakeLists.txt
index 0fed2d2..12b4d20 100644
--- a/library/src/CMakeLists.txt
+++ b/library/src/CMakeLists.txt
@@ -138,15 +138,9 @@ set( rocfft_source
 # for cache serialization.  We also want to use a static SQLite,
 # and distro static libraries aren't typically built
 # position-independent.
-include( FetchContent )
 
 # embed SQLite
-FetchContent_Declare(sqlite_local
-  URL https://sqlite.org/2021/sqlite-amalgamation-3360000.zip
-  URL_HASH SHA256=999826fe4c871f18919fdb8ed7ec9dd8217180854dd1fe21eea96aed36186729
-)
-FetchContent_MakeAvailable(sqlite_local)
-add_library( sqlite3 STATIC ${sqlite_local_SOURCE_DIR}/sqlite3.c )
+add_library( sqlite3 STATIC ${sql_DIR}/sqlite3.c )
 set_target_properties( sqlite3 PROPERTIES
   C_VISIBILITY_PRESET "hidden"
   VISIBILITY_INLINES_HIDDEN ON
@@ -203,7 +197,7 @@ endif()
 target_include_directories( rocfft
   PRIVATE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/library/src/include>
   $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/library/src/device>
-  ${sqlite_local_SOURCE_DIR}
+  ${sql_DIR}
   PUBLIC  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/library/include>
   $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
   $<INSTALL_INTERFACE:include>
