--- cmake/onnxruntime.cmake.orig	2021-08-06 12:36:32.720081500 +0200
+++ cmake/onnxruntime.cmake	2021-08-13 18:53:30.310868700 +0200
@@ -120,7 +120,8 @@
     onnxruntime_common
     onnxruntime_mlas
     onnxruntime_flatbuffers
-    ${onnxruntime_EXTERNAL_LIBRARIES})
+    ${onnxruntime_EXTERNAL_LIBRARIES}
+    -liconv)
 
 if (onnxruntime_ENABLE_LANGUAGE_INTEROP_OPS)
   target_link_libraries(onnxruntime PRIVATE onnxruntime_language_interop onnxruntime_pyop)
--- cmake/onnxruntime_python.cmake.orig	2021-08-06 12:36:32.725148600 +0200
+++ cmake/onnxruntime_python.cmake	2021-08-13 18:54:37.085622000 +0200
@@ -106,6 +106,7 @@
     onnxruntime_mlas
     onnxruntime_flatbuffers
     ${pybind11_lib}
+    -liconv
 )
 
 if (onnxruntime_ENABLE_LANGUAGE_INTEROP_OPS)
--- cmake/onnxruntime_unittests.cmake.orig	2021-08-13 19:11:58.645461300 +0200
+++ cmake/onnxruntime_unittests.cmake	2021-08-13 19:14:18.373814800 +0200
@@ -603,7 +603,7 @@
 AddTest(
   TARGET onnxruntime_test_all
   SOURCES ${all_tests} ${onnxruntime_unittest_main_src}
-  LIBS onnx_test_runner_common ${onnxruntime_test_providers_libs}  ${onnxruntime_test_common_libs}  re2::re2 onnx_test_data_proto
+  LIBS onnx_test_runner_common ${onnxruntime_test_providers_libs}  ${onnxruntime_test_common_libs}  re2::re2 onnx_test_data_proto -liconv
   DEPENDS ${all_dependencies}
 )
 
@@ -723,7 +723,7 @@
   target_compile_options(onnx_test_runner PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--compiler-options /utf-8>"
           "$<$<NOT:$<COMPILE_LANGUAGE:CUDA>>:/utf-8>")
 endif()
-target_link_libraries(onnx_test_runner PRIVATE onnx_test_runner_common ${GETOPT_LIB_WIDE} ${onnx_test_libs})
+target_link_libraries(onnx_test_runner PRIVATE onnx_test_runner_common ${GETOPT_LIB_WIDE} ${onnx_test_libs} -liconv)
 target_include_directories(onnx_test_runner PRIVATE ${ONNXRUNTIME_ROOT})
 set_target_properties(onnx_test_runner PROPERTIES FOLDER "ONNXRuntimeTest")
 
