diff -Naur apache-log4cxx-0.10.0.orig/src/examples/cpp/console.cpp apache-log4cxx-0.10.0/src/examples/cpp/console.cpp
--- apache-log4cxx-0.10.0.orig/src/examples/cpp/console.cpp	2008-04-01 00:34:52.000000000 +0200
+++ apache-log4cxx-0.10.0/src/examples/cpp/console.cpp	2008-05-06 05:40:52.000000000 +0200
@@ -15,7 +15,10 @@
  * limitations under the License.
  */
 
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <stdint.h>
 #include <log4cxx/logger.h>
 #include <log4cxx/consoleappender.h>
 #include <log4cxx/simplelayout.h>
diff -Naur apache-log4cxx-0.10.0.orig/src/main/cpp/inputstreamreader.cpp apache-log4cxx-0.10.0/src/main/cpp/inputstreamreader.cpp
--- apache-log4cxx-0.10.0.orig/src/main/cpp/inputstreamreader.cpp	2008-04-01 00:34:09.000000000 +0200
+++ apache-log4cxx-0.10.0/src/main/cpp/inputstreamreader.cpp	2008-05-06 05:32:31.000000000 +0200
@@ -21,6 +21,8 @@
 #include <log4cxx/helpers/pool.h>
 #include <log4cxx/helpers/bytebuffer.h>
 
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;
 
diff -Naur apache-log4cxx-0.10.0.orig/src/main/cpp/socketoutputstream.cpp apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp
--- apache-log4cxx-0.10.0.orig/src/main/cpp/socketoutputstream.cpp	2008-04-01 00:34:09.000000000 +0200
+++ apache-log4cxx-0.10.0/src/main/cpp/socketoutputstream.cpp	2008-05-06 05:35:55.000000000 +0200
@@ -20,6 +20,8 @@
 #include <log4cxx/helpers/socket.h>
 #include <log4cxx/helpers/bytebuffer.h>
 
+#include <cstring>
+
 using namespace log4cxx;
 using namespace log4cxx::helpers;