--- cmake/onnxruntime.cmake.orig	2022-10-07 16:39:05.935164330 +0200
+++ cmake/onnxruntime.cmake	2022-10-07 16:39:56.702750071 +0200
@@ -195,6 +195,7 @@
 target_link_libraries(onnxruntime PRIVATE
     ${onnxruntime_INTERNAL_LIBRARIES}
     ${onnxruntime_EXTERNAL_LIBRARIES}
+    -liconv
 )
 
 set_property(TARGET onnxruntime APPEND_STRING PROPERTY LINK_FLAGS ${ONNXRUNTIME_SO_LINK_FLAG} ${onnxruntime_DELAYLOAD_FLAGS})
--- cmake/onnxruntime_python.cmake.orig	2022-10-07 16:40:36.134428297 +0200
+++ cmake/onnxruntime_python.cmake	2022-10-07 16:40:46.916340311 +0200
@@ -156,6 +156,7 @@
     onnxruntime_common
     onnxruntime_flatbuffers
     ${pybind11_lib}
+    -liconv
 )
 
 if (onnxruntime_ENABLE_LANGUAGE_INTEROP_OPS)
--- cmake/onnxruntime_unittests.cmake.orig	2022-10-07 16:41:33.413960900 +0200
+++ cmake/onnxruntime_unittests.cmake	2022-10-07 16:42:35.780451977 +0200
@@ -664,7 +664,7 @@
   SOURCES ${all_tests} ${onnxruntime_unittest_main_src}
   LIBS
     onnx_test_runner_common ${onnxruntime_test_providers_libs} ${onnxruntime_test_common_libs}
-    onnx_test_data_proto nlohmann_json::nlohmann_json
+    onnx_test_data_proto nlohmann_json::nlohmann_json -liconv
   DEPENDS ${all_dependencies}
 )
 if(NOT MSVC)
@@ -790,7 +790,7 @@
   )
 endif()
 
-target_link_libraries(onnx_test_runner PRIVATE onnx_test_runner_common ${GETOPT_LIB_WIDE} ${onnx_test_libs})
+target_link_libraries(onnx_test_runner PRIVATE onnx_test_runner_common ${GETOPT_LIB_WIDE} ${onnx_test_libs} -liconv)
 target_include_directories(onnx_test_runner PRIVATE ${ONNXRUNTIME_ROOT})
 if (onnxruntime_USE_ROCM)
   target_include_directories(onnx_test_runner PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/amdgpu/onnxruntime ${CMAKE_CURRENT_BINARY_DIR}/amdgpu/orttraining)
