commit 07ab6e565f939c54fff6580fc8463ea61662871a
Author: Gregory L. Lee <lee218@llnl.gov>
Date:   Tue May 20 14:53:35 2014 -0700

    re-boostrap to update configure

diff --git a/configure b/configure
index 6c4af7d..30901ea 100755
--- a/configure
+++ b/configure
@@ -15529,7 +15529,7 @@ fi
 done
 test -n "$MPICC" || MPICC="$CC"
 
-  for ac_prog in mpig++ mpiicpc mpxlC mpixlC
+  for ac_prog in mpig++ mpiCC mpicxx mpiicpc mpxlC mpixlC
 do
   # Extract the first word of "$ac_prog", so it can be a program name with args.
 set dummy $ac_prog; ac_word=$2
