--- src/Standard/Standard_CLocaleSentry.hxx.ORIG	2017-08-11 07:51:11.000000000 +0200
+++ src/Standard/Standard_CLocaleSentry.hxx	2017-08-18 01:07:11.639206272 +0200
@@ -30,7 +30,9 @@
   #endif
 
   //! We check _GNU_SOURCE for glibc extensions here and it is always defined by g++ compiler.
-  #if defined(_GNU_SOURCE) && !defined(__ANDROID__)
+  //! Musl libc does not provide xlocale.h
+  //! glibc 2.26+ does not provide xlocale.h (anymore)
+  #if defined(_GNU_SOURCE) && defined(__GLIBC__) && __GLIBC__ <= 2 && __GLIBC_MINOR__ < 26 && !defined(__ANDROID__)
     #define HAVE_XLOCALE_H
   #endif
 #endif // ifndef HAVE_LOCALE_H
