diff --git a/lib/spack/spack/vendor/altgraph/__init__.py b/lib/spack/spack/vendor/altgraph/__init__.py
index 45ce7bfe5f8..0fb21d77884 100644
--- a/lib/spack/spack/vendor/altgraph/__init__.py
+++ b/lib/spack/spack/vendor/altgraph/__init__.py
@@ -139,9 +139,8 @@
   @contributor: U{Reka Albert <http://www.phys.psu.edu/~ralbert/>}
 
 """
-import pkg_resources
 
-__version__ = pkg_resources.require("altgraph")[0].version
+__version__ = "0.17.3"
 
 
 class GraphError(ValueError):
