diff -r -u a/clients/benchmarks/CMakeLists.txt b/clients/benchmarks/CMakeLists.txt
--- a/clients/benchmarks/CMakeLists.txt	2021-08-14 18:54:53.356456513 -0600
+++ b/clients/benchmarks/CMakeLists.txt	2021-08-14 18:55:25.125354419 -0600
@@ -19,6 +19,7 @@
 target_link_libraries( rocsolver-bench PRIVATE
   cblas
   lapack
+  blas
   Threads::Threads
   hip::device
   rocsolver-common
diff -r -u a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
--- a/clients/gtest/CMakeLists.txt	2021-08-14 18:54:53.356456513 -0600
+++ b/clients/gtest/CMakeLists.txt	2021-08-14 18:55:16.581112850 -0600
@@ -89,6 +89,7 @@
 target_link_libraries( rocsolver-test PRIVATE
   cblas
   lapack
+  blas
   GTest::GTest
   hip::device
   rocsolver-common
