index f5efcf7..71e9a3e 100644
--- a/setup.py
+++ b/setup.py
@@ -40,13 +40,13 @@ setuptools.setup(
     license=module.__license__,
     description=module.__doc__,
     platforms='all',
-    long_description=read('readme.md'),
+    long_description="See https://github.com/uburuntu/throttler/blob/master/readme.md",
     long_description_content_type='text/markdown',
     url='https://github.com/uburuntu/{}'.format(module_name),
     download_url='https://github.com/uburuntu/{}/archive/master.zip'.format(module_name),
     packages=setuptools.find_packages(exclude=['examples', 'tests']),
     install_requires=[],
-    extras_require={'dev': load_requirements('requirements-dev.txt')},
+    extras_require={},
     keywords=['asyncio', 'aio-throttle', 'aiothrottle', 'aiothrottler', 'aiothrottling',
               'asyncio-throttle', 'rate-limit', 'rate-limiter', 'throttling', 'throttle', 'throttler'],
     classifiers=[
