--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java
+++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelConfiguration.java
@@ -150,6 +150,13 @@
       builder.put("PATH", null);
       builder.put("LD_LIBRARY_PATH", null);
     }
+
+    Map<String, String> spackEnv = System.getenv();
+    for (String envName : spackEnv.keySet()) {
+      if (envName.startsWith("SPACK_")) {
+        builder.put(envName, spackEnv.get(envName));
+      }
+    }
   }
 
   private static PathFragment determineShellExecutable(OS os, PathFragment fromOption) {
