--- a/compile.sh
+++ b/compile.sh
@@ -92,7 +92,7 @@
 log "Building output/bazel"
 # We set host and target platform directly since the defaults in @bazel_tools
 # have not yet been generated.
-bazel_build "src:bazel${EXE_EXT}" \
+CC=$SPACK_CC CXX=$SPACK_CXX bazel_build "src:bazel${EXE_EXT}" \
   --host_platform=//tools/platforms:host_platform \
   --platforms=//tools/platforms:target_platform \
   || fail "Could not build Bazel"
