--- a/src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileAction.java.orig    2020-06-08 13:42:14.035342560 -0400
+++ b/src/main/java/com/google/devtools/build/lib/rules/cpp/CppCompileAction.java 2020-06-08 13:42:25.149375458 -0400
@@ -963,7 +963,7 @@
           // are, it's probably due to a non-hermetic #include, & we should stop
           // the build with an error.
           if (execPath.startsWith(execRoot)) {
-            execPathFragment = execPath.relativeTo(execRoot); // funky but tolerable path
+             //  execPathFragment = execPath.relativeTo(execRoot); // funky but tolerable path
           } else {
             problems.add(execPathFragment.getPathString());
             continue;
