diff --git a/spack-expanded-archive/tools/cpp/unix_cc_configure.bzl.orig b/spack-expanded-archive/tools/cpp/unix_cc_configure.bzl
old mode 100644
new mode 100755
index aa46f55..ae3c689
--- spack-expanded-archive/tools/cpp/unix_cc_configure.bzl
+++ spack-expanded-archive/tools/cpp/unix_cc_configure.bzl
@@ -117,9 +117,23 @@ def get_escaped_cxx_inc_directories(repository_ctx, cc):
   else:
     inc_dirs = result.stderr[index1 + 1:index2].strip()
 
-  return [escape_string(repository_ctx.path(_cxx_inc_convert(p)))
-          for p in inc_dirs.split("\n")]
+  default_inc_directories = [
+    escape_string(repository_ctx.path(_cxx_inc_convert(p)))
+          for p in inc_dirs.split("\n")
+  ]
 
+  env = repository_ctx.os.environ
+  if "SPACK_DEPENDENCIES" in env:
+    for dep in env["SPACK_DEPENDENCIES"].split(":"):
+      path = dep + "/include"
+      # path = repository_ctx.os.path.join(dep, "include")
+      # if not repository_ctx.os.path.exists(path):
+      #   continue
+      default_inc_directories.append(
+        repository_ctx.path(_cxx_inc_convert(path))
+      )
+
+  return default_inc_directories
 
 def _add_option_if_supported(repository_ctx, cc, option):
   """Checks that `option` is supported by the C compiler. Doesn't %-escape the option."""
