diff --git a/tools/cpp/unix_cc_configure.bzl.orig b/tools/cpp/unix_cc_configure.bzl
index f67b9b5..cc952e0 100755
--- a/tools/cpp/unix_cc_configure.bzl.orig
+++ b/tools/cpp/unix_cc_configure.bzl
@@ -147,9 +147,18 @@ def get_escaped_cxx_inc_directories(repository_ctx, cc, additional_flags = []):
   else:
     inc_dirs = result.stderr[index1 + 1:index2].strip()
 
-  return [_prepare_include_path(repository_ctx, _cxx_inc_convert(p))
-          for p in inc_dirs.split("\n")]
+  default_inc_directories = [
+    _prepare_include_path(repository_ctx, _cxx_inc_convert(p))
+      for p in inc_dirs.split("\n")
+  ]
 
+  env = repository_ctx.os.environ
+  if "SPACK_DEPENDENCIES" in env:
+    for dep in env["SPACK_DEPENDENCIES"].split(":"):
+      path = dep + "/include"
+      default_inc_directories.append(path)
+
+  return default_inc_directories
 
 def _is_option_supported(repository_ctx, cc, option):
   """Checks that `option` is supported by the C compiler. Doesn't %-escape the option."""
