diff -Naur a/tools/cpp/unix_cc_configure.bzl b/tools/cpp/unix_cc_configure.bzl
--- a/tools/cpp/unix_cc_configure.bzl	1980-01-01 00:00:00.000000000 -0800
+++ b/tools/cpp/unix_cc_configure.bzl	2022-06-30 23:35:14.000000000 -0700
@@ -417,10 +417,7 @@
         bazel_linklibs,
         False,
     ), ":")
-    gold_or_lld_linker_path = (
-        _find_linker_path(repository_ctx, cc, "lld", is_clang) or
-        _find_linker_path(repository_ctx, cc, "gold", is_clang)
-    )
+    gold_or_lld_linker_path = None
     cc_path = repository_ctx.path(cc)
     if not str(cc_path).startswith(str(repository_ctx.path(".")) + "/"):
         # cc is outside the repository, set -B
@@ -531,7 +528,6 @@
                     # Security hardening requires optimization.
                     # We need to undef it as some distributions now have it enabled by default.
                     "-U_FORTIFY_SOURCE",
-                    "-fstack-protector",
                     # All warnings are enabled. Maybe enable -Werror as well?
                     "-Wall",
                     # Enable a few more warnings that aren't part of -Wall.
