--- portage.orig/portage/search/pairs.cc	2018-07-16 13:42:12.000000000 -0600
+++ portage/portage/search/pairs.cc	2020-01-09 15:32:43.289085000 -0700
@@ -323,8 +323,8 @@
   // check for completely outside source boxes
   bool outside = false;
   for (size_t m=0;m<dim;m++) {
-    if (y[m][j]<=cminmax[0][m]) outside=true;
-    if (y[m][j]>=cminmax[1][m]) outside=true;
+    if (y[m][j]<=cminmax[0][m]) { outside=true; break; }
+    if (y[m][j]>=cminmax[1][m]) { outside=true; break; }
   }
   if (outside) return pairlist;
 
@@ -347,8 +347,8 @@
     // check that y is contained
     bool inside = true;
     for(size_t m=0; m<dim; m++) {
-      if (y[m][j] <= xll[m]) inside = false;
-      if (y[m][j] >= xur[m]) inside = false;
+      if (y[m][j] <= xll[m]) { inside = false; break; }
+      if (y[m][j] >= xur[m]) { inside = false; break; }
     }
 
     // add pair: put x's in this y-cell onto neighbor list, if inside
