diff --git a/cmake/PythonModules.cmake b/cmake/PythonModules.cmake
index 96727cc..51a970c 100755
--- a/cmake/PythonModules.cmake
+++ b/cmake/PythonModules.cmake
@@ -53,7 +53,7 @@ function(py_add_module NAME)
     )
 
 endfunction()
-set(PYTHON_SEARCH_VERSIONS 2.7 3.5 3.6 3.7 3.8 3.9)
+set(PYTHON_SEARCH_VERSIONS 3.5 3.6 3.7 3.8 3.9)
 
 set(_PYTHON_VERSIONS)
 foreach(PYTHON_VERSION ${PYTHON_SEARCH_VERSIONS})
