diff --git a/CMake/common/CommonCompiler.cmake b/CMake/common/CommonCompiler.cmake
index 42d40e5..37f29ab 100644
--- a/CMake/common/CommonCompiler.cmake
+++ b/CMake/common/CommonCompiler.cmake
@@ -110,6 +110,9 @@ if(CMAKE_COMPILER_IS_GCC OR CMAKE_COMPILER_IS_CLANG)
     if(GCC_COMPILER_VERSION VERSION_GREATER 4.5)
       list(APPEND COMMON_C_FLAGS -fmax-errors=5)
     endif()
+    if(GCC_COMPILER_VERSION VERSION_GREATER 8.0)
+      list(APPEND COMMON_CXX_FLAGS -faligned-new)
+    endif()
   endif()
 
   list(APPEND COMMON_CXX_FLAGS_RELEASE -Wuninitialized)
