diff -ru magma-2.5.0-orig/CMakeLists.txt magma-2.5.0/CMakeLists.txt
--- magma-2.5.0-orig/CMakeLists.txt	2019-01-02 11:18:39.000000000 -0800
+++ magma-2.5.0/CMakeLists.txt	2019-04-03 15:58:01.871234891 -0700
@@ -440,18 +440,20 @@
 # compile MAGMA sparse library
 
 # sparse doesn't have Fortran at the moment, so no need for above shenanigans
-include_directories( sparse/include )
-include_directories( sparse/control )
-include_directories( testing )
-cuda_add_library( magma_sparse ${libsparse_all} )
-target_link_libraries( magma_sparse
-	magma
+if (MAGMA_SPARSE)
+  include_directories( sparse/include )
+  include_directories( sparse/control )
+  include_directories( testing )
+  cuda_add_library( magma_sparse ${libsparse_all} )
+  target_link_libraries( magma_sparse
+	  magma
 	${LAPACK_LIBRARIES}
 	${CUDA_CUDART_LIBRARY}
 	${CUDA_CUBLAS_LIBRARIES}
 	${CUDA_cusparse_LIBRARY}
-)
-set( LIBS_SPARSE ${LIBS} magma_sparse )
+  )
+  set( LIBS_SPARSE ${LIBS} magma_sparse )
+endif()
 
 
 # ----------------------------------------
@@ -480,23 +482,31 @@
 
 # ----------------------------------------
 # compile each sparse tester
-set( CMAKE_RUNTIME_OUTPUT_DIRECTORY sparse/testing )
-foreach( TEST ${sparse_testing_all} )
+if (MAGMA_SPARSE)
+  set( CMAKE_RUNTIME_OUTPUT_DIRECTORY sparse/testing )
+  foreach( TEST ${sparse_testing_all} )
 	string( REGEX REPLACE "\\.(cpp|f90|F90)"     "" EXE ${TEST} )
 	string( REGEX REPLACE "sparse/testing/" "" EXE ${EXE} )
 	#message( "${TEST} --> ${EXE}" )
 	add_executable( ${EXE} ${TEST} )
 	target_link_libraries( ${EXE} ${LIBS_SPARSE} )
-endforeach()
+  endforeach()
+endif()
 
 
 # ----------------------------------------
 # what to install
-install( TARGETS magma magma_sparse ${blas_fix}
+set(MAGMA_TARGETS magma)
+set(MAGMA_HEADERS_PATTERNS include/*.h)
+if (MAGMA_SPARSE)
+  set(MAGMA_TARGETS ${MAGMA_TARGETS} magma_sparse)
+  set(MAGMA_HEADERS_PATTERNS ${MAGMA_HEADERS_PATTERNS} sparse/include/*.h)
+endif()
+install( TARGETS ${MAGMA_TARGETS} ${blas_fix}
          RUNTIME DESTINATION bin
          LIBRARY DESTINATION lib
          ARCHIVE DESTINATION lib )
-file( GLOB headers include/*.h sparse/include/*.h )
+file( GLOB headers ${MAGMA_HEADERS_PATTERNS} )
 install( FILES ${headers}
          DESTINATION include )
 
@@ -509,4 +519,6 @@
 message( STATUS "    NFLAGS       ${CUDA_NVCC_FLAGS}" )
 message( STATUS "    FFLAGS       ${CMAKE_Fortran_FLAGS}" )
 message( STATUS "    LIBS         ${LIBS}" )
-message( STATUS "    LIBS_SPARSE  ${LIBS_SPARSE}" )
+if (MAGMA_SPARSE)
+  message( STATUS "    LIBS_SPARSE  ${LIBS_SPARSE}" )
+endif()
