--- a/numpy/distutils/fcompiler/gnu.py.orig	2015-08-17 11:31:52.000000000 -0700
+++ b/numpy/distutils/fcompiler/gnu.py	2015-08-17 11:32:48.000000000 -0700
@@ -252,7 +252,7 @@
     #       GNU Fortran 95 (GCC) 4.2.0 20060218 (experimental)
     #       GNU Fortran (GCC) 4.3.0 20070316 (experimental)
 
-    possible_executables = ['gfortran', 'f95']
+    possible_executables = ['fc', 'f95']
     executables = {
         'version_cmd'  : ["<F90>", "--version"],
         'compiler_f77' : [None, "-Wall", "-g", "-ffixed-form",
