--- a/ambertools_tmpdir/AmberTools/src/fftw-3.3/config.guess	2020-09-30 13:28:45.035344970 -0700
+++ b/ambertools_tmpdir/AmberTools/src/fftw-3.3/config.guess	2020-09-30 13:28:55.534923364 -0700
@@ -858,6 +858,9 @@
     i*86:Minix:*:*)
 	echo ${UNAME_MACHINE}-pc-minix
 	exit ;;
+    aarch64:Linux:*:*)
+	echo aarch64-unknown-linux-gnu
+	exit ;;
     alpha:Linux:*:*)
 	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo` in
 	  EV5)   UNAME_MACHINE=alphaev5 ;;
--- a/ambertools_tmpdir/AmberTools/src/xblas/config/config.guess	2020-09-30 15:30:11.573779015 -0700
+++ b/ambertools_tmpdir/AmberTools/src/xblas/config/config.guess	2020-09-30 15:30:46.932366326 -0700
@@ -863,6 +863,9 @@
     i*86:Minix:*:*)
 	echo ${UNAME_MACHINE}-pc-minix
 	exit ;;
+    aarch64:Linux:*:*)
+       echo aarch64-unknown-linux-gnu
+       exit ;;
     alpha:Linux:*:*)
 	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo` in
 	  EV5)   UNAME_MACHINE=alphaev5 ;;
