diff --git a/shared_lib/Makefile b/shared_lib/Makefile
index 6be951030..c8e06e32f 100644
--- a/shared_lib/Makefile
+++ b/shared_lib/Makefile
@@ -41,7 +41,6 @@ SHARED_LINKED_LIB := $(SPDK_ROOT_DIR)/build/lib/libspdk.so
 SHARED_REALNAME_LIB := $(SHARED_LINKED_LIB).$(SO_SUFFIX_ALL)
 IGNORED_LIBS := libspdk.so
 IGNORED_LIBS += libspdk_spdk_mock.so
-IGNORED_LIBS += libspdk_env_dpdk.so
 
 SPDK_SHARED_LIBS := $(filter-out $(IGNORED_LIBS), $(sort $(notdir $(wildcard $(SPDK_ROOT_DIR)/build/lib/*.so))))

