diff --git a/mk/spdk.common.mk b/mk/spdk.common.mk
index ca11c49ab..5ac255527 100644
--- a/mk/spdk.common.mk
+++ b/mk/spdk.common.mk
@@ -279,7 +279,8 @@ INSTALL_SHARED_LIB=\
 		perm_mode=644; \
 	fi; \
 	install -m $$perm_mode "$(SHARED_REALNAME_LIB)" "$(DESTDIR)$(libdir)/"; \
-	$(call spdk_install_lib_symlink,$(notdir $(SHARED_REALNAME_LIB)),$(notdir $(SHARED_LINKED_LIB)));
+	$(call spdk_install_lib_symlink,$(notdir $(SHARED_REALNAME_LIB)),$(notdir $(SHARED_LINKED_LIB))); \
+        $(call spdk_install_lib_symlink,$(notdir $(SHARED_REALNAME_LIB)),$(notdir $(SHARED_LINKED_LIB).$(SO_VER)));
 
 # Install an app binary
 INSTALL_APP=\

