--- a/alps/config/run_test.cmake  2020-12-10 10:05:48.925076699 +0900
+++ b/alps/config/run_test.cmake  2020-12-10 11:37:18.180959480 +0900
@@ -3,7 +3,7 @@
 #      (See accompanying file LICENSE_1_0.txt or copy at
 #          http://www.boost.org/LICENSE_1_0.txt)

-find_program(cmd_path ${cmd} ${binarydir} ${dllexedir})
+find_program(cmd_path ${cmd} ${binarydir} ${dllexedir} NO_SYSTEM_ENVIRONMENT_PATH)

 find_file(input_path ${input}.input ${binarydir} ${sourcedir})
 if(NOT input_path)
