diff --git a/alps/src/alps/CMakeLists.txt b/alps/src/alps/CMakeLists.txt
index ae73f13..0d1ba34 100644
--- a/alps/src/alps/CMakeLists.txt
+++ b/alps/src/alps/CMakeLists.txt
@@ -93,7 +93,7 @@ if (Boost_FOUND)
     set(ALPS_LINK_LIBS ${ALPS_LINK_LIBS} ${PYTHON_LIBRARY} ${PYTHON_EXTRA_LIBS})
   endif(PYTHONLIBS_FOUND)
   if(MPI_FOUND)
-    set(ALPS_LINK_LIBS ${ALPS_LINK_LIBS} ${MPI_LIBRARIES} ${MPI_EXTRA_LIBRARY})
+    set(ALPS_LINK_LIBS ${ALPS_LINK_LIBS} ${MPI_LIBRARIES})
   endif(MPI_FOUND)
   target_link_libraries(alps ${ALPS_LINK_LIBS})
 else (Boost_FOUND)
