diff --git a/cmake/VerifyCompiler.cmake b/cmake/VerifyCompiler.cmake
index 846072f..281ccdc 100644
--- a/cmake/VerifyCompiler.cmake
+++ b/cmake/VerifyCompiler.cmake
@@ -22,8 +22,6 @@
 
 list(APPEND CMAKE_PREFIX_PATH /opt/rocm /opt/rocm/hip)
 if(CMAKE_CXX_COMPILER MATCHES ".*nvcc$" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-    list(APPEND CMAKE_MODULE_PATH /opt/rocm/hip/cmake)
-    find_package(hip QUIET CONFIG PATHS /opt/rocm)
     if(NOT hip_FOUND)
         find_package(HIP REQUIRED)
     endif()
