--- WRF-4.2/phys/module_mp_SBM_polar_radar.F	2020-04-23 22:38:37.000000000 +0530
+++ WRF-4.2/phys/module_mp_SBM_polar_radar_aocc.F	2020-12-29 11:45:52.329495585 +0530
@@ -1534,7 +1534,7 @@
                     (1.0d0-(1.0d0-fract_volume_water)*ratc))
 ! new change 18.01.09                                         (start)
       if(fract_volume_water.gt.1.0d-10) then
-         t=derf((1.0d0-fract_volume_water)/fract_volume_water-0.2d0)
+         t=erf((1.0d0-fract_volume_water)/fract_volume_water-0.2d0)
       else
          t=1.0d0
       endif
