--- pyext/setup.py.in.orig      2016-09-30 09:13:02.000000000 +0200
+++ pyext/setup.py.in   2016-09-30 09:13:21.000000000 +0200
@@ -34,7 +34,7 @@
 extns = [ext("util"),  ext("core", statics=static_files, depends=core_depends)]

 ## Enable building of ROOT extension if ROOT is present
-if os.environ.has_key("BUILD_ROOTCOMPAT"):
+if "BUILD_ROOTCOMPAT" in os.environ:
     try:
         # import ROOT
         # TODO: Need to test for and use root-config in configure
