--- pyext/setup.py.in.orig	2017-02-01 15:24:41.000000000 +0100
+++ pyext/setup.py.in	2017-02-01 15:25:04.000000000 +0100
@@ -34,7 +34,7 @@
 extns = [ext("util"),  ext("core", statics=static_files, depends=core_depends)]
 
 ## Enable building of ROOT extension if ROOT is present
-if os.environ.has_key("BUILD_ROOTCOMPAT"):
+if "BUILD_ROOTCOMPAT" in os.environ:
     try:
         # import ROOT
         # TODO: Need to test for and use root-config in configure
