diff --git a/src/umpire/ResourceManager.hpp b/src/umpire/ResourceManager.hpp
index 168655c..a04240a 100644
--- a/src/umpire/ResourceManager.hpp
+++ b/src/umpire/ResourceManager.hpp
@@ -13,6 +13,8 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+#include <cstdlib>
+#include <stdexcept>
 
 #include "camp/resource.hpp"
 #include "umpire/Allocator.hpp"
diff --git a/src/umpire/op/MemoryOperation.hpp b/src/umpire/op/MemoryOperation.hpp
index d5888ed..3b11607 100644
--- a/src/umpire/op/MemoryOperation.hpp
+++ b/src/umpire/op/MemoryOperation.hpp
@@ -8,6 +8,8 @@
 #define UMPIRE_MemoryOperation_HPP
 
 #include <cstddef>
+#include <cstdlib>
+#include <stdexcept>
 
 #include "camp/resource.hpp"
 #include "umpire/util/AllocationRecord.hpp"
