diff --git a/umpire-config.cmake.in b/umpire-config.cmake.in
index a98ad5fe..4e54e173 100644
--- a/umpire-config.cmake.in
+++ b/umpire-config.cmake.in
@@ -7,6 +7,13 @@
 get_filename_component(UMPIRE_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 set(UMPIRE_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include")
 
+if (NOT TARGET camp)
+  if (NOT DEFINED camp_DIR)
+    set(camp_DIR @CMAKE_INSTALL_PREFIX@/lib/cmake/camp)
+  endif ()
+  find_package(camp REQUIRED)
+endif ()
+
 set(Umpire_VERSION_MAJOR  @Umpire_VERSION_MAJOR@)
 set(Umpire_VERSION_MINOR @Umpire_VERSION_MINOR@)
 set(Umpire_VERSION_PATCH @Umpire_VERSION_PATCH@)
