diff --git a/cmake/SetupUmpireThirdParty.cmake b/cmake/SetupUmpireThirdParty.cmake
index fc351369..6e82ab6e 100644
--- a/cmake/SetupUmpireThirdParty.cmake
+++ b/cmake/SetupUmpireThirdParty.cmake
@@ -80,29 +80,40 @@ if (UMPIRE_ENABLE_SQLITE_EXPERIMENTAL)
   find_package(SQLite3 REQUIRED)
 endif()
 
-set(UMPIRE_NEEDS_BLT_TPLS False)
-if (UMPIRE_ENABLE_MPI OR UMPIRE_ENABLE_HIP OR UMPIRE_ENABLE_OPENMP OR UMPIRE_ENABLE_CUDA)
-  set(UMPIRE_NEEDS_BLT_TPLS True)
-
-  if (NOT BLT_EXPORTED)
-    set(BLT_EXPORTED On CACHE BOOL "" FORCE)
-    blt_import_library(NAME          blt_stub EXPORTABLE On)
-    set_target_properties(blt_stub PROPERTIES EXPORT_NAME blt::blt_stub)
-    install(TARGETS blt_stub
-      EXPORT               bltTargets)
-    blt_export_tpl_targets(EXPORT bltTargets NAMESPACE blt)
-    install(EXPORT bltTargets
-      DESTINATION  lib/cmake/umpire)
-  elseif (UMPIRE_ENABLE_MPI)
+
+#################################################
+# use bonafide cmake targets for openmp and mpi
+#################################################
+set(umpire_mpi_deps "")
+if (UMPIRE_ENABLE_MPI)
+  if(ENABLE_FIND_MPI)
+      set (umpire_mpi_deps MPI::MPI_CXX)
+  endif()
+endif()
+
+set(umpire_openmp_deps "")
+if (UMPIRE_ENABLE_OPENMP)
+  set (umpire_openmp_deps OpenMP::OpenMP_CXX)
+endif ()
+
+#################################################
+# export necessary blt targets
+#################################################
+
+set(UMPIRE_BLT_TPL_DEPS_EXPORTS)
+
+blt_list_append(TO UMPIRE_BLT_TPL_DEPS_EXPORTS ELEMENTS cuda cuda_runtime IF UMPIRE_ENABLE_CUDA)
+blt_list_append(TO UMPIRE_BLT_TPL_DEPS_EXPORTS ELEMENTS blt_hip blt_hip_runtime IF UMPIRE_ENABLE_HIP)
+
+foreach(dep ${UMPIRE_BLT_TPL_DEPS_EXPORTS})
     # If the target is EXPORTABLE, add it to the export set
-    get_target_property(_is_imported mpi IMPORTED)
+    get_target_property(_is_imported ${dep} IMPORTED)
     if(NOT ${_is_imported})
-      install(TARGETS              mpi
-        EXPORT               ${arg_EXPORT})
-      # Namespace target to avoid conflicts
-      set_target_properties(mpi PROPERTIES EXPORT_NAME blt::mpi)
-      install(EXPORT bltTargets
-        DESTINATION  lib/cmake/umpire)
+        install(TARGETS              ${dep}
+                EXPORT               umpire-targets
+                DESTINATION          lib/cmake/umpire)
+        # Namespace target to avoid conflicts
+        set_target_properties(${dep} PROPERTIES EXPORT_NAME umpire::blt_tpl_exports_${dep})
     endif()
-  endif()
-endif()
+endforeach()
+
diff --git a/src/umpire/interface/c_fortran/CMakeLists.txt b/src/umpire/interface/c_fortran/CMakeLists.txt
index ab675adf..22e4dc90 100644
--- a/src/umpire/interface/c_fortran/CMakeLists.txt
+++ b/src/umpire/interface/c_fortran/CMakeLists.txt
@@ -43,7 +43,7 @@ set_source_files_properties(
 set(umpire_interface_c_fortran_depends camp umpire_util)
 blt_list_append( TO umpire_interface_c_fortran_depends ELEMENTS cuda_runtime IF UMPIRE_ENABLE_CUDA )
-blt_list_append( TO umpire_interface_c_fortran_depends ELEMENTS blt::hip_runtime IF UMPIRE_ENABLE_HIP )
-blt_list_append( TO umpire_interface_c_fortran_depends ELEMENTS mpi IF UMPIRE_ENABLE_MPI )
+blt_list_append( TO umpire_interface_c_fortran_depends ELEMENTS blt_hip_runtime IF UMPIRE_ENABLE_HIP )
+blt_list_append( TO umpire_interface_c_fortran_depends ELEMENTS ${umpire_mpi_deps} IF UMPIRE_ENABLE_MPI )
 
 blt_add_library(
   NAME umpire_interface
diff --git a/src/umpire/util/CMakeLists.txt b/src/umpire/util/CMakeLists.txt
index 551f8188..02eb13e4 100644
--- a/src/umpire/util/CMakeLists.txt
+++ b/src/umpire/util/CMakeLists.txt
@@ -73,14 +73,15 @@ endif()
 
 if (UMPIRE_ENABLE_MPI)
   set (umpire_util_depends
-    ${umpire_util_depends}
-    mpi)
+       ${umpire_util_depends}
+       ${umpire_mpi_deps}
+    )
 endif ()
 
 if (UMPIRE_ENABLE_OPENMP_TARGET)
   set (umpire_util_depends
-    ${umpire_util_depends}
-    openmp)
+       ${umpire_util_depends}
+       ${umpire_openmp_deps})
 endif ()
 
 if (UMPIRE_ENABLE_HIP)
