diff --git a/src/tpl/CMakeLists.txt b/src/tpl/CMakeLists.txt
index 6803f0a4..60269f88 100644
--- a/src/tpl/CMakeLists.txt
+++ b/src/tpl/CMakeLists.txt
@@ -50,7 +50,13 @@ target_include_directories(
   umpire_tpl_json
   INTERFACE
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/tpl>
-  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>)
+
+blt_convert_to_system_includes(TARGET umpire_tpl_json)
+
+target_include_directories(
+  umpire_tpl_json
+  INTERFACE
   $<INSTALL_INTERFACE:include>)
 
 install(FILES
@@ -64,8 +70,6 @@ install(TARGETS
   LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib)
 
-blt_patch_target(NAME umpire_tpl_json
-                 TREAT_INCLUDES_AS_SYSTEM ON)
 
 #
 # CLI11 Option Parsing Headers
@@ -82,7 +86,13 @@ target_include_directories(
   umpire_tpl_CLI11
   INTERFACE
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/tpl>
-  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>)
+
+blt_convert_to_system_includes(TARGET umpire_tpl_CLI11)
+
+target_include_directories(
+  umpire_tpl_CLI11
+  INTERFACE
   $<INSTALL_INTERFACE:include>)
 
 install(FILES
@@ -96,9 +106,6 @@ install(TARGETS
   LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib)
 
-blt_patch_target(NAME umpire_tpl_CLI11
-                 TREAT_INCLUDES_AS_SYSTEM ON)
-
 add_subdirectory(umpire/judy)
 
 if (NOT TARGET camp)
@@ -153,7 +160,14 @@ target_include_directories(
   umpire_tpl_fmt
   INTERFACE
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/tpl>
-  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>)
+
+# Avoid warnings from fmt (so we can still use -Werror)
+blt_convert_to_system_includes(TARGET umpire_tpl_fmt)
+
+target_include_directories(
+  umpire_tpl_fmt
+  INTERFACE
   $<INSTALL_INTERFACE:include>)
 
 if (C_COMPILER_FAMILY_IS_XL)
@@ -168,21 +182,17 @@ if (C_COMPILER_FAMILY_IS_PGI)
   set(_fmt_warning_disable_flag
    "--diag_suppress 1625;--diag_suppress 185;--diag_suppress 811;--diag_suppress 186")
    
-   if (ENABLE_FORTRAN)
+  if (ENABLE_FORTRAN)
     target_compile_options(umpire_tpl_fmt
       INTERFACE 
       $<$<NOT:$<COMPILE_LANGUAGE:Fortran>>:${_fmt_warning_disable_flag}>)
-    else ()
-      target_compile_options(umpire_tpl_fmt
-        INTERFACE 
-        ${_fmt_warning_disable_flag})
-    endif ()
+  else ()
+    target_compile_options(umpire_tpl_fmt
+      INTERFACE 
+      ${_fmt_warning_disable_flag})
+  endif ()
 endif ()
 
-# Avoid warnings from fmt (so we can still use -Werror)
-blt_patch_target(NAME umpire_tpl_fmt
-  TREAT_INCLUDES_AS_SYSTEM ON)
-
 if (C_COMPILER_FAMILY_IS_GNU)
   target_compile_options(umpire_tpl_fmt
     INTERFACE 
diff --git a/src/umpire/interface/c_fortran/CMakeLists.txt b/src/umpire/interface/c_fortran/CMakeLists.txt
index cf0d6932..ab675adf 100644
--- a/src/umpire/interface/c_fortran/CMakeLists.txt
+++ b/src/umpire/interface/c_fortran/CMakeLists.txt
@@ -57,15 +57,22 @@ target_include_directories(
   PUBLIC
   $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>
   $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
-  $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:Fortran>:${CMAKE_Fortran_MODULE_DIRECTORY}>>
-  $<INSTALL_INTERFACE:$<$<COMPILE_LANGUAGE:Fortran>:include/umpire>> # for Fortran module files
   $<INSTALL_INTERFACE:include>)
 
 install(FILES
   ${umpire_interface_c_fortran_headers}
   DESTINATION include/umpire/interface/c_fortran)
 
-install(DIRECTORY
-  ${CMAKE_Fortran_MODULE_DIRECTORY}/
-  DESTINATION include/umpire
-  FILES_MATCHING PATTERN "*.mod")
+if(UMPIRE_ENABLE_FORTRAN)
+  target_include_directories(
+    umpire_interface
+    PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_Fortran_MODULE_DIRECTORY}>
+    $<INSTALL_INTERFACE:include/umpire>)
+
+  install(DIRECTORY
+    ${CMAKE_Fortran_MODULE_DIRECTORY}/
+    DESTINATION include/umpire
+    FILES_MATCHING PATTERN "*.mod")
+endif()
+
