diff -ruN spack-src/cmake/SetupCompilerFlags.cmake spack-src-patched/cmake/SetupCompilerFlags.cmake
--- spack-src/cmake/SetupCompilerFlags.cmake	2022-10-12 08:05:03.538390165 -0700
+++ spack-src-patched/cmake/SetupCompilerFlags.cmake	2022-10-12 09:47:56.317645003 -0700
@@ -11,20 +11,20 @@
 
 message(STATUS "Checking for std::filesystem")
 
-include(CheckCXXSourceCompiles)
-check_cxx_source_compiles(
-  "#include <iostream>
-  #include <filesystem>
-
-  int main(int, char**)
-  {
+blt_check_code_compiles(CODE_COMPILES UMPIRE_ENABLE_FILESYSTEM
+                        SOURCE_STRING
+[=[
+#include <iostream>
+#include <filesystem>
 
+int main(int, char**)
+{
     auto path = std::filesystem::path(\".\");
     (void)(path);
 
     return 0;
-  }"
-  UMPIRE_ENABLE_FILESYSTEM)
+}
+]=])
 
 if (UMPIRE_ENABLE_FILESYSTEM)
   message(STATUS "std::filesystem found")
