diff --git a/external/dyninst/instructionAPI/src/InstructionDecoder-aarch64.C b/external/dyninst/instructionAPI/src/InstructionDecoder-aarch64.C
index 096e46ad2..5bae7b6fa 100644
--- a/external/dyninst/instructionAPI/src/InstructionDecoder-aarch64.C
+++ b/external/dyninst/instructionAPI/src/InstructionDecoder-aarch64.C
@@ -28,6 +28,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */

+#include <array>
 #include "InstructionDecoder-aarch64.h"

 namespace Dyninst
diff --git a/external/dyninst/instructionAPI/src/InstructionDecoder-amdgpu-vega.C b/external/dyninst/instructionAPI/src/InstructionDecoder-amdgpu-vega.C
index 582c055e5..4dc706edb 100644
--- a/external/dyninst/instructionAPI/src/InstructionDecoder-amdgpu-vega.C
+++ b/external/dyninst/instructionAPI/src/InstructionDecoder-amdgpu-vega.C
@@ -28,6 +28,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */

+#include <array>
 #include "Ternary.h"
 #include "InstructionDecoder-amdgpu-vega.h"

diff --git a/external/timemory/source/timemory/backends/gperftools.hpp b/external/timemory/source/timemory/backends/gperftools.hpp
index 9177da5..d4c6d74 100644
--- a/external/timemory/source/timemory/backends/gperftools.hpp
+++ b/external/timemory/source/timemory/backends/gperftools.hpp
@@ -26,6 +26,7 @@

 #include <iostream>
 #include <sstream>
+#include <cstdint>
 #include <string>

 #if defined(TIMEMORY_USE_GPERFTOOLS) || defined(TIMEMORY_USE_GPERFTOOLS_TCMALLOC)
