diff -Naur lz4-1.8.1.2/programs/Makefile lz4-1.8.1.2x/programs/Makefile
--- lz4-1.8.1.2/programs/Makefile	2018-01-14 00:21:43.000000000 -0800
+++ lz4-1.8.1.2x/programs/Makefile	2018-03-26 01:13:54.467096931 -0700
@@ -50,6 +50,7 @@
             -Wswitch-enum -Wdeclaration-after-statement -Wstrict-prototypes \
             -Wpointer-arith -Wstrict-aliasing=1
 CFLAGS   += $(DEBUGFLAGS) $(MOREFLAGS)
+LDFLAGS   = -lrt
 FLAGS     = $(CFLAGS) $(CPPFLAGS) $(LDFLAGS)
 
 LZ4_VERSION=$(LIBVER)
