diff -ur spack-src/Release.org/include/pplx/pplxlinux.h spack-src/Release/include/pplx/pplxlinux.h
--- spack-src/Release.org/include/pplx/pplxlinux.h	2020-03-12 13:52:40.161917503 +0900
+++ spack-src/Release/include/pplx/pplxlinux.h	2020-03-12 14:05:50.834896829 +0900
@@ -240,6 +240,11 @@
     {
     public:
         _PPLXIMP virtual void schedule( TaskProc_t proc, _In_ void* param);
+#if defined(__APPLE__)
+    virtual ~apple_scheduler() {}
+#else
+    virtual ~linux_scheduler() {}
+#endif
     };
 
 } // namespace details
diff -ur spack-src/Release.org/libs/websocketpp/websocketpp/transport/asio/connection.hpp spack-src/Release/libs/websocketpp/websocketpp/transport/asio/connection.hpp
--- spack-src/Release.org/libs/websocketpp/websocketpp/transport/asio/connection.hpp	2020-03-12 13:52:40.201921703 +0900
+++ spack-src/Release/libs/websocketpp/websocketpp/transport/asio/connection.hpp	2020-03-12 14:09:13.586186467 +0900
@@ -422,7 +422,7 @@
         m_io_service = io_service;
 
         if (config::enable_multithreading) {
-            m_strand = lib::make_shared<boost::asio::strand>(
+            m_strand = lib::make_shared<boost::asio::io_service::strand>(
                 lib::ref(*io_service));
 
             m_async_read_handler = m_strand->wrap(lib::bind(
diff -ur spack-src/Release.org/src/uri/uri.cpp spack-src/Release/src/uri/uri.cpp
--- spack-src/Release.org/src/uri/uri.cpp	2020-03-12 13:52:40.241925902 +0900
+++ spack-src/Release/src/uri/uri.cpp	2020-03-12 13:53:58.610152210 +0900
@@ -22,12 +22,12 @@
     // canonicalize components first
 
     // convert scheme to lowercase
-    std::transform(m_scheme.begin(), m_scheme.end(), m_scheme.begin(), [this](utility::char_t c) {
+    std::transform(m_scheme.begin(), m_scheme.end(), m_scheme.begin(), [](utility::char_t c) {
         return (utility::char_t)tolower(c);
     });
 
     // convert host to lowercase
-    std::transform(m_host.begin(), m_host.end(), m_host.begin(), [this](utility::char_t c) {
+    std::transform(m_host.begin(), m_host.end(), m_host.begin(), [](utility::char_t c) {
         return (utility::char_t)tolower(c);
     });
 
