Index: spack-src/tools/build/src/tools/python.jam
===================================================================
--- spack-src/tools/build/src/tools/python.jam
+++ spack-src/tools/build/src/tools/python.jam
@@ -501,6 +501,10 @@ local rule probe ( python-cmd )
                 sys.$(s) = [ SUBST $(output) "\\<$(s)=([^$(nl)]+)" $1 ] ;
             }
         }
+         # Try to get python abiflags
+        full-cmd = $(python-cmd)" -c \"from sys import abiflags; print(abiflags, end='')\"" ;
+
+        sys.abiflags = [ SHELL $(full-cmd) ] ;
         return $(output) ;
     }
 }
@@ -510,7 +514,7 @@ local rule probe ( python-cmd )
 # have a value based on the information given.
 #
 local rule compute-default-paths ( target-os : version ? : prefix ? :
-    exec-prefix ? )
+    exec-prefix ? : abiflags ? )
 {
     exec-prefix ?= $(prefix) ;
 
@@ -547,7 +551,7 @@ local rule compute-default-paths ( targe
     }
     else
     {
-        local default-include-path = $(prefix)/include/python$(version) ;
+        local default-include-path = $(prefix)/include/python$(version)$(abiflags) ;
         if ! [ path.exists $(default-include-path) ] && [ path.exists $(default-include-path)m ]
         {
             default-include-path = $(default-include-path)m ;
@@ -798,7 +802,7 @@ local rule configure ( version ? : cmd-o
                     exec-prefix = $(sys.exec_prefix) ;
 
                     compute-default-paths $(target-os) : $(sys.version) :
-                        $(sys.prefix) : $(sys.exec_prefix) ;
+                        $(sys.prefix) : $(sys.exec_prefix) : $(sys.abiflags) ;
 
                     version = $(sys.version) ;
                     interpreter-cmd ?= $(cmd) ;
