diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5904f78..a0b0c92 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -204,7 +204,7 @@ else()
         set( tensile_fork "ROCmSoftwarePlatform" CACHE STRING "Tensile fork to use" )
         file (STRINGS "tensilelite_tag.txt" read_tensile_tag)
         set( tensile_tag ${read_tensile_tag} CACHE STRING "Tensile tag to download" )
-        virtualenv_install("git+https://github.com/${tensile_fork}/hipBLASLt.git@${tensile_tag}#subdirectory=tensilelite")
+        virtualenv_install("git+https://github.com/ROCm/hipBLASLt.git@modify-tensilelite-spack-6.4#subdirectory=tensilelite")
 
         message (STATUS "using GIT Tensile fork=${tensile_fork} from branch=${tensile_tag}")
       endif()
diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index aac8506..345147f 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -123,6 +123,7 @@ target_include_directories(hipsparselt
                            PUBLIC  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/library/include>
                                    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
                                    $<BUILD_INTERFACE:${HIP_INCLUDE_DIRS}>
+                                   $<BUILD_INTERFACE:${HIPSPARSE_INCLUDE_DIRS}>
                                    $<INSTALL_INTERFACE:include>
 )
 
