diff -Nur netcdf-4.3.3/CMakeLists.txt netcdf-4.3.3.mpi/CMakeLists.txt
--- netcdf-4.3.3/CMakeLists.txt	2015-02-12 16:44:35.000000000 -0500
+++ netcdf-4.3.3.mpi/CMakeLists.txt	2015-10-14 16:44:41.176300658 -0400
@@ -753,6 +753,7 @@
     SET(USE_PARALLEL OFF CACHE BOOL "")
     MESSAGE(STATUS "Cannot find HDF5 library built with parallel support. Disabling parallel build.")
   ELSE()
+    FIND_PACKAGE(MPI REQUIRED)
     SET(USE_PARALLEL ON CACHE BOOL "")
     SET(STATUS_PARALLEL "ON")
   ENDIF()
diff -Nur netcdf-4.3.3/liblib/CMakeLists.txt netcdf-4.3.3.mpi/liblib/CMakeLists.txt
--- netcdf-4.3.3/liblib/CMakeLists.txt	2015-02-12 16:44:35.000000000 -0500
+++ netcdf-4.3.3.mpi/liblib/CMakeLists.txt	2015-10-14 16:44:57.757793634 -0400
@@ -71,6 +71,10 @@
   SET(TLL_LIBS ${TLL_LIBS} ${CURL_LIBRARY})
 ENDIF()
 
+IF(USE_PARALLEL)
+  SET(TLL_LIBS ${TLL_LIBS} ${MPI_C_LIBRARIES})
+ENDIF()
+
 IF(USE_HDF4)
   SET(TLL_LIBS ${TLL_LIBS} ${HDF4_LIBRARIES})
 ENDIF()
