diff -Naru a/src/qs_kpp1_env_methods.F b/src/qs_kpp1_env_methods.F
--- a/src/qs_kpp1_env_methods.F	2021-11-20 14:35:36.103103400 +0530
+++ b/src/qs_kpp1_env_methods.F	2023-06-14 12:00:52.350584708 +0530
@@ -220,7 +220,6 @@
                                                             output_unit
       LOGICAL                                            :: gapw, gapw_xc, lsd, my_calc_forces
       REAL(KIND=dp)                                      :: alpha, energy_hartree, energy_hartree_1c
-      REAL(KIND=dp), DIMENSION(:, :, :, :), POINTER      :: vxg
       TYPE(atomic_kind_type), DIMENSION(:), POINTER      :: atomic_kind_set
       TYPE(cp_logger_type), POINTER                      :: logger
       TYPE(cp_para_env_type), POINTER                    :: para_env
@@ -361,7 +360,8 @@
 
       CALL xc_calc_2nd_deriv(v_xc, p_env%kpp1_env%deriv_set, p_env%kpp1_env%rho_set, &
                              rho1_r_pw, rho1_g_pw, auxbas_pw_pool, xc_section, .FALSE., &
-                             NULL(vxg), lsd_singlets, do_excitations, do_triplet, do_tddft, &
+                             lsd_singlets=lsd_singlets, do_excitations=do_excitations, &
+                             do_triplet=do_triplet, do_tddft=do_tddft, &
                              compute_virial=calc_virial, virial_xc=virial)
 
       DO ispin = 1, nspins
