diff --git a/frontend/plugin/statistics/gtkmm/src/StatisticsDialog.cc b/frontend/plugin/statistics/gtkmm/src/StatisticsDialog.cc
index ee7d56f..608fddb 100644
--- a/frontend/plugin/statistics/gtkmm/src/StatisticsDialog.cc
+++ b/frontend/plugin/statistics/gtkmm/src/StatisticsDialog.cc
@@ -604,7 +604,7 @@ StatisticsDialog::on_history_delete_all()
         );
     Gtk::MessageDialog mb_ask( *this, msg, true, Gtk::MESSAGE_WARNING, Gtk::BUTTONS_YES_NO, false );
     mb_ask.set_title( _("Warning") );
-    mb_ask.get_widget_for_response( Gtk::RESPONSE_NO )->grab_default();
+    //mb_ask.get_widget_for_response( Gtk::RESPONSE_NO )->grab_default();
     if( mb_ask.run() == Gtk::RESPONSE_YES )
     {
         mb_ask.hide();
@@ -630,7 +630,7 @@ StatisticsDialog::on_history_delete_all()
                 );
             Gtk::MessageDialog mb_error( *this, msg, true, Gtk::MESSAGE_ERROR, Gtk::BUTTONS_YES_NO, false );
             mb_error.set_title( _("Error") );
-            mb_error.get_widget_for_response( Gtk::RESPONSE_NO )->grab_default();
+            //mb_error.get_widget_for_response( Gtk::RESPONSE_NO )->grab_default();
             if( mb_error.run() != Gtk::RESPONSE_YES )
                 break;
         }
