diff --git a/configure.ac b/configure.ac
index a8a437f..6128cd2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -949,8 +949,8 @@ dnl
 dnl Internationalization
 dnl
 
-AM_GNU_GETTEXT_VERSION([0.17])
-AM_GNU_GETTEXT([external])
+dnl AM_GNU_GETTEXT_VERSION([0.17])
+dnl AM_GNU_GETTEXT([external])
 
 dnl Hack because windows gettext package is SEVERELY broken...
 if test "x$platform_os_win32" = "xyes"
@@ -963,9 +963,10 @@ then
         BUILD_INCLUDED_LIBINTL=no
 fi
 
-GETTEXT_PACKAGE=workrave
-AC_SUBST(GETTEXT_PACKAGE)
-AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Define to the gettext domain name.])
+dnl GETTEXT_PACKAGE=workrave
+dnl AC_SUBST(GETTEXT_PACKAGE)
+dnl AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Define to the gettext domain name.])
+AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [Gettext package])
 IT_PROG_INTLTOOL([0.40.0])
 
 
