--- oneTBB-2021.2.0/include/oneapi/tbb/detail/_config.h.orig	2021-05-21 14:01:51.073986600 +0200
+++ oneTBB-2021.2.0/include/oneapi/tbb/detail/_config.h	2021-01-27 21:03:49.000000000 +0100
@@ -308,7 +308,7 @@
                                                             || _LIBCPP_VERSION >= 3700 && (!__TBB_MACOS_TARGET_VERSION || __TBB_MACOS_TARGET_VERSION >= 101200))
 
 
-#define __TBB_TSX_INTRINSICS_PRESENT 0
+#define __TBB_TSX_INTRINSICS_PRESENT ((__RTM__ || _MSC_VER>=1700 || __INTEL_COMPILER) && !__ANDROID__)
 
 #define __TBB_WAITPKG_INTRINSICS_PRESENT ((__INTEL_COMPILER >= 1900 || __TBB_GCC_VERSION >= 110000 || __TBB_CLANG_VERSION >= 120000) && !__ANDROID__)
 
