From cdea634c02e1549408cf137e2766e8e3ee0c2b14 Mon Sep 17 00:00:00 2001
From: Chris Green <artists@fnal.gov>
Date: Thu, 7 Mar 2019 11:34:58 -0600
Subject: [PATCH] Remove superfluous semicolons to satisfy -pedantic.

---
 include/tbb/machine/gcc_generic.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/tbb/machine/gcc_generic.h b/include/tbb/machine/gcc_generic.h
index bd698a4..d970653 100644
--- a/include/tbb/machine/gcc_generic.h
+++ b/include/tbb/machine/gcc_generic.h
@@ -107,9 +107,9 @@ __TBB_MACHINE_DEFINE_ATOMICS(8,int64_t)
 #undef __TBB_MACHINE_DEFINE_ATOMICS
 
 namespace tbb{ namespace internal { namespace gcc_builtins {
-    inline int clz(unsigned int x){ return __builtin_clz(x);};
-    inline int clz(unsigned long int x){ return __builtin_clzl(x);};
-    inline int clz(unsigned long long int x){ return __builtin_clzll(x);};
+    inline int clz(unsigned int x){ return __builtin_clz(x);}
+    inline int clz(unsigned long int x){ return __builtin_clzl(x);}
+    inline int clz(unsigned long long int x){ return __builtin_clzll(x);}
 }}}
 //gcc __builtin_clz builtin count _number_ of leading zeroes
 static inline intptr_t __TBB_machine_lg( uintptr_t x ) {
-- 
2.20.1

