*** build/linux.gcc.inc.orig	2017-01-10 16:54:01.000000000 -0800
--- build/linux.gcc.inc	2017-01-10 16:54:04.000000000 -0800
***************
*** 49,57 ****
  endif
  
  # gcc 4.8 and later support RTM intrinsics, but require command line switch to enable them
! ifneq (,$(shell gcc -dumpversion | egrep  "^(4\.[8-9]|[5-9])"))
!     RTM_KEY = -mrtm
! endif
  
  ifeq ($(cfg), release)
          CPLUS_FLAGS = $(ITT_NOTIFY) -g -O2 -DUSE_PTHREAD
--- 49,57 ----
  endif
  
  # gcc 4.8 and later support RTM intrinsics, but require command line switch to enable them
! #ifneq (,$(shell gcc -dumpversion | egrep  "^(4\.[8-9]|[5-9])"))
! #    RTM_KEY = -mrtm
! #endif
  
  ifeq ($(cfg), release)
          CPLUS_FLAGS = $(ITT_NOTIFY) -g -O2 -DUSE_PTHREAD