diff --git a/Makefile b/Makefile
index c43f1e2..8b3bd86 100644
--- a/Makefile
+++ b/Makefile
@@ -14,12 +14,11 @@ endif
 CXXFLAGS+=$(OPTFLAGS)

 CUR_DIRECTORY=$(shell pwd)
-DRAM_LIB=$(CUR_DIRECTORY)/../DRAMSim2
-NV_LIB=$(CUR_DIRECTORY)/../NVDIMMSim/src
-#NV_LIB=$(CUR_DIRECTORY)/../FNVSim
+DRAM_LIB=$(CUR_DIRECTORY)/DRAMSim2
+NV_LIB=$(CUR_DIRECTORY)/NVDIMMSim

 INCLUDES=-I$(DRAM_LIB) -I$(NV_LIB)
-LIBS=-L${DRAM_LIB} -L${NV_LIB} -ldramsim -lnvdsim -Wl,-rpath ${DRAM_LIB} -Wl,-rpath ${NV_LIB}
+LIBS=-L${DRAM_LIB} -L${NV_LIB} -ldramsim -lnvdsim -Wl,-rpath -Wl,${DRAM_LIB} -Wl,-rpath -Wl,${NV_LIB}

 EXE_NAME=HybridSim
 LIB_NAME=libhybridsim.so
