--- numpy-1.20.0/numpy/distutils/fcompiler/__init__.py.org	2021-02-04 17:38:42.000000000 +0900
+++ numpy-1.20.0/numpy/distutils/fcompiler/__init__.py	2021-02-04 18:29:21.000000000 +0900
@@ -314,7 +314,12 @@
             if not exe_from_environ:
                 possibles = [f90, f77] + self.possible_executables
             else:
-                possibles = [exe_from_environ] + self.possible_executables
+                matching_executables = []
+                for e in exe_from_environ:
+                    for p in self.possible_executables:
+                        if p in e:
+                            matching_executables.append(e)
+                possibles = [matching_executables] + self.possible_executables

             seen = set()
             unique_possibles = []
