--- spack-src/src/basic/score_matrix.cpp.org	2019-12-10 14:49:46.698520161 +0900
+++ spack-src/src/basic/score_matrix.cpp	2019-12-10 14:50:04.740418478 +0900
@@ -36,7 +36,7 @@
 
 const double INT2_MAX = std::numeric_limits<double>::max();
 
-const char DNA_scores[5 * 5] = {
+const signed char DNA_scores[5 * 5] = {
 	2, -3, -3, -3, -3,
 	-3,2,-3,-3,-3,
 	-3,-3,2,-3,-3,
@@ -317,4 +317,4 @@
 	constants[3] = lambda;
 	constants[4] = K;
 	constants_ = constants;
-}
\ ファイル末尾に改行がありません
+}
