--- diamond-0.9.25/src/basic/score_matrix.cpp.org	2019-12-10 15:16:46.549126025 +0900
+++ diamond-0.9.25/src/basic/score_matrix.cpp	2019-12-10 15:17:08.721466092 +0900
@@ -36,7 +36,7 @@
 
 const double INT2_MAX = std::numeric_limits<double>::max();
 
-const char DNA_scores[5 * 5] = {
+const signed char DNA_scores[5 * 5] = {
 	2, -3, -3, -3, -3,
 	-3,2,-3,-3,-3,
 	-3,-3,2,-3,-3,
@@ -317,4 +317,4 @@
 	constants[3] = lambda;
 	constants[4] = K;
 	constants_ = constants;
-}
\ No newline at end of file
+}
