--- a/Lib/distutils/unixccompiler.py	2003-06-02 05:27:40.000000000 +1000
+++ b/Lib/distutils/unixccompiler.py	2017-05-13 13:52:45.554213616 +1000
@@ -208,7 +208,8 @@
         elif compiler[:3] == "gcc" or compiler[:3] == "g++":
           return "-Wl,-R" + dir
         else:
-            return "-R" + dir
+            # Patched by spack to use gcc syntax by default:
+            return "-Wl,-R" + dir
 
     def library_option(self, lib):
         return "-l" + lib
