--- a/Lib/distutils/unixccompiler.py	2009-05-09 21:55:12.000000000 +1000
+++ b/Lib/distutils/unixccompiler.py	2017-05-13 14:30:18.077518999 +1000
@@ -299,10 +299,8 @@
                 else:
                     return "-Wl,-R" + dir
             else:
-                # No idea how --enable-new-dtags would be passed on to
-                # ld if this system was using GNU ld.  Don't know if a
-                # system like this even exists.
-                return "-R" + dir
+                # Patched by spack to use gcc syntax by default:
+                return "-Wl,-R" + dir
 
     def library_option(self, lib):
         return "-l" + lib
