--- a/Lib/distutils/unixccompiler.py	2014-06-30 12:05:26.000000000 +1000
+++ b/Lib/distutils/unixccompiler.py	2017-05-11 13:43:58.112063016 +1000
@@ -209,7 +209,7 @@
         return "-L" + dir
 
     def _is_gcc(self, compiler_name):
-        return "gcc" in compiler_name or "g++" in compiler_name
+        return True
 
     def runtime_library_dir_option(self, dir):
         # XXX Hackish, at the very least.  See Python bug #445902:
