--- a/cmake/fiat_compiler_warnings.cmake	2023-08-21 08:29:06.000000000 -0600
+++ b/cmake/fiat_compiler_warnings.cmake	2023-08-21 08:37:40.000000000 -0600
@@ -5,15 +5,17 @@
   ecbuild_add_c_flags("-Wextra"                                NO_FAIL)
   ecbuild_add_c_flags("-Wno-unused-parameter"                  NO_FAIL)
   ecbuild_add_c_flags("-Wno-unused-variable"                   NO_FAIL)
-  ecbuild_add_c_flags("-Wno-gnu-zero-variadic-macro-arguments" NO_FAIL)
+  if (CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
+    ecbuild_add_c_flags("-Wno-gnu-zero-variadic-macro-arguments" NO_FAIL)
+  endif()
 endif()
 
 # Always disable some warnings
 ecbuild_add_c_flags("-Wno-deprecated-declarations" NO_FAIL)
-if( CMAKE_C_COMPILER_ID MATCHES Intel )
-  ecbuild_add_c_flags("-diag-disable=279")   # controlling expression is constant
-  ecbuild_add_c_flags("-diag-disable=11076") # inline limits
-endif()
+#if( CMAKE_C_COMPILER_ID MATCHES Intel )
+#  ecbuild_add_c_flags("-diag-disable=279")   # controlling expression is constant
+#  ecbuild_add_c_flags("-diag-disable=11076") # inline limits
+#endif()
 if( CMAKE_Fortran_COMPILER_ID MATCHES Cray )
   ecbuild_add_fortran_flags("-hnomessage=878") # A module named ... has already been directly or indirectly use associated into this scope
   ecbuild_add_fortran_flags("-hnomessage=867") # Module ... has no public objects declared in the module, therefore nothing can be use associated from the module.
--- a/src/fiat/CMakeLists.txt
+++ b/src/fiat/CMakeLists.txt
@@ -26,10 +26,10 @@ endif()

 ### Compilation flags

-if( CMAKE_Fortran_COMPILER_ID MATCHES "Intel" )
-  ## To disable checking of argument correctness of dummy mpi symbols
-  ecbuild_add_fortran_flags( -nowarn nointerfaces )
-endif()
+#if( CMAKE_Fortran_COMPILER_ID MATCHES "Intel" )
+#  ## To disable checking of argument correctness of dummy mpi symbols
+#  ecbuild_add_fortran_flags( -nowarn nointerfaces )
+#endif()

 if( CMAKE_Fortran_COMPILER_ID MATCHES "GNU" )
   ecbuild_add_fortran_flags( -ffree-line-length-none )
