--- percept.orig/src/percept/MeshType.hpp	2017-12-08 09:16:42.080818000 -0700
+++ percept/src/percept/MeshType.hpp	2017-12-08 09:17:25.568819000 -0700
@@ -11,6 +11,7 @@
 #include <percept/Percept.hpp>
 
 #include <array>
+#include <memory>
 #include <stk_mesh/base/Field.hpp>
 #include <stk_mesh/base/Entity.hpp>
 #include <stk_mesh/base/Bucket.hpp>
@@ -62,7 +63,7 @@
       typedef Kokkos::LayoutRight   SecondaryDataLayout;
 #endif
 
-	#elif KOKKOS_HAVE_OPENMP
+	#elif defined(KOKKOS_HAVE_OPENMP)
   	  typedef Kokkos::OpenMP     ExecSpace;
   	  typedef Kokkos::OpenMP     MemSpace;
   	  typedef Kokkos::LayoutLeft DataLayout;
