diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7be6ac1..a09781d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@ MESSAGE("-- OPENNURBS_LIBRARY = ${OPENNURBS_LIBRARY}")
 
 
 # gtest
-FIND_LIBRARY(GTEST_LIBRARY NAMES libgtest.a PATHS ${PERCEPT_TPLS_INSTALL_DIR}/lib NO_DEFAULT_PATH)
+FIND_LIBRARY(GTEST_LIBRARY NAMES libgtest.a PATHS ${PERCEPT_TPLS_INSTALL_DIR}/lib64 NO_DEFAULT_PATH)
 FIND_PATH(GTEST_INCLUDES gtest.h PATHS ${PERCEPT_TPLS_INSTALL_DIR}/include/gtest )
 MESSAGE("-- GTEST_LIBRARY = ${GTEST_LIBRARY}")
 MESSAGE("-- GTEST_INCLUDES = ${GTEST_INCLUDES}")
@@ -97,13 +97,6 @@ IF (ENABLE_INSTALL)
     SET (INSTALL_PERCEPT TRUE)
 ENDIF()
 
-IF (INSTALL_PERCEPT)
-  set(BINARY_INSTALL_DIR bin)
-  set(INCLUDE_INSTALL_DIR include)
-  set(LIB_INSTALL_DIR lib)
-  INCLUDE(CMakePackageConfigHelpers)
-ENDIF ()
-
 MESSAGE("Setting and checking of compilers:")
 SET(CMAKE_CXX_COMPILER ${Trilinos_CXX_COMPILER} )
 SET(CMAKE_C_COMPILER ${Trilinos_C_COMPILER} )
@@ -269,6 +262,8 @@ MESSAGE("-- HEADER = ${HEADER}")
 FILE (GLOB ADAPT_SOURCE src/adapt/main/*.cpp)
 MESSAGE("-- ADAPT_SOURCE = ${ADAPT_SOURCE}")
 
+file(GLOB TRANSFER_SOURCE src/percept/mesh_transfer/*.cpp)
+
 FILE (GLOB PERCEPT_UTEST_SOURCE test/unit_tests/*.cpp)
 MESSAGE("-- PERCEPT_UTEST_SOURCE = ${PERCEPT_UTEST_SOURCE}")
 
@@ -330,8 +325,11 @@ IF (NOT STK_PERCEPT_LITE)
   TARGET_LINK_LIBRARIES(${percept_lib_name} ${Trilinos_LIBRARIES} ${YAML_LIBRARY} ${OPENNURBS_LIBRARY} ${BOOST_LIBRARIES})
 #  TARGET_LINK_LIBRARIES(${percept_lib_name} ${GTEST_LIBRARY})
 
+  add_executable(mesh_transfer ${TRANSFER_SOURCE})
+  target_link_libraries(mesh_transfer ${percept_lib_name})
+
   SET_PROPERTY(
-    TARGET ${percept_lib_name} ${adapt_ex_name} ${percept_utest_name} ${percept_rtest_name}
+    TARGET ${percept_lib_name} mesh_transfer ${adapt_ex_name} ${percept_utest_name} ${percept_rtest_name}
     PROPERTY COMPILE_DEFINITIONS STK_PERCEPT_LITE=0 STK_PERCEPT_HAS_GEOMETRY STK_PERCEPT_USE_INTREPID
   )
 
@@ -342,5 +340,15 @@ IF (NOT STK_PERCEPT_LITE)
 
 ENDIF()
 
+IF (INSTALL_PERCEPT)
+  install(TARGETS ${percept_ex_name} ${adapt_ex_name}
+          ${percept_htest_name} ${percept_rtest_name} ${percept_utest_name}
+          mesh_transfer percept
+          RUNTIME DESTINATION bin
+          ARCHIVE DESTINATION lib
+          LIBRARY DESTINATION lib)
+  #install(DIRECTORY include/ DESTINATION include)
+  include(CMakePackageConfigHelpers)
+ENDIF ()
 
 MESSAGE("\nAnd CMake says...:")
