diff --git a/sbin/start-master.sh b/sbin/start-master.sh
index 36fe4b4abe..343714e2a4 100755
--- a/sbin/start-master.sh
+++ b/sbin/start-master.sh
@@ -63,6 +63,6 @@ if [ "$SPARK_MASTER_WEBUI_PORT" = "" ]; then
   SPARK_MASTER_WEBUI_PORT=8080
 fi
 
-"${SPARK_HOME}/sbin"/spark-daemon.sh start $CLASS 1 \
+exec "${SPARK_HOME}/sbin"/spark-daemon.sh start $CLASS 1 \
   --host $SPARK_MASTER_HOST --port $SPARK_MASTER_PORT --webui-port $SPARK_MASTER_WEBUI_PORT \
   $ORIGINAL_ARGS
