--- libmonitor-20130218/configure	2013-02-17 23:08:32.000000000 -0800
+++ libmonitor-20130218-fixes/configure	2013-02-18 10:34:05.237918411 -0800
@@ -3600,13 +3600,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+    CFLAGS="-g -O0"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+    CFLAGS="-O0"
   else
     CFLAGS=
   fi
