--- a/configure	2017-05-12 16:03:01.000000000 -0500
+++ b/configure	2017-05-12 16:01:58.000000000 -0500
@@ -29273,8 +29273,8 @@
   for flag in \
    -Wall -Wstrict-prototypes -Werror=declaration-after-statement \
    -Werror=missing-prototypes -Werror=implicit-function-declaration \
-   -Werror=pointer-arith -Werror=init-self -Werror=format-security \
-   -Werror=format=2 -Werror=missing-include-dirs; do
+   -Werror=pointer-arith -Werror=init-self \
+   -Werror=missing-include-dirs; do
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports flag $flag in envvar CFLAGS" >&5
 $as_echo_n "checking if $CC supports flag $flag in envvar CFLAGS... " >&6; }
