--- a/gio/gfile.c     2018-06-11 15:28:30.527667202 -0500
+++ b/gio/gfile.c     2018-03-12 11:23:37.000000000 -0500
@@ -3014,11 +3014,11 @@
 
   /* Try a 1MiB buffer for improved throughput. If that fails, use the default
    * pipe size. See: https://bugzilla.gnome.org/791457 */
-  buffer_size = fcntl (buffer[1], F_SETPIPE_SZ, 1024 * 1024);
+  buffer_size = -1;
   if (buffer_size <= 0)
     {
       int errsv;
-      buffer_size = fcntl (buffer[1], F_GETPIPE_SZ);
+      buffer_size = 65536;
       errsv = errno;
 
       if (buffer_size <= 0)
