diff --git a/tensorboard/defs/web.bzl b/tensorboard/defs/web.bzl
index dfca0bfd21b5..fb981493b25d 100644
--- a/tensorboard/defs/web.bzl
+++ b/tensorboard/defs/web.bzl
@@ -180,6 +180,7 @@ def _tf_web_library(ctx):
         tools=ctx.files._tsc,
         inputs=ts_inputs,
         outputs=ts_outputs,
+       use_default_shell_env=True,
         executable=ctx.executable._execrooter,
         arguments=(
             [er_config.path] +
