diff -ru a/tensorboard/plugins/projector/vz_projector/BUILD b/tensorboard/plugins/projector/vz_projector/BUILD
--- a/tensorboard/plugins/projector/vz_projector/BUILD	2021-10-13 11:04:47.000000000 -0500
+++ b/tensorboard/plugins/projector/vz_projector/BUILD	2021-12-02 14:27:59.275004056 -0600
@@ -64,6 +64,7 @@
         "@npm//@polymer/decorators",
         "@npm//@polymer/polymer",
         "@npm//numeric",
+        "@npm//@types/three",
         "@npm//three",
         "@npm//umap-js",
     ],
