diff -ru a/tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_component.ts b/tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_component.ts
--- a/tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_component.ts	2021-10-13 11:04:47.000000000 -0500
+++ b/tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_component.ts	2021-12-01 20:39:45.512831971 -0600
@@ -79,6 +79,6 @@
 
   handleFocusOut() {
     clearTimeout(this.timeOutId);
-    this.timeOutId = setTimeout(this.resetFocus.bind(this), 0);
+    this.timeOutId = window.setTimeout(this.resetFocus.bind(this), 0);
   }
 }
