diff --git a/src/stgen.py b/src/stgen.py
index b818ea5..3cb45e6 100644
--- a/src/stgen.py
+++ b/src/stgen.py
@@ -227,7 +227,7 @@ class StGen:
             number = min(5+numpy.ceil(2*n),100)

         if number > 0:
-            isi = self.rng.exponential(1.0/rate, number)*1000.0
+            isi = self.rng.exponential(1.0/rate, int(number)) * 1000.0
             if number > 1:
                 spikes = numpy.add.accumulate(isi)
             else:
@@ -301,7 +301,7 @@ class StGen:
             number = min(5+numpy.ceil(2*n),100)

         if number > 0:
-            isi = self.rng.gamma(a, b, number)*1000.0
+            isi = self.rng.gamma(a, b, int(number))*1000.0
             if number > 1:
                 spikes = numpy.add.accumulate(isi)
             else:
