diff -Naru a/scripts/site_scons/alm/env.py b/scripts/site_scons/alm/env.py
--- a/scripts/site_scons/alm/env.py	2024-10-11 04:15:23.000000000 +0000
+++ b/scripts/site_scons/alm/env.py	2025-05-12 18:17:06.390404980 +0000
@@ -73,6 +73,8 @@
                     'SPACK_SHORT_SPEC',
                     'SPACK_SYSTEM_DIRS',
                     'SPACK_MANAGED_DIRS',
+                    'SPACK_COMPILER_WRAPPER_PATH',
+                    'SPACK_CC_LINKER_ARG',
                 ]
 
     # add these keys to mydict if they exist in environ
