--- spack-src/lib/bloomfilter/BloomFilter.hpp.org	2018-10-17 07:04:45.000000000 +0900
+++ spack-src/lib/bloomfilter/BloomFilter.hpp	2020-07-16 15:41:03.607766127 +0900
@@ -230,7 +230,7 @@
 
 	void writeHeader(std::ostream& out) const {
 		FileHeader header;
-		strncpy(header.magic, "BlOOMFXX", 8);
+		memcpy(header.magic, "BlOOMFXX", 8);
 		char magic[9];
 		strncpy(magic, header.magic, 8);
 		magic[8] = '\0';
