--- spack-src/CMakeLists.txt.bak	2020-05-14 19:42:00.000000000 +0900
+++ spack-src/CMakeLists.txt	2020-12-11 14:02:52.645857641 +0900
@@ -97,6 +97,11 @@
 # Cython generated code emits way to many warnings at CHECKIN and EVERYTHING
 set(BUILD_WARNING_LEVEL "PRODUCTION")
 
+# This must be synchronized with the definition in
+# cmake_modules/DefineOptions.cmake.
+set(ARROW_ARMV8_ARCH
+    "armv8-a"
+    CACHE STRING "Arm64 arch and extensions: armv8-a, armv8-a or armv8-a+crc+crypto")
 include(SetupCxxFlags)
 
 # Add common flags
