From 8cb20248c8045e2e8ffc9302a119159ce53148fe Mon Sep 17 00:00:00 2001
From: Ben Olson <macslayer@argentina.eecs.utk.edu>
Date: Mon, 10 Dec 2018 12:46:21 -0500
Subject: [PATCH] Fixed modifications

---
 include/jemalloc/jemalloc_macros.h.in | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/jemalloc/jemalloc_macros.h.in b/include/jemalloc/jemalloc_macros.h.in
index aee55438c..1bfccefb7 100644
--- a/include/jemalloc/jemalloc_macros.h.in
+++ b/include/jemalloc/jemalloc_macros.h.in
@@ -19,17 +19,17 @@
      ((int)(((size_t)(a) < (size_t)INT_MAX) ? ffs((int)(a))-1 :	\
      ffs((int)(((size_t)(a))>>32))+31))
 #endif
-#define MALLOCX_ZERO	((int)0x40)
+#define MALLOCX_ZERO	((int)0x1)
 /*
  * Bias tcache index bits so that 0 encodes "automatic tcache management", and 1
  * encodes MALLOCX_TCACHE_NONE.
  */
-#define MALLOCX_TCACHE(tc)	((int)(((tc)+2) << 8))
+#define MALLOCX_TCACHE(tc)	((int)(((tc)+2) << 2))
 #define MALLOCX_TCACHE_NONE	MALLOCX_TCACHE(-1)
 /*
  * Bias arena index bits so that 0 encodes "use an automatically chosen arena".
  */
-#define MALLOCX_ARENA(a)	((((int)(a))+1) << 20)
+#define MALLOCX_ARENA(a)	((((int)(a))+1) << 14)
 
 /*
  * Use as arena index in "arena.<i>.{purge,decay,dss}" and
@@ -43,12 +43,12 @@
  *   mallctl("arena." STRINGIFY(MALLCTL_ARENAS_ALL) ".purge", NULL, NULL, NULL,
  *       0);
  */
-#define MALLCTL_ARENAS_ALL	4096
+#define MALLCTL_ARENAS_ALL	262144
 /*
  * Use as arena index in "stats.arenas.<i>.*" mallctl interfaces to select
  * destroyed arenas.
  */
-#define MALLCTL_ARENAS_DESTROYED	4097
+#define MALLCTL_ARENAS_DESTROYED	262145
 
 #if defined(__cplusplus) && defined(JEMALLOC_USE_CXX_THROW)
 #  define JEMALLOC_CXX_THROW throw()
