diff --git a/util/chplenv/chpl_llvm.py b/util/chplenv/chpl_llvm.py
index f0fd495f28..95dc9c3f67 100755
--- a/util/chplenv/chpl_llvm.py
+++ b/util/chplenv/chpl_llvm.py
@@ -866,13 +866,14 @@ def get_clang_prgenv_args():
         os.environ['PE_CHAPEL_PKGCONFIG_LIBS'] = gather_pe_chpl_pkgconfig_libs()

         # Use cc --cray-print-opts=... to get arguments from compiler driver
-
+        # find the actual cc in case something like spack has wrapped it
+        real_cc = os.path.join(os.environ["CRAYPE_DIR"], "bin", "cc")
         # Get compilation arguments
-        opts = run_command(['cc', '--cray-print-opts=cflags'])
+        opts = run_command([real_cc, '--cray-print-opts=cflags'])
         comp_args.extend(opts.split())

         # Get link arguments
-        opts = run_command(['cc', '--cray-print-opts=libs'])
+        opts = run_command([real_cc, '--cray-print-opts=libs'])
         link_args.extend(opts.split())

     return (comp_args, link_args)
