diff -Naur a/setup.py b/setup.py
--- a/setup.py	2020-02-06 15:40:26.000000000 -0600
+++ b/setup.py	2020-02-06 15:41:17.000000000 -0600
@@ -27,10 +27,12 @@
     "future>=0.16.0,<=0.18.2",
     "tabulate>=0.8.2,<=0.8.3",
     "ipaddress>=1.0.22",
-    "enum34>=1.1.6",
     "PyYAML>=5.1.2",
 ]
 
+if sys.version_info < (3, 4):
+    REQUIRES.append("enum34>=1.1.6")
+
 if sys.version_info[0] == 2:
     REQUIRES.append("configparser>=3.5.0,<=3.8.1")
 
