diff --git a/ultraliser/data/meshes/mesh/Mesh.cpp b/ultraliser/data/meshes/mesh/Mesh.cpp
index 694eb8c..08b83c9 100644
--- a/ultraliser/data/meshes/mesh/Mesh.cpp
+++ b/ultraliser/data/meshes/mesh/Mesh.cpp
@@ -129,7 +129,7 @@ Mesh::Mesh(vtkSmartPointer<vtkPolyData> polyData)
     for (int64_t i = 0; i < polygonalData->GetNumberOfPolys(); i++)
     {
         // Get the triangle array
-        vtkIdType* pointsArray;
+        const vtkIdType* pointsArray;
         vtkIdType numberPoints = 3;
         int n = cellArray->GetNextCell(numberPoints, pointsArray);
 
