diff -Naur ADIOS-1.10.0/tests/genarray/Makefile.am ADIOS-1.10.0-patched/tests/genarray/Makefile.am
--- ADIOS-1.10.0/tests/genarray/Makefile.am	2016-08-04 09:38:29.000000000 -0400
+++ ADIOS-1.10.0-patched/tests/genarray/Makefile.am	2017-02-08 16:54:29.096684832 -0500
@@ -36,10 +36,10 @@
 
 gwrite_genarray.fh: genarray3d.xml
 	rm -f gwrite_genarray.fh gread_genarray.fh
-	$(top_builddir)/utils/gpp/gpp.py $(srcdir)/genarray3d.xml
+	@PYTHON_EXECUTABLE@ $(top_builddir)/utils/gpp/gpp.py $(srcdir)/genarray3d.xml
 gread_genarray.fh: genarray3d.xml
 	rm -f gwrite_genarray.fh gread_genarray.fh
-	$(top_builddir)/utils/gpp/gpp.py $(srcdir)/genarray3d.xml
+	@PYTHON_EXECUTABLE@ $(top_builddir)/utils/gpp/gpp.py $(srcdir)/genarray3d.xml
 
 FCLINK = \
 	$(LIBTOOL) --mode=link --tag F77 $(FC) \
