diff --git a/src/core/adios_internals_mxml.c b/src/core/adios_internals_mxml.c
index 513fd45c..b9296050 100644
--- a/src/core/adios_internals_mxml.c
+++ b/src/core/adios_internals_mxml.c
@@ -2173,8 +2173,9 @@ int adios_parse_config (const char * config, MPI_Comm comm)
     char * buffer = NULL;
     //#if HAVE_MPI
     int buffer_size = 0;
-    int rank;
-    MPI_Comm_rank (comm, &rank);
+    int rank = 0;
+    if (comm != MPI_COMM_NULL)
+        MPI_Comm_rank (comm, &rank);
     init_comm = comm;
     if (rank == 0)
     {
