--- a/src/gallium/auxiliary/util/u_debug_stack.c	2019-11-07 17:57:36.000000000 -0700
+++ b/src/gallium/auxiliary/util/u_debug_stack.c	2020-07-23 15:30:46.033145497 -0600
@@ -46,7 +46,7 @@
 #include "os/os_thread.h"
 #include "u_hash_table.h"
 
-struct util_hash_table* symbols_hash;
+static struct util_hash_table* symbols_hash;
 static mtx_t symbols_mutex = _MTX_INITIALIZER_NP;
 
 static unsigned hash_ptr(void* p)
--- a/src/gallium/auxiliary/util/u_debug_symbol.c	2019-11-07 17:58:53.000000000 -0700
+++ b/src/gallium/auxiliary/util/u_debug_symbol.c	2020-07-23 15:31:06.400146072 -0600
@@ -270,7 +270,7 @@
    debug_printf("\t%s\n", buf);
 }
 
-struct util_hash_table* symbols_hash;
+static struct util_hash_table* symbols_hash;
 static mtx_t symbols_mutex = _MTX_INITIALIZER_NP;
 
 static unsigned hash_ptr(void* p)
