--- misc/cgo/testcshared/test.bash.orig	2016-11-19 00:00:11.917000000 +0000
+++ misc/cgo/testcshared/test.bash	2016-11-19 00:00:22.081000000 +0000
@@ -107,7 +107,7 @@
 
 # test0: exported symbols in shared lib are accessible.
 # TODO(iant): using _shared here shouldn't really be necessary.
-$(go env CC) ${GOGCCFLAGS} -I ${installdir} -o testp main0.c libgo.$libext
+$(go env CC) ${GOGCCFLAGS} -I ${installdir} -o testp main0.c ./libgo.$libext
 binpush testp
 
 output=$(run LD_LIBRARY_PATH=. ./testp)
