--- a/src/Imakefile	2021-01-27 19:34:12.168968075 +0900
+++ b/src/Imakefile	2021-01-27 19:41:52.664965600 +0900
@@ -1,8 +1,3 @@
-#define CBFLIB_LOCAL
-#define CQRLIB_LOCAL
-#define CVECTOR_LOCAL
-#define NEARTREE_LOCAL
-#define XFORMSLIB_LOCAL
 ###########################################################################
 #                               RasMol 2.7.5                              #
 #                                                                         #
