--- a/src/Imakefile	2021-01-27 19:34:12.168968075 +0900
+++ b/src/Imakefile	2021-01-27 19:41:52.664965600 +0900
@@ -1,8 +1,3 @@
-#define CBFLIB_LOCAL
-#define CQRLIB_LOCAL
-#define CVECTOR_LOCAL
-#define NEARTREE_LOCAL
-#define XFORMSLIB_LOCAL
 ###########################################################################
 #                               RasMol 2.7.5                              #
 #                                                                         #
@@ -168,13 +163,7 @@
 XFORMSLIB_OBJDEP = $(XFORMSLIB_DIR)/lib/.libs/libforms.a
 #undef NO_XFORMS_BUILD
 #else
-ifeq ($(strip $(XFORMSLIB_DIR)),)
-#ifdef MACOSX_FINK
-XFORMSLIB_DIR = /sw
-#else
-XFORMSLIB_DIR = /usr/local
-#endif
-endif
+XFORMSLIB_DIR = SPACK_XFORMS
 XFORMSLIB_INCLUDE_DIR = $(XFORMSLIB_DIR)/include
 XFORMSLIB_LDLIB = -lXpm -L$(XFORMSLIB_DIR)/lib -lforms
 #endif
@@ -194,9 +183,7 @@
 CBFLIB_LDLIB = $(CBFLIB_DIR)/lib/libcbf.a
 CBFLIB_OBJDEP = $(CBFLIB_DIR)/lib/libcbf.a
 #else
-ifeq ($(strip $(CBFLIB_DIR)),)
-CBFLIB_DIR = $(PKGDIR)
-endif
+CBFLIB_DIR = SPACK_CBF
 CBFLIB_INCLUDE_DIR = $(CBFLIB_DIR)/include/cbflib
 CBFLIB_LDLIB = -L$(CBFLIB_DIR)/lib -lcbf
 CBFLIB_OBJDEP =
@@ -216,9 +203,7 @@
 CQRLIB_SRC = $(CQRLIB_DIR)/cqrlib.c
 CQRLIB_OBJDEP = $(CQRLIB_DIR)/cqrlib.o
 #else
-ifeq ($(strip $(CQRLIB_DIR)),)
-CQRLIB_DIR = $(PKGDIR)
-endif
+CQRLIB_DIR = SPACK_CQR
 CQRLIB_INCLUDE_DIR = $(CQRLIB_DIR)/include
 CQRLIB_LDLIB = -L$(CQRLIB_DIR)/lib -lCQRlib
 CQRLIB_SRC =
@@ -239,9 +224,7 @@
 CVECTOR_SRC = $(CVECTOR_DIR)/CVector.c
 CVECTOR_OBJDEP = $(CVECTOR_DIR)/CVector.o
 #else
-ifeq ($(strip $(CVECTOR_DIR)),)
-CVECTOR_DIR = $(PKGDIR)
-endif
+CVECTOR_DIR = SPACK_CVECTOR
 CVECTOR_INCLUDE_DIR = $(CVECTOR_DIR)/include
 CVECTOR_LDLIB = -L$(CVECTOR_DIR)/lib -lCVector
 CVECTOR_SRC =
@@ -262,9 +245,7 @@
 NEARTREE_SRC = -DCNEARTREE_FORCEFLIP $(NEARTREE_DIR)/CNearTree.c
 NEARTREE_OBJDEP = $(NEARTREE_DIR)/CNearTree.o
 #else
-ifeq ($(strip $(NEARTREE_DIR)),)
-NEARTREE_DIR = $(PKGDIR)
-endif
+NEARTREE_DIR = SPACK_NEARTREE
 NEARTREE_INCLUDE_DIR = $(NEARTREE_DIR)/include
 NEARTREE_LDLIB = -L$(NEARTREE_DIR)/lib -lCNearTree
 NEARTREE_SRC =
