diff --git a/cmake/FindPyQt5.py b/cmake/FindPyQt5.py
index 6a55a0f801..d6eda1fa1e 100644
--- a/cmake/FindPyQt5.py
+++ b/cmake/FindPyQt5.py
@@ -40,6 +40,7 @@ except ImportError:
     import sys
     cfg = sipconfig.Configuration()
     sip_dir = cfg.default_sip_dir
+    pyqt_prefix = os.sep.join(PyQt5.QtCore.__file__.split(os.sep)[0:-5])
     if sys.platform.startswith('freebsd'):
         py_version = str(sys.version_info.major) + str(sys.version_info.minor)
         sip_dir = sip_dir.replace(py_version, '')
@@ -53,9 +54,9 @@ except ImportError:
         'pyqt_version': PyQt5.QtCore.PYQT_VERSION,
         'pyqt_version_str': PyQt5.QtCore.PYQT_VERSION_STR,
         'pyqt_sip_flags': PyQt5.QtCore.PYQT_CONFIGURATION['sip_flags'],
-        'pyqt_mod_dir': os.path.join(cfg.default_mod_dir, "PyQt5"),
-        'pyqt_sip_dir': sip_dir,
-        'pyqt_bin_dir': cfg.default_bin_dir,
+        'pyqt_mod_dir': os.path.dirname(PyQt5.QtCore.__file__),
+        'pyqt_sip_dir': os.path.join(pyqt_prefix,'share','sip','PyQt5'),
+        'pyqt_bin_dir': os.path.join(pyqt_prefix,'bin'),
     }
     pyqtcfg = sipconfig.Configuration([cfg])
 
