--- a/src/common/Polyline.cc	2016-04-28 14:38:09.000000000 +0200
+++ b/src/common/Polyline.cc	2016-09-14 13:31:35.784617803 +0200
@@ -31,2 +30,0 @@
-#include "TeCoord2D.h"
-#include "TeGeometryAlgorithms.h"
--- a/src/decoders/GribRegularInterpretor.cc	2016-04-28 14:38:09.000000000 +0200
+++ b/src/decoders/GribRegularInterpretor.cc	2016-09-14 13:43:41.673614590 +0200
@@ -2083,2 +2083,2 @@
-                            if (isnan(val1)) {
-                                if (isnan(val2)) {
+                            if (std::isnan(val1)) {
+                                if (std::isnan(val2)) {
@@ -2090 +2090 @@
-                                if (isnan(val2)) {
+                                if (std::isnan(val2)) {
@@ -2101 +2101 @@
-                            if (isnan(val) || isinf(val) || isinf(-val)) {
+                            if (std::isnan(val) || std::isinf(val) || std::isinf(-val)) {
@@ -2105 +2105 @@
-                        if (isnan(val))
+                        if (std::isnan(val))
--- a/src/decoders/GribSatelliteInterpretor.cc	2016-04-28 14:38:09.000000000 +0200
+++ b/src/decoders/GribSatelliteInterpretor.cc	2016-09-14 13:48:55.243699910 +0200
@@ -33,5 +32,0 @@
-#include "TeProjection.h"
-#include "TeDataTypes.h"
-#include "TeRasterParams.h"
-#include "TeDecoderMemory.h"
-#include "TeRasterRemap.h"
--- a/src/decoders/NetcdfGeoMatrixInterpretor.cc	2016-04-28 14:38:09.000000000 +0200
+++ b/src/decoders/NetcdfGeoMatrixInterpretor.cc	2016-09-14 13:52:37.481201085 +0200
@@ -93 +93 @@
-				if ( !isnan(*d) ) {
+				if ( !std::isnan(*d) ) {
--- a/src/decoders/NetcdfOrcaInterpretor.cc	2016-04-28 14:38:09.000000000 +0200
+++ b/src/decoders/NetcdfOrcaInterpretor.cc	2016-09-14 13:51:16.248650570 +0200
@@ -210,2 +210,2 @@
-							if ( isnan(val1) ) {
-								if ( isnan(val2) ) {
+							if ( std::isnan(val1) ) {
+								if ( std::isnan(val2) ) {
@@ -218 +218 @@
-								if ( isnan(val2) ) {
+								if ( std::isnan(val2) ) {
@@ -226 +226 @@
-							if (isnan(val) || isinf(val) || isinf(-val) ) {
+							if (std::isnan(val) || std::isinf(val) || std::isinf(-val) ) {
@@ -230 +230 @@
-						if (isnan(val) ) val = missing;
+						if (std::isnan(val) ) val = missing;
@@ -296 +296 @@
-           if (isnan(value) ) 
+           if (std::isnan(value) ) 
--- a/src/terralib/kernel/TeCentroid.cpp	2016-04-28 14:38:09.000000000 +0200
+++ b/src/terralib/kernel/TeCentroid.cpp	2016-09-14 14:17:31.675996554 +0200
@@ -23,0 +24,2 @@
+#include "TeGeometryAlgorithms.h"
+
@@ -30 +31,0 @@
-#include "TeGeometryAlgorithms.h"
--- a/src/terralib/kernel/TeDatabase.h	2014-11-07 17:39:24.000000000 +0100
+++ b/src/terralib/kernel/TeDatabase.h	2016-09-14 14:20:01.041100590 +0200
@@ -33,0 +34 @@
+#include "TeGeometry.h"
@@ -38 +38,0 @@
-#include "TeGeometry.h"
--- a/src/terralib/kernel/TeOverlayUtils.h	2014-11-07 17:39:24.000000000 +0100
+++ b/src/terralib/kernel/TeOverlayUtils.h	2016-09-14 14:21:51.649920405 +0200
@@ -37,0 +38,2 @@
+#include "TeGeometry.h"
+
@@ -44 +45,0 @@
-#include "TeGeometry.h"
