--- include-what-you-use/CMakeLists.txt	2019-10-24 15:47:52.000000000 -0400
+++ spack-src/CMakeLists.txt	2020-03-26 07:16:53.284593661 -0400
@@ -60,21 +60,22 @@
   # Use only major.minor.patch for the resource directory structure; some
   # platforms include suffix in LLVM_VERSION.
   set(llvm_ver ${LLVM_VERSION_MAJOR}.${LLVM_VERSION_MINOR}.${LLVM_VERSION_PATCH})
-  set(clang_headers_src ${CMAKE_PREFIX_PATH}/lib/clang/${llvm_ver}/include)
-  set(clang_headers_dst ${CMAKE_BINARY_DIR}/lib/clang/${llvm_ver}/include)
+  set(clang_headers_src "${LLVM_INSTALL_PREFIX}/lib/clang/${llvm_ver}/include")
+  set(clang_headers_dst "${CMAKE_BINARY_DIR}/lib/clang/${llvm_ver}/include")
 
-  file(GLOB_RECURSE in_files RELATIVE ${clang_headers_src} ${clang_headers_src}/*)
+  file(GLOB_RECURSE in_files RELATIVE "${clang_headers_src}"
+    "${clang_headers_src}/*")
 
   set(out_files)
   foreach (file ${in_files})
-    set(src ${clang_headers_src}/${file})
-    set(dst ${clang_headers_dst}/${file})
+    set(src "${clang_headers_src}/${file}")
+    set(dst "${clang_headers_dst}/${file}")
 
-    add_custom_command(OUTPUT ${dst}
-      DEPENDS ${src}
-      COMMAND ${CMAKE_COMMAND} -E copy_if_different ${src} ${dst}
+    add_custom_command(OUTPUT "${dst}"
+      DEPENDS "${src}"
+      COMMAND ${CMAKE_COMMAND} -E copy_if_different "${src}" "${dst}"
       COMMENT "Copying clang's ${file}...")
-    list(APPEND out_files ${dst})
+    list(APPEND out_files "${dst}")
   endforeach()
 
   add_custom_target(clang-resource-headers ALL DEPENDS ${out_files})
