diff --git a/configure b/configure
index 6c0f43c..bd7f627 100755
--- a/configure
+++ b/configure
@@ -29,7 +29,7 @@ fi
 if [ "$INCLUDE_DIR" ] || [ "$LIB_DIR" ]; then
   echo "Found INCLUDE_DIR and/or LIB_DIR!"
   PKG_CFLAGS="-I$INCLUDE_DIR $PKG_CFLAGS"
-  PKG_LIBS="-L$LIB_DIR $PKG_LIBS"
+  PKG_LIBS="-L$LIB_DIR -Wl,-rpath,$LIB_DIR $PKG_LIBS"
 elif [ "$PKGCONFIG_CFLAGS" ] || [ "$PKGCONFIG_LIBS" ]; then
   echo "Found mysql_config/mariadb_config cflags and libs!"
   PKG_CFLAGS=${PKGCONFIG_CFLAGS}
