--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRuleClassProvider.java
+++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRuleClassProvider.java
@@ -185,7 +185,7 @@ public class BazelRuleClassProvider {
 
         Map<String, String> spackEnv = System.getenv();
         for (String envName : spackEnv.keySet()) {
-          if (envName.startsWith("SPACK_")) {
+          if ((envName.startsWith("SPACK_")) || (envName.equals("fcc_ENV")) || (envName.equals("FCC_ENV"))) {
             env.put(envName, spackEnv.get(envName));
           }
         }
