diff --color=auto --color=never -Naur a/src/main/java/com/google/devtools/build/lib/rules/cpp/HeaderDiscovery.java b/src/main/java/com/google/devtools/build/lib/rules/cpp/HeaderDiscovery.java
--- a/src/main/java/com/google/devtools/build/lib/rules/cpp/HeaderDiscovery.java	1980-01-01 00:00:00
+++ b/src/main/java/com/google/devtools/build/lib/rules/cpp/HeaderDiscovery.java	2025-05-20 15:05:12
@@ -174,7 +174,7 @@
           // here should never be absolute. If they are, it's probably due to a non-hermetic
           // #include,
           // and we should stop the build with an error.
-          absolutePathProblems.add(execPathFragment.getPathString());
+          // absolutePathProblems.add(execPathFragment.getPathString());
           continue;
         }
       }
