--- a/Makefile
+++ b/Makefile
@@ -2007,9 +2007,9 @@ gen-pc:
 	-echo 'Name: rocksdb' >> rocksdb.pc
 	-echo 'Description: An embeddable persistent key-value store for fast storage' >> rocksdb.pc
 	-echo Version: $(shell ./build_tools/version.sh full) >> rocksdb.pc
-	-echo 'Libs: -L$${libdir} $(EXEC_LDFLAGS) -lrocksdb' >> rocksdb.pc
+	-echo 'Libs: -L$${libdir} $(subst ','\'',$(EXEC_LDFLAGS)) -lrocksdb' >> rocksdb.pc
 	-echo 'Libs.private: $(PLATFORM_LDFLAGS)' >> rocksdb.pc
-	-echo 'Cflags: -I$${includedir} $(PLATFORM_CXXFLAGS)' >> rocksdb.pc
+	-echo 'Cflags: -I$${includedir}' >> rocksdb.pc
 	-echo 'Requires: $(subst ",,$(ROCKSDB_PLUGIN_PKGCONFIG_REQUIRES))' >> rocksdb.pc
 
 #-------------------------------------------------
