diff --git a/setup.py b/setup.py
index 82061c6..ff342af 100644
--- a/setup.py
+++ b/setup.py
@@ -1,7 +1,13 @@
 #!/usr/bin/env python
 
 """Setup script for the pyparsing module distribution."""
-from setuptools import setup
+
+# Setuptools depends on pyparsing (via packaging) as of version 34, so allow
+# installing without it to avoid bootstrap problems.
+try:
+    from setuptools import setup
+except ImportError:
+    from distutils.core import setup
 
 import sys
 import os
 
