diff -Nuar a/setup.py b/setup.py
--- a/setup.py	2017-02-20 18:45:45.000000000 -0600
+++ b/setup.py	2017-02-20 18:47:59.000000000 -0600
@@ -1,13 +1,16 @@
 #!/usr/bin/env python
 
 """Setup script for the pyparsing module distribution."""
-from setuptools import setup
+try:
+    from setuptools import setup
+except ImportError:
+    from distutils.core import setup
 
 import sys
 import os
 
 from pyparsing import __version__ as pyparsing_version
-    
+
 modules = ["pyparsing",]
 
 setup(# Distribution meta-data
