--- spack-src/CMakeLists.txt.bak	2019-02-05 19:20:28.000000000 +0900
+++ spack-src/CMakeLists.txt	2020-08-17 16:54:33.476417139 +0900
@@ -51,7 +51,7 @@
 # SIMD and FMA instructions set-------------------------------------------------
 if (NEON)
   message(STATUS "Using NEON instructions!")
-  set(PACKED_INSTR "-mfpu=neon ")
+  set(PACKED_INSTR " ")
 else()
   if (SSE AND (NOT (AVX OR AVX2) ))
     message(STATUS "Using SSE instructions!")
