--- a/FindSalomeHDF5.cmake	2020-02-14 09:27:51.918313640 +0100
+++ b/FindSalomeHDF5.cmake.patched	2020-02-14 09:28:38.658223715 +0100
@@ -69,15 +69,15 @@
     SET(_suffix "-shared")
     SET(HDF5_LIBRARIES "${_target_prefix}hdf5${_suffix}")
   ENDIF()
-  GET_PROPERTY(_lib_lst TARGET ${HDF5_LIBRARIES} PROPERTY IMPORTED_LINK_INTERFACE_LIBRARIES_NOCONFIG)
-  FOREACH(s ${_lib_lst})
-    STRING(FIND "${s}" "mpi." _res)   # should cover WIN(?) and LINUX
-    IF(_res GREATER -1)
-      GET_FILENAME_COMPONENT(_tmp "${s}" PATH)     # go up to levels
-      GET_FILENAME_COMPONENT(MPI_ROOT_DIR_EXP "${_tmp}" PATH)
-      BREAK()
-    ENDIF()
-  ENDFOREACH()
+#  GET_PROPERTY(_lib_lst TARGET ${HDF5_LIBRARIES} PROPERTY IMPORTED_LINK_INTERFACE_LIBRARIES_NOCONFIG)
+#  FOREACH(s ${_lib_lst})
+#    STRING(FIND "${s}" "mpi." _res)   # should cover WIN(?) and LINUX
+#    IF(_res GREATER -1)
+#      GET_FILENAME_COMPONENT(_tmp "${s}" PATH)     # go up to levels
+#      GET_FILENAME_COMPONENT(MPI_ROOT_DIR_EXP "${_tmp}" PATH)
+#      BREAK()
+#    ENDIF()
+#  ENDFOREACH()
   IF(NOT SalomeHDF5_FIND_QUIETLY)
     MESSAGE(STATUS "HDF5 was compiled with MPI: ${MPI_ROOT_DIR_EXP}")
   ENDIF()  
