--- a/cmake/atlas_compile_flags.cmake	2023-07-10 04:13:02.000000000 -0600
+++ b/cmake/atlas_compile_flags.cmake	2023-08-21 14:29:10.000000000 -0600
@@ -21,6 +21,8 @@
 
 endif()
 
+ecbuild_add_cxx_flags("-fp-speculation=safe")
+
 if( CMAKE_CXX_COMPILER_ID MATCHES NVHPC )
   ecbuild_add_cxx_flags("--diag_suppress declared_but_not_referenced --display_error_number" NAME atlas_cxx_disable_warnings )
   # For all the variables with side effects (constructor/dectructor functionality)
