diff -r -u a/clients/benchmarks/CMakeLists.txt b/clients/benchmarks/CMakeLists.txt
--- a/clients/benchmarks/CMakeLists.txt	2021-11-12 12:22:24.359556397 -0700
+++ b/clients/benchmarks/CMakeLists.txt	2021-11-12 14:21:31.246604351 -0700
@@ -52,6 +52,8 @@
 target_link_libraries( rocblas-bench PRIVATE rocblas_fortran_client roc::rocblas lapack cblas )
 if(LINK_BLIS)
   target_link_libraries( rocblas-bench PRIVATE ${BLIS_LIBRARY} )
+else()
+  target_link_libraries( rocblas-bench PRIVATE blas )
 endif()
 
 if( CUDA_FOUND )
diff -r -u a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
--- a/clients/gtest/CMakeLists.txt	2021-11-12 12:22:24.359556397 -0700
+++ b/clients/gtest/CMakeLists.txt	2021-11-12 14:20:59.057676192 -0700
@@ -132,6 +132,8 @@
 target_link_libraries( rocblas-test PRIVATE rocblas_fortran_client roc::rocblas lapack cblas ${GTEST_LIBRARIES} )
 if(LINK_BLIS)
   target_link_libraries( rocblas-test PRIVATE ${BLIS_LIBRARY} )
+else()
+  target_link_libraries( rocblas-test PRIVATE blas )
 endif()
 
 
