--- a/config/ymake	2015-03-16 22:21:42.000000000 +0100
+++ b/config/ymake	2016-10-14 13:44:49.530646098 +0200
@@ -537,0 +538,3 @@
+# We want to have our own definitions for spack
+set sysincs = Spack
+
--- a/Configure	2015-03-16 22:22:17.000000000 +0100
+++ b/Configure	2016-10-14 13:49:42.157631106 +0200
@@ -1137,5 +1137,13 @@
-    if (! -d $incs[1]) then
-      echo "  *** Warning: <$incs[1]> does not exist"
-      echo ""
-      goto proc_locincdir
-    else 
+
+    # We don't want our path(s) to be preprocessed by cpp
+    # inside ymake script. That is why we pass them in quotes (')
+    # to this script. But if we do so, the following condition
+    # is always false. That is why we comment it out and promise
+    # to pass only correct path(s). You might want to do the same
+    # thing for the libraries search path(s).
+
+    # if (! -d $incs[1]) then
+    #   echo "  *** Warning: <$incs[1]> does not exist"
+    #   echo ""
+    #   goto proc_locincdir
+    # else 
@@ -1143 +1151 @@
-    endif
+    # endif
