diff --git a/CMakeLists.txt b/CMakeLists.txt
index aeb0442..8283e5c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,11 +56,18 @@ set( CMAKE_CXX_FLAGS "${flags_str} ${CMAKE_CXX_FLAGS}")
 
 set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -L${ROCM_PATH}/lib")
 include_directories(${ROCM_PATH}/include)
-link_libraries(numa hsa-runtime64 pthread)
+find_package(hsa-runtime64 REQUIRED)
+
+find_library(NUMA numa)
+find_path(NUMA_INCLUDE_DIR numa.h)
+
+link_libraries(${NUMA} hsa-runtime64::hsa-runtime64 pthread)
 set (CMAKE_RUNTIME_OUTPUT_DIRECTORY .)
 add_executable(TransferBench src/client/Client.cpp)
 target_include_directories(TransferBench PRIVATE src/header src/client src/client/Presets)
 
+target_include_directories(TransferBench PRIVATE ${NUMA_INCLUDE_DIR})
+
 find_package(ROCM 0.8 REQUIRED PATHS ${ROCM_PATH})
 include(ROCMInstallTargets)
 include(ROCMCreatePackage)
