--- a/configure
+++ b/configure
@@ -3307,8 +3307,6 @@ ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
-CFLAGS="${CFLAGS} -fno-strict-aliasing"
-
 # Expand $ac_aux_dir to an absolute path.
 am_aux_dir=`cd "$ac_aux_dir" && pwd`
 
@@ -4161,6 +4159,9 @@ ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
 
+
+CFLAGS="${CFLAGS} -fno-strict-aliasing"
+
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
--- a/configure.ac
+++ b/configure.ac
@@ -61,6 +61,7 @@ AC_CONFIG_HEADERS([config.h])
 ##
 SAVE_CFLAGS="${CFLAGS}"
 AC_LANG_PUSH([C])
+AC_LANG_COMPILER_REQUIRE
 CFLAGS="${CFLAGS} -fno-strict-aliasing"
 
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([
