--- a/libhdf5/hdf5internal.c
+++ b/libhdf5/hdf5internal.c
@@ -163,7 +163,7 @@ find_var_dim_max_length(NC_GRP_INFO_T *grp, int varid, int dimid,
                 if (var->dimids[d] == dimid)
                     *maxlen = *maxlen > h5dimlen[d] ? *maxlen : h5dimlen[d];
 
-#ifdef USE_PARALLEL
+#ifdef USE_PARALLEL4
 	    /* If we are doing parallel I/O in collective mode (with
 	     * either pnetcdf or HDF5), then communicate with all
 	     * other tasks in the collective and find out which has
