diff --git a/libmetis/pmetis.c b/libmetis/pmetis.c
index 9174aa3..f8ced79 100644
--- a/libmetis/pmetis.c
+++ b/libmetis/pmetis.c
@@ -163,8 +163,6 @@ idx_t MlevelRecursiveBisection(ctrl_t *ctrl, graph_t *graph, idx_t nparts,
   real_t wsum, *tpwgts2;
 
   if ((nvtxs = graph->nvtxs) == 0) {
-    printf("\t***Cannot bisect a graph with 0 vertices!\n"
-           "\t***You are trying to partition a graph into too many parts!\n");
     return 0;
   }
 
