--- org/abinit-9.4.2/configure	2021-04-30 18:25:20.000000000 +0900
+++ spack-src/configure	2021-10-14 16:05:13.000000000 +0900
@@ -13019,6 +13019,30 @@
 
   # Do some sanity checking of the arguments
 
+    cc_info_string=`${CC} --version 2>&1 | head -n 1`
+  abi_result=`echo "${cc_info_string}" | grep 'FCC'`
+  if test "${abi_result}" = ""; then
+    abi_result="no"
+    cc_info_string=""
+    abi_cc_vendor="unknown"
+    abi_cc_version="unknown"
+  else
+
+$as_echo "#define CC_FUJITSU 1" >>confdefs.h
+
+    abi_cc_vendor="fujitsu"
+    abi_cc_version=`echo "${abi_result}" | sed -e 's/[^0-9]*\([0-9]\)/\1/' -e 's/ .*$//'`
+    if test "${abi_cc_version}" = "${abi_result}"; then
+      abi_cc_version="unknown"
+    fi
+    abi_result="yes"
+  fi
+
+  fi
+  if test "${abi_cc_vendor}" = "unknown"; then
+
+  # Do some sanity checking of the arguments
+
     cc_info_string=`${CC} --version 2>/dev/null | head -n 1`
   abi_result=`echo "${cc_info_string}" | grep '[Cc]lang'`
   if test "${abi_result}" = ""; then
@@ -13891,6 +13915,30 @@
 
   # Do some sanity checking of the arguments
 
+    cxx_info_string=`${CXX} --version 2>&1 | head -n 1`
+  abi_result=`echo "${cxx_info_string}" | grep 'FCC'`
+  if test "${abi_result}" = ""; then
+    abi_result="no"
+    cxx_info_string=""
+    abi_cxx_vendor="unknown"
+    abi_cxx_version="unknown"
+  else
+
+$as_echo "#define CXX_FUJITSU 1" >>confdefs.h
+
+    abi_cxx_vendor="fujitsu"
+    abi_cxx_version=`echo "${abi_result}" | sed -e 's/[^0-9]*\([0-9]\)/\1/' -e 's/ .*$//'`
+    if test "${abi_cxx_version}" = "${abi_result}"; then
+      abi_cxx_version="unknown"
+    fi
+    abi_result="yes"
+  fi
+
+  fi
+  if test "${abi_cxx_vendor}" = "unknown"; then
+
+  # Do some sanity checking of the arguments
+
     cxx_info_string=`${CXX} --version 2>/dev/null | head -n 1`
   abi_result=`echo "${cxx_info_string}" | grep '[Cc]lang'`
   if test "${abi_result}" = ""; then
@@ -14697,6 +14745,32 @@
   fi
   echo "${fc_info_string}" >>"${tmp_fc_info_file}"
 
+  if test "${abi_fc_vendor}" = "unknown"; then
+
+  # Do some sanity checking of the arguments
+
+    fc_info_string=`${FC} --version 2>&1 | head -n 1 `
+  abi_result=`echo "${fc_info_string}" | grep 'FRT'`
+  if test "${abi_result}" = ""; then
+    abi_result="no"
+    fc_info_string=""
+    abi_fc_vendor="unknown"
+    abi_fc_version="unknown"
+  else
+
+$as_echo "#define FC_FUJITSU 1" >>confdefs.h
+
+    abi_fc_vendor="fujitsu"
+    abi_fc_version=`echo "${abi_result}" | sed -e 's/[^0-9]*\([0-9]\)/\1/' -e 's/ .*$//'`
+    if test "${abi_fc_version}" = "${abi_result}"; then
+      abi_fc_version="unknown"
+    fi
+    abi_result="yes"
+  fi
+
+  fi
+  echo "${fc_info_string}" >>"${tmp_fc_info_file}"
+
  if test "${abi_fc_vendor}" = "unknown"; then
 
   # Do some sanity checking of the arguments
@@ -15049,6 +15123,7 @@
 # gfortran 4.3 outputs lines setting COLLECT_GCC_OPTIONS, COMPILER_PATH,
 # LIBRARY_PATH; skip all such settings.
 ac_fc_v_output=`eval $ac_link 5>&1 2>&1 |
+  sed -r "s/(\-L)(\/[^ ]+)+(\/bin\/\.\.\/lib64\/nofjobj)//g" |
   sed '/^Driving:/d; /^Configured with:/d;
       '"/^[_$as_cr_Letters][_$as_cr_alnum]*=/d"`
 $as_echo "$ac_fc_v_output" >&5
@@ -15157,6 +15232,7 @@
 # gfortran 4.3 outputs lines setting COLLECT_GCC_OPTIONS, COMPILER_PATH,
 # LIBRARY_PATH; skip all such settings.
 ac_fc_v_output=`eval $ac_link 5>&1 2>&1 |
+  sed -r "s/(\-L)(\/[^ ]+)+(\/bin\/\.\.\/lib64\/nofjobj)//g" |
   sed '/^Driving:/d; /^Configured with:/d;
       '"/^[_$as_cr_Letters][_$as_cr_alnum]*=/d"`
 $as_echo "$ac_fc_v_output" >&5
@@ -15982,6 +16058,13 @@
       openmp='-qopenmp'
       CFLAGS_PIC='-fPIC'
       ;;
+    fujitsu)
+      abi_cc_vendor_hnt="fujitsu"
+      abi_cc_version_hnt="default"
+      abi_sys_spec_hnt="default"
+      CFLAGS_PIC='-fPIC'
+      openmp='-fopenmp'
+      ;;
     llvm)
       abi_cc_vendor_hnt="llvm"
       abi_cc_version_hnt="default"
@@ -16050,6 +16133,12 @@
       CXX_LDFLAGS_HINTS='-static-libgcc -static-intel'
       CXXFLAGS_PIC='-fPIC'
       ;;
+    fujitsu)
+      abi_cxx_vendor_hnt="fujitsu"
+      abi_cxx_version_hnt="default"
+      abi_sys_spec_hnt="default"
+      CXXFLAGS_PIC='-fPIC'
+      ;;
     llvm)
       abi_cxx_vendor_hnt="llvm"
       abi_cxx_version_hnt="default"
@@ -16222,6 +16311,27 @@
       FCFLAGS_HINTS='-Mextend'
       FC_LDFLAGS_HINTS=''
       ;;
+    fujitsu)
+      abi_fc_vendor_hnt="fujitsu"
+      abi_sys_spec_hnt="default"
+      FCFLAGS_FIXEDFORM='-Fixed -X7'
+      FCFLAGS_FREEFORM='-Free -X9'
+      FCFLAGS_MODDIR='-M ../mods'
+      FCFLAGS_PIC='-KPIC'
+      FC_LDFLAGS_HINTS=''
+      case "${abi_fc_version}" in
+         4.*)
+            abi_fc_version_hnt="4.x"
+            FCFLAGS_OPENMP='-Kopenmp'
+            FCFLAGS_HINTS=''
+            ;;
+         *)
+            abi_fc_version_hnt="default"
+            FCFLAGS_HINTS='-Am -Ee -Ep'
+            FCFLAGS_OPENMP='--openmp'
+            ;;
+      esac
+      ;;
   esac   # [case: abi_fc_vendor, indent: 0, item: True]
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: ${abi_fc_vendor_hnt}/${abi_fc_version_hnt}/${abi_sys_spec_hnt}" >&5
@@ -16878,6 +16988,25 @@
           ;;
       esac   # [case: abi_optim_flavor, indent: 2, item: True]
       ;;
+    fujitsu)
+      abi_cc_vendor_opt="fujitsu"
+      abi_cc_version_opt="default"
+      abi_cpu_spec_opt="default"
+      case "${abi_optim_flavor}" in
+        safe)
+          abi_optim_flavor_opt="safe"
+          CFLAGS_OPTIM="-O2"
+          ;;
+        standard)
+          abi_optim_flavor_opt="standard"
+          CFLAGS_OPTIM="-O2"
+          ;;
+        aggressive)
+          abi_optim_flavor_opt="aggressive"
+          CFLAGS_OPTIM="-Kfast"
+          ;;
+      esac   # [case: abi_optim_flavor, indent: 2, item: True]
+      ;;
     llvm)
       abi_cc_vendor_opt="llvm"
       abi_cc_version_opt="default"
@@ -17031,6 +17160,25 @@
           ;;
       esac   # [case: abi_optim_flavor, indent: 2, item: True]
       ;;
+    fujitsu)
+      abi_cxx_vendor_opt="fujitsu"
+      abi_cxx_version_opt="default"
+      abi_cpu_spec_opt="default"
+      case "${abi_optim_flavor}" in
+        safe)
+          abi_optim_flavor_opt="safe"
+          CXXFLAGS_OPTIM="-O2"
+          ;;
+        standard)
+          abi_optim_flavor_opt="standard"
+          CXXFLAGS_OPTIM="-O2"
+          ;;
+        aggressive)
+          abi_optim_flavor_opt="aggressive"
+          CXXFLAGS_OPTIM="-Kfast"
+          ;;
+      esac   # [case: abi_optim_flavor, indent: 2, item: True]
+      ;;
     llvm)
       abi_cxx_vendor_opt="llvm"
       abi_cxx_version_opt="default"
@@ -17100,11 +17248,11 @@
           ;;
         standard)
           abi_optim_flavor_opt="standard"
-          FCFLAGS_OPTIM="-O2 -mtune=native -march=native"
+          FCFLAGS_OPTIM="-O2"
           ;;
         aggressive)
           abi_optim_flavor_opt="aggressive"
-          FCFLAGS_OPTIM="-O3 -mtune=native -march=native -funroll-loops -faggressive-function-elimination"
+          FCFLAGS_OPTIM="-O3 -funroll-loops -faggressive-function-elimination"
           ;;
       esac   # [case: abi_optim_flavor, indent: 2, item: True]
       ;;
@@ -17302,6 +17450,46 @@
           ;;
       esac   # [case: abi_optim_flavor, indent: 2, item: True]
       ;;
+    fujitsu)
+      abi_fc_vendor_opt="fujitsu"
+      abi_cpu_spec_opt="default"
+      case "${abi_fc_version}" in
+          4.*)
+	      abi_fc_version_opt="4.X"
+	      case "${abi_optim_flavor}" in
+		safe)
+		  abi_optim_flavor_opt="safe"
+		  FCFLAGS_OPTIM="-O2 -Koptmsg=2 -Nlst=t"
+		  ;;
+		standard)
+		  abi_optim_flavor_opt="standard"
+		  FCFLAGS_OPTIM="-O2 -Koptmsg=2 -Nlst=t"
+		  ;;
+		aggressive)
+		  abi_optim_flavor_opt="aggressive"
+		  FCFLAGS_OPTIM="-Kfast -Koptmsg=2 -Nlst=t"
+		  ;;
+	      esac
+	      ;;
+          *)
+	      abi_fc_version_opt="default"
+	      case "${abi_optim_flavor}" in
+		safe)
+		  abi_optim_flavor_opt="safe"
+		  FCFLAGS_OPTIM="-Of -X9 -Ps -Wv,-md"
+		  ;;
+		standard)
+		  abi_optim_flavor_opt="standard"
+		  FCFLAGS_OPTIM="-Of -X9 -Ps -Wv,-md"
+		  ;;
+		aggressive)
+		  abi_optim_flavor_opt="aggressive"
+		  FCFLAGS_OPTIM="-Of -X9 -Ps -Wv,-md"
+		  ;;
+	      esac
+	      ;;
+      esac
+      ;;
   esac   # [case: abi_fc_vendor, indent: 0, item: True]
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: ${abi_fc_vendor_opt}/${abi_fc_version_opt}/${abi_cpu_spec_opt}" >&5
