diff --git a/src/vtkh/filters/CMakeLists.txt b/src/vtkh/filters/CMakeLists.txt
index 4a42d61..f586155 100644
--- a/src/vtkh/filters/CMakeLists.txt
+++ b/src/vtkh/filters/CMakeLists.txt
@@ -41,6 +41,11 @@ set(vtkh_filters_deps vtkh_core vtkh_utils )
 if(CUDA_FOUND)
     # triggers cuda compile
     list(APPEND vtkh_filters_deps cuda)
+    
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 10.0.0)
+        set(particle_cuda_src "${CMAKE_CURRENT_SOURCE_DIR}/Lagrangian.cpp")
+        set_source_files_properties(${particle_cuda_src} PROPERTIES COMPILE_FLAGS "-Xptxas --opt-level=0")
+    endif()
 endif()
 
 
