From ed64417746c570a00b0ba6c2c8091de4845f9ed1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tiziano=20M=C3=BCller?= <tiziano.mueller@chem.uzh.ch>
Date: Tue, 6 Oct 2020 17:48:12 +0200
Subject: [PATCH 4/5] fix building c code with pgcc

---
 bsd/i386.make | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bsd/i386.make b/bsd/i386.make
index 759ed84..9018c1b 100644
--- a/bsd/i386.make
+++ b/bsd/i386.make
@@ -130,7 +130,7 @@ VECTOR = -Mvect=assoc,recog,noaltcode,cachesize:$(CSIZE)$(VECTOR4)
 MACHTY = p7-32
 MACH = -tp $(MACHTY) $(TIME)
 OPTOI = -m32 -march=i486 -malign-double 
-GCCOPTS = -ffast-math -funroll-loops -fexpensive-optimizations
+GCCOPTS = -fast
 OPTFLAGO = $(OPTOI) -O3 $(GCCOPTS)
 # Flags for portland compiler.
 #
-- 
2.26.2

