diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index 24e25ab..1907108 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -72,6 +72,7 @@ include(src/CMakeLists.txt)
 # Create hipBLASLt library
 add_library(hipblaslt ${hipblaslt_source} ${hipblaslt_headers_public})
 add_library(roc::hipblaslt ALIAS hipblaslt)
+target_include_directories( hipblaslt PRIVATE ${MSGPACK_DIR}/include )
 
 # Target compile definitions
 if(NOT BUILD_CUDA)
@@ -159,7 +160,9 @@ if(NOT BUILD_CUDA)
 endif()
 
 if(HIPBLASLT_ENABLE_MARKER)
-  target_link_libraries(hipblaslt PRIVATE -lroctx64)
+  find_path(ROCTRACER_INCLUDE_DIR "roctracer/roctx.h")
+  target_include_directories(hipblaslt PRIVATE ${ROCTRACER_INCLUDE_DIR})
+  target_link_libraries(hipblaslt PRIVATE ${rocTracer})
 endif()
 
 if( NOT BUILD_CUDA AND NOT LEGACY_HIPBLAS_DIRECT )
diff --git a/tensilelite/Tensile/Source/client/CMakeLists.txt b/tensilelite/Tensile/Source/client/CMakeLists.txt
index 8dc7401..e6d0ba9 100644
--- a/tensilelite/Tensile/Source/client/CMakeLists.txt
+++ b/tensilelite/Tensile/Source/client/CMakeLists.txt
@@ -64,7 +64,9 @@ target_include_directories(TensileClient PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/inc
 target_link_libraries(TensileClient PRIVATE TensileHost Boost::program_options Boost::filesystem rocm_smi)
 
 if(Tensile_ENABLE_MARKER)
-    target_link_libraries(TensileClient PRIVATE -lroctx64)
+    find_path(ROCTRACER_INCLUDE_DIR "roctracer/roctx.h")
+    target_include_directories(TensileClient PRIVATE ${ROCTRACER_INCLUDE_DIR})
+    target_link_libraries(TensileClient PRIVATE ${rocTracer})
 endif()
 
 if(TENSILE_USE_OPENMP)
diff --git a/tensilelite/Tensile/cmake/TensileConfig.cmake b/tensilelite/Tensile/cmake/TensileConfig.cmake
index 7627780..ac3560a 100644
--- a/tensilelite/Tensile/cmake/TensileConfig.cmake
+++ b/tensilelite/Tensile/cmake/TensileConfig.cmake
@@ -234,6 +234,8 @@ function(TensileCreateLibraryFiles
     set(Options ${Options} "--build-id=${Tensile_BUILD_ID}")
   endif()
 
+  set(Options ${Options} "--offload-bundler=$ENV{TENSILE_ROCM_OFFLOAD_BUNDLER_PATH}")
+
   set(CommandLine ${VIRTUALENV_BIN_DIR}/${VIRTUALENV_PYTHON_EXENAME} ${Script} ${Options} ${Tensile_LOGIC_PATH} ${Tensile_OUTPUT_PATH} HIP)
   message(STATUS "Tensile_CREATE_COMMAND: ${CommandLine}")
 
