diff -rupN nest-2.4.2/models/tsodyks2_connection.h nest-2.4.2_mod/models/tsodyks2_connection.h
--- nest-2.4.2/models/tsodyks2_connection.h	2014-08-19 11:46:16.000000000 +0200
+++ nest-2.4.2_mod/models/tsodyks2_connection.h	2017-02-02 15:57:58.374342403 +0100
@@ -157,8 +157,18 @@ void Tsodyks2Connection::send(Event& e,
   double_t u_decay = (tau_fac_ < 1.0e-10) ? 0.0 : std::exp(-h/tau_fac_);
 
   // now we compute spike number n+1
-  x_= 1. + (x_ -x_*u_ -1.)*x_decay; // Eq. 5 from reference [3]
-  u_= U_+u_*(1.-U_)*u_decay;       // Eq. 4 from [3] 
+  // x_= 1. + (x_ -x_*u_ -1.)*x_decay; // Eq. 5 from reference [3]
+  // change obreitwi:
+  if (t_lastspike > 0)
+  {
+    x_= 1. + (x_ -x_*u_ -1.)*x_decay; // Eq. 5 from reference [3]
+    u_= U_+u_*(1.-U_)*u_decay;        // Eq. 4 from [3] 
+  }
+  else
+  {
+    x_=1.;
+    u_= U_;                           // Eq. 4 from [3] 
+  }
 
   // We use the current values for the spike number n.
   e.set_receiver(*target_);
