diff -r -u a/clients/benchmarks/CMakeLists.txt b/clients/benchmarks/CMakeLists.txt
--- a/clients/benchmarks/CMakeLists.txt	2021-10-28 14:14:41.379987882 -0600
+++ b/clients/benchmarks/CMakeLists.txt	2021-10-28 13:37:50.409696036 -0600
@@ -53,7 +53,7 @@
     $<BUILD_INTERFACE:${HIP_INCLUDE_DIRS}>
 )
 
-target_link_libraries( hipblas-bench PRIVATE hipblas_fortran_client roc::hipblas cblas lapack)
+target_link_libraries( hipblas-bench PRIVATE hipblas_fortran_client roc::hipblas cblas lapack blas)
 
 # need mf16c flag for float->half convertion
 target_compile_options( hipblas-bench PRIVATE -mf16c)
diff -r -u a/clients/gtest/CMakeLists.txt b/clients/gtest/CMakeLists.txt
--- a/clients/gtest/CMakeLists.txt	2021-10-28 14:14:41.379987882 -0600
+++ b/clients/gtest/CMakeLists.txt	2021-10-28 13:37:34.609274623 -0600
@@ -146,7 +146,7 @@
     ${ROCM_PATH}/hsa/include
 )
 
-target_link_libraries( hipblas-test PRIVATE roc::hipblas cblas lapack ${GTEST_LIBRARIES} ${Boost_LIBRARIES} hipblas_fortran_client )
+target_link_libraries( hipblas-test PRIVATE roc::hipblas cblas lapack blas ${GTEST_LIBRARIES} ${Boost_LIBRARIES} hipblas_fortran_client )
 
 # need mf16c flag for float->half convertion
 target_compile_options( hipblas-test PRIVATE -mf16c )
