--- ./Makefile	2013-11-30 08:54:00.000000000 +0100
+++ ./Makefile	2020-02-25 12:44:03.889985931 +0100
@@ -92,14 +92,14 @@
 #################
 
 gcc:
-	make $(APP)  CC="gcc" CFLAGS="-O2 -Wall -Werror"
+	make $(APP)  CC="gcc" CFLAGS="-O2 -Wall "
 
 all:
-	make $(APP)  CC="gcc" CFLAGS="-O2 -Wall -Werror"
-	make $(APP2) CC="gcc" CFLAGS="-O2 -Wall -Werror"
-	make $(APP3) CC="gcc" CFLAGS="-O2 -Wall -Werror"
-	make $(APP4) CC="gcc" CFLAGS="-O2 -Wall -Werror"
-	make $(APP5) CC="gcc" CFLAGS="-O2 -Wall -Werror"
+	make $(APP)  CC="gcc" CFLAGS="-O2 -Wall "
+	make $(APP2) CC="gcc" CFLAGS="-O2 -Wall "
+	make $(APP3) CC="gcc" CFLAGS="-O2 -Wall "
+	make $(APP4) CC="gcc" CFLAGS="-O2 -Wall "
+	make $(APP5) CC="gcc" CFLAGS="-O2 -Wall "
 
 
 ###################
