--- wonton-1.2.11-orig/cmake/wontonConfig.cmake.in	2021-01-20 13:25:56.000000000 -0700
+++ wonton-1.2.11/cmake/wontonConfig.cmake.in	2022-09-12 16:28:11.104950000 -0600
@@ -12,10 +12,6 @@
 set(WONTON_VERSION_MINOR @WONTON_VERSION_PATCH@ CACHE STRING "Patch version of WONTON")
 
 
-# Compiler Definitions
-set(CMAKE_C_COMPILER @CMAKE_C_COMPILER@ CACHE FILEPATH "C Compiler used for compiling WONTON")
-set(CMAKE_CXX_COMPILER @CMAKE_CXX_COMPILER@ CACHE FILEPATH "CXX Compiler used for compiling WONTON")
-
 # Wonton installation path
 set(WONTON_ROOT @CMAKE_INSTALL_PREFIX@ CACHE PATH "Path to Wonton installation")
 
@@ -65,6 +61,7 @@
 endif ()
 
 if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.15)
+  set(SAVED_CMAKE_FIND_PACKAGE_PREFER_CONFIG "${CMAKE_FIND_PACKAGE_PREFER_CONFIG}")
   set(CMAKE_FIND_PACKAGE_PREFER_CONFIG TRUE)  # search for PkgConfig.cmake files first
 endif ()
 
@@ -135,6 +132,10 @@
 # Restore original CMAKE_MODULE_PATH
 set(CMAKE_MODULE_PATH ${SAVED_CMAKE_MODULE_PATH})
 
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.15)
+  set(CMAKE_FIND_PACKAGE_PREFER_CONFIG "${SAVED_CMAKE_FIND_PACKAGE_PREFER_CONFIG}")
+endif ()
+
 #
 # Import wonton targets
 #
