--- fltk-1.3.3/test/menubar.cxx	2012-02-18 17:29:30.000000000 +0900
+++ fltk-1.3.3/test/menubar_b.cxx	2019-08-06 14:16:55.733089438 +0900
@@ -127,10 +127,10 @@
     {"Italic",	0, 0, 0, 0, 0, FL_ITALIC, 14},
     {"BoldItalic",0,0,0, 0, 0, FL_BOLD+FL_ITALIC, 14},
     {"Small",	0, 0, 0, 0, 0, FL_BOLD+FL_ITALIC, 10},
-    {"Emboss",	0, 0, 0, 0, FL_EMBOSSED_LABEL},
-    {"Engrave",	0, 0, 0, 0, FL_ENGRAVED_LABEL},
-    {"Shadow",	0, 0, 0, 0, FL_SHADOW_LABEL},
-    {"@->",	0, 0, 0, 0, FL_SYMBOL_LABEL},
+    {"Emboss",	0, 0, 0, 0, (uchar)FL_EMBOSSED_LABEL},
+    {"Engrave",	0, 0, 0, 0, (uchar)FL_ENGRAVED_LABEL},
+    {"Shadow",	0, 0, 0, 0, (uchar)FL_SHADOW_LABEL},
+    {"@->",	0, 0, 0, 0, (uchar)FL_SYMBOL_LABEL},
     {0},
   {"&International",0,0,0,FL_SUBMENU},
     {"Sharp Ess",0x0000df},
