diff --git a/config/BuildSystem/config/framework.py b/config/BuildSystem/config/framework.py
index 259f4a35fac..06c54bfedee 100644
--- a/config/BuildSystem/config/framework.py
+++ b/config/BuildSystem/config/framework.py
@@ -582,6 +582,8 @@ class Framework(config.base.Configure, script.LanguageProcessor):
       lines = [s for s in lines if s.find(' stack subq instruction is too different from dwarf stack size') < 0]
       lines = [s for s in lines if s.find('could not create compact unwind') < 0]
       lines = [s for s in lines if s.find('ld: warning: -undefined dynamic_lookup may not work with chained fixups') < 0]
+      lines = [s for s in lines if s.find('ld: warning: duplicate -rpath ') < 0]
+      lines = [s for s in lines if s.find('ld: warning: -multiply_defined is obsolete') < 0]
       # Nvidia linker
       lines = [s for s in lines if s.find('nvhpc.ld contains output sections') < 0]
       # Intel dpcpp linker
