diff --git a/include/petscsys.h b/include/petscsys.h
index 933d30bfc8..4c4c991251 100644
--- a/include/petscsys.h
+++ b/include/petscsys.h
@@ -294,7 +294,7 @@ M*/
 .seealso: PetscReal, PetscScalar, PetscComplex, PetscInt, MPIU_REAL, MPIU_SCALAR, MPIU_COMPLEX
 M*/
 
-#if defined(PETSC_HAVE_STDINT_H) && defined(PETSC_HAVE_INTTYPES_H) && defined(PETSC_HAVE_MPI_INT64_T) /* MPI_INT64_T is not guaranteed to be a macro */
+#if defined(PETSC_HAVE_STDINT_H) && defined(PETSC_HAVE_INTTYPES_H)
 typedef int64_t PetscInt64;
 # define MPIU_INT64 MPI_INT64_T
 # define PetscInt64_FMT PRId64
