diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3bd1ce0..dc72b16 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,6 +80,8 @@ include_directories(src)
 include_directories(src/include)
 include_directories(src/collectives)
 include_directories(src/collectives/device)
+include_directories(${NUMACTL_DIR}/include)
+link_directories(${NUMACTL_DIR}/lib)
 
 set(CU_SOURCES
     src/collectives/device/all_reduce.cu
