diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1deb1ba..f2bbf30 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -117,6 +117,8 @@ include_directories(src)
 include_directories(src/include)
 include_directories(src/collectives)
 include_directories(src/collectives/device)
+include_directories(${NUMACTL_DIR}/include)
+link_directories(${NUMACTL_DIR}/lib)
 
 set(CU_SOURCES
     src/collectives/device/all_reduce.cu
@@ -241,8 +243,8 @@ if("${HIP_COMPILER}" MATCHES "hcc")
   endif()
 endif()
 
-target_include_directories(rccl PRIVATE ${ROCM_PATH}/rocm_smi/include)
-target_link_libraries(rccl PRIVATE hip::device dl -lrocm_smi64 -L${ROCM_PATH}/rocm_smi/lib)
+target_include_directories(rccl PRIVATE ${ROCM_SMI_DIR}/include)
+target_link_libraries(rccl PRIVATE hip::device dl -lrocm_smi64 -L${ROCM_SMI_DIR}/lib)
 target_link_libraries(rccl INTERFACE hip::host)
 
 #Setup librccl.so version
