From d929b8a0c4fcaaf2ce37cff335b89455db043780 Mon Sep 17 00:00:00 2001
From: Renjith Ravindran <Renjith.RavindranKannath@amd.com>
Date: Fri, 19 May 2023 22:36:11 +0000
Subject: [PATCH] Updating with hsa include path in CMakeList

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9c139ef..f53cedf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -134,6 +134,10 @@ include_directories(${PROJECT_BINARY_DIR}/src/include) # for hipified header fil
 include_directories(src)
 include_directories(src/collectives)
 include_directories(src/collectives/device)
+include_directories(${NUMACTL_DIR}/include)
+include_directories(${HSA_INCLUDE_PATH})
+link_directories(${NUMACTL_DIR}/lib)
+
 
 if (BUILD_ALLREDUCE_ONLY)
   add_definitions(-DBUILD_ALLREDUCE_ONLY)
-- 
2.17.1

