From 5d40f88d3af525326aaeae052f0e7345445336d8 Mon Sep 17 00:00:00 2001
From: Renjith Ravindran <Renjith.RavindranKannath@amd.com>
Date: Wed, 21 Sep 2022 05:42:54 +0000
Subject: [PATCH] 5.2.3 changes for spack

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 79e0a35..21f2a54 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -123,6 +123,8 @@ include_directories(src)
 include_directories(src/include)
 include_directories(src/collectives)
 include_directories(src/collectives/device)
+include_directories(${NUMACTL_DIR}/include)
+link_directories(${NUMACTL_DIR}/lib)
 
 # Create git version file
 # _git_version.cpp is a dummy output to force re-run prior to build
-- 
2.17.1

