--- a/configure
+++ b/configure
@@ -22237,6 +22237,8 @@ _LT_EOF
 	  whole_archive_flag_spec_F77=
 	  tmp_sharedflag='--shared' ;;
 	nagfor*)			# NAGFOR 5.3
+	  whole_archive_flag_spec_F77='${wl}--whole-archive`for conv in $convenience\"\"; do test  -n \"$conv\" && new_convenience=\"$new_convenience,$conv\"; done; func_echo_all \"$new_convenience\"` ${wl}--no-whole-archive'
+	  compiler_needs_object_F77=yes
 	  tmp_sharedflag='-Wl,-shared' ;;
 	xl[cC]* | bgxl[cC]* | mpixl[cC]*) # IBM XL C 8.0 on PPC (deal with xlf below)
 	  tmp_sharedflag='-qmkshrobj'
