diff --git a/config/ApplicationUnitTest.cmake b/config/ApplicationUnitTest.cmake
index a0a79858..0c47b72a 100644
--- a/config/ApplicationUnitTest.cmake
+++ b/config/ApplicationUnitTest.cmake
@@ -249,7 +249,7 @@ macro( aut_register_test )
   endif(VERBOSE_DEBUG)

   # Look for python, which is used to drive application unit tests
-  if( NOT PYTHONINTERP_FOUND )
+  if( NOT Python_Interpreter_FOUND )
      # python should have been found when vendor_libraries.cmake was run.
     message( FATAL_ERROR "Draco requires python. Python not found in PATH.")
   endif()
@@ -289,7 +289,7 @@ macro( aut_register_test )
   if (${PYTHON_TEST})
     add_test(
       NAME ${ctestname_base}${argname}
-      COMMAND "${PYTHON_EXECUTABLE}"
+      COMMAND "${Python_EXECUTABLE}"
       ${aut_DRIVER}
       ${SHARED_ARGUMENTS}
       )
diff --git a/config/draco-config-install.cmake.in b/config/draco-config-install.cmake.in
index c5bf1c75..a16f72f4 100644
--- a/config/draco-config-install.cmake.in
+++ b/config/draco-config-install.cmake.in
@@ -107,8 +107,9 @@ set( WITH_CUDA "@WITH_CUDA@" )
 #endif()

 # Python
-set( PYTHONINTERP_FOUND "@PYTHONINTERP_FOUND@" )
-set( PYTHON_EXECUTABLE "@PYTHON_EXECUTABLE@" )
+set( Python_FOUND "@Python_FOUND@" )
+set( Python_Interpreter_FOUND "@Python_Interpreter_FOUND@" )
+set( Python_EXECUTABLE "@Python_EXECUTABLE@" )

 ## ---------------------------------------------------------------------------
 ## Set useful general variables
diff --git a/config/vendor_libraries.cmake b/config/vendor_libraries.cmake
index c3e079bc..6b393eb4 100644
--- a/config/vendor_libraries.cmake
+++ b/config/vendor_libraries.cmake
@@ -16,7 +16,7 @@ include( setupMPI ) # defines the macros setupMPILibrariesUnix|Windows
 macro( setupPython )

   message( STATUS "Looking for Python...." )
-  find_package(PythonInterp 2.7 QUIET REQUIRED)
+  find_package(Python 3.5 QUIET REQUIRED COMPONENTS Interpreter)
   #  PYTHONINTERP_FOUND - Was the Python executable found
   #  PYTHON_EXECUTABLE  - path to the Python interpreter
   set_package_properties( PythonInterp PROPERTIES
@@ -25,8 +25,8 @@ macro( setupPython )
     TYPE REQUIRED
     PURPOSE "Required for running tests and accessing features that rely on matplotlib."
     )
-  if( PYTHONINTERP_FOUND )
-    message( STATUS "Looking for Python....found ${PYTHON_EXECUTABLE}" )
+  if( Python_Interpreter_FOUND )
+    message( STATUS "Looking for Python....found ${Python_EXECUTABLE}" )
   else()
     message( STATUS "Looking for Python....not found" )
   endif()
