diff --git a/setup.py b/setup.py
index ee44bad..414bb94 100644
--- a/setup.py
+++ b/setup.py
@@ -22,7 +22,6 @@ def setup_package():
       packages=find_packages(exclude=('tests', 'docs')),
       setup_requires=[
         'setuptools_scm>=1.15.6',
-        'pyscaffold>=2.5a0,<2.6a0',
       ] + sphinx,
       install_requires=[
         'clustershell',
@@ -33,7 +32,6 @@ def setup_package():
         'seaborn',
         'six',
       ],
-      use_pyscaffold=True,
       entry_points='''
       [console_scripts]
       whatismyuc = bb5.uc:main
