diff --git a/tests/kfdtest/CMakeLists.txt b/tests/kfdtest/CMakeLists.txt
index 5c5db26..f9f6bbc 100644
--- a/tests/kfdtest/CMakeLists.txt
+++ b/tests/kfdtest/CMakeLists.txt
@@ -148,6 +148,12 @@ include_directories(${PROJECT_SOURCE_DIR}/../../include)
 
 include_directories(${DRM_INCLUDE_DIRS})
 
+if( DEFINED ENV{NUMA_PATH} )
+	set ( NUMA_PATH $ENV{NUMA_PATH} )
+	include_directories(${NUMA_PATH}/include)
+	link_directories(${NUMA_PATH}/lib)
+endif()
+
 set (SRC_FILES gtest-1.6.0/gtest-all.cpp
 
   src/AqlQueue.cpp
