From 0060fd48c12a59a080974ca3754bf0eab9ab6d35 Mon Sep 17 00:00:00 2001
From: Howard Pritchard <howardp@lanl.gov>
Date: Tue, 24 Nov 2020 15:14:41 -0700
Subject: [PATCH] exit prior to running configure

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

diff --git a/autogen.sh b/autogen.sh
index bbca69dc..4de1e783 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -112,6 +112,8 @@ if test -z "$*"; then
         echo "to pass any to it, please specify them on the $0 command line."
 fi
 
+exit 0
+
 echo "Generating configuration files for $package, please wait...."
 
 echo "  $ACLOCAL $ACLOCAL_FLAGS"
-- 
2.18.2

