From 93ab69cac72cc2d13cfd4b7efcc235bdbca2b9f5 Mon Sep 17 00:00:00 2001
From: Afzal Patel <afzal.patel@amd.com>
Date: Wed, 17 Jan 2024 11:44:18 -0800
Subject: [PATCH] Add hip library path to ghv flags so libamdhip64 can be found

---
 config/makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config/makefile b/config/makefile
index 627d117..a453865 100644
--- a/config/makefile
+++ b/config/makefile
@@ -38,7 +38,7 @@ all: header config-mk
 MPI = $(MFEM_USE_MPI:NO=)
 GHV_CXX ?= $(MFEM_CXX)
 GHV = get_hypre_version
-GHV_FLAGS = $(subst @MFEM_DIR@,$(if $(MFEM_DIR),$(MFEM_DIR),..),$(HYPRE_OPT))
+GHV_FLAGS = $(subst @MFEM_DIR@,$(if $(MFEM_DIR),$(MFEM_DIR),..),$(HYPRE_OPT)) $(HIP_LIB)
 SMX = $(if $(MFEM_USE_PUMI:NO=),MFEM_USE_SIMMETRIX)
 SMX_PATH = $(PUMI_DIR)/include/gmi_sim.h
 SMX_FILE = $(subst @MFEM_DIR@,$(if $(MFEM_DIR),$(MFEM_DIR),..),$(SMX_PATH))
--
2.25.1