--- modules/pmg_tk/startup/apbs_tools.py.orig	2014-05-14 01:07:32.000000000 +0400
+++ modules/pmg_tk/startup/apbs_tools.py	2014-05-14 01:15:35.090032827 +0400
@@ -612,7 +612,8 @@
         # Set up the Main page
         page = self.notebook.add('Main')
         group = Pmw.Group(page,tag_text='Main options')
-        group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        #group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        group.grid(padx = 10, pady = 5, sticky=(N, S, E, W))
         self.selection = Pmw.EntryField(group.interior(),
                                         labelpos='w',
                                         label_text='Selection to use: ',
@@ -856,7 +864,8 @@
         page.grid_columnconfigure(5,weight=1)
         page = self.notebook.add('Program Locations')
         group = Pmw.Group(page,tag_text='Locations')
-        group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        #group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        group.grid(padx = 10, pady = 5, sticky=(N, S, E, W))
         def quickFileValidation(s):
             if s == '': return Pmw.PARTIAL
             elif os.path.isfile(s): return Pmw.OK
@@ -955,7 +964,8 @@

         page = self.notebook.add('Temp File Locations')
         group = Pmw.Group(page,tag_text='Locations')
-        group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        #group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        group.grid(padx = 10, pady = 5, sticky=(N, S, E, W))
         self.pymol_generated_pqr_filename = Pmw.EntryField(group.interior(),
                                                            labelpos = 'w',
                                                            label_pyclass = FileDialogButtonClassFactory.get(self.setPymolGeneratedPqrFilename),
@@ -1003,17 +1013,20 @@
         page = self.notebook.add('Visualization (1)')
         group = VisualizationGroup(page,tag_text='Visualization',visgroup_num=1)
         self.visualization_group_1 = group
-        group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        #group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        group.grid(padx = 10, pady = 5, sticky=(N, S, E, W))

         page = self.notebook.add('Visualization (2)')
         group = VisualizationGroup(page,tag_text='Visualization',visgroup_num=2)
         self.visualization_group_2 = group
-        group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        #group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        group.grid(padx = 10, pady = 5, sticky=(N, S, E, W))

         # Create a couple of other empty pages
         page = self.notebook.add('About')
         group = Pmw.Group(page, tag_text='About PyMOL APBS Tools')
-        group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        #group.pack(fill = 'both', expand = 1, padx = 10, pady = 5)
+        group.grid(padx = 10, pady = 5, sticky=(N, S, E, W))
         text = """This plugin integrates PyMOL (http://PyMOL.org/) with APBS (http://www.poissonboltzmann.org/apbs/).

 Documentation may be found at
@@ -2271,7 +2284,8 @@
             self.update_buttonbox = Pmw.ButtonBox(self.mm_group.interior(), padx=0)
             self.update_buttonbox.pack(side=LEFT)
             self.update_buttonbox.add('Update',command=self.refresh)
-            self.mm_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=TOP)
+            #self.mm_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=TOP)
+            self.mm_group.grid(padx = 4, pady = 5, sticky=(N, S, E, W))

             self.ms_group = Pmw.Group(self.interior(),tag_text='Molecular Surface')
             self.ms_buttonbox = Pmw.ButtonBox(self.ms_group.interior(), padx=0)
@@ -2322,7 +2336,8 @@
             bars = (self.mol_surf_low,self.mol_surf_middle,self.mol_surf_high)
             Pmw.alignlabels(bars)
             for bar in bars: bar.pack(side=LEFT)
-            self.ms_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=LEFT)
+            #self.ms_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=LEFT)
+            self.ms_group.grid(padx = 4, pady = 5, sticky=(N, S, E, W))

             self.fl_group = Pmw.Group(self.interior(),tag_text='Field Lines')
             self.fl_buttonbox = Pmw.ButtonBox(self.fl_group.interior(), padx=0)
@@ -2337,7 +2352,8 @@
                                   text = """Follows same coloring as surface.""",
                                   )
             label.pack()
-            self.fl_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=TOP)
+            #self.fl_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=TOP)
+            self.fl_group.grid(padx = 4, pady = 5, sticky=(N, S, E, W))

             self.pi_group = Pmw.Group(self.interior(),tag_text='Positive Isosurface')
             self.pi_buttonbox = Pmw.ButtonBox(self.pi_group.interior(), padx=0)
@@ -2356,7 +2372,8 @@
                                             entryfield_validate = {'validator' : 'real', 'min':0}
                                             )
             self.pos_surf_val.pack(side=LEFT)
-            self.pi_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=LEFT)
+            #self.pi_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=LEFT)
+            self.pi_group.grid(padx = 4, pady = 5, sticky=(N, S, E, W))

             self.ni_group = Pmw.Group(self.interior(),tag_text='Negative Isosurface')
             self.ni_buttonbox = Pmw.ButtonBox(self.ni_group.interior(), padx=0)
@@ -2375,7 +2392,8 @@
                                             entryfield_validate = {'validator' : 'real', 'max':0}
                                             )
             self.neg_surf_val.pack(side=LEFT)
-            self.ni_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=LEFT)
+            #self.ni_group.pack(fill = 'both', expand = 1, padx = 4, pady = 5, side=LEFT)
+            self.ni_group.grid(padx = 4, pady = 5, sticky=(N, S, E, W))



