diff -Naur pytime.f pytime.f
--- pytime.f	2018-08-23 11:24:27.000000000 -0500
+++ pytime.f	2018-08-23 11:46:08.189999826 -0500
@@ -1,4 +1,3 @@
- 
 C*********************************************************************
  
 C...PYTIME
@@ -59,17 +58,18 @@
 C      IDATI(6)=ISEC
  
 C...Example 4: GNU LINUX libU77, SunOS.
-C      CALL IDATE(IDTEMP)
-C      IDATI(1)=IDTEMP(3)
-C      IDATI(2)=IDTEMP(2)
-C      IDATI(3)=IDTEMP(1)
-C      CALL ITIME(IDTEMP)
-C      IDATI(4)=IDTEMP(1)
-C      IDATI(5)=IDTEMP(2)
-C      IDATI(6)=IDTEMP(3)
+      CALL IDATE(IDTEMP)
+      IDATI(1)=IDTEMP(3)
+      IDATI(2)=IDTEMP(2)
+      IDATI(3)=IDTEMP(1)
+      CALL ITIME(IDTEMP)
+      IDATI(4)=IDTEMP(1)
+      IDATI(5)=IDTEMP(2)
+      IDATI(6)=IDTEMP(3)
  
 C...Common code to ensure right century.
       IDATI(1)=2000+MOD(IDATI(1),100)
  
       RETURN
       END
+ 
diff -Naur upevnt.f upevnt.f
--- upevnt.f	2018-08-23 11:24:27.000000000 -0500
+++ upevnt.f	2018-08-23 11:46:08.189999826 -0500
@@ -1,56 +1,3 @@
-
-C...Old example: handles a simple Pythia 6.4 initialization file.
- 
-c      SUBROUTINE UPINIT
- 
-C...Double precision and integer declarations.
-c      IMPLICIT DOUBLE PRECISION(A-H, O-Z)
-c      IMPLICIT INTEGER(I-N)
- 
-C...Commonblocks.
-c      COMMON/PYDAT1/MSTU(200),PARU(200),MSTJ(200),PARJ(200)
-c      COMMON/PYPARS/MSTP(200),PARP(200),MSTI(200),PARI(200)
-c      SAVE /PYDAT1/,/PYPARS/
- 
-C...User process initialization commonblock.
-c      INTEGER MAXPUP
-c      PARAMETER (MAXPUP=100)
-c      INTEGER IDBMUP,PDFGUP,PDFSUP,IDWTUP,NPRUP,LPRUP
-c      DOUBLE PRECISION EBMUP,XSECUP,XERRUP,XMAXUP
-c      COMMON/HEPRUP/IDBMUP(2),EBMUP(2),PDFGUP(2),PDFSUP(2),
-c     &IDWTUP,NPRUP,XSECUP(MAXPUP),XERRUP(MAXPUP),XMAXUP(MAXPUP),
-c     &LPRUP(MAXPUP)
-c      SAVE /HEPRUP/
- 
-C...Read info from file.
-c      IF(MSTP(161).GT.0) THEN
-c        READ(MSTP(161),*,END=110,ERR=110) IDBMUP(1),IDBMUP(2),EBMUP(1),
-c     &  EBMUP(2),PDFGUP(1),PDFGUP(2),PDFSUP(1),PDFSUP(2),IDWTUP,NPRUP
-c        DO 100 IPR=1,NPRUP
-c          READ(MSTP(161),*,END=110,ERR=110) XSECUP(IPR),XERRUP(IPR),
-c     &    XMAXUP(IPR),LPRUP(IPR)
-c  100   CONTINUE
-c        RETURN
-C...Error or prematurely reached end of file.
-c  110   WRITE(MSTU(11),5000)
-c        STOP
- 
-C...Else not implemented.
-c      ELSE
-c        WRITE(MSTU(11),5100)
-c        STOP
-c      ENDIF
- 
-C...Format for error printout.
-c 5000 FORMAT(1X,'Error: UPINIT routine failed to read information'/
-c     &1X,'Execution stopped!')
-c 5100 FORMAT(1X,'Error: You have not implemented UPINIT routine'/
-c     &1X,'Dummy routine in PYTHIA file called instead.'/
-c     &1X,'Execution stopped!')
- 
-c      RETURN
-c      END
- 
 C*********************************************************************
  
 C...UPEVNT
@@ -120,3 +67,54 @@
  
       RETURN
       END
+
+C...Old example: handles a simple Pythia 6.4 event file.
+ 
+c      SUBROUTINE UPEVNT
+ 
+C...Double precision and integer declarations.
+c      IMPLICIT DOUBLE PRECISION(A-H, O-Z)
+c      IMPLICIT INTEGER(I-N)
+ 
+C...Commonblocks.
+c      COMMON/PYDAT1/MSTU(200),PARU(200),MSTJ(200),PARJ(200)
+c      COMMON/PYPARS/MSTP(200),PARP(200),MSTI(200),PARI(200)
+c      SAVE /PYDAT1/,/PYPARS/
+ 
+C...User process event common block.
+c      INTEGER MAXNUP
+c      PARAMETER (MAXNUP=500)
+c      INTEGER NUP,IDPRUP,IDUP,ISTUP,MOTHUP,ICOLUP
+c      DOUBLE PRECISION XWGTUP,SCALUP,AQEDUP,AQCDUP,PUP,VTIMUP,SPINUP
+c      COMMON/HEPEUP/NUP,IDPRUP,XWGTUP,SCALUP,AQEDUP,AQCDUP,IDUP(MAXNUP),
+c     &ISTUP(MAXNUP),MOTHUP(2,MAXNUP),ICOLUP(2,MAXNUP),PUP(5,MAXNUP),
+c     &VTIMUP(MAXNUP),SPINUP(MAXNUP)
+c      SAVE /HEPEUP/
+ 
+C...Read info from file.
+c      IF(MSTP(162).GT.0) THEN
+c        READ(MSTP(162),*,END=110,ERR=110) NUP,IDPRUP,XWGTUP,SCALUP,
+c     &  AQEDUP,AQCDUP
+c        DO 100 I=1,NUP
+c          READ(MSTP(162),*,END=110,ERR=110) IDUP(I),ISTUP(I),
+c     &    MOTHUP(1,I),MOTHUP(2,I),ICOLUP(1,I),ICOLUP(2,I),
+c     &    (PUP(J,I),J=1,5),VTIMUP(I),SPINUP(I)
+c  100   CONTINUE
+c        RETURN
+C...Special when reached end of file or other error.
+c  110   NUP=0
+ 
+C...Else not implemented.
+c      ELSE
+c        WRITE(MSTU(11),5000)
+c        STOP
+c      ENDIF
+ 
+C...Format for error printout.
+c 5000 FORMAT(1X,'Error: You have not implemented UPEVNT routine'/
+c     &1X,'Dummy routine in PYTHIA file called instead.'/
+c     &1X,'Execution stopped!')
+ 
+c      RETURN
+c      END
+ 
diff -Naur upinit.f upinit.f
--- upinit.f	2018-08-23 11:24:27.000000000 -0500
+++ upinit.f	2018-08-23 11:46:08.190999819 -0500
@@ -1,4 +1,3 @@
- 
 C*********************************************************************
  
 C...UPINIT
@@ -64,3 +63,56 @@
  
       RETURN
       END
+
+C...Old example: handles a simple Pythia 6.4 initialization file.
+ 
+c      SUBROUTINE UPINIT
+ 
+C...Double precision and integer declarations.
+c      IMPLICIT DOUBLE PRECISION(A-H, O-Z)
+c      IMPLICIT INTEGER(I-N)
+ 
+C...Commonblocks.
+c      COMMON/PYDAT1/MSTU(200),PARU(200),MSTJ(200),PARJ(200)
+c      COMMON/PYPARS/MSTP(200),PARP(200),MSTI(200),PARI(200)
+c      SAVE /PYDAT1/,/PYPARS/
+ 
+C...User process initialization commonblock.
+c      INTEGER MAXPUP
+c      PARAMETER (MAXPUP=100)
+c      INTEGER IDBMUP,PDFGUP,PDFSUP,IDWTUP,NPRUP,LPRUP
+c      DOUBLE PRECISION EBMUP,XSECUP,XERRUP,XMAXUP
+c      COMMON/HEPRUP/IDBMUP(2),EBMUP(2),PDFGUP(2),PDFSUP(2),
+c     &IDWTUP,NPRUP,XSECUP(MAXPUP),XERRUP(MAXPUP),XMAXUP(MAXPUP),
+c     &LPRUP(MAXPUP)
+c      SAVE /HEPRUP/
+ 
+C...Read info from file.
+c      IF(MSTP(161).GT.0) THEN
+c        READ(MSTP(161),*,END=110,ERR=110) IDBMUP(1),IDBMUP(2),EBMUP(1),
+c     &  EBMUP(2),PDFGUP(1),PDFGUP(2),PDFSUP(1),PDFSUP(2),IDWTUP,NPRUP
+c        DO 100 IPR=1,NPRUP
+c          READ(MSTP(161),*,END=110,ERR=110) XSECUP(IPR),XERRUP(IPR),
+c     &    XMAXUP(IPR),LPRUP(IPR)
+c  100   CONTINUE
+c        RETURN
+C...Error or prematurely reached end of file.
+c  110   WRITE(MSTU(11),5000)
+c        STOP
+ 
+C...Else not implemented.
+c      ELSE
+c        WRITE(MSTU(11),5100)
+c        STOP
+c      ENDIF
+ 
+C...Format for error printout.
+c 5000 FORMAT(1X,'Error: UPINIT routine failed to read information'/
+c     &1X,'Execution stopped!')
+c 5100 FORMAT(1X,'Error: You have not implemented UPINIT routine'/
+c     &1X,'Dummy routine in PYTHIA file called instead.'/
+c     &1X,'Execution stopped!')
+ 
+c      RETURN
+c      END
+ 
diff -Naur upveto.f upveto.f
--- upveto.f	2018-08-23 11:24:27.000000000 -0500
+++ upveto.f	2018-08-23 11:46:08.190999819 -0500
@@ -1,54 +1,3 @@
-
-C...Old example: handles a simple Pythia 6.4 event file.
- 
-c      SUBROUTINE UPEVNT
- 
-C...Double precision and integer declarations.
-c      IMPLICIT DOUBLE PRECISION(A-H, O-Z)
-c      IMPLICIT INTEGER(I-N)
- 
-C...Commonblocks.
-c      COMMON/PYDAT1/MSTU(200),PARU(200),MSTJ(200),PARJ(200)
-c      COMMON/PYPARS/MSTP(200),PARP(200),MSTI(200),PARI(200)
-c      SAVE /PYDAT1/,/PYPARS/
- 
-C...User process event common block.
-c      INTEGER MAXNUP
-c      PARAMETER (MAXNUP=500)
-c      INTEGER NUP,IDPRUP,IDUP,ISTUP,MOTHUP,ICOLUP
-c      DOUBLE PRECISION XWGTUP,SCALUP,AQEDUP,AQCDUP,PUP,VTIMUP,SPINUP
-c      COMMON/HEPEUP/NUP,IDPRUP,XWGTUP,SCALUP,AQEDUP,AQCDUP,IDUP(MAXNUP),
-c     &ISTUP(MAXNUP),MOTHUP(2,MAXNUP),ICOLUP(2,MAXNUP),PUP(5,MAXNUP),
-c     &VTIMUP(MAXNUP),SPINUP(MAXNUP)
-c      SAVE /HEPEUP/
- 
-C...Read info from file.
-c      IF(MSTP(162).GT.0) THEN
-c        READ(MSTP(162),*,END=110,ERR=110) NUP,IDPRUP,XWGTUP,SCALUP,
-c     &  AQEDUP,AQCDUP
-c        DO 100 I=1,NUP
-c          READ(MSTP(162),*,END=110,ERR=110) IDUP(I),ISTUP(I),
-c     &    MOTHUP(1,I),MOTHUP(2,I),ICOLUP(1,I),ICOLUP(2,I),
-c     &    (PUP(J,I),J=1,5),VTIMUP(I),SPINUP(I)
-c  100   CONTINUE
-c        RETURN
-C...Special when reached end of file or other error.
-c  110   NUP=0
- 
-C...Else not implemented.
-c      ELSE
-c        WRITE(MSTU(11),5000)
-c        STOP
-c      ENDIF
- 
-C...Format for error printout.
-c 5000 FORMAT(1X,'Error: You have not implemented UPEVNT routine'/
-c     &1X,'Dummy routine in PYTHIA file called instead.'/
-c     &1X,'Execution stopped!')
- 
-c      RETURN
-c      END
- 
 C*********************************************************************
  
 C...UPVETO
@@ -98,3 +47,4 @@
  
       RETURN
       END
+ 
--- pystrf.f.orig	2014-05-14 18:12:02.000000001 +0200
+++ pystrf.f	2014-05-14 18:10:23.000000001 +0200
@@ -805,7 +805,9 @@
  
 C...Closed string: random initial breakup flavour, pT and vertex.
       ELSE
-        KFL(3)=INT(1D0+(2D0+PARJ(2))*PYR(0))*(-1)**INT(PYR(0)+0.5D0)
+        RNDM1=PYR(0)
+        RNDM2=PYR(0)
+        KFL(3)=INT(1D0+(2D0+PARJ(2))*RNDM1)*(-1)**INT(RNDM2+0.5D0)
         IBMO=0
   770   CALL PYKFDI(KFL(3),0,KFL(1),KDUMP)
 C.. Closed string: first vertex diq attempt => enforced second
