diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2df9e2..8afcce5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,8 +1,4 @@
 cmake_minimum_required(VERSION 2.8)
-set(CMAKE_BUILD_TYPE  "Release")
-SET(CMAKE_C_FLAGS   "-w -g -O3 -funroll-loops")
-set(CMAKE_MACOSX_RPATH 1)
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
 project(Commons)
 
@@ -13,7 +9,6 @@ if(OPENMP)
   find_package(OpenMP)
   if (OPENMP_FOUND)
     set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
   endif()
 endif()
 FIND_LIBRARY(          M_LIB m)
@@ -29,7 +24,6 @@ TARGET_LINK_LIBRARIES( Commons ${LIBRARIES})
 
 INSTALL(
   TARGETS Commons
-  RUNTIME DESTINATION "$ENV{HOME}/lib"
-  LIBRARY DESTINATION "$ENV{HOME}/lib")
+  LIBRARY DESTINATION ${CMAKE_INSTALL_PREFIX}/lib)
 
-INSTALL(               FILES   ${header_files} DESTINATION "$ENV{HOME}/include")
+INSTALL(FILES ${header_files} DESTINATION ${CMAKE_INSTALL_PREFIX}/include)
