diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index 762580d..1692578 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -81,7 +81,10 @@ function( rocblas_library_settings lib_target_ )
   else()
     target_link_libraries( ${lib_target_} PRIVATE hip::device -lstdc++fs --rtlib=compiler-rt --unwindlib=libgcc)
     if (BUILD_SHARED_LIBS)
-      target_link_libraries(${lib_target_} PRIVATE -lroctx64)
+      find_path(ROCTRACER_INCLUDE_DIR "roctracer/roctx.h")
+      target_include_directories(${lib_target_} PUBLIC ${ROCTRACER_INCLUDE_DIR})
+      find_library(ROCTX_LIB NAMES roctx64)
+      target_link_libraries(${lib_target_} PRIVATE ${ROCTX_LIB})
     endif()
   endif()
     target_link_libraries( ${lib_target_} PRIVATE Threads::Threads )
