diff --git a/clients/CMakeLists.txt b/clients/CMakeLists.txt
index 6194286..292bd8d 100644
--- a/clients/CMakeLists.txt
+++ b/clients/CMakeLists.txt
@@ -178,8 +178,8 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
 
   # Find the package ROCmSMI
   if(NOT WIN32)
-      find_package(ROCmSMI REQUIRED)
-      list( APPEND COMMON_LINK_LIBS rocm_smi )
+      find_package(rocm_smi CONFIG REQUIRED)
+      list( APPEND COMMON_LINK_LIBS rocm_smi64 )
   endif()
 
   find_package( GTest REQUIRED )
