--- qt-x11-free-3.3.8b/src/tools/qmap.h	2008-01-15 13:09:13.000000000 -0600
+++ qt-x11-free-3.3.8b-fixes/src/tools/qmap.h	2015-07-08 15:47:34.757565247 -0500
@@ -52,6 +52,7 @@
 #ifndef QT_NO_STL
 #include <iterator>
 #include <map>
+#include <cstddef>
 #endif
 
 //#define QT_CHECK_MAP_RANGE
--- qt-x11-free-3.3.8b/src/tools/qvaluelist.h	2008-01-15 13:09:13.000000000 -0600
+++ qt-x11-free-3.3.8b-fixes/src/tools/qvaluelist.h	2015-07-08 15:47:34.758565247 -0500
@@ -50,6 +50,7 @@
 #ifndef QT_NO_STL
 #include <iterator>
 #include <list>
+#include <cstddef>
 #endif
 
 //#define QT_CHECK_VALUELIST_RANGE
--- qt-x11-free-3.3.8b/src/tools/qvaluevector.h	2008-01-15 13:09:13.000000000 -0600
+++ qt-x11-free-3.3.8b-fixes/src/tools/qvaluevector.h	2015-07-08 15:47:34.758565247 -0500
@@ -47,6 +47,7 @@
 
 #ifndef QT_NO_STL
 #include <vector>
+#include <cstddef>
 #endif
 
 template <class T>
--- qt-x11-free-3.3.8b/configure	2008-01-15 13:09:15.000000000 -0600
+++ qt-x11-free-3.3.8b-fixes/configure	2015-07-08 15:49:03.379560633 -0500
@@ -2339,7 +2339,7 @@
         else
             echo "Do you accept the terms of the $TheLicense? \c"
         fi
-        read acceptance
+        acceptance=yes
         echo
 	if [ "$acceptance" = yes ]; then
 	    break
@@ -2397,7 +2397,7 @@
 	else
 	    echo "Do you accept the terms of $affix license? \c"
 	fi
-	read acceptance
+        acceptance=yes
 	echo
 	if [ "$acceptance" = "yes" ]; then
 	    break
@@ -2443,7 +2443,7 @@
 	else
 	    echo "Do you accept the terms of the license? \c"
 	fi
-	read acceptance
+        acceptance=yes
 	echo
 	if [ "$acceptance" = "yes" ]; then
 	    break
@@ -2524,7 +2524,7 @@
 	else
 	    echo "Do you accept the terms of the $Platform License? \c"
 	fi
-	read acceptance
+        acceptance=yes
 	echo
 	if [ "$acceptance" = "yes" ]; then
 	    break
