diff -r -u a/tools/linguist/linguist/messagemodel.cpp b/tools/linguist/linguist/messagemodel.cpp
--- a/tools/linguist/linguist/messagemodel.cpp	2015-05-07 10:14:39.000000000 -0400
+++ b/tools/linguist/linguist/messagemodel.cpp	2019-03-20 09:30:21.000000000 -0400
@@ -183,7 +183,7 @@
         if (ContextItem *c = one->findContext(oc->context())) {
             for (int j = 0; j < oc->messageCount(); ++j) {
                 MessageItem *m = oc->messageItem(j);
-                if (c->findMessage(m->text(), m->comment()) >= 0)
+                if (c->findMessage(m->text(), m->comment()))
                     ++inBoth;
             }
         }
