diff --git a/configure.pri b/configure.pri
index be1b166080..13fdf0b91e 100644
--- a/configure.pri
+++ b/configure.pri
@@ -796,10 +796,10 @@ defineTest(qtConfOutput_preparePaths) {
         "static const char qt_configure_installation     [12+11]  = \"qt_instdate=2012-12-20\";" \
         "" \
         "/* Installation Info */" \
-        "static const char qt_configure_prefix_path_str  [12+256] = \"qt_prfxpath=$$config.input.prefix\";" \
+        "static const char qt_configure_prefix_path_str  [12+1024] = \"qt_prfxpath=$$config.input.prefix\";" \
         "$${LITERAL_HASH}ifdef QT_BUILD_QMAKE" \
-        "static const char qt_configure_ext_prefix_path_str   [12+256] = \"qt_epfxpath=$$config.input.extprefix\";" \
-        "static const char qt_configure_host_prefix_path_str  [12+256] = \"qt_hpfxpath=$$config.input.hostprefix\";" \
+        "static const char qt_configure_ext_prefix_path_str   [12+1024] = \"qt_epfxpath=$$config.input.extprefix\";" \
+        "static const char qt_configure_host_prefix_path_str  [12+1024] = \"qt_hpfxpath=$$config.input.hostprefix\";" \
         "$${LITERAL_HASH}endif" \
         "" \
         "static const short qt_configure_str_offsets[] = {" \
