diff --git a/Makefile b/Makefile
index 967ab1bb6..98666e853 100644
--- a/Makefile
+++ b/Makefile
@@ -42,6 +42,9 @@ SUBDIRS_ALL = $(SUBDIRS) test ctest utest exports benchmark ../laswp ../bench cp
 .PHONY : all libs netlib $(RELA) test ctest shared install
 .NOTPARALLEL : all libs $(RELA) prof lapack-test install blas-test
 
+shared: libs netlib $(RELA)
+tests: libs netlib $(RELA) shared
+
 all :: libs netlib $(RELA) tests shared
 	@echo
 	@echo " OpenBLAS build complete. ($(LIB_COMPONENTS))"
