diff -ur OpenBLAS-0.3.6/f_check OpenBLAS-0.3.6_fix/f_check
--- OpenBLAS-0.3.6/f_check	2019-04-30 02:22:19.000000000 +0900
+++ OpenBLAS-0.3.6_fix/f_check	2019-06-21 17:05:07.809067385 +0900
@@ -69,7 +69,11 @@
 	    $bu       = "_";
 	}
 
-	if ($data =~ /GNU/) {
+        if ($data =~ /Fujitsu/) {
+            $vendor = FUJITSU;
+            $openmp = "-openmp";
+        } elsif ($data =~ /GNU/) {
+#	if ($data =~ /GNU/ && $data !~ /Fujitsu/) {
 
 	    $data =~ /(\d)\.(\d).(\d)/;
 	    $major = $1;
