--- a/main/REDItoolDenovo.py	2020-07-16 17:35:46.008030346 -0500
+++ b/main/REDItoolDenovo.py	2020-07-16 17:38:39.700035490 -0500
@@ -768,7 +768,7 @@
 for j in ridxinfo.split('\n'): #MOD
 	l=(j.strip()).split('\t')
 	if l[0] in ['*', '']: continue  #MOD
- 	if int(l[2])+int(l[3]) > 0: rrefs[l[0]]=int(l[1])
+	if int(l[2])+int(l[3]) > 0: rrefs[l[0]]=int(l[1])
 frefs=[]
 fidxinfo=open(fastafile+'.fai')
 for j in fidxinfo:
