diff --git a/packages/amesos/src/Amesos_Superludist.cpp b/packages/amesos/src/Amesos_Superludist.cpp
index f0542faebe..ddca911356 100644
--- a/packages/amesos/src/Amesos_Superludist.cpp
+++ b/packages/amesos/src/Amesos_Superludist.cpp
@@ -135,7 +135,7 @@ Amesos_Superludist::Amesos_Superludist(const Epetra_LinearProblem &prob) :
   Equil_ = true;
   ColPerm_ = "MMD_AT_PLUS_A";
   perm_c_ = 0;
-  RowPerm_ = "LargeDiag";
+  RowPerm_ = "LargeDiag_MC64";
   perm_r_ = 0;
   IterRefine_ = "DOUBLE";
   ReplaceTinyPivot_ = true;
@@ -472,7 +472,7 @@ int Amesos_Superludist::Factor()
     }
 
     if( RowPerm_ == "NATURAL" ) PrivateSuperluData_->options_.RowPerm = (rowperm_t)NATURAL;
-    if( RowPerm_ == "LargeDiag" ) PrivateSuperluData_->options_.RowPerm = LargeDiag;
+    if( RowPerm_ == "LargeDiag_MC64" ) PrivateSuperluData_->options_.RowPerm = LargeDiag_MC64;
     else if( ColPerm_ == "MY_PERMR" ) {
       PrivateSuperluData_->options_.RowPerm = MY_PERMR;
       PrivateSuperluData_->ScalePermstruct_.perm_r = perm_r_;
diff --git a/packages/amesos/test/TestOptions/TestSuperludist.cpp b/packages/amesos/test/TestOptions/TestSuperludist.cpp
index dc1235f62d..de2ab9459a 100644
--- a/packages/amesos/test/TestOptions/TestSuperludist.cpp
+++ b/packages/amesos/test/TestOptions/TestSuperludist.cpp
@@ -35,9 +35,9 @@
 //   16                       false/true     false                  true           2
 //   17 SamePattern           true           false                  true           10
 //   18 RowPerm - NATURAL     true           false                  false          10
-//   19 RowPerm - LargeDiag   true           false                  false          10
+//   19 RowPerm - LargeDiag_MC64 true        false                  false          10
 //   20 RowPerm - NATURAL     true           false                  false          10
-//   21 RowPerm - LargeDiag   true           false                  false          10
+//   21 RowPerm - LargeDiag_MC64 true        false                  false          10
 //   22 RowPerm - TinyPivot=t true           false                  false          10
 //   23 RowPerm - TinyPivot=f true           false                  false          10
 //
@@ -703,7 +703,7 @@ int TestSuperludist( Epetra_CrsMatrix *& Amat,
       ParamList.set( "AddZeroToDiag", false );
       Teuchos::ParameterList& SuperludistParams = ParamList.sublist("Superludist") ;
       SuperludistParams.set( "ReuseSymbolic", false );
-      SuperludistParams.set( "RowPerm", "LargeDiag" );
+      SuperludistParams.set( "RowPerm", "LargeDiag_MC64" );
       SuperludistParams.set( "MaxProcesses", 10 );
       //  ParamList.print( std::cerr, 10 ) ;
 
@@ -772,7 +772,7 @@ int TestSuperludist( Epetra_CrsMatrix *& Amat,
       ParamList.set( "AddZeroToDiag", false );
       Teuchos::ParameterList& SuperludistParams = ParamList.sublist("Superludist") ;
       SuperludistParams.set( "ReuseSymbolic", true );
-      SuperludistParams.set( "RowPerm", "LargeDiag" );
+      SuperludistParams.set( "RowPerm", "LargeDiag_MC64" );
       SuperludistParams.set( "MaxProcesses", 10 );
       //  ParamList.print( std::cerr, 10 ) ;
 
