--- a/src/fgetln.c	2020-08-07 09:17:09.456951858 -0700
+++ b/src/fgetln.c	2020-08-07 09:16:12.471731317 -0700
@@ -25,6 +25,7 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "config.h"
 #include <stdio.h>
 #include <sys/cdefs.h>
 #include <sys/types.h>
--- a/src/fpurge.c	2020-08-07 09:17:00.871918633 -0700
+++ b/src/fpurge.c	2020-08-07 09:15:55.436665389 -0700
@@ -24,6 +24,7 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "config.h"
 #include <errno.h>
 #include <stdio.h>
 #include <stdio_ext.h>
--- a/src/funopen.c	2020-08-07 09:16:52.566886490 -0700
+++ b/src/funopen.c	2020-08-07 09:17:25.206012812 -0700
@@ -24,6 +24,7 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "config.h"
 #include <sys/cdefs.h>
 #include <sys/types.h>
 #include <errno.h>
--- a/src/getpeereid.c	2020-08-07 09:25:13.822834376 -0700
+++ b/src/getpeereid.c	2020-08-07 09:25:27.802889361 -0700
@@ -24,6 +24,7 @@
  * ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "config.h"
 #include <sys/cdefs.h>
 
 #include <sys/param.h>
--- a/src/progname.c	2020-08-07 09:23:02.168316940 -0700
+++ b/src/progname.c	2020-08-07 09:23:09.840346776 -0700
@@ -30,6 +30,7 @@
  * <https://sourceware.org/ml/libc-alpha/2006-03/msg00125.html>.
  */
 
+#include "config.h"
 #include <errno.h>
 #include <string.h>
 #include <stdlib.h>
--- a/src/setproctitle.c	2020-08-07 09:21:49.003033771 -0700
+++ b/src/setproctitle.c	2020-08-07 09:22:09.161111787 -0700
@@ -22,6 +22,7 @@
  * USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#include "config.h"
 #include <errno.h>
 #include <stddef.h>
 #include <stdarg.h>
