--- a/sys_linux/g77_gcc.conf	2021-10-14 16:24:57.912819100 +0900
+++ b/sys_linux/g77_gcc.conf	2021-10-14 16:25:00.840831262 +0900
@@ -5,7 +5,7 @@
 #           X2DRIV (/xdisp and /figdisp).
 # The arguments needed by the C compiler to locate X-window include files.
  
-   XINCL="-I/usr/X11R6/include"
+   XINCL=""
 
 # Optional: Needed by XMDRIV (/xmotif).
 # The arguments needed by the C compiler to locate Motif, Xt and
@@ -23,7 +23,7 @@
 # The arguments needed by the C compiler to locate Tcl, Tk and
 # X-window include files.
  
-   TK_INCL="-I/usr/include $XINCL"
+   TK_INCL="@TK_INCL@"
 
 # Optional: Needed by RVDRIV (/xrv).
 # The arguments needed by the C compiler to locate Rivet, Tcl, Tk and
@@ -34,35 +34,35 @@
 # Mandatory.
 # The FORTRAN compiler to use.
  
-   FCOMPL="g77"
+   FCOMPL="@FCOMPL@"
 
 # Mandatory.
 # The FORTRAN compiler flags to use when compiling the pgplot library.
 # (NB. makemake prepends -c to $FFLAGC where needed)
  
-   FFLAGC="-u -Wall -fPIC -O"
+   FFLAGC="@FFLAGC@"
 
 # Mandatory.
 # The FORTRAN compiler flags to use when compiling fortran demo programs.
 # This may need to include a flag to tell the compiler not to treat
 # backslash characters as C-style escape sequences
  
-   FFLAGD="-fno-backslash"
+   FFLAGD="@FFLAGD@"
 
 # Mandatory.
 # The C compiler to use.
  
-   CCOMPL="gcc"
+   CCOMPL="@CCOMPL@"
 
 # Mandatory.
 # The C compiler flags to use when compiling the pgplot library.
  
-   CFLAGC="-Wall -fPIC -DPG_PPU -O"
+   CFLAGC="@CFLAGC@"
 
 # Mandatory.
 # The C compiler flags to use when compiling C demo programs.
  
-   CFLAGD="-Wall -O"
+   CFLAGD="@CFLAGD@"
 
 # Optional: Only needed if the cpgplot library is to be compiled.
 # The flags to use when running pgbind to create the C pgplot wrapper
@@ -74,7 +74,7 @@
 # The library-specification flags to use when linking normal pgplot
 # demo programs.
  
-   LIBS="-L/usr/X11R6/lib -lX11"
+   LIBS="@LIBS@"
 
 # Optional: Needed by XMDRIV (/xmotif).
 # The library-specification flags to use when linking motif
@@ -92,7 +92,7 @@
 # The library-specification flags to use when linking Tk demo programs.
 # Note that you may need to append version numbers to -ltk and -ltcl.
  
-   TK_LIBS="-L/usr/lib -ltk -ltcl $LIBS -ldl"
+   TK_LIBS="@TK_LIBS@"
 
 # Mandatory.
 # On systems that have a ranlib utility, put "ranlib" here. On other
@@ -108,16 +108,16 @@
 # Optional: Needed if SHARED_LIB is set.
 # How to create a shared library from a trailing list of object files.
  
-   SHARED_LD="gcc -shared  -o $SHARED_LIB"
+   SHARED_LD="@SHARED_LD@"
 
 # Optional:
 # On systems such as Solaris 2.x, that allow specification of the
 # libraries that a shared library needs to be linked with when a
 # program that uses it is run, this variable should contain the
 # library-specification flags used to specify these libraries to
 # $SHARED_LD
  
-   SHARED_LIB_LIBS=""
+   SHARED_LIB_LIBS="@SHARED_LIB_LIBS@"
 
 # Optional:
 # Compiler name used on Next systems to compile objective-C files.
