--- OpenFOAM-v1806/wmake/scripts/have_kahip-ORIG	2018-06-28 16:39:32.000000000 +0200
+++ OpenFOAM-v1806/wmake/scripts/have_kahip	2018-08-11 13:37:18.250219013 +0200
@@ -83,10 +83,10 @@
         # FOAM_EXT_LIBBIN is allowed to be unset
         library=$(findFirstFile \
             $FOAM_EXT_LIBBIN/$library \
-            $METIS_ARCH_PATH/lib/$static \
-            $METIS_ARCH_PATH/lib/$library \
-            $METIS_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/$static \
-            $METIS_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/$library \
+            $KAHIP_ARCH_PATH/lib/$static \
+            $KAHIP_ARCH_PATH/lib/$library \
+            $KAHIP_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/$static \
+            $KAHIP_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH/$library \
         )
     elif isSystem "$KAHIP_ARCH_PATH"
     then
