diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 1526e4f..d39a80b 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -83,11 +83,12 @@ function(add_rocwmma_test TEST_TARGET TEST_SOURCE)
 
   list(APPEND TEST_SOURCE ${ARGN})
   add_executable(${TEST_TARGET} ${TEST_SOURCE})
-  target_link_libraries(${TEST_TARGET} rocwmma gtest)
+  target_link_libraries(${TEST_TARGET} rocwmma gtest ${ROCM_SMI_DIR}/lib)
   target_link_libraries(${TEST_TARGET} OpenMP::OpenMP_CXX "-L${HIP_CLANG_ROOT}/lib" "-Wl,-rpath=${HIP_CLANG_ROOT}/lib")
   target_include_directories(${TEST_TARGET} PRIVATE
                              ${CMAKE_CURRENT_SOURCE_DIR}
-                             ${ROCWMMA_TEST_INCLUDE_DIRS})
+                             ${ROCWMMA_TEST_INCLUDE_DIRS}
+		             ${ROCM_SMI_DIR}/include)
   # Use offload compress flag if it is supported
   if(BUILD_OFFLOAD_COMPRESS AND CXX_COMPILER_SUPPORTS_OFFLOAD_COMPRESS)
     target_compile_options(${TEST_TARGET} PRIVATE "--offload-compress" )
