diff --git a/configure.ac b/configure.ac
index cd9ff25..79bc511 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1859,7 +1859,9 @@ for plat in $platforms; do
         ;;
 
     x11)
-        PKG_CHECK_MODULES([XCB_DRI2], [x11-xcb xcb xcb-dri2 >= $XCBDRI2_REQUIRED xcb-xfixes])
+        if test "x$enable_dri" = "xyes"; then
+            PKG_CHECK_MODULES([XCB_DRI2], [x11-xcb xcb xcb-dri2 >= $XCBDRI2_REQUIRED xcb-xfixes])
+        fi
         DEFINES="$DEFINES -DHAVE_X11_PLATFORM"
         ;;
