diff --git a/meson.build b/meson.build
index d975b0d..cbd4a2c 100644
--- a/meson.build
+++ b/meson.build
@@ -1387,12 +1387,14 @@ if with_platform_x11
       dep_xshmfence = dependency('xshmfence', version : '>= 1.1')
     endif
   endif
-  if with_glx == 'dri'
+  if with_glx == 'dri' or with_glx == 'gallium-xlib'
+    dep_glproto = dependency('glproto', version : '>= 1.4.14')
+  endif
+  if with_glx == 'dri' 
     if with_dri_platform == 'drm'
       dep_dri2proto = dependency('dri2proto', version : '>= 2.8')
       dep_xxf86vm = dependency('xxf86vm')
     endif
-    dep_glproto = dependency('glproto', version : '>= 1.4.14')
   endif
   if (with_egl or (
       with_gallium_vdpau or with_gallium_xvmc or with_gallium_xa or
diff --git a/src/gallium/state_trackers/glx/xlib/meson.build b/src/gallium/state_trackers/glx/xlib/meson.build
index f4ee754..34b93c9 100644
--- a/src/gallium/state_trackers/glx/xlib/meson.build
+++ b/src/gallium/state_trackers/glx/xlib/meson.build
@@ -23,5 +23,5 @@ libxlib = static_library(
   files('glx_api.c', 'glx_getproc.c', 'glx_usefont.c', 'xm_api.c', 'xm_st.c'),
   c_args : c_vis_args,
   include_directories : [inc_common, inc_mapi, inc_mesa],
-  dependencies : [dep_x11, dep_xext, dep_xcb],
+  dependencies : [dep_x11, dep_xext, dep_xcb, dep_glproto],
 )
