--- a/Makefile	2018-12-08 05:36:48.000000000 -0600
+++ b/Makefile	2019-10-04 13:08:37.095337704 -0500
@@ -1,12 +1,6 @@
-CUDAPATH=/usr/local/cuda
-
-# Have this point to an old enough gcc (for nvcc)
-GCCPATH=/usr
-
 NVCC=nvcc
-CCPATH=${GCCPATH}/bin
 
 drv:
-	PATH=${PATH}:.:${CCPATH}:${PATH} ${NVCC} -I${CUDAPATH}/include -arch=compute_30 -ptx compare.cu -o compare.ptx
-	g++ -O3 -Wno-unused-result -I${CUDAPATH}/include -c gpu_burn-drv.cpp
-	g++ -o gpu_burn gpu_burn-drv.o -O3 -lcuda -L${CUDAPATH}/lib64 -L${CUDAPATH}/lib -Wl,-rpath=${CUDAPATH}/lib64 -Wl,-rpath=${CUDAPATH}/lib -lcublas -lcudart -o gpu_burn
+	${NVCC} -arch=compute_30 -ptx compare.cu -o compare.ptx
+	g++ -O3 -Wno-unused-result -c gpu_burn-drv.cpp
+	g++ -o gpu_burn gpu_burn-drv.o -O3 -lcuda -lcublas -lcudart -o gpu_burn
