diff -up ftgl-2.1.3~rc5/demo/Makefile.am.ldflags ftgl-2.1.3~rc5/demo/Makefile.am
--- ftgl-2.1.3~rc5/demo/Makefile.am.ldflags	2008-06-02 03:10:10.000000000 +0200
+++ ftgl-2.1.3~rc5/demo/Makefile.am	2010-02-14 15:02:19.670081290 +0100
@@ -9,14 +9,14 @@ simple_SOURCES = \
     simple.cpp \
     $(NULL)
 simple_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-simple_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+simple_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS)
 simple_LDADD = ../src/libftgl.la
 
 c_demo_SOURCES = \
     c-demo.c \
     $(NULL)
 c_demo_CFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-c_demo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+c_demo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GLUT_LIBS) -lm
 c_demo_LDADD = ../src/libftgl.la
 
 FTGLDemo_SOURCES = \
@@ -27,7 +27,7 @@ FTGLDemo_SOURCES = \
 	trackball.h \
 	$(NULL)
 FTGLDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-FTGLDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+FTGLDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS)
 FTGLDemo_LDADD = ../src/libftgl.la
 
 FTGLMFontDemo_SOURCES = \
@@ -38,7 +38,7 @@ FTGLMFontDemo_SOURCES = \
 	trackball.h \
 	$(NULL)
 FTGLMFontDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-FTGLMFontDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+FTGLMFontDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS)
 FTGLMFontDemo_LDADD = ../src/libftgl.la
 
 NULL =
diff -up ftgl-2.1.3~rc5/demo/Makefile.in.ldflags ftgl-2.1.3~rc5/demo/Makefile.in
--- ftgl-2.1.3~rc5/demo/Makefile.in.ldflags	2008-06-12 16:33:01.000000000 +0200
+++ ftgl-2.1.3~rc5/demo/Makefile.in	2010-02-14 15:02:01.866831398 +0100
@@ -242,14 +242,14 @@ simple_SOURCES = \
     $(NULL)
 
 simple_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-simple_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+simple_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS)
 simple_LDADD = ../src/libftgl.la
 c_demo_SOURCES = \
     c-demo.c \
     $(NULL)
 
 c_demo_CFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-c_demo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+c_demo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS) -lm
 c_demo_LDADD = ../src/libftgl.la
 FTGLDemo_SOURCES = \
 	FTGLDemo.cpp \
@@ -260,7 +260,7 @@ FTGLDemo_SOURCES = \
 	$(NULL)
 
 FTGLDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-FTGLDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+FTGLDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS)
 FTGLDemo_LDADD = ../src/libftgl.la
 FTGLMFontDemo_SOURCES = \
 	FTGLMFontDemo.cpp \
@@ -271,7 +271,7 @@ FTGLMFontDemo_SOURCES = \
 	$(NULL)
 
 FTGLMFontDemo_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-FTGLMFontDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+FTGLMFontDemo_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS)
 FTGLMFontDemo_LDADD = ../src/libftgl.la
 NULL = 
 all: all-am
diff -up ftgl-2.1.3~rc5/test/Makefile.am.ldflags ftgl-2.1.3~rc5/test/Makefile.am
--- ftgl-2.1.3~rc5/test/Makefile.am.ldflags	2010-02-14 14:43:14.819077822 +0100
+++ ftgl-2.1.3~rc5/test/Makefile.am	2010-02-14 14:43:49.227081230 +0100
@@ -51,7 +51,7 @@ AM_CPPFLAGS = \
     $(NULL)
 
 CXXTest_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-CXXTest_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) -lcppunit
+CXXTest_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS) -lcppunit
 CXXTest_LDADD = ../src/libftgl.la
 
 CTest_SOURCES = \
@@ -64,7 +64,7 @@ CTest_CPPFLAGS = \
     -I$(top_srcdir)/src/FTFont \
     -I$(top_srcdir)/src/FTLayout
 CTest_CFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-CTest_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+CTest_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS)
 CTest_LDADD = ../src/libftgl.la
 
 NULL =
diff -up ftgl-2.1.3~rc5/test/Makefile.in.ldflags ftgl-2.1.3~rc5/test/Makefile.in
--- ftgl-2.1.3~rc5/test/Makefile.in.ldflags	2010-02-14 14:43:20.365077958 +0100
+++ ftgl-2.1.3~rc5/test/Makefile.in	2010-02-14 14:44:13.653078013 +0100
@@ -294,7 +294,7 @@ AM_CPPFLAGS = \
     $(NULL)
 
 CXXTest_CXXFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-CXXTest_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) -lcppunit
+CXXTest_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS) -lcppunit
 CXXTest_LDADD = ../src/libftgl.la
 CTest_SOURCES = \
     CTest.c \
@@ -308,7 +308,7 @@ CTest_CPPFLAGS = \
     -I$(top_srcdir)/src/FTLayout
 
 CTest_CFLAGS = $(FT2_CFLAGS) $(GL_CFLAGS)
-CTest_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS)
+CTest_LDFLAGS = $(FT2_LIBS) $(GLUT_LIBS) $(GL_LIBS)
 CTest_LDADD = ../src/libftgl.la
 NULL = 
 all: all-am
