--- FTGL/include/FTTextureGlyph.h   2018-07-26 08:30:55.144488976 +0000
+++ FTGL/include/FTTextureGlyph.h   2018-07-26 08:29:42.072489186 +0000
@@ -52,7 +52,7 @@
          * Reset the currently active texture to zero to get into a known state before
          * drawing a string. This is to get round possible threading issues.
          */
-        static void FTTextureGlyph::ResetActiveTexture(){ activeTextureID = 0;}
+        static void ResetActiveTexture(){ activeTextureID = 0;}
         
     private:
         /**
