diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2989bfe57..5fed42b44 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,7 +110,7 @@ include(${CMAKE_SOURCE_DIR}/cmake/set_rpath.cmake)
 # OpenMP
 #-------------------------------------------------------------------------------
 
-if (OPENMP OR BUILD_FASTFARM OR BUILD_OPENFAST_CPP_API)
+if (OPENMP)
   if (OPENMP)
     FIND_PACKAGE(OpenMP REQUIRED)
   else()
