--- julia/src/gc.c
+++ julia/src/gc.c
@@ -162,7 +162,7 @@
 // A region is contiguous storage for up to REGION_PG_COUNT naturally aligned GC_PAGE_SZ pages
 // It uses a very naive allocator (see malloc_page & free_page)
 #if defined(_P64) && !defined(_COMPILER_MICROSOFT_)
-#define REGION_PG_COUNT 16*8*4096 // 8G because virtual memory is cheap
+#define REGION_PG_COUNT 8*4096 // 512M
 #else
 #define REGION_PG_COUNT 8*4096 // 512M
 #endif
