--- a/contrib/normalize_triplet.py        2023-02-01 10:12:23.457386000 +0100
+++ b/contrib/normalize_triplet.py        2023-02-01 09:26:11.877554000 +0100
@@ -118,6 +118,7 @@
             "9":  "libgfortran5",
             "10": "libgfortran5",
             "11": "libgfortran5",
+            "12": "libgfortran5",
         }[list(filter(lambda x: re.match("\d+\.\d+(\.\d+)?", x), sys.argv[2].split()))[-1].split('.')[0]]

 if cxx_abi == "blank_cxx_abi":

