diff --git a/Source/Checks/cm_cxx_features.cmake b/Source/Checks/cm_cxx_features.cmake
index fb68ed78c9..c6c1ba667f 100644
--- a/Source/Checks/cm_cxx_features.cmake
+++ b/Source/Checks/cm_cxx_features.cmake
@@ -17,2 +17,6 @@ function(cm_check_cxx_feature name)
     set(check_output "${OUTPUT}")
+    # Filter out libhugetlbfs warnings
+    string(REGEX REPLACE "[^\n]*libhugetlbfs [^\n]*: WARNING[^\n]*" "" check_output "${check_output}")
+    # Filter out icpc warnings
+    string(REGEX REPLACE "[^\n]*icpc: command line warning #10121: overriding [^\n]*" "" check_output "${check_output}")
     # Filter out MSBuild output that looks like a warning.
