diff --git a/DDDigi/CMakeLists.txt b/DDDigi/CMakeLists.txt
index e6fb1096..88eb5c92 100644
--- a/DDDigi/CMakeLists.txt
+++ b/DDDigi/CMakeLists.txt
@@ -34,12 +34,10 @@ target_include_directories(DDDigi
 
 FIND_PACKAGE(TBB QUIET)
 if(TBB_FOUND)
-  dd4hep_print( "|++> TBB_INCLUDE_DIR -->  ${TBB_INCLUDE_DIR}")
-  dd4hep_print( "|++> TBB_LIBRARY     -->  ${TBB_LIBRARY}")
+  dd4hep_print( "|++> TBB_IMPORTED_TARGETS -->  ${TBB_IMPORTED_TARGETS}")
   dd4hep_print( "|++> TBB found. DDDigi will run multi threaded.")
   target_compile_definitions(DDDigi PUBLIC DD4HEP_USE_TBB)
-  target_link_libraries(DDDigi ${TBB_LIBRARY})
-  target_include_directories(DDDigi ${TBB_INCLUDE_DIRS})
+  target_link_libraries(DDDigi PUBLIC ${TBB_IMPORTED_TARGETS})
 else()
   dd4hep_print( "|++> TBB not found. DDDigi will only work single threaded.")
 endif()
diff --git a/DDDigi/src/DigiKernel.cpp b/DDDigi/src/DigiKernel.cpp
index d62c6694..f2c2e86c 100644
--- a/DDDigi/src/DigiKernel.cpp
+++ b/DDDigi/src/DigiKernel.cpp
@@ -91,7 +91,7 @@ public:
   DigiEventAction*  action = 0;
   Wrapper(DigiContext& c, DigiEventAction* a)
     : context(c), action(a) {}
-  Wrapper(Wrapper&& copy) = delete;
+  Wrapper(Wrapper&& copy) = default;
   Wrapper(const Wrapper& copy) = default;
   Wrapper& operator=(Wrapper&& copy) = delete;
   Wrapper& operator=(const Wrapper& copy) = delete;
@@ -111,7 +111,7 @@ class DigiKernel::Processor {
   DigiKernel& kernel;
 public: 
   Processor(DigiKernel& k) : kernel(k) {}
-  Processor(Processor&& l) = delete;
+  Processor(Processor&& l) = default;
   Processor(const Processor& l) = default;
   void operator()()  const {
     int todo = 1;
