diff --git a/xcbgen/align.py b/xcbgen/align.py
index d4c12ee..cc39764 100644
--- a/xcbgen/align.py
+++ b/xcbgen/align.py
@@ -2,7 +2,7 @@
 This module contains helper classes for alignment arithmetic and checks
 '''
 
-from fractions import gcd
+import math
 
 class Alignment(object):
 
@@ -73,14 +73,14 @@ class Alignment(object):
     def combine_with(self, other):
         # returns the alignment that is guaranteed when
         # both, self or other, can happen
-        new_align = gcd(self.align, other.align)
+        new_align = math.gcd(self.align, other.align)
         new_offset_candidate1 = self.offset % new_align
         new_offset_candidate2 = other.offset % new_align
         if new_offset_candidate1 == new_offset_candidate2:
             new_offset = new_offset_candidate1
         else:
             offset_diff = abs(new_offset_candidate2 - new_offset_candidate1)
-            new_align = gcd(new_align, offset_diff)
+            new_align = math.gcd(new_align, offset_diff)
             new_offset_candidate1 = self.offset % new_align
             new_offset_candidate2 = other.offset % new_align
             assert new_offset_candidate1 == new_offset_candidate2
