diff -uprN a/script/build_gcc.sh b/script/build_gcc.sh
--- a/script/build_gcc.sh	2023-02-02 16:42:26.000000000 +0900
+++ b/script/build_gcc.sh	2023-02-09 18:16:17.000000000 +0900
@@ -8,7 +8,7 @@ GXX_COMMAND=${CXX_COMPILER:-"g++"}
 mkdir -p ./build
 cd ./build
 if [ "${QULACS_OPT_FLAGS:-"__UNSET__"}" = "__UNSET__" ]; then
-  cmake -G "Unix Makefiles" -D CMAKE_C_COMPILER=$GCC_COMMAND -D CMAKE_CXX_COMPILER=$GXX_COMMAND -D CMAKE_BUILD_TYPE=Release -D USE_GPU:STR=No -D USE_TEST="${USE_TEST:-No}" -D COVERAGE="${COVERAGE:-No}" ..
+  cmake -G "Unix Makefiles" -D CMAKE_C_COMPILER=$GCC_COMMAND -D CMAKE_CXX_COMPILER=$GXX_COMMAND -D CMAKE_BUILD_TYPE=Release -D USE_GPU:STR=No -D USE_TEST="${USE_TEST:-No}" -D COVERAGE="${COVERAGE:-No}" -D PYTHON_EXECUTABLE=$PYTHONHOME/bin/python ..
 else
   cmake -G "Unix Makefiles" -D CMAKE_C_COMPILER=$GCC_COMMAND -D CMAKE_CXX_COMPILER=$GXX_COMMAND -D OPT_FLAGS="${QULACS_OPT_FLAGS}" -D CMAKE_BUILD_TYPE=Release -D USE_GPU:STR=No -D USE_TEST="${USE_TEST:-No}" -D COVERAGE="${COVERAGE:-No}" ..
 fi
