--- spack-src/src/third_party/Random123/features/gccfeatures.h.bak	2020-01-11 00:45:52.000000000 +0900
+++ spack-src/src/third_party/Random123/features/gccfeatures.h	2020-09-25 13:17:53.218231843 +0900
@@ -34,7 +34,7 @@
 
 #define R123_GNUC_VERSION (__GNUC__*10000 + __GNUC_MINOR__*100 + __GNUC_PATCHLEVEL__)
 
-#if !defined(__x86_64__) && !defined(__i386__) && !defined(__powerpc__)
+#if !defined(__x86_64__) && !defined(__i386__) && !defined(__powerpc__) && !defined(__aarch64__)
 #  error "This code has only been tested on x86 and powerpc platforms."
 #include <including_a_nonexistent_file_will_stop_some_compilers_from_continuing_with_a_hopeless_task>
 { /* maybe an unbalanced brace will terminate the compilation */
