diff --git a/bluepyopt/ephys/simulators.py b/bluepyopt/ephys/simulators.py
index e71ad8b..3c93237 100644
--- a/bluepyopt/ephys/simulators.py
+++ b/bluepyopt/ephys/simulators.py
@@ -89,6 +89,12 @@ class NrnSimulator(object):
             NrnSimulator._nrn_disable_banner()
             self.banner_disabled = True
 
+        # certain mpi libraries (hpe-mpt) use PMI_RANK env variable to initialize
+        # MPI before calling MPI_Init (which is undesirable). Unset this variable
+        # if exist to avoid issue with loading neuron and mechanism library.
+        if 'PMI_RANK' in os.environ:
+            os.environ.pop("PMI_RANK")
+
         import neuron  # NOQA
 
         return neuron
