--- scotch_6.0.4.orig/src/libscotch/common.h	2015-03-01 10:14:02.000000000 +0100
+++ scotch_6.0.4/src/libscotch/common.h	2017-03-27 13:07:18.644221999 +0200
@@ -306,6 +306,8 @@
   ThreadLaunchStartFunc     stafptr;              /*+ Pointer to start routine +*/
   ThreadLaunchJoinFunc      joifptr;              /*+ Pointer to join routine  +*/
   ThreadBarrier             barrdat;              /*+ Barrier data structure   +*/
+#else
+  int                       thrdnbr;    /* dummy for non-threaded */
 #endif /* ((defined COMMON_PTHREAD) || (defined SCOTCH_PTHREAD)) */
 } ThreadGroupHeader;
 
