--- a/conf/confpetsc.py	2023-10-31 00:35:21.000000000 -0500
+++ b/conf/confpetsc.py	2025-04-22 23:56:40.015502466 -0500
@@ -334,7 +334,7 @@
         ldshared = getenv('LDSHARED', ldshared)
         ldflags = getenv('LDFLAGS', cflags + ' ' + (ldflags or ''))
         ldcmd = split_quoted(ld) + split_quoted(ldflags)
-        ldshared = [flg for flg in split_quoted(ldshared) if flg not in ldcmd and (flg.find('/lib/spack/env')<0)]
+        ldshared = [flg for flg in split_quoted(ldshared) if flg not in ldcmd and (flg.find('/lib/spack/env')<0) and (flg.find('/libexec/spack')<0)]
         ldshared = str.join(' ', ldshared)
         #
         def get_flags(cmd):
