diff --git a/openmp/runtime/cmake/LibompHandleFlags.cmake b/openmp/runtime/cmake/LibompHandleFlags.cmake
index 9e19e59ba17d..f92fa12d851a 100644
--- a/openmp/runtime/cmake/LibompHandleFlags.cmake
+++ b/openmp/runtime/cmake/LibompHandleFlags.cmake
@@ -144,7 +144,8 @@ function(libomp_get_libflags libflags)
   endif()
   set(libflags_local ${libflags_local} ${LIBOMP_LIBFLAGS})
   libomp_setup_flags(libflags_local)
-  set(${libflags} ${libflags_local} PARENT_SCOPE)
+  libomp_string_to_list("${libflags_local}" libflags_local_list)
+  set(${libflags} ${libflags_local_list} PARENT_SCOPE)
 endfunction()

 # Fortran flags
