diff --git a/lldb/include/lldb/Utility/LLDBAssert.h_org b/lldb/include/lldb/Utility/LLDBAssert.h
index aeef3e5..2f14ff3 100644
--- a/lldb/include/lldb/Utility/LLDBAssert.h_org
+++ b/lldb/include/lldb/Utility/LLDBAssert.h
@@ -14,7 +14,7 @@
 #ifndef NDEBUG
 #define lldbassert(x) assert(x)
 #else
-#if defined(__clang__)
+#if defined(__clang__) && !defined(__CLANG_FUJITSU)
 // __FILE_NAME__ is a Clang-specific extension that functions similar to
 // __FILE__ but only renders the last path component (the filename) instead of
 // an invocation dependent full path to that file.

diff --git runtimes/CMakeLists.txt_org runtimes/CMakeLists.txt
--- a/runtimes/CMakeLists.txt_org
+++ b/runtimes/CMakeLists.txt
@@ -6,2 +6,2 @@
 include(${LLVM_COMMON_CMAKE_UTILS}/Modules/CMakePolicy.cmake
   NO_POLICY_SCOPE)

+string(REPLACE "-Nclang" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+string(REPLACE "-Nnofjprof" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+string(REPLACE "-Nfjprof" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
 project(Runtimes C CXX ASM)

 list(INSERT CMAKE_MODULE_PATH 0
