--- spack-src/config.make.orig	2017-06-09 21:15:21.000000000 +0900
+++ spack-src/config.make	2022-02-02 17:04:31.000000000 +0900
@@ -16,6 +16,8 @@
     H5Z_ZFP_BASE := ../src
 else ifeq ($(PWD_BASE),H5Z-ZFP)
     H5Z_ZFP_BASE := ./src
+else 
+    H5Z_ZFP_BASE := ./src
 endif
 H5Z_ZFP_PLUGIN := $(H5Z_ZFP_BASE)/plugin
 H5Z_ZFP_VERSINFO := $(shell grep '^\#define H5Z_FILTER_ZFP_VERSION_[MP]' $(H5Z_ZFP_BASE)/H5Zzfp_plugin.h | cut -d' ' -f3 | tr '\n' '.' | cut -d'.' -f-3 2>/dev/null)
