diff --git a/config.make b/config.make
index d782bd1..1c53ab8 100644
--- a/config.make
+++ b/config.make
@@ -128,18 +128,37 @@ ZFP_INC = $(ZFP_HOME)/inc
 else
 ZFP_INC = $(ZFP_HOME)/include
 endif
+ifeq ($(wildcard $(ZFP_HOME)/lib),)
+ZFP_LIB = $(ZFP_HOME)/lib64
+else
 ZFP_LIB = $(ZFP_HOME)/lib
+endif
+
+# Check if specified individually the HDF5 include directory,
+# library directory and bin directory separated by commas, i.e. HDF5_HOME=INC,LIB,BIN
 
-HDF5_INC = $(HDF5_HOME)/include
-HDF5_LIB = $(HDF5_HOME)/lib
-HDF5_BIN = $(HDF5_HOME)/bin
+ifneq (,$(findstring ",",$(HDF5_HOME)))
+  HDF5_INC = $(shell echo $(HDF5_HOME) | awk -F'[,]' '{print $$1}')
+  HDF5_LIB = $(shell echo $(HDF5_HOME) | awk -F'[,]' '{print $$2}')
+  HDF5_BIN = $(shell echo $(HDF5_HOME) | awk -F'[,]' '{print $$3}')
+  MAKEVARS =
+else
+  HDF5_INC = $(HDF5_HOME)/include
+  ifeq ($(wildcard $(HDF5_HOME)/lib),)
+    HDF5_LIB = $(HDF5_HOME)/lib64
+  else
+    HDF5_LIB = $(HDF5_HOME)/lib
+  endif
+  HDF5_BIN = $(HDF5_HOME)/bin
+  MAKEVARS = HDF5_HOME=$(HDF5_HOME)
+endif
 
 ifeq ($(PREFIX),)
     PREFIX := $(shell pwd)/install
 endif
 INSTALL ?= install
 
-MAKEVARS = ZFP_HOME=$(ZFP_HOME) HDF5_HOME=$(HDF5_HOME) PREFIX=$(PREFIX)
+MAKEVARS += ZFP_HOME=$(ZFP_HOME)  PREFIX=$(PREFIX)
 
 .SUFFIXES:
 .SUFFIXES: .c .F90 .h .o .mod
