--- spack-src/src/Makefile.orig	2022-02-02 17:42:03.000000000 +0900
+++ spack-src/src/Makefile	2022-02-02 17:42:34.000000000 +0900
@@ -54,7 +54,7 @@
 	$(INSTALL) libh5zzfp.a $(PREFIX)/lib
 	$(INSTALL) -m 644 H5Zzfp.h H5Zzfp_lib.h H5Zzfp_plugin.h H5Zzfp_props.h $(PREFIX)/include
 ifneq ($(FC),)
-	$(INSTALL) -m 644 H5Zzfp_props_f.mod $(PREFIX)/include
+	$(INSTALL) -m 644 *.[mM][oO][dD] $(PREFIX)/include
 endif
 
 clean:
