diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc4d98e..f1dea96 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -164,7 +164,7 @@ endif()
 message (STATUS "CPACK_GENERATOR ${CPACK_GENERATOR}" )
 
 if ( NOT DEFINED CMAKE_PACKAGING_INSTALL_PREFIX )
-    set (CMAKE_PACKAGING_INSTALL_PREFIX "/opt/rocm" )
+    set (CMAKE_PACKAGING_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" )
 endif ()
 
 ################################################################################
@@ -369,16 +369,16 @@ if (RVS_ROCBLAS EQUAL 1)
   set(ROCBLAS_INC_DIR "${CMAKE_BINARY_DIR}/rvs_rblas-src/build/release/rocblas-install")
   set(ROCBLAS_LIB_DIR "${CMAKE_BINARY_DIR}/rvs_rblas-src/build/release/rocblas-install/lib/")
 else()
-  set(ROCBLAS_INC_DIR "${ROCM_PATH}/include")
-  set(ROCBLAS_LIB_DIR "${ROCM_PATH}/lib")
+  set(ROCBLAS_INC_DIR "${ROCBLAS_DIR}/include")
+  set(ROCBLAS_LIB_DIR "${ROCBLAS_DIR}/lib")
 endif()
 
 if (RVS_ROCMSMI EQUAL 1)
   set(ROCM_SMI_INC_DIR "${CMAKE_BINARY_DIR}/rvs_smi-src/include")
   set(ROCM_SMI_LIB_DIR "${CMAKE_BINARY_DIR}/rvs_smi-build")
 else()
-  set(ROCM_SMI_INC_DIR "${ROCM_PATH}/rocm_smi/include")
-  set(ROCM_SMI_LIB_DIR "${ROCM_PATH}/rocm_smi/lib")
+  set(ROCM_SMI_INC_DIR "${ROCM_SMI_DIR}/include")
+  set(ROCM_SMI_LIB_DIR "${ROCM_SMI_DIR}/lib")
 endif()
 set(ROCM_SMI_LIB "rocm_smi64" CACHE STRING "rocm_smi library name")
