From 3b4094b9cf1beec0b870e9ca0c3e51ed9b10dc39 Mon Sep 17 00:00:00 2001
From: John Jolly <john.jolly@gmail.com>
Date: Fri, 5 Jun 2020 23:21:18 -0600
Subject: [PATCH] Add ISLHOME option

---
 Makefile.am  | 4 ++++
 configure.ac | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 30667a8..541cb99 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -16,6 +16,8 @@ core_libs    += -liegenlib -lisl -lgmp
 core_libdirs  = -Lomega/code_gen/obj -Lomega/omega_lib/obj
 core_libdirs += -L$(ROSEHOME)/lib -L$(BOOSTHOME)/lib
 
+core_libdirs += -L$(ISLHOME)/lib
+
 core_libdirs += -L$(IEGENHOME)/lib
 core_libdirs += -L$(IEGENHOME)/../lib/installed/lib
 
@@ -30,6 +32,8 @@ core_includes += -I$(srcdir)/include/chill
 core_includes += -I$(ROSEHOME)/include/rose
 core_includes += -I$(BOOSTHOME)/include
 
+core_includes += -I$(ISLHOME)/include
+
 core_includes += -I$(IEGENHOME)/include/iegenlib
 core_includes += -I$(IEGENHOME)/../lib/installed/include
 
diff --git a/configure.ac b/configure.ac
index 8e163cd..e146839 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,6 +65,13 @@ AC_ARG_WITH([iegen],[
     ],[
     AC_SUBST([IEGENHOME], ["${IEGENHOME}"])])
 
+AC_ARG_WITH([isl],[
+    AS_HELP_STRING([--with-isl],[set path to isl])
+    ],[
+    AC_SUBST([ISLHOME], [$withval])
+    ],[
+    AC_SUBST([ISLHOME], ["${ISLHOME}"])])
+
 AC_ARG_WITH([omega],[
     AS_HELP_STRING([--with-omega],[set omega home])],[
     AC_SUBST([OMEGAHOME], [$withval])],[
-- 
2.25.1

