From 336dfb3a1c314b6eec23ab4f99114be94c5e518e Mon Sep 17 00:00:00 2001
From: John Jolly <john.jolly@gmail.com>
Date: Fri, 5 Jun 2020 23:45:39 -0600
Subject: [PATCH] Add GMPHOME option

---
 Makefile.am  | 4 ++++
 configure.ac | 7 +++++++
 2 files changed, 11 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 541cb99..7dd92b0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -16,6 +16,8 @@ core_libs    += -liegenlib -lisl -lgmp
 core_libdirs  = -Lomega/code_gen/obj -Lomega/omega_lib/obj
 core_libdirs += -L$(ROSEHOME)/lib -L$(BOOSTHOME)/lib
 
+core_libdirs += -L$(GMPHOME)/lib
+
 core_libdirs += -L$(ISLHOME)/lib
 
 core_libdirs += -L$(IEGENHOME)/lib
@@ -32,6 +34,8 @@ core_includes += -I$(srcdir)/include/chill
 core_includes += -I$(ROSEHOME)/include/rose
 core_includes += -I$(BOOSTHOME)/include
 
+core_includes += -I$(GSLHOME)/include
+
 core_includes += -I$(ISLHOME)/include
 
 core_includes += -I$(IEGENHOME)/include/iegenlib
diff --git a/configure.ac b/configure.ac
index e146839..18ecf76 100644
--- a/configure.ac
+++ b/configure.ac
@@ -58,6 +58,13 @@ AC_ARG_WITH([boost],[
     ],[
     AC_SUBST([BOOSTHOME], ["${BOOSTHOME}"])])
 
+AC_ARG_WITH([gmp],[
+    AS_HELP_STRING([--with-gmp],[set path to gmp])
+    ],[
+    AC_SUBST([GMPHOME], [$withval])
+    ],[
+    AC_SUBST([GMPHOME], ["${GMPHOME}"])])
+
 AC_ARG_WITH([iegen],[
     AS_HELP_STRING([--with-iegen],[set path to iegenlib])
     ],[
-- 
2.25.1

