From 15c3cc127d8f48324b00fd654d9630724eedcfcd Mon Sep 17 00:00:00 2001
From: John Jolly <john.jolly@gmail.com>
Date: Fri, 5 Jun 2020 23:54:13 -0600
Subject: [PATCH] Add GCC libquadmath for rose

---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 7dd92b0..a8daba1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -7,7 +7,7 @@ core_cxxflags   =-g --std=c++11
 
 ## Core Libraries ##
 core_libs     = -lcodegen -lomega
-core_libs    += -lm -lrose -lrt -lutil -ldl
+core_libs    += -lm -lrose -lrt -lutil -ldl -lquadmath
 core_libs    += -lboost_date_time -lboost_filesystem -lboost_program_options
 core_libs    += -lboost_regex -lboost_system -lboost_wave -lboost_iostreams
 
-- 
2.25.1

