--- spack-src/arch/configure.defaults.bak	2020-04-24 04:31:06.000000000 +0900
+++ spack-src/arch/configure.defaults	2020-12-11 15:33:51.215405970 +0900
@@ -168,7 +168,7 @@
 RANLIB              = ranlib
 
 ########################################################################################################################
-#ARCH    Linux x86_64, gfortran   # serial serial_NO_GRIB2 dmpar dmpar_NO_GRIB2
+#ARCH    Linux aarch64, gfortran   # serial serial_NO_GRIB2 dmpar dmpar_NO_GRIB2
 #
 COMPRESSION_LIBS    = CONFIGURE_COMP_L
 COMPRESSION_INC     = CONFIGURE_COMP_I
