diff --git a/src/ambfor/makefile b/src/ambfor/makefile
--- a/src/ambfor/makefile
+++ b/src/ambfor/makefile
@@ -15,7 +15,7 @@
 CFLAGS = -c -g ${EXT} -Wno-implicit-function-declaration
 #FC = g77 
 FC = gfortran
-FFLAGS = -c -g -ffast-math -funroll-loops -O3
+FFLAGS = -c -g -ffast-math -funroll-loops -O3 -fallow-argument-mismatch
 LIBS =  -lm
 LDR = ${FC} -g
 EXT :=
@@ -46,7 +46,7 @@
 #mpi:    FFLAGS +=  -frecord-marker=4
 mpi:	FC = ${OPENMPI}/bin/mpif77
 mpi:	CC = ${OPENMPI}/bin/mpicc
-mpi:	FFLAGS = -c -g -ffast-math -funroll-loops -O3
+mpi:	FFLAGS = -c -g -ffast-math -funroll-loops -O3 -fallow-argument-mismatch
 mpi:	ambfor.mpi ambmd.mpi
 
 ambfor:	ambfor.o qvdw.o alloc.o optimise.o lbfgs.o $(OBJS)
